--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegate.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegate.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 @protocol RCTBackedTextInputViewProtocol;
 
@@ -25,6 +25,13 @@
                                // Dismisses keyboard if true
 - (void)textInputDidReturn;
 
+#if TARGET_OS_OSX // [macOS
+- (void)automaticSpellingCorrectionDidChange:(BOOL)enabled;
+- (void)continuousSpellCheckingDidChange:(BOOL)enabled;
+- (void)grammarCheckingDidChange:(BOOL)enabled;
+- (void)submitOnKeyDownIfNeeded:(NSEvent *)event;
+#endif // macOS]
+
 - (BOOL)textInputShouldSubmitOnReturn; // Checks whether to submit when return is pressed and emits an event if true.
 
 /*
@@ -32,15 +39,26 @@
  * reject the change completely. To change the replacement, return the changed version of the `text`. To accept the
  * change, return `text` argument as-is. To reject the change, return `nil`.
  */
-- (NSString *)textInputShouldChangeText:(NSString *)text inRange:(NSRange)range;
+- (nullable NSString *)textInputShouldChangeText:(NSString *)text inRange:(NSRange)range; // [macOS]
 - (void)textInputDidChange;
 
 - (void)textInputDidChangeSelection;
 
+- (BOOL)textInputShouldHandleDeleteBackward:(id<RCTBackedTextInputViewProtocol>)sender; // Return `YES` to have the deleteBackward event handled normally. Return `NO` to disallow it and handle it yourself. // [macOS]
+#if TARGET_OS_OSX // [macOS
+- (BOOL)textInputShouldHandleDeleteForward:(id<RCTBackedTextInputViewProtocol>)sender; // Return `YES` to have the deleteForward event handled normally. Return `NO` to disallow it and handle it yourself.
+- (BOOL)textInputShouldHandleKeyEvent:(NSEvent *)event; // Return `YES` to have the key event handled normally. Return `NO` to disallow it and handle it yourself.
+- (BOOL)hasKeyDownEventOrKeyUpEvent:(NSString *)key;
+- (NSDragOperation)textInputDraggingEntered:(id<NSDraggingInfo>)draggingInfo;
+- (void)textInputDraggingExited:(id<NSDraggingInfo>)draggingInfo;
+- (BOOL)textInputShouldHandleDragOperation:(id<NSDraggingInfo>)draggingInfo;
+- (void)textInputDidCancel;  // Handle `Escape` key press.
+- (BOOL)textInputShouldHandlePaste:(id<RCTBackedTextInputViewProtocol>)sender; // Return `YES` to have the paste event handled normally. Return `NO` to disallow it and handle it yourself.
+#endif // macOS]
+
 @optional
+- (void)scrollViewDidScroll:(RCTUIScrollView *)scrollView; // [macOS]
 
-- (void)scrollViewDidScroll:(UIScrollView *)scrollView;
-
 @end
 
 NS_ASSUME_NONNULL_END
