--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/components/legacyviewmanagerinterop/RCTLegacyViewManagerInteropCoordinator.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.GB5RGbAAA3	2026-03-06 18:50:53
@@ -7,7 +7,7 @@
 
 #import <Foundation/Foundation.h>
 #import <React/RCTBridgeModuleDecorator.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 #include <folly/dynamic.h>
 
 NS_ASSUME_NONNULL_BEGIN
@@ -25,20 +25,20 @@
                           bridgeProxy:(nullable RCTBridgeProxy *)bridgeProxy
                 bridgelessInteropData:(RCTBridgeModuleDecorator *)bridgelessInteropData;
 
-- (UIView *)createPaperViewWithTag:(NSInteger)tag;
+- (RCTPlatformView *)createPaperViewWithTag:(NSInteger)tag; // [macOS]
 
 - (void)addObserveForTag:(NSInteger)tag usingBlock:(InterceptorBlock)block;
 
 - (void)removeObserveForTag:(NSInteger)tag;
 
-- (void)setProps:(NSDictionary<NSString *, id> *)props forView:(UIView *)view;
+- (void)setProps:(NSDictionary<NSString *, id> *)props forView:(RCTPlatformView *)view; // [macOS]
 
 - (NSString *)componentViewName;
 
 - (void)handleCommand:(NSString *)commandName
                  args:(NSArray *)args
              reactTag:(NSInteger)tag
-            paperView:(UIView *)paperView;
+            paperView:(RCTPlatformView *)paperView; // [macOS]
 @end
 
 NS_ASSUME_NONNULL_END
