--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/Image.ios.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Image/Image.ios.js	2026-03-06 15:22:52
@@ -33,7 +33,7 @@
 function getSize(
   uri: string,
   success?: (width: number, height: number) => void,
-  failure?: (error: unknown) => void,
+  failure?: (error: mixed) => void,
 ): void | Promise<ImageSize> {
   const promise = NativeImageLoaderIOS.getSize(uri).then(([width, height]) => ({
     width,
@@ -56,7 +56,7 @@
   uri: string,
   headers: {[string]: string, ...},
   success?: (width: number, height: number) => void,
-  failure?: (error: unknown) => void,
+  failure?: (error: mixed) => void,
 ): void | Promise<ImageSize> {
   const promise = NativeImageLoaderIOS.getSizeWithHeaders(uri, headers);
   if (typeof success !== 'function') {
@@ -153,7 +153,7 @@
     'aria-disabled': ariaDisabled,
     'aria-expanded': ariaExpanded,
     'aria-selected': ariaSelected,
-    'aria-hidden': ariaHidden,
+    accessibilityRole, // [macOS]
     src,
     ...restProps
   } = props;
@@ -165,10 +165,6 @@
     expanded: ariaExpanded ?? props.accessibilityState?.expanded,
     selected: ariaSelected ?? props.accessibilityState?.selected,
   };
-
-  // In order for `aria-hidden` to work on iOS we must set `accessible` to false (`accessibilityElementsHidden` is not enough).
-  const accessible =
-    ariaHidden !== true && (props.alt !== undefined ? true : props.accessible);
   const accessibilityLabel = props['aria-label'] ?? props.accessibilityLabel;
 
   const actualRef = useWrapRefWithImageAttachedCallbacks(forwardedRef);
@@ -179,8 +175,9 @@
         return (
           <ImageViewNativeComponent
             accessibilityState={_accessibilityState}
+            accessibilityRole={accessibilityRole ?? 'image'} // [macOS]
             {...restProps}
-            accessible={accessible}
+            accessible={props.alt !== undefined ? true : props.accessible}
             accessibilityLabel={accessibilityLabel ?? props.alt}
             ref={actualRef}
             style={style}
