--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/RCTAnimatedImage.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Image/RCTAnimatedImage.h	2026-03-06 15:22:52
@@ -5,17 +5,21 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 @protocol RCTAnimatedImage <NSObject>
 @property (nonatomic, assign, readonly) NSUInteger animatedImageFrameCount;
 @property (nonatomic, assign, readonly) NSUInteger animatedImageLoopCount;
 
-- (nullable UIImage *)animatedImageFrameAtIndex:(NSUInteger)index;
+- (nullable RCTPlatformImage *)animatedImageFrameAtIndex:(NSUInteger)index; // [macOS]
 - (NSTimeInterval)animatedImageDurationAtIndex:(NSUInteger)index;
 
 @end
 
-@interface RCTAnimatedImage : UIImage <RCTAnimatedImage>
-
+@interface RCTAnimatedImage : RCTPlatformImage <RCTAnimatedImage> // [macOS]
+// [macOS
+// This is a known initializer for UIImage, but needs to be exposed publicly for macOS since
+// this is not a known initializer for NSImage
+- (nullable instancetype)initWithData:(NSData *)data scale:(CGFloat)scale;
+// macOS]
 @end
