--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/RCTImageView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Image/RCTImageView.h	2026-03-06 15:22:52
@@ -7,26 +7,24 @@
 
 #import <React/RCTResizeMode.h>
 #import <React/RCTView.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 @class RCTBridge;
 @class RCTImageSource;
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
 @interface RCTImageView : RCTView
 
 - (instancetype)initWithBridge:(RCTBridge *)bridge NS_DESIGNATED_INITIALIZER;
 
 @property (nonatomic, assign) UIEdgeInsets capInsets;
-@property (nonatomic, strong) UIImage *defaultImage;
+@property (nonatomic, strong) RCTPlatformImage *defaultImage; // [macOS]
 @property (nonatomic, assign) UIImageRenderingMode renderingMode;
 @property (nonatomic, copy) NSArray<RCTImageSource *> *imageSources;
 @property (nonatomic, assign) CGFloat blurRadius;
 @property (nonatomic, assign) RCTResizeMode resizeMode;
 @property (nonatomic, copy) NSString *internal_analyticTag;
 
+#if TARGET_OS_OSX // [macOS
+@property (nonatomic, copy) NSColor *tintColor;
+#endif // macOS]
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
