--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/RCTImageView.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Image/RCTImageView.mm	2026-03-06 15:22:52
@@ -17,8 +17,6 @@
 #import <React/RCTUtils.h>
 #import <React/UIView+React.h>
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 /**
  * Determines whether an image of `currentSize` should be reloaded for display
  * at `idealSize`.
@@ -35,7 +33,57 @@
       heightMultiplier > upscaleThreshold || heightMultiplier < downscaleThreshold;
 }
 
+#if TARGET_OS_OSX // [macOS
 /**
+ * Implements macOS equivalent behavior of UIViewContentModeScaleAspectFill.
+ * Used for RCTResizeModeCover support.
+ */
+static NSImage *RCTFillImagePreservingAspectRatio(NSImage *originalImage, NSSize targetSize, CGFloat windowScale)
+{
+  RCTAssertParam(originalImage);
+  if (!originalImage) {
+    return nil;
+  }
+
+  NSSize originalImageSize = originalImage.size;
+  if (NSEqualSizes(originalImageSize, NSZeroSize) ||
+      NSEqualSizes(originalImageSize, targetSize) ||
+      [[originalImage representations] count] == 0) {
+    return originalImage;
+  }
+
+  CGFloat scaleX = targetSize.width / originalImageSize.width;
+  CGFloat scaleY = targetSize.height / originalImageSize.height;
+  CGFloat scale = 1.0;
+
+  if (scaleX < scaleY) {
+    // clamped width
+    scale = scaleY;
+  }
+  else {
+    // clamped height
+    scale = scaleX;
+  }
+
+  NSSize newSize = NSMakeSize(RCTRoundPixelValue(originalImageSize.width * scale),
+                              RCTRoundPixelValue(originalImageSize.height * scale));
+  NSImage *newImage = [[NSImage alloc] initWithSize:newSize];
+
+  for (NSImageRep *imageRep in [originalImage representations]) {
+    NSImageRep *newImageRep = [imageRep copy];
+    NSSize newImageRepSize = NSMakeSize(RCTRoundPixelValue(imageRep.size.width * scale),
+                                        RCTRoundPixelValue(imageRep.size.height * scale));
+
+    newImageRep.size = newImageRepSize;
+
+    [newImage addRepresentation:newImageRep];
+  }
+
+  return newImage;
+}
+#endif // macOS]
+
+/**
  * See RCTConvert (ImageSource). We want to send down the source as a similar
  * JSON parameter.
  */
@@ -79,6 +127,8 @@
   // Whether the latest change of props requires the image to be reloaded
   BOOL _needsReload;
 
+  RCTPlatformImage *_image; // [macOS]
+
   RCTUIImageViewAnimated *_imageView;
 
   RCTImageURLLoaderRequest *_loaderRequest;
@@ -86,12 +136,20 @@
 
 - (instancetype)initWithBridge:(RCTBridge *)bridge
 {
+#if !TARGET_OS_OSX // [macOS]
   if ((self = [super initWithFrame:CGRectZero])) {
+#else // [macOS
+  if ((self = [super initWithFrame:NSZeroRect])) {
+#endif // macOS]
     _bridge = bridge;
+#if TARGET_OS_OSX // [macOS
+    self.wantsLayer = YES;
+#endif // macOS]
     _imageView = [RCTUIImageViewAnimated new];
     _imageView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
     [self addSubview:_imageView];
 
+#if !TARGET_OS_OSX // [macOS]
     NSNotificationCenter *center = [NSNotificationCenter defaultCenter];
     [center addObserver:self
                selector:@selector(clearImageIfDetached)
@@ -106,6 +164,7 @@
 
                    name:UISceneDidEnterBackgroundNotification
                  object:nil];
+#endif // [macOS]
   }
   return self;
 }
@@ -116,7 +175,7 @@
 
 RCT_NOT_IMPLEMENTED(-(instancetype)initWithFrame : (CGRect)frame)
 
-- (void)updateWithImage:(UIImage *)image
+- (void)updateWithImage:(RCTPlatformImage *)image // [macOS]
 {
   if (!image) {
     _imageView.image = nil;
@@ -124,17 +183,39 @@
   }
 
   // Apply rendering mode
+#if !TARGET_OS_OSX // [macOS]
   if (_renderingMode != image.renderingMode) {
     image = [image imageWithRenderingMode:_renderingMode];
   }
+#endif // [macOS]
 
   if (_resizeMode == RCTResizeModeRepeat) {
+#if !TARGET_OS_OSX // [macOS]
     image = [image resizableImageWithCapInsets:_capInsets resizingMode:UIImageResizingModeTile];
+#else // [macOS
+    image.capInsets = _capInsets;
+    image.resizingMode = NSImageResizingModeTile;
+  } else if (_resizeMode == RCTResizeModeCover) {
+    if (!NSEqualSizes(self.bounds.size, NSZeroSize)) {
+      image = RCTFillImagePreservingAspectRatio(image, self.bounds.size, self.window.backingScaleFactor ?: 1.0);
+    }
+#endif // macOS]
   } else if (!UIEdgeInsetsEqualToEdgeInsets(UIEdgeInsetsZero, _capInsets)) {
     // Applying capInsets of 0 will switch the "resizingMode" of the image to "tile" which is undesired
+#if !TARGET_OS_OSX // [macOS]
     image = [image resizableImageWithCapInsets:_capInsets resizingMode:UIImageResizingModeStretch];
+#else // [macOS
+    image.capInsets = _capInsets;
+    image.resizingMode = NSImageResizingModeStretch;
+#endif // macOS]
   }
 
+#if TARGET_OS_OSX // [macOS
+  if ((_renderingMode == UIImageRenderingModeAlwaysTemplate) != [image isTemplate]) {
+    [image setTemplate:(_renderingMode == UIImageRenderingModeAlwaysTemplate)];
+  }
+#endif // macOS]
+
   // Apply trilinear filtering to smooth out missized images
   _imageView.layer.minificationFilter = kCAFilterTrilinear;
   _imageView.layer.magnificationFilter = kCAFilterTrilinear;
@@ -142,17 +223,18 @@
   _imageView.image = image;
 }
 
-- (void)setImage:(UIImage *)image
+- (void)setImage:(RCTPlatformImage *)image // [macOS]
 {
   image = image ?: _defaultImage;
   if (image != self.image) {
+    _image = image; // [macOS]
     [self updateWithImage:image];
   }
 }
 
-- (UIImage *)image
+- (RCTPlatformImage *)image // [macOS]
 {
-  return _imageView.image;
+  return _image ?: _imageView.image; // [macOS]
 }
 
 - (void)setBlurRadius:(CGFloat)blurRadius
@@ -202,9 +284,21 @@
     if (_resizeMode == RCTResizeModeRepeat) {
       // Repeat resize mode is handled by the UIImage. Use scale to fill
       // so the repeated image fills the UIImageView.
+#if !TARGET_OS_OSX // [macOS]
       _imageView.contentMode = UIViewContentModeScaleToFill;
+#else // [macOS
+      _imageView.imageScaling = NSImageScaleAxesIndependently;
+#endif // macOS]
     } else {
+#if !TARGET_OS_OSX // [macOS]
       _imageView.contentMode = (UIViewContentMode)resizeMode;
+#else // [macOS
+      // This relies on having previously resampled the image to a size that exceeds the image view.
+      if (resizeMode == RCTResizeModeCover) {
+        resizeMode = RCTResizeModeCenter;
+      }
+      _imageView.imageScaling = (NSImageScaling)resizeMode;
+#endif // macOS]
     }
 
     if ([self shouldReloadImageSourceAfterResize]) {
@@ -239,6 +333,7 @@
   }
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)clearImageIfDetached
 {
   if (!self.window) {
@@ -247,6 +342,7 @@
     _imageSource = nil;
   }
 }
+#endif // [macOS]
 
 - (BOOL)hasMultipleSources
 {
@@ -264,7 +360,11 @@
     return nil;
   }
 
+#if !TARGET_OS_OSX // [macOS]
   const CGFloat scale = RCTScreenScale();
+#else // [macOS
+  const CGFloat scale = self.window != nil ? self.window.backingScaleFactor : [NSScreen mainScreen].backingScaleFactor;
+#endif // macOS]
   const CGFloat targetImagePixels = size.width * size.height * scale * scale;
 
   RCTImageSource *bestSource = nil;
@@ -321,19 +421,23 @@
     }
 
     __weak RCTImageView *weakSelf = self;
-    RCTImageLoaderPartialLoadBlock partialLoadHandler = ^(UIImage *image) {
+    RCTImageLoaderPartialLoadBlock partialLoadHandler = ^(RCTPlatformImage *image) { // [macOS]
       [weakSelf imageLoaderLoadedImage:image error:nil forImageSource:source partial:YES];
     };
 
     CGSize imageSize = self.bounds.size;
+#if !TARGET_OS_OSX // [macOS]
     CGFloat imageScale = RCTScreenScale();
+#else // [macOS
+    CGFloat imageScale = self.window != nil ? self.window.backingScaleFactor : [NSScreen mainScreen].backingScaleFactor;
+#endif // macOS]
     if (!UIEdgeInsetsEqualToEdgeInsets(_capInsets, UIEdgeInsetsZero)) {
       // Don't resize images that use capInsets
       imageSize = CGSizeZero;
       imageScale = source.scale;
     }
 
-    RCTImageLoaderCompletionBlockWithMetadata completionHandler = ^(NSError *error, UIImage *loadedImage, id metadata) {
+    RCTImageLoaderCompletionBlockWithMetadata completionHandler = ^(NSError *error, RCTPlatformImage *loadedImage, id metadata) { // [macOS]
       [weakSelf imageLoaderLoadedImage:loadedImage error:error forImageSource:source partial:NO];
     };
 
@@ -360,7 +464,7 @@
   }
 }
 
-- (void)imageLoaderLoadedImage:(UIImage *)loadedImage
+- (void)imageLoaderLoadedImage:(RCTPlatformImage *)loadedImage // [macOS]
                          error:(NSError *)error
                 forImageSource:(RCTImageSource *)source
                        partial:(BOOL)isPartialLoad
@@ -390,7 +494,7 @@
   }
 
   __weak RCTImageView *weakSelf = self;
-  void (^setImageBlock)(UIImage *) = ^(UIImage *image) {
+  void (^setImageBlock)(RCTPlatformImage *) = ^(RCTPlatformImage *image) { // [macOS]
     RCTImageView *strongSelf = weakSelf;
     if (!strongSelf) {
       return;
@@ -408,7 +512,7 @@
       }
     } else {
       if (strongSelf->_onLoad) {
-        RCTImageSource *sourceLoaded = [source imageSourceWithSize:image.size scale:image.scale];
+        RCTImageSource *sourceLoaded = [source imageSourceWithSize:image.size scale:UIImageGetScale(image)]; // [macOS]
         strongSelf->_onLoad(onLoadParamsForSource(sourceLoaded));
       }
       if (strongSelf->_onLoadEnd) {
@@ -421,7 +525,7 @@
     // Blur on a background thread to avoid blocking interaction
     CGFloat blurRadius = self.blurRadius;
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-      UIImage *blurredImage = RCTBlurredImageWithRadius(loadedImage, blurRadius);
+      RCTPlatformImage *blurredImage = RCTBlurredImageWithRadius(loadedImage, blurRadius); // [macOS]
       RCTExecuteOnMainQueue(^{
         setImageBlock(blurredImage);
       });
@@ -448,13 +552,26 @@
     [self reloadImage];
   } else if ([self shouldReloadImageSourceAfterResize]) {
     CGSize imageSize = self.image.size;
-    CGFloat imageScale = self.image.scale;
+    CGFloat imageScale = UIImageGetScale(self.image); // [macOS]
+#if !TARGET_OS_OSX // [macOS]
     CGSize idealSize = RCTTargetSize(
         imageSize, imageScale, frame.size, RCTScreenScale(), RCTResizeModeFromUIViewContentMode(self.contentMode), YES);
+#else // [macOS
+    CGSize idealSize = RCTTargetSize(
+        imageSize, imageScale, frame.size, RCTScreenScale(), RCTResizeModeFromUIViewContentMode(_imageView.contentMode), YES);
+#endif // macOS]
 
     // Don't reload if the current image or target image size is close enough
-    if (!RCTShouldReloadImageForSizeChange(imageSize, idealSize) ||
-        !RCTShouldReloadImageForSizeChange(_targetSize, idealSize)) {
+    if ((!RCTShouldReloadImageForSizeChange(imageSize, idealSize) ||
+         !RCTShouldReloadImageForSizeChange(_targetSize, idealSize)) // [macOS
+#if TARGET_OS_OSX
+         // Since macOS doen't support UIViewContentModeScaleAspectFill, we have to manually resample the image
+         // If we're in cover mode we need to ensure that the image is re-sampled to the correct size when the container size (shrinking 
+         // being the most obvious case) otherwise we will end up in a state an image will not properly scale inside its container
+         && (RCTResizeModeFromUIViewContentMode(_imageView.contentMode) != RCTResizeModeCover ||
+            (imageSize.width == idealSize.width && imageSize.height == idealSize.height))
+#endif
+         ) { // macOS]
       return;
     }
 
@@ -482,6 +599,9 @@
   }
 }
 
+#if TARGET_OS_OSX // [macOS
+#define didMoveToWindow viewDidMoveToWindow
+#endif
 - (void)didMoveToWindow
 {
   [super didMoveToWindow];
@@ -497,12 +617,37 @@
     [self reloadImage];
   }
 }
+    
+#if TARGET_OS_OSX // [macOS
+- (void)viewDidChangeBackingProperties
+{
+  [self reloadImage];
+}
+  
+- (RCTPlatformView *)reactAccessibilityElement
+{
+  return _imageView;
+}
 
+- (NSCell *)reactAccessibilityElementCell
+  {
+    return _imageView.cell;
+  }
+
+- (NSColor *)tintColor
+{
+  return _imageView.tintColor;
+}
+
+- (void)setTintColor:(NSColor *)tintColor
+{
+  [_imageView setTintColor:tintColor];
+}
+#endif // macOS]
+
 - (void)dealloc
 {
   [_imageLoader trackURLImageDidDestroy:_loaderRequest];
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
