--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/RCTImageLoader.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Image/RCTImageLoader.mm	2026-03-06 15:22:52
@@ -26,11 +26,51 @@
 
 using namespace facebook::react;
 
-static NSInteger RCTImageBytesForImage(UIImage *image)
+static NSInteger RCTImageBytesForImage(RCTPlatformImage *image) // [macOS]
 {
-  NSInteger singleImageBytes = (NSInteger)(image.size.width * image.size.height * image.scale * image.scale * 4);
+  CGFloat imageScale = 1.0;
+#if !TARGET_OS_OSX // [macOS] no .scale prop on NSImage
+  imageScale = image.scale;
+#endif // [macOS]
+  NSInteger singleImageBytes = (NSInteger)(image.size.width * image.size.height * imageScale * imageScale * 4);
+#if !TARGET_OS_OSX // [macOS]
   return image.images ? image.images.count * singleImageBytes : singleImageBytes;
+#else // [macOS
+    return singleImageBytes;
+#endif // macOS]
+}
+
+#if TARGET_OS_OSX // [macOS
+
+/**
+ * Github #1611 - We can't depend on RCTUIKit here, because this file's podspec (React-RCTImage) doesn't
+ * take a dependency on RCTUIKit's pod `React-Core`. Let's just copy the methods we want to shim here
+ */
+
+static NSData *NSImageDataForFileType(NSImage *image, NSBitmapImageFileType fileType, NSDictionary<NSString *, id> *properties)
+{
+  RCTAssert(image.representations.count == 1, @"Expected only a single representation since UIImage only supports one.");
+
+  NSBitmapImageRep *imageRep = (NSBitmapImageRep *)image.representations.firstObject;
+  if (![imageRep isKindOfClass:[NSBitmapImageRep class]]) {
+    RCTAssert([imageRep isKindOfClass:[NSBitmapImageRep class]], @"We need an NSBitmapImageRep to create an image.");
+    return nil;
+  }
+
+  return [imageRep representationUsingType:fileType properties:properties];
+}
+
+
+NSData *UIImagePNGRepresentation(NSImage *image) {
+  return NSImageDataForFileType(image, NSBitmapImageFileTypePNG, @{});
+}
+
+NSData *UIImageJPEGRepresentation(NSImage *image, CGFloat compressionQuality) {
+  return NSImageDataForFileType(image,
+                                NSBitmapImageFileTypeJPEG,
+                                @{NSImageCompressionFactor: @(compressionQuality)});
 }
+#endif // macOS]
 
 static auto currentRequestCount = std::atomic<uint64_t>(0);
 
@@ -49,7 +89,7 @@
 
 @end
 
-@implementation UIImage (React)
+@implementation RCTPlatformImage (React) // [macOS]
 
 - (NSInteger)reactDecodedImageBytes
 {
@@ -305,7 +345,7 @@
   return nil;
 }
 
-static UIImage *RCTResizeImageIfNeeded(UIImage *image, CGSize size, CGFloat scale, RCTResizeMode resizeMode)
+static RCTPlatformImage *RCTResizeImageIfNeeded(RCTPlatformImage *image, CGSize size, CGFloat scale, RCTResizeMode resizeMode) // [macOS]
 {
   if (CGSizeEqualToSize(size, CGSizeZero) || CGSizeEqualToSize(image.size, CGSizeZero) ||
       CGSizeEqualToSize(image.size, size)) {
@@ -331,16 +371,16 @@
     RCTImageLoaderPartialLoadBlock partialLoadHandler,
     RCTImageLoaderCompletionBlockWithMetadata completionHandler)
 {
-  if ([loadHandler
-          respondsToSelector:@selector
-          (loadImageForURL:size:scale:resizeMode:progressHandler:partialLoadHandler:completionHandlerWithMetadata:)]) {
+  if ([loadHandler respondsToSelector:@selector(loadImageForURL:
+                                                           size:scale:resizeMode:progressHandler:partialLoadHandler
+                                                               :completionHandlerWithMetadata:)]) {
     return [loadHandler loadImageForURL:imageURL
                                    size:size
                                   scale:scale
                              resizeMode:resizeMode
                         progressHandler:progressHandler
                      partialLoadHandler:partialLoadHandler
-          completionHandlerWithMetadata:^(NSError *error, UIImage *image, id metadata) {
+          completionHandlerWithMetadata:^(NSError *error, RCTPlatformImage *image, id metadata) { // [macOS]
             completionHandler(error, image, metadata);
           }];
   } else {
@@ -350,7 +390,7 @@
                              resizeMode:resizeMode
                         progressHandler:progressHandler
                      partialLoadHandler:partialLoadHandler
-                      completionHandler:^(NSError *error, UIImage *image) {
+                      completionHandler:^(NSError *error, RCTPlatformImage *image) { // [macOS]
                         completionHandler(error, image, nil);
                       }];
   }
@@ -410,17 +450,17 @@
                                                     completionBlock:(RCTImageLoaderCompletionBlock)completionBlock
 {
   RCTImageURLLoaderRequest *request = [self loadImageWithURLRequest:imageURLRequest
-                                                               size:size
-                                                              scale:scale
-                                                            clipped:clipped
-                                                         resizeMode:resizeMode
-                                                           priority:priority
-                                                        attribution:{}
-                                                      progressBlock:progressBlock
-                                                   partialLoadBlock:partialLoadBlock
-                                                    completionBlock:^(NSError *error, UIImage *image, id metadata) {
-                                                      completionBlock(error, image);
-                                                    }];
+      size:size
+      scale:scale
+      clipped:clipped
+      resizeMode:resizeMode
+      priority:priority
+      attribution:{}
+      progressBlock:progressBlock
+      partialLoadBlock:partialLoadBlock
+      completionBlock:^(NSError *error, RCTPlatformImage *image, id metadata) { // [macOS]
+        completionBlock(error, image);
+      }];
   return ^{
     [request cancel];
   };
@@ -490,6 +530,12 @@
   });
 }
 
+// [macOS
+- (NSInteger)activeTasks {
+  return _activeTasks;
+}
+// macOS]
+
 /**
  * This returns either an image, or raw image data, depending on the loading
  * path taken. This is useful if you want to skip decoding, e.g. when preloading
@@ -548,7 +594,7 @@
   }
 
   if (cacheResult && partialLoadHandler) {
-    UIImage *image = [[self imageCache] imageForUrl:request.URL.absoluteString
+    RCTPlatformImage *image = [[self imageCache] imageForUrl:request.URL.absoluteString // [macOS]
                                                size:size
                                               scale:scale
                                          resizeMode:resizeMode];
@@ -570,7 +616,7 @@
 
         // If we've received an image, we should try to set it synchronously,
         // if it's data, do decoding on a background thread.
-        if (RCTIsMainQueue() && ![imageOrData isKindOfClass:[UIImage class]]) {
+        if (RCTIsMainQueue() && ![imageOrData isKindOfClass:[RCTPlatformImage class]]) { // [macOS]
           // Most loaders do not return on the main thread, so caller is probably not
           // expecting it, and may do expensive post-processing in the callback
           dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
@@ -601,7 +647,7 @@
                  attribution:attributionCopy
              progressHandler:progressHandler
           partialLoadHandler:partialLoadHandler
-           completionHandler:^(NSError *error, UIImage *image, id metadata) {
+           completionHandler:^(NSError *error, RCTPlatformImage *image, id metadata) { // [macOS]
              completionHandler(error, image, metadata, nil);
            }];
     }
@@ -614,7 +660,7 @@
         resizeMode,
         progressHandler,
         partialLoadHandler,
-        ^(NSError *error, UIImage *image, id metadata) {
+        ^(NSError *error, RCTPlatformImage *image, id metadata) {
           completionHandler(error, image, metadata, nil);
         });
     return [[RCTImageURLLoaderRequest alloc] initWithRequestId:nil imageURL:request.URL cancellationBlock:cb];
@@ -645,7 +691,7 @@
                    attribution:attributionCopy
                progressHandler:progressHandler
             partialLoadHandler:partialLoadHandler
-             completionHandler:^(NSError *error, UIImage *image, id metadata) {
+             completionHandler:^(NSError *error, RCTPlatformImage *image, id metadata) { // [macOS]
                completionHandler(error, image, metadata, nil);
              }];
         cancelLoadLocal = loaderRequest.cancellationBlock;
@@ -658,7 +704,7 @@
             resizeMode,
             progressHandler,
             partialLoadHandler,
-            ^(NSError *error, UIImage *image, id metadata) {
+            ^(NSError *error, RCTPlatformImage *image, id metadata) { // [macOS]
               completionHandler(error, image, metadata, nil);
             });
       }
@@ -666,7 +712,7 @@
       cancelLoad = cancelLoadLocal;
       [cancelLoadLock unlock];
     } else {
-      UIImage *image;
+      RCTPlatformImage *image; // [macOS]
       if (cacheResult) {
         image = [[strongSelf imageCache] imageForUrl:request.URL.absoluteString
                                                 size:size
@@ -856,7 +902,7 @@
           return;
         }
 
-        if (!imageOrData || [imageOrData isKindOfClass:[UIImage class]]) {
+        if (!imageOrData || [imageOrData isKindOfClass:[RCTPlatformImage class]]) { // [macOS]
           [cancelLoadLock lock];
           cancelLoad = nil;
           [cancelLoadLock unlock];
@@ -864,7 +910,7 @@
           return;
         }
 
-        RCTImageLoaderCompletionBlock decodeCompletionHandler = ^(NSError *error_, UIImage *image) {
+        RCTImageLoaderCompletionBlock decodeCompletionHandler = ^(NSError *error_, RCTPlatformImage *image) { // [macOS]
           if (cacheResult && image) {
             // Store decoded image in cache
             [[strongSelf imageCache] addImageToCache:image
@@ -916,7 +962,7 @@
   return NO;
 }
 
-- (void)trackURLImageVisibilityForRequest:(RCTImageURLLoaderRequest *)loaderRequest imageView:(UIView *)imageView
+- (void)trackURLImageVisibilityForRequest:(RCTImageURLLoaderRequest *)loaderRequest imageView:(RCTUIView *)imageView // [macOS]
 {
   if (!loaderRequest || !imageView) {
     return;
@@ -969,7 +1015,7 @@
   }
 
   auto cancelled = std::make_shared<std::atomic<int>>(0);
-  void (^completionHandler)(NSError *, UIImage *) = ^(NSError *error, UIImage *image) {
+  void (^completionHandler)(NSError *, RCTPlatformImage *) = ^(NSError *error, RCTPlatformImage *image) { // [macOS]
     if (RCTIsMainQueue()) {
       // Most loaders do not return on the main thread, so caller is probably not
       // expecting it, and may do expensive post-processing in the callback
@@ -1004,10 +1050,10 @@
       dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
         if (!std::atomic_load(cancelled.get())) {
           // Decompress the image data (this may be CPU and memory intensive)
-          UIImage *image = RCTDecodeImageWithData(data, size, scale, resizeMode);
+          RCTPlatformImage *image = RCTDecodeImageWithData(data, size, scale, resizeMode); // [macOS]
 
-#if RCT_DEV
-          CGSize imagePixelSize = RCTSizeInPixels(image.size, image.scale);
+#if !TARGET_OS_OSX && RCT_DEV // [macOS]
+          CGSize imagePixelSize = RCTSizeInPixels(image.size, UIImageGetScale(image)); // [macOS]
           CGSize screenPixelSize = RCTSizeInPixels(RCTScreenSize(), RCTScreenScale());
           if (imagePixelSize.width * imagePixelSize.height > screenPixelSize.width * screenPixelSize.height) {
             RCTLogInfo(
@@ -1099,10 +1145,16 @@
               break;
           }
         } else {
-          UIImage *image = imageOrData;
+          RCTPlatformImage *image = imageOrData; // [macOS]
+#if !TARGET_OS_OSX // [macOS]
+          CGFloat imageScale = image.scale;
+#else // [macOS
+          // Trust -[NSImage size] on macOS since an image is a collection of representations instead of a thin wrapper around a CGImage
+          CGFloat imageScale = 1.0;
+#endif // macOS]
           size = (CGSize){
-              image.size.width * image.scale,
-              image.size.height * image.scale,
+              image.size.width * imageScale, // [macOS]
+              image.size.height * imageScale, // [macOS]
           };
         }
         callback(error, size);
@@ -1187,7 +1239,7 @@
 {
   __block RCTImageLoaderCancellationBlock requestToken;
   requestToken = [self loadImageWithURLRequest:request
-                                      callback:^(NSError *error, UIImage *image) {
+                                      callback:^(NSError *error, RCTPlatformImage *image) { // [macOS]
                                         if (error) {
                                           [delegate URLRequest:requestToken didCompleteWithError:error];
                                           return;
@@ -1195,7 +1247,7 @@
 
                                         NSString *mimeType = nil;
                                         NSData *imageData = nil;
-                                        if (RCTImageHasAlpha(image.CGImage)) {
+                                        if (RCTUIImageHasAlpha(image)) { // [macOS]
                                           mimeType = @"image/png";
                                           imageData = UIImagePNGRepresentation(image);
                                         } else {
@@ -1229,8 +1281,10 @@
   return std::make_shared<facebook::react::NativeImageLoaderIOSSpecJSI>(params);
 }
 
-RCT_EXPORT_METHOD(
-    getSize : (NSString *)uri resolve : (RCTPromiseResolveBlock)resolve reject : (RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(getSize
+                  : (NSString *)uri resolve
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (RCTPromiseRejectBlock)reject)
 {
   NSURLRequest *request = [RCTConvert NSURLRequest:uri];
   [self getImageSizeForURLRequest:request
@@ -1246,9 +1300,11 @@
                             }];
 }
 
-RCT_EXPORT_METHOD(
-    getSizeWithHeaders : (NSString *)uri headers : (NSDictionary *)headers resolve : (RCTPromiseResolveBlock)
-        resolve reject : (RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(getSizeWithHeaders
+                  : (NSString *)uri headers
+                  : (NSDictionary *)headers resolve
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (RCTPromiseRejectBlock)reject)
 {
   NSURL *URL = [RCTConvert NSURL:uri];
   NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
@@ -1265,15 +1321,20 @@
                             }];
 }
 
-RCT_EXPORT_METHOD(
-    prefetchImage : (NSString *)uri resolve : (RCTPromiseResolveBlock)resolve reject : (RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(prefetchImage
+                  : (NSString *)uri resolve
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (RCTPromiseRejectBlock)reject)
 {
   [self prefetchImageWithMetadata:uri queryRootName:nil rootTag:0 resolve:resolve reject:reject];
 }
 
-RCT_EXPORT_METHOD(
-    prefetchImageWithMetadata : (NSString *)uri queryRootName : (NSString *)queryRootName rootTag : (double)
-        rootTag resolve : (RCTPromiseResolveBlock)resolve reject : (RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(prefetchImageWithMetadata
+                  : (NSString *)uri queryRootName
+                  : (NSString *)queryRootName rootTag
+                  : (double)rootTag resolve
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (RCTPromiseRejectBlock)reject)
 {
   NSURLRequest *request = [RCTConvert NSURLRequest:uri];
   [self loadImageWithURLRequest:request
@@ -1288,7 +1349,7 @@
                                 }
                   progressBlock:nil
                partialLoadBlock:nil
-                completionBlock:^(NSError *error, UIImage *image, id completionMetadata) {
+                completionBlock:^(NSError *error, RCTPlatformImage *image, id completionMetadata) { // [macOS]
                   if (error) {
                     reject(@"E_PREFETCH_FAILURE", nil, error);
                     return;
@@ -1297,8 +1358,10 @@
                 }];
 }
 
-RCT_EXPORT_METHOD(
-    queryCache : (NSArray *)uris resolve : (RCTPromiseResolveBlock)resolve reject : (RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(queryCache
+                  : (NSArray *)uris resolve
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (RCTPromiseRejectBlock)reject)
 {
   resolve([self getImageCacheStatus:uris]);
 }
