--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/RCTImageCache.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Image/RCTImageCache.mm	2026-03-06 15:22:52
@@ -46,6 +46,7 @@
     _decodedImageCache.totalCostLimit = RCTImageCacheTotalCostLimit;
     _cacheStaleTimes = [NSMutableDictionary new];
 
+#if !TARGET_OS_OSX // [macOS]
     [[NSNotificationCenter defaultCenter] addObserver:self
                                              selector:@selector(clearCache)
                                                  name:UIApplicationDidReceiveMemoryWarningNotification
@@ -54,11 +55,13 @@
                                              selector:@selector(clearCache)
                                                  name:UIApplicationWillResignActiveNotification
                                                object:nil];
+#endif // [macOS]
   }
 
   return self;
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)clearCache
 {
   [_decodedImageCache removeAllObjects];
@@ -66,8 +69,9 @@
     [_cacheStaleTimes removeAllObjects];
   }
 }
+#endif // [macOS]
 
-- (void)addImageToCache:(UIImage *)image forKey:(NSString *)cacheKey
+- (void)addImageToCache:(RCTPlatformImage *)image forKey:(NSString *)cacheKey // [macOS]
 {
   if (!image) {
     return;
@@ -78,7 +82,7 @@
   }
 }
 
-- (UIImage *)imageForUrl:(NSString *)url size:(CGSize)size scale:(CGFloat)scale resizeMode:(RCTResizeMode)resizeMode
+- (RCTPlatformImage *)imageForUrl:(NSString *)url size:(CGSize)size scale:(CGFloat)scale resizeMode:(RCTResizeMode)resizeMode // [macOS]
 {
   NSString *cacheKey = RCTCacheKeyForImage(url, size, scale, resizeMode);
   @synchronized(_cacheStaleTimes) {
@@ -95,7 +99,7 @@
   return [_decodedImageCache objectForKey:cacheKey];
 }
 
-- (void)addImageToCache:(UIImage *)image
+- (void)addImageToCache:(RCTPlatformImage *)image // [macOS]
                     URL:(NSString *)url
                    size:(CGSize)size
                   scale:(CGFloat)scale
