--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/RCTImageEditingManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Image/RCTImageEditingManager.mm	2026-03-06 15:22:52
@@ -15,7 +15,7 @@
 #import <React/RCTImageUtils.h>
 #import <React/RCTLog.h>
 #import <React/RCTUtils.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import "RCTImagePlugins.h"
 
@@ -38,9 +38,11 @@
  *        be scaled down to `displaySize` rather than `size`.
  *        All units are in px (not points).
  */
-RCT_EXPORT_METHOD(
-    cropImage : (NSURLRequest *)imageRequest cropData : (JS::NativeImageEditor::Options &)cropData successCallback : (
-        RCTResponseSenderBlock)successCallback errorCallback : (RCTResponseSenderBlock)errorCallback)
+RCT_EXPORT_METHOD(cropImage
+                  : (NSURLRequest *)imageRequest cropData
+                  : (JS::NativeImageEditor::Options &)cropData successCallback
+                  : (RCTResponseSenderBlock)successCallback errorCallback
+                  : (RCTResponseSenderBlock)errorCallback)
 {
   CGRect rect = {
     [RCTConvert CGPoint:@{
@@ -58,7 +60,7 @@
 
   [[_moduleRegistry moduleForName:"ImageLoader"]
       loadImageWithURLRequest:imageRequest
-                     callback:^(NSError *error, UIImage *image) {
+                     callback:^(NSError *error, RCTPlatformImage *image) { // [macOS]
                        if (error) {
                          errorCallback(@[ RCTJSErrorFromNSError(error) ]);
                          return;
@@ -68,7 +70,7 @@
                        CGSize targetSize = rect.size;
                        CGRect targetRect = {{-rect.origin.x, -rect.origin.y}, image.size};
                        CGAffineTransform transform = RCTTransformFromTargetRect(image.size, targetRect);
-                       UIImage *croppedImage = RCTTransformImage(image, targetSize, image.scale, transform);
+                       RCTPlatformImage *croppedImage = RCTTransformImage(image, targetSize, UIImageGetScale(image), transform); // [macOS]
 
                        // Scale image
                        if (cropDataCopy.displaySize()) {
@@ -79,7 +81,7 @@
                          RCTResizeMode resizeMode = [RCTConvert RCTResizeMode:cropDataCopy.resizeMode() ?: @"contain"];
                          targetRect = RCTTargetRect(croppedImage.size, targetSize, 1, resizeMode);
                          transform = RCTTransformFromTargetRect(croppedImage.size, targetRect);
-                         croppedImage = RCTTransformImage(croppedImage, targetSize, image.scale, transform);
+                         croppedImage = RCTTransformImage(croppedImage, targetSize, UIImageGetScale(image), transform); // [macOS]
                        }
 
                        // Store image
