--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/RCTImageStoreManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Image/RCTImageStoreManager.mm	2026-03-06 15:22:52
@@ -12,7 +12,9 @@
 
 #import <FBReactNativeSpec/FBReactNativeSpec.h>
 #import <ImageIO/ImageIO.h>
+#if !TARGET_OS_OSX // [macOS]
 #import <MobileCoreServices/UTType.h>
+#endif // [macOS]
 #import <React/RCTAssert.h>
 #import <React/RCTImageUtils.h>
 #import <React/RCTLog.h>
@@ -48,7 +50,7 @@
 {
   dispatch_async(_methodQueue, ^{
     [self removeImageForTag:imageTag];
-    if (block != nullptr) {
+    if (block) {
       block();
     }
   });
@@ -58,7 +60,7 @@
 {
   RCTAssertThread(_methodQueue, @"Must be called on RCTImageStoreManager thread");
 
-  if (_store == nullptr) {
+  if (!_store) {
     _store = [NSMutableDictionary new];
     _id = 0;
   }
@@ -84,7 +86,7 @@
   });
 }
 
-- (void)storeImage:(UIImage *)image withBlock:(void (^)(NSString *imageTag))block
+- (void)storeImage:(RCTPlatformImage *)image withBlock:(void (^)(NSString *imageTag))block // [macOS]
 {
   RCTAssertParam(block);
   dispatch_async(_methodQueue, ^{
@@ -106,12 +108,13 @@
 }
 
 // TODO (#5906496): Name could be more explicit - something like getBase64EncodedDataForTag:?
-RCT_EXPORT_METHOD(
-    getBase64ForTag : (NSString *)imageTag successCallback : (RCTResponseSenderBlock)
-        successCallback errorCallback : (RCTResponseSenderBlock)errorCallback)
+RCT_EXPORT_METHOD(getBase64ForTag
+                  : (NSString *)imageTag successCallback
+                  : (RCTResponseSenderBlock)successCallback errorCallback
+                  : (RCTResponseSenderBlock)errorCallback)
 {
   NSData *imageData = _store[imageTag];
-  if (imageData == nullptr) {
+  if (!imageData) {
     errorCallback(
         @[ RCTJSErrorFromNSError(RCTErrorWithMessage([NSString stringWithFormat:@"Invalid imageTag: %@", imageTag])) ]);
     return;
@@ -122,15 +125,16 @@
   });
 }
 
-RCT_EXPORT_METHOD(
-    addImageFromBase64 : (NSString *)base64String successCallback : (RCTResponseSenderBlock)
-        successCallback errorCallback : (RCTResponseSenderBlock)errorCallback)
+RCT_EXPORT_METHOD(addImageFromBase64
+                  : (NSString *)base64String successCallback
+                  : (RCTResponseSenderBlock)successCallback errorCallback
+                  : (RCTResponseSenderBlock)errorCallback)
 
 {
   // Dispatching to a background thread to perform base64 decoding
   dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
     NSData *imageData = [[NSData alloc] initWithBase64EncodedString:base64String options:0];
-    if (imageData != nullptr) {
+    if (imageData) {
       dispatch_async(self->_methodQueue, ^{
         successCallback(@[ [self _storeImageData:imageData] ]);
       });
@@ -162,14 +166,14 @@
 
     NSString *imageTag = request.URL.absoluteString;
     NSData *imageData = self->_store[imageTag];
-    if (imageData == nullptr) {
+    if (!imageData) {
       NSError *error = RCTErrorWithMessage([NSString stringWithFormat:@"Invalid imageTag: %@", imageTag]);
       [delegate URLRequest:cancellationBlock didCompleteWithError:error];
       return;
     }
 
     CGImageSourceRef sourceRef = CGImageSourceCreateWithData((__bridge CFDataRef)imageData, NULL);
-    if (sourceRef == nullptr) {
+    if (!sourceRef) {
       NSError *error =
           RCTErrorWithMessage([NSString stringWithFormat:@"Unable to decode data for imageTag: %@", imageTag]);
       [delegate URLRequest:cancellationBlock didCompleteWithError:error];
@@ -195,7 +199,7 @@
 
 - (void)cancelRequest:(id)requestToken
 {
-  if (requestToken != nullptr) {
+  if (requestToken) {
     ((void (^)(void))requestToken)();
   }
 }
@@ -210,7 +214,7 @@
 
 @implementation RCTImageStoreManager (Deprecated)
 
-- (NSString *)storeImage:(UIImage *)image
+- (NSString *)storeImage:(RCTPlatformImage *)image // [macOS]
 {
   RCTAssertMainQueue();
   RCTLogWarn(
@@ -222,7 +226,7 @@
   return imageTag;
 }
 
-- (UIImage *)imageForTag:(NSString *)imageTag
+- (RCTPlatformImage *)imageForTag:(NSString *)imageTag // [macOS]
 {
   RCTAssertMainQueue();
   RCTLogWarn(
@@ -231,17 +235,17 @@
   dispatch_sync(_methodQueue, ^{
     imageData = self->_store[imageTag];
   });
-  return [UIImage imageWithData:imageData];
+  return UIImageWithData(imageData); // [macOS]
 }
 
-- (void)getImageForTag:(NSString *)imageTag withBlock:(void (^)(UIImage *image))block
+- (void)getImageForTag:(NSString *)imageTag withBlock:(void (^)(RCTPlatformImage *image))block // [macOS]
 {
   RCTAssertParam(block);
   dispatch_async(_methodQueue, ^{
     NSData *imageData = self->_store[imageTag];
     dispatch_async(dispatch_get_main_queue(), ^{
       // imageWithData: is not thread-safe, so we can't do this on methodQueue
-      block([UIImage imageWithData:imageData]);
+      block(UIImageWithData(imageData)); // [macOS]
     });
   });
 }
