--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/RCTImageURLLoader.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Image/RCTImageURLLoader.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTBridge.h>
 #import <React/RCTResizeMode.h>
@@ -13,13 +13,13 @@
 NS_ASSUME_NONNULL_BEGIN
 
 typedef void (^RCTImageLoaderProgressBlock)(int64_t progress, int64_t total);
-typedef void (^RCTImageLoaderPartialLoadBlock)(UIImage *image);
-typedef void (^RCTImageLoaderCompletionBlock)(NSError *_Nullable error, UIImage *_Nullable image);
+typedef void (^RCTImageLoaderPartialLoadBlock)(RCTPlatformImage *image); // [macOS]
+typedef void (^RCTImageLoaderCompletionBlock)(NSError *_Nullable error, RCTPlatformImage *_Nullable image); // [macOS]
 // Metadata is passed as a id in an additional parameter because there are forks of RN without this parameter,
 // and the complexity of RCTImageLoader would make using protocols here difficult to typecheck.
 typedef void (^RCTImageLoaderCompletionBlockWithMetadata)(
     NSError *_Nullable error,
-    UIImage *_Nullable image,
+    RCTPlatformImage *_Nullable image, // [macOS]
     id _Nullable metadata);
 typedef dispatch_block_t RCTImageLoaderCancellationBlock;
 
