--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/LinkingIOS/RCTLinkingManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/LinkingIOS/RCTLinkingManager.mm	2026-03-06 15:22:52
@@ -87,8 +87,10 @@
   [self sendEventWithName:@"url" body:notification.userInfo];
 }
 
-RCT_EXPORT_METHOD(
-    openURL : (NSURL *)URL resolve : (RCTPromiseResolveBlock)resolve reject : (RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(openURL
+                  : (NSURL *)URL resolve
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (RCTPromiseRejectBlock)reject)
 {
   [RCTSharedApplication() openURL:URL
       options:@{}
@@ -112,8 +114,10 @@
       }];
 }
 
-RCT_EXPORT_METHOD(
-    canOpenURL : (NSURL *)URL resolve : (RCTPromiseResolveBlock)resolve reject : (__unused RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(canOpenURL
+                  : (NSURL *)URL resolve
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (__unused RCTPromiseRejectBlock)reject)
 {
   if (RCTRunningInAppExtension()) {
     // Technically Today widgets can open urls, but supporting that would require
@@ -151,7 +155,9 @@
 RCT_EXPORT_METHOD(getInitialURL : (RCTPromiseResolveBlock)resolve reject : (__unused RCTPromiseRejectBlock)reject)
 {
   NSURL *initialURL = nil;
-  if (self.bridge.launchOptions[UIApplicationLaunchOptionsURLKey] != nullptr) {
+#pragma clang diagnostic push // [macOS]
+#pragma clang diagnostic ignored "-Wdeprecated-declarations" // [macOS]
+  if (self.bridge.launchOptions[UIApplicationLaunchOptionsURLKey]) {
     initialURL = self.bridge.launchOptions[UIApplicationLaunchOptionsURLKey];
   } else {
     NSDictionary *userActivityDictionary =
@@ -160,6 +166,7 @@
       initialURL = ((NSUserActivity *)userActivityDictionary[@"UIApplicationLaunchOptionsUserActivityKey"]).webpageURL;
     }
   }
+#pragma clang diagnostic pop // [macOS]
   resolve(RCTNullIfNil(initialURL.absoluteString));
 }
 
@@ -177,9 +184,11 @@
       }];
 }
 
-RCT_EXPORT_METHOD(
-    sendIntent : (NSString *)action extras : (NSArray *_Nullable)extras resolve : (RCTPromiseResolveBlock)
-        resolve reject : (RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(sendIntent
+                  : (NSString *)action extras
+                  : (NSArray *_Nullable)extras resolve
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (RCTPromiseRejectBlock)reject)
 {
   RCTLogError(@"Not implemented: %@", NSStringFromSelector(_cmd));
 }
