--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Network/__tests__/XMLHttpRequest-test.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Network/__tests__/XMLHttpRequest-test.js	2026-03-06 15:22:52
@@ -21,7 +21,8 @@
 jest.mock('../../Utilities/GlobalPerformanceLogger');
 let requestId = 1;
 function setRequestId(id: number) {
-  if (Platform.OS === 'ios') {
+  if (Platform.OS === 'ios' || Platform.OS === 'macos') {
+    // [macOS]
     return;
   }
   requestId = id;
@@ -169,7 +170,7 @@
     xhr.open('GET', 'blabla');
     xhr.send();
     setRequestId(4);
-    // $FlowFixMe[incompatible-type]
+    // $FlowFixMe[incompatible-call]
     xhr.__didCompleteResponse(requestId, 'Generic error');
 
     expect(xhr.readyState).toBe(xhr.DONE);
@@ -191,7 +192,7 @@
     xhr.open('GET', 'blabla');
     xhr.send();
     setRequestId(5);
-    // $FlowFixMe[incompatible-type]
+    // $FlowFixMe[incompatible-call]
     xhr.__didCompleteResponse(requestId, null);
 
     expect(xhr.readyState).toBe(xhr.DONE);
