--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTFabricModalHostViewController.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTFabricModalHostViewController.mm	2026-03-06 15:22:52
@@ -15,9 +15,10 @@
   RCTSurfaceTouchHandler *_touchHandler;
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (instancetype)init
 {
-  if ((self = [super init]) == nullptr) {
+  if (!(self = [super init])) {
     return nil;
   }
   _touchHandler = [RCTSurfaceTouchHandler new];
@@ -40,10 +41,12 @@
   [_touchHandler attachToView:self.view];
 }
 
+#if TARGET_OS_IOS // [visionOS]
 - (UIStatusBarStyle)preferredStatusBarStyle
 {
   return [RCTUIStatusBarManager() statusBarStyle];
 }
+#endif // [visionOS]
 
 - (void)viewDidDisappear:(BOOL)animated
 {
@@ -51,17 +54,19 @@
   _lastViewBounds = CGRectZero;
 }
 
+#if TARGET_OS_IOS // [visionOS]
 - (BOOL)prefersStatusBarHidden
 {
   return [RCTUIStatusBarManager() isStatusBarHidden];
 }
+#endif // [visionOS]
 
 #if RCT_DEV && TARGET_OS_IOS
 - (UIInterfaceOrientationMask)supportedInterfaceOrientations
 {
   UIInterfaceOrientationMask appSupportedOrientationsMask =
       [RCTSharedApplication() supportedInterfaceOrientationsForWindow:RCTKeyWindow()];
-  if ((_supportedInterfaceOrientations & appSupportedOrientationsMask) == 0u) {
+  if (!(_supportedInterfaceOrientations & appSupportedOrientationsMask)) {
     RCTLogError(
         @"Modal was presented with 0x%x orientations mask but the application only supports 0x%x."
         @"Add more interface orientations to your app's Info.plist to fix this."
@@ -74,5 +79,6 @@
   return _supportedInterfaceOrientations;
 }
 #endif // RCT_DEV
+#endif // [macOS]
 
 @end
