--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm	2026-03-06 15:22:52
@@ -21,6 +21,7 @@
 
 using namespace facebook::react;
 
+#if !TARGET_OS_OSX // [macOS]
 static UIInterfaceOrientationMask supportedOrientationsMask(ModalHostViewSupportedOrientationsMask mask)
 {
   UIInterfaceOrientationMask supportedOrientations = 0;
@@ -93,11 +94,10 @@
       : ModalHostViewEventEmitter::OnOrientationChangeOrientation::Landscape;
   return {orientation};
 }
+#endif // [macOS]
 
 @interface RCTModalHostViewComponentView () <RCTFabricModalHostViewControllerDelegate>
 
-@property (nonatomic, weak) UIView *accessibilityFocusedView;
-
 @end
 
 @implementation RCTModalHostViewComponentView {
@@ -106,9 +106,9 @@
   BOOL _shouldAnimatePresentation;
   BOOL _shouldPresent;
   BOOL _isPresented;
-  BOOL _modalInPresentation;
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (instancetype)initWithFrame:(CGRect)frame
 {
   if (self = [super initWithFrame:frame]) {
@@ -116,7 +116,6 @@
     _shouldAnimatePresentation = YES;
 
     _isPresented = NO;
-    _modalInPresentation = YES;
   }
 
   return self;
@@ -128,7 +127,7 @@
     _viewController = [RCTFabricModalHostViewController new];
     _viewController.modalTransitionStyle = UIModalTransitionStyleCoverVertical;
     _viewController.delegate = self;
-    _viewController.modalInPresentation = _modalInPresentation;
+    _viewController.modalInPresentation = YES;
   }
   return _viewController;
 }
@@ -152,9 +151,7 @@
 {
   BOOL shouldBePresented = !_isPresented && _shouldPresent && self.window;
   if (shouldBePresented) {
-    [self saveAccessibilityFocusedView];
     self.viewController.presentationController.delegate = self;
-    self.viewController.modalInPresentation = _modalInPresentation;
 
     _isPresented = YES;
     [self presentViewController:self.viewController
@@ -172,7 +169,7 @@
     _isPresented = NO;
     // To animate dismissal of view controller, snapshot of
     // view hierarchy needs to be added to the UIViewController.
-    UIView *snapshot = [self.viewController.view snapshotViewAfterScreenUpdates:NO];
+    RCTPlatformView *snapshot = [self.viewController.view snapshotViewAfterScreenUpdates:NO]; // [macOS]
     if (_shouldPresent) {
       [self.viewController.view addSubview:snapshot];
     }
@@ -185,8 +182,6 @@
                        if (eventEmitter) {
                          eventEmitter->onDismiss(ModalHostViewEventEmitter::OnDismiss{});
                        }
-
-                       [self restoreAccessibilityFocusedView];
                      }];
   }
 }
@@ -213,25 +208,8 @@
 {
   [super didMoveToSuperview];
   [self ensurePresentedOnlyIfNeeded];
-}
-
-- (void)saveAccessibilityFocusedView
-{
-  id focusedElement = UIAccessibilityFocusedElement(nil);
-  if (focusedElement && [focusedElement isKindOfClass:[UIView class]]) {
-    self.accessibilityFocusedView = (UIView *)focusedElement;
-  }
 }
 
-- (void)restoreAccessibilityFocusedView
-{
-  id viewToFocus = self.accessibilityFocusedView;
-  if (viewToFocus) {
-    UIAccessibilityPostNotification(UIAccessibilityScreenChangedNotification, viewToFocus);
-    self.accessibilityFocusedView = nil;
-  }
-}
-
 #pragma mark - RCTFabricModalHostViewControllerDelegate
 
 - (void)boundsDidChange:(CGRect)newBounds
@@ -279,8 +257,7 @@
   self.viewController.modalPresentationStyle = presentationConfiguration(newProps);
 
   if (oldViewProps.allowSwipeDismissal != newProps.allowSwipeDismissal) {
-    _modalInPresentation = !newProps.allowSwipeDismissal;
-    self.viewController.modalInPresentation = _modalInPresentation;
+    self.viewController.modalInPresentation = !newProps.allowSwipeDismissal;
   }
 
   _shouldPresent = newProps.visible;
@@ -295,12 +272,12 @@
   _state = std::static_pointer_cast<const ModalHostViewShadowNode::ConcreteState>(state);
 }
 
-- (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
+- (void)mountChildComponentView:(RCTUIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index // [macOS]
 {
   [self.viewController.view insertSubview:childComponentView atIndex:index];
 }
 
-- (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
+- (void)unmountChildComponentView:(RCTUIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index // [macOS]
 {
   [childComponentView removeFromSuperview];
 }
@@ -324,6 +301,7 @@
     eventEmitter->onRequestClose({});
   }
 }
+#endif // [macOS]
 
 @end
 
