--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/Pressable/Pressable.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Components/Pressable/Pressable.js	2026-03-06 15:22:52
@@ -14,6 +14,16 @@
   LayoutChangeEvent,
   MouseEvent,
 } from '../../Types/CoreEventTypes';
+// [macOS
+import type {
+  BlurEvent,
+  DragEvent,
+  FocusEvent,
+  HandledKeyEvent,
+  KeyEvent,
+} from '../../Types/CoreEventTypes';
+// macOS]
+import type {DraggedTypesType} from '../View/DraggedType'; // [macOS]
 import type {ViewProps} from '../View/ViewPropTypes';
 
 import {PressabilityDebugView} from '../../Pressability/PressabilityDebug';
@@ -80,47 +90,149 @@
   /**
    * Called when this view's layout changes.
    */
-  onLayout?: ?(event: LayoutChangeEvent) => unknown,
+  onLayout?: ?(event: LayoutChangeEvent) => mixed,
 
   /**
    * Called when the hover is activated to provide visual feedback.
    */
-  onHoverIn?: ?(event: MouseEvent) => unknown,
+  onHoverIn?: ?(event: MouseEvent) => mixed,
 
   /**
    * Called when the hover is deactivated to undo visual feedback.
    */
-  onHoverOut?: ?(event: MouseEvent) => unknown,
+  onHoverOut?: ?(event: MouseEvent) => mixed,
 
   /**
    * Called when a long-tap gesture is detected.
    */
-  onLongPress?: ?(event: GestureResponderEvent) => unknown,
+  onLongPress?: ?(event: GestureResponderEvent) => mixed,
 
   /**
    * Called when a single tap gesture is detected.
    */
-  onPress?: ?(event: GestureResponderEvent) => unknown,
+  onPress?: ?(event: GestureResponderEvent) => mixed,
 
   /**
    * Called when a touch is engaged before `onPress`.
    */
-  onPressIn?: ?(event: GestureResponderEvent) => unknown,
+  onPressIn?: ?(event: GestureResponderEvent) => mixed,
   /**
    * Called when the press location moves.
    */
-  onPressMove?: ?(event: GestureResponderEvent) => unknown,
+  onPressMove?: ?(event: GestureResponderEvent) => mixed,
 
   /**
    * Called when a touch is released before `onPress`.
    */
-  onPressOut?: ?(event: GestureResponderEvent) => unknown,
+  onPressOut?: ?(event: GestureResponderEvent) => mixed,
 
+  // [macOS
   /**
-   * Whether to prevent any other native components from becoming responder
-   * while this pressable is responder.
+   * Called after the element is focused.
    */
-  blockNativeResponder?: ?boolean,
+  onFocus?: ?(event: FocusEvent) => void,
+
+  /**
+   * Called after the element loses focus.
+   */
+  onBlur?: ?(event: BlurEvent) => void,
+
+  /**
+   * Called after a key down event is detected.
+   */
+  onKeyDown?: ?(event: KeyEvent) => void,
+
+  /**
+   * Called after a key up event is detected.
+   */
+  onKeyUp?: ?(event: KeyEvent) => void,
+
+  /**
+   * Array of keys to receive key down events for. These events have their default native behavior prevented.
+   *
+   * @platform macos
+   */
+  keyDownEvents?: ?Array<HandledKeyEvent>,
+
+  /**
+   * Array of keys to receive key up events for. These events have their default native behavior prevented.
+   *
+   * @platform macos
+   */
+  keyUpEvents?: ?Array<HandledKeyEvent>,
+
+  /**
+   * Specifies whether the view should receive the mouse down event when the
+   * containing window is in the background.
+   *
+   * @platform macos
+   */
+  acceptsFirstMouse?: ?boolean,
+
+  /**
+   * Specifies whether clicking and dragging the view can move the window. This is useful
+   * to disable in Button like components like Pressable where mouse the user should still
+   * be able to click and drag off the view to cancel the click without accidentally moving the window.
+   *
+   * @platform macos
+   */
+  mouseDownCanMoveWindow?: ?boolean,
+
+  /**
+   * Specifies whether system focus ring should be drawn when the view has keyboard focus.
+   *
+   * @platform macos
+   */
+  enableFocusRing?: ?boolean,
+
+  /**
+   * Specifies whether the view ensures it is vibrant on top of other content.
+   * For more information, see the following apple documentation:
+   * https://developer.apple.com/documentation/appkit/nsview/1483793-allowsvibrancy
+   * https://developer.apple.com/documentation/appkit/nsvisualeffectview#1674177
+   *
+   * @platform macos
+   */
+  allowsVibrancy?: ?boolean,
+
+  /**
+   * Specifies the Tooltip for the Pressable.
+   * @platform macos
+   */
+  tooltip?: ?string,
+
+  /**
+   * Fired when a file is dragged into the Pressable via the mouse.
+   *
+   * @platform macos
+   */
+  onDragEnter?: (event: DragEvent) => void,
+
+  /**
+   * Fired when a file is dragged out of the Pressable via the mouse.
+   *
+   * @platform macos
+   */
+  onDragLeave?: (event: DragEvent) => void,
+
+  /**
+   * Fired when a file is dropped on the Pressable via the mouse.
+   *
+   * @platform macos
+   */
+  onDrop?: (event: DragEvent) => void,
+
+  /**
+   * The types of dragged files that the Pressable will accept.
+   *
+   * Possible values for `draggedTypes` are:
+   *
+   * - `'fileUrl'`
+   *
+   * @platform macos
+   */
+  draggedTypes?: ?DraggedTypesType,
+  // macOS]
 
   /**
    * Either view styles or a function that receives a boolean reflecting whether
@@ -189,7 +301,6 @@
     'aria-expanded': ariaExpanded,
     'aria-label': ariaLabel,
     'aria-selected': ariaSelected,
-    blockNativeResponder,
     cancelable,
     children,
     delayHoverIn,
@@ -207,6 +318,14 @@
     onPressIn,
     onPressMove,
     onPressOut,
+    // [macOS
+    onKeyDown,
+    onKeyUp,
+    keyDownEvents,
+    acceptsFirstMouse,
+    mouseDownCanMoveWindow,
+    enableFocusRing,
+    // macOS]
     pressRetentionOffset,
     style,
     testOnly_pressed,
@@ -243,9 +362,10 @@
   };
 
   const accessibilityLiveRegion =
-    ariaLive === 'off' ? 'none' : (ariaLive ?? props.accessibilityLiveRegion);
+    ariaLive === 'off' ? 'none' : ariaLive ?? props.accessibilityLiveRegion;
 
   const accessibilityLabel = ariaLabel ?? props.accessibilityLabel;
+
   const restPropsWithDefaults: React.ElementConfig<typeof View> = {
     ...restProps,
     ...android_rippleConfig?.viewProps,
@@ -255,9 +375,15 @@
     accessibilityLiveRegion,
     accessibilityLabel,
     accessibilityState: _accessibilityState,
-    focusable: focusable !== false,
+    focusable: focusable !== false && !disabled, // macOS]
     accessibilityValue,
     hitSlop,
+    // [macOS
+    acceptsFirstMouse: acceptsFirstMouse !== false && !disabled,
+    enableFocusRing: enableFocusRing !== false && !disabled,
+    keyDownEvents: keyDownEvents ?? [{key: ' '}, {key: 'Enter'}],
+    mouseDownCanMoveWindow: false,
+    // macOS]
   };
 
   const config = useMemo(
@@ -301,12 +427,14 @@
           onPressOut(event);
         }
       },
-      blockNativeResponder,
+      // [macOS
+      onKeyDown,
+      onKeyUp,
+      // macOS]
     }),
     [
       android_disableSound,
       android_rippleConfig,
-      blockNativeResponder,
       cancelable,
       delayHoverIn,
       delayHoverOut,
@@ -322,6 +450,10 @@
       onPressIn,
       onPressMove,
       onPressOut,
+      // [macOS
+      onKeyDown,
+      onKeyUp,
+      // macOS]
       pressRetentionOffset,
       setPressed,
       shouldUpdatePressed,
