--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/VirtualText/RCTVirtualTextViewManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/VirtualText/RCTVirtualTextViewManager.mm	2026-03-06 15:22:52
@@ -9,13 +9,11 @@
 #import <React/RCTVirtualTextView.h>
 #import <React/RCTVirtualTextViewManager.h>
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 @implementation RCTVirtualTextViewManager
 
 RCT_EXPORT_MODULE(RCTVirtualText)
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
   return [RCTVirtualTextView new];
 }
@@ -26,5 +24,3 @@
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
