--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/ScrollView/RCTScrollContentViewManager.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/ScrollView/RCTScrollContentViewManager.m	2026-03-06 15:22:52
@@ -7,8 +7,6 @@
 
 #import "RCTScrollContentViewManager.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import "RCTScrollContentShadowView.h"
 #import "RCTScrollContentView.h"
 
@@ -16,6 +14,8 @@
 
 RCT_EXPORT_MODULE()
 
+RCT_EXPORT_OSX_VIEW_PROPERTY(inverted, BOOL) // [macOS]
+
 - (RCTScrollContentView *)view
 {
   return [RCTScrollContentView new];
@@ -27,5 +27,3 @@
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
