--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/ScrollView/RCTScrollContentView.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/ScrollView/RCTScrollContentView.m	2026-03-06 15:22:52
@@ -7,20 +7,34 @@
 
 #import "RCTScrollContentView.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTAssert.h>
 #import <React/UIView+React.h>
 
+#if TARGET_OS_OSX // [macOS
+#import <React/RCTUIManager.h>
+#import "RCTScrollContentLocalData.h"
+#endif // macOS]
+
 #import "RCTScrollView.h"
 
 @implementation RCTScrollContentView
+#if TARGET_OS_OSX // [macOS
+- (BOOL)isFlipped
+{
+  return !self.inverted;
+}
+#endif // macOS]
 
 - (void)reactSetFrame:(CGRect)frame
 {
   [super reactSetFrame:frame];
 
+#if !TARGET_OS_OSX // [macOS]
   RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview;
+#else // [macOS
+  // macOS also has a NSClipView in its hierarchy
+  RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview.superview;
+#endif // macOS]
 
   if (!scrollView) {
     return;
@@ -29,8 +43,35 @@
   RCTAssert([scrollView isKindOfClass:[RCTScrollView class]], @"Unexpected view hierarchy of RCTScrollView component.");
 
   [scrollView updateContentSizeIfNeeded];
+#if TARGET_OS_OSX // [macOS
+  // On macOS scroll indicators may float over the content view like they do in iOS
+  // or depending on system preferences they may be outside of the content view
+  // which means the clip view will be smaller than the scroll view itself.
+  // In such cases the content view layout must shrink accordingly otherwise
+  // the contents will overflow causing the scroll indicators to appear unnecessarily.
+  NSScrollView *platformScrollView = [scrollView scrollView];
+  if ([platformScrollView scrollerStyle] == NSScrollerStyleLegacy) {
+    BOOL contentHasHeight = platformScrollView.contentSize.height > 0;
+    CGFloat horizontalScrollerHeight = ([platformScrollView hasHorizontalScroller] && contentHasHeight) ? NSHeight([[platformScrollView horizontalScroller] frame]) : 0;
+    CGFloat verticalScrollerWidth = [platformScrollView hasVerticalScroller] ? NSWidth([[platformScrollView verticalScroller] frame]) : 0;
+
+    RCTScrollContentLocalData *localData = [[RCTScrollContentLocalData alloc] initWithVerticalScrollerWidth:verticalScrollerWidth horizontalScrollerHeight:horizontalScrollerHeight];
+
+    [[[scrollView bridge] uiManager] setLocalData:localData forView:self];
+  }
+
+  if ([platformScrollView accessibilityRole] == NSAccessibilityTableRole) {
+      NSMutableArray *subViews = [[NSMutableArray alloc] initWithCapacity:[[self subviews] count]];
+      for (NSView *view in [self subviews]) {
+          if ([view isKindOfClass:[RCTView class]]) {
+            [subViews addObject:view];
+          }
+      }
+
+      [platformScrollView setAccessibilityRows:subViews];
+  }
+
+#endif // macOS]
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
