--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/ScrollView/RCTScrollView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/ScrollView/RCTScrollView.h	2026-03-06 15:22:52
@@ -5,10 +5,8 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIScrollView.h>
+#import <React/RCTUIKit.h> // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTAutoInsetsProtocol.h>
 #import <React/RCTDefines.h>
 #import <React/RCTEventDispatcherProtocol.h>
@@ -17,11 +15,17 @@
 
 @protocol UIScrollViewDelegate;
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTScrollView : RCTView<UIScrollViewDelegate, RCTScrollableProtocol, RCTAutoInsetsProtocol>
+@interface RCTScrollView : RCTView <
+#if TARGET_OS_IPHONE // [macOS
+	UIScrollViewDelegate,
+#endif
+	RCTScrollableProtocol, RCTAutoInsetsProtocol
+> // macOS]
 
 - (instancetype)initWithEventDispatcher:(id<RCTEventDispatcherProtocol>)eventDispatcher NS_DESIGNATED_INITIALIZER;
 
+@property (nonatomic, readonly) RCTBridge *bridge;
+
 /**
  * The `RCTScrollView` may have at most one single subview. This will ensure
  * that the scroll view's `contentSize` will be efficiently set to the size of
@@ -29,12 +33,12 @@
  * efficiently since it will have already been computed by the off-main-thread
  * layout system.
  */
-@property (nonatomic, readonly) UIView *contentView;
+@property (nonatomic, readonly) RCTUIView *contentView; // [macOS]
 
 /**
  * The underlying scrollView (TODO: can we remove this?)
  */
-@property (nonatomic, readonly) UIScrollView *scrollView;
+@property (nonatomic, readonly) RCTUIScrollView *scrollView; // [macOS]
 
 @property (nonatomic, assign) UIEdgeInsets contentInset;
 @property (nonatomic, assign) BOOL automaticallyAdjustContentInsets;
@@ -54,8 +58,10 @@
 /** Focus area of newly-activated text input relative to the window to compare against UIKeyboardFrameBegin/End */
 @property (nonatomic, assign) CGRect firstResponderFocus;
 /** newly-activated text input outside of the scroll view */
-@property (nonatomic, weak) UIView *firstResponderViewOutsideScrollView;
+@property (nonatomic, weak) RCTPlatformView *firstResponderViewOutsideScrollView; // [macOS]
 
+@property (nonatomic, assign) BOOL hasOverlayStyleIndicator; // [macOS]
+
 // NOTE: currently these event props are only declared so we can export the
 // event names to JS - we don't call the blocks directly because scroll events
 // need to be coalesced before sending, for performance reasons.
@@ -65,14 +71,21 @@
 @property (nonatomic, copy) RCTDirectEventBlock onScrollEndDrag;
 @property (nonatomic, copy) RCTDirectEventBlock onMomentumScrollBegin;
 @property (nonatomic, copy) RCTDirectEventBlock onMomentumScrollEnd;
+@property (nonatomic, copy) RCTDirectEventBlock onPreferredScrollerStyleDidChange; // [macOS]
 
+@property (nonatomic, copy) RCTDirectEventBlock onInvertedDidChange; // [macOS]
+
+- (void)flashScrollIndicators; // [macOS]
+
 @end
 
-@interface UIView (RCTScrollView)
+#if !TARGET_OS_OSX // [macOS]
+@interface UIView (RCTScrollView) // [macOS]
 
 - (void)reactUpdateResponderOffsetForScrollView:(RCTScrollView *)scrollView;
 
 @end
+#endif // [macOS]
 
 @interface RCTScrollView (Internal)
 
@@ -80,6 +93,4 @@
 
 @end
 
-RCT_EXTERN void RCTSendFakeScrollEvent(id<RCTEventDispatcherProtocol> eventDispatcher, NSNumber *reactTag);
-
-#endif // RCT_REMOVE_LEGACY_ARCH
+RCT_EXTERN void RCTSendFakeScrollEvent(id<RCTEventDispatcherProtocol> eventDispatcher, NSNumber *reactTag);
\ No newline at end of file
