--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/ScrollView/RCTScrollViewManager.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/ScrollView/RCTScrollViewManager.h	2026-03-06 15:22:52
@@ -8,18 +8,16 @@
 #import <React/RCTConvert.h>
 #import <React/RCTViewManager.h>
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
+#if !TARGET_OS_OSX // [macOS]
+@interface RCTConvert (UIScrollView)
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTConvert(UIScrollView)
-
+#if TARGET_OS_IOS // [visionOS]
 + (UIScrollViewKeyboardDismissMode)UIScrollViewKeyboardDismissMode:(id)json;
+#endif // [visionOS]
 
 @end
+#endif // [macOS]
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
 @interface RCTScrollViewManager : RCTViewManager
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
