--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/ScrollView/RCTScrollContentShadowView.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/ScrollView/RCTScrollContentShadowView.m	2026-03-06 15:22:52
@@ -7,14 +7,30 @@
 
 #import "RCTScrollContentShadowView.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <yoga/Yoga.h>
 
+#if TARGET_OS_OSX // [macOS
+#import "RCTScrollContentLocalData.h"
+#endif // macOS]
+
 #import "RCTUtils.h"
 
 @implementation RCTScrollContentShadowView
 
+#if TARGET_OS_OSX // [macOS
+- (void)setLocalData:(RCTScrollContentLocalData *)localData
+{
+  RCTAssert(
+      [localData isKindOfClass:[RCTScrollContentLocalData class]],
+      @"Local data object for `RCTScrollContentView` must be `RCTScrollContentLocalData` instance.");
+
+  super.marginEnd = (YGValue){localData.verticalScrollerWidth, YGUnitPoint};
+  super.marginBottom = (YGValue){localData.horizontalScrollerHeight, YGUnitPoint};
+
+  [self didSetProps:@[@"marginEnd", @"marginBottom"]];
+}
+#endif // macOS]
+
 - (void)layoutWithMetrics:(RCTLayoutMetrics)layoutMetrics layoutContext:(RCTLayoutContext)layoutContext
 {
   if (layoutMetrics.layoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
@@ -31,5 +47,3 @@
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
