--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/ScrollView/RCTScrollViewManager.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/ScrollView/RCTScrollViewManager.m	2026-03-06 15:22:52
@@ -7,15 +7,15 @@
 
 #import "RCTScrollViewManager.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import "RCTBridge.h"
 #import "RCTScrollView.h"
 #import "RCTShadowView.h"
 #import "RCTUIManager.h"
 
+#if !TARGET_OS_OSX // [macOS]
 @implementation RCTConvert (UIScrollView)
 
+#if TARGET_OS_IOS // [visionOS]
 RCT_ENUM_CONVERTER(
     UIScrollViewKeyboardDismissMode,
     (@{
@@ -27,6 +27,7 @@
     }),
     UIScrollViewKeyboardDismissModeNone,
     integerValue)
+#endif // [visionOS]
 
 RCT_ENUM_CONVERTER(
     UIScrollViewIndicatorStyle,
@@ -50,36 +51,41 @@
     integerValue)
 
 @end
+#endif // [macOS]
 
 @implementation RCTScrollViewManager
 
 RCT_EXPORT_MODULE()
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
   return [[RCTScrollView alloc] initWithEventDispatcher:self.bridge.eventDispatcher];
 }
 
 RCT_EXPORT_VIEW_PROPERTY(alwaysBounceHorizontal, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(alwaysBounceVertical, BOOL)
-RCT_EXPORT_VIEW_PROPERTY(bounces, BOOL)
-RCT_EXPORT_VIEW_PROPERTY(bouncesZoom, BOOL)
-RCT_EXPORT_VIEW_PROPERTY(canCancelContentTouches, BOOL)
+RCT_EXPORT_NOT_OSX_VIEW_PROPERTY(bounces, BOOL) // [macOS]
+RCT_EXPORT_NOT_OSX_VIEW_PROPERTY(bouncesZoom, BOOL) // [macOS]
+RCT_EXPORT_NOT_OSX_VIEW_PROPERTY(canCancelContentTouches, BOOL) // [macOS]
 RCT_EXPORT_VIEW_PROPERTY(centerContent, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(maintainVisibleContentPosition, NSDictionary)
 RCT_EXPORT_VIEW_PROPERTY(automaticallyAdjustContentInsets, BOOL)
+#if !TARGET_OS_OSX // [macOS]
 RCT_EXPORT_VIEW_PROPERTY(automaticallyAdjustKeyboardInsets, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(decelerationRate, CGFloat)
 RCT_EXPORT_VIEW_PROPERTY(endDraggingSensitivityMultiplier, CGFloat)
 RCT_EXPORT_VIEW_PROPERTY(directionalLockEnabled, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(indicatorStyle, UIScrollViewIndicatorStyle)
+#if TARGET_OS_IOS // [visionOS]
 RCT_EXPORT_VIEW_PROPERTY(keyboardDismissMode, UIScrollViewKeyboardDismissMode)
+#endif // [visionOS]
+#endif // [macOS]
 RCT_EXPORT_VIEW_PROPERTY(maximumZoomScale, CGFloat)
 RCT_EXPORT_VIEW_PROPERTY(minimumZoomScale, CGFloat)
 RCT_EXPORT_VIEW_PROPERTY(scrollEnabled, BOOL)
-RCT_EXPORT_VIEW_PROPERTY(pagingEnabled, BOOL)
-RCT_REMAP_VIEW_PROPERTY(pinchGestureEnabled, scrollView.pinchGestureEnabled, BOOL)
-RCT_EXPORT_VIEW_PROPERTY(scrollsToTop, BOOL)
+RCT_EXPORT_NOT_OSX_VIEW_PROPERTY(pagingEnabled, BOOL) // [macOS]
+RCT_REMAP_NOT_OSX_VIEW_PROPERTY(pinchGestureEnabled, scrollView.pinchGestureEnabled, BOOL) // [macOS]
+RCT_EXPORT_NOT_OSX_VIEW_PROPERTY(scrollsToTop, BOOL) // [macOS]
 RCT_EXPORT_VIEW_PROPERTY(showsHorizontalScrollIndicator, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(showsVerticalScrollIndicator, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(scrollEventThrottle, NSTimeInterval)
@@ -101,9 +107,13 @@
 RCT_EXPORT_VIEW_PROPERTY(onScrollEndDrag, RCTDirectEventBlock)
 RCT_EXPORT_VIEW_PROPERTY(onMomentumScrollBegin, RCTDirectEventBlock)
 RCT_EXPORT_VIEW_PROPERTY(onMomentumScrollEnd, RCTDirectEventBlock)
+RCT_EXPORT_OSX_VIEW_PROPERTY(onInvertedDidChange, RCTDirectEventBlock) // [macOS]
+RCT_EXPORT_OSX_VIEW_PROPERTY(onPreferredScrollerStyleDidChange, RCTDirectEventBlock) // [macOS]
 RCT_EXPORT_VIEW_PROPERTY(inverted, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(automaticallyAdjustsScrollIndicatorInsets, BOOL)
+#if !TARGET_OS_OSX // [macOS]
 RCT_EXPORT_VIEW_PROPERTY(contentInsetAdjustmentBehavior, UIScrollViewContentInsetAdjustmentBehavior)
+#endif // [macOS]
 
 // overflow is used both in css-layout as well as by react-native. In css-layout
 // we always want to treat overflow as scroll but depending on what the overflow
@@ -131,12 +141,15 @@
       }];
 }
 
-RCT_EXPORT_METHOD(
-    scrollTo : (nonnull NSNumber *)reactTag offsetX : (CGFloat)x offsetY : (CGFloat)y animated : (BOOL)animated)
+RCT_EXPORT_METHOD(scrollTo
+                  : (nonnull NSNumber *)reactTag offsetX
+                  : (CGFloat)x offsetY
+                  : (CGFloat)y animated
+                  : (BOOL)animated)
 {
   [self.bridge.uiManager
-      addUIBlock:^(__unused RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry) {
-        UIView *view = viewRegistry[reactTag];
+      addUIBlock:^(__unused RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTUIView *> *viewRegistry) { // [macOS]
+        RCTUIView *view = viewRegistry[reactTag]; // [macOS]
         if ([view conformsToProtocol:@protocol(RCTScrollableProtocol)]) {
           [(id<RCTScrollableProtocol>)view scrollToOffset:(CGPoint){x, y} animated:animated];
         } else {
@@ -152,8 +165,8 @@
 RCT_EXPORT_METHOD(scrollToEnd : (nonnull NSNumber *)reactTag animated : (BOOL)animated)
 {
   [self.bridge.uiManager
-      addUIBlock:^(__unused RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry) {
-        UIView *view = viewRegistry[reactTag];
+      addUIBlock:^(__unused RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTUIView *> *viewRegistry) { // [macOS]
+        RCTUIView *view = viewRegistry[reactTag]; // [macOS]
         if ([view conformsToProtocol:@protocol(RCTScrollableProtocol)]) {
           [(id<RCTScrollableProtocol>)view scrollToEnd:animated];
         } else {
@@ -169,8 +182,8 @@
 RCT_EXPORT_METHOD(zoomToRect : (nonnull NSNumber *)reactTag withRect : (CGRect)rect animated : (BOOL)animated)
 {
   [self.bridge.uiManager
-      addUIBlock:^(__unused RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry) {
-        UIView *view = viewRegistry[reactTag];
+      addUIBlock:^(__unused RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTUIView *> *viewRegistry) { // [macOS]
+        RCTUIView *view = viewRegistry[reactTag]; // [macOS]
         if ([view conformsToProtocol:@protocol(RCTScrollableProtocol)]) {
           [(id<RCTScrollableProtocol>)view zoomToRect:rect animated:animated];
         } else {
@@ -193,10 +206,8 @@
           return;
         }
 
-        [view.scrollView flashScrollIndicators];
+        [view flashScrollIndicators]; // [macOS]
       }];
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
