--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/AppDelegate/RCTAppSetupUtils.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/AppDelegate/RCTAppSetupUtils.h	2026-03-06 15:22:52
@@ -40,20 +40,17 @@
 
 std::unique_ptr<facebook::react::JSExecutorFactory> RCTAppSetupJsExecutorFactoryForOldArch(
     RCTBridge *bridge,
-    const std::shared_ptr<facebook::react::RuntimeScheduler> &runtimeScheduler)
-    __attribute__((deprecated(
-        "RCTAppSetupJsExecutorFactoryForOldArch(RCTBridge *, RuntimeScheduler) is deprecated and will be removed when we remove the legacy architecture.")));
-;
+    const std::shared_ptr<facebook::react::RuntimeScheduler> &runtimeScheduler);
 
 #endif // __cplusplus
 
 RCT_EXTERN_C_BEGIN
 
-void RCTAppSetupPrepareApp(UIApplication *application, BOOL turboModuleEnabled) __attribute__((deprecated(
-    "RCTAppSetupPrepareApp(UIApplication, BOOL) is deprecated and it's signature will change when we remove the legacy arch")));
-UIView *
-RCTAppSetupDefaultRootView(RCTBridge *bridge, NSString *moduleName, NSDictionary *initialProperties, BOOL fabricEnabled)
-    __attribute__((deprecated(
-        "RCTAppSetupDefaultRootView(RCTBridge *, NSString *, NSDictionary *, BOOL) is deprecated and it's signature will change when we remove the legacy arch")));
+void RCTAppSetupPrepareApp(UIApplication *application, BOOL turboModuleEnabled);
+RCTUIView *RCTAppSetupDefaultRootView( // [macOS]
+    RCTBridge *bridge,
+    NSString *moduleName,
+    NSDictionary *initialProperties,
+    BOOL fabricEnabled);
 
 RCT_EXTERN_C_END
