--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/AppDelegate/RCTAppDelegate.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/AppDelegate/RCTAppDelegate.h	2026-03-06 15:22:52
@@ -5,7 +5,9 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTBridgeDelegate.h>
+#import <React/RCTConvert.h>
+#import <React/RCTUIKit.h> // [macOS]
 #import "RCTDefaultReactNativeFactoryDelegate.h"
 #import "RCTReactNativeFactory.h"
 #import "RCTRootViewFactory.h"
@@ -61,13 +63,16 @@
  */
 __attribute__((deprecated(
     "RCTAppDelegate is deprecated and will be removed in a future version of React Native. Use `RCTReactNativeFactory` instead.")))
-@interface RCTAppDelegate : RCTDefaultReactNativeFactoryDelegate<UIApplicationDelegate>
+#if !TARGET_OS_OSX // [macOS]
+@interface RCTAppDelegate : RCTDefaultReactNativeFactoryDelegate <UIApplicationDelegate>
+#else // [macOS
+@interface RCTAppDelegate : RCTDefaultReactNativeFactoryDelegate <NSApplicationDelegate>
+#endif // macOS]
 
 /// The window object, used to render the UViewControllers
-@property (nonatomic, strong, nonnull) UIWindow *window;
+@property (nonatomic, strong, nonnull) RCTPlatformWindow *window; // [macOS]
 
-@property (nonatomic, nullable) RCTBridge *bridge
-    __attribute__((deprecated("The bridge is deprecated and will be removed when removing the legacy architecture.")));
+@property (nonatomic, nullable) RCTBridge *bridge;
 @property (nonatomic, strong, nullable) NSString *moduleName;
 @property (nonatomic, strong, nullable) NSDictionary *initialProps;
 @property (nonatomic, strong) RCTReactNativeFactory *reactNativeFactory;
@@ -75,9 +80,7 @@
 /// If `automaticallyLoadReactNativeWindow` is set to `true`, the React Native window will be loaded automatically.
 @property (nonatomic, assign) BOOL automaticallyLoadReactNativeWindow;
 
-@property (nonatomic, nullable) RCTSurfacePresenterBridgeAdapter *bridgeAdapter __attribute__((
-    deprecated("The bridge adapter is deprecated and will be removed when removing the legacy architecture.")));
-;
+@property (nonatomic, nullable) RCTSurfacePresenterBridgeAdapter *bridgeAdapter;
 
 - (RCTRootViewFactory *)rootViewFactory;
 
