--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/AppDelegate/RCTAppSetupUtils.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/AppDelegate/RCTAppSetupUtils.mm	2026-03-06 15:22:52
@@ -31,30 +31,35 @@
 
 void RCTAppSetupPrepareApp(UIApplication *application, BOOL turboModuleEnabled)
 {
-  RCTEnableTurboModule(YES);
+  RCTEnableTurboModule(turboModuleEnabled);
 
 #if DEBUG
+#if !TARGET_OS_OSX // [macOS]
   // Disable idle timer in dev builds to avoid putting application in background and complicating
   // Metro reconnection logic. Users only need this when running the application using our CLI tooling.
   application.idleTimerDisabled = YES;
+#endif // macOS]
 #endif
 }
 
-UIView *
+RCTUIView * // [macOS]
 RCTAppSetupDefaultRootView(RCTBridge *bridge, NSString *moduleName, NSDictionary *initialProperties, BOOL fabricEnabled)
 {
-  id<RCTSurfaceProtocol> surface = [[RCTFabricSurface alloc] initWithBridge:bridge
-                                                                 moduleName:moduleName
-                                                          initialProperties:initialProperties];
-  UIView *rootView = [[RCTSurfaceHostingProxyRootView alloc] initWithSurface:surface];
-  [surface start];
-  return rootView;
+  if (fabricEnabled) {
+    id<RCTSurfaceProtocol> surface = [[RCTFabricSurface alloc] initWithBridge:bridge
+                                                                   moduleName:moduleName
+                                                            initialProperties:initialProperties];
+    RCTUIView *rootView = [[RCTSurfaceHostingProxyRootView alloc] initWithSurface:surface]; // [macOS]
+    [surface start];
+    return rootView;
+  }
+  return [[RCTRootView alloc] initWithBridge:bridge moduleName:moduleName initialProperties:initialProperties];
 }
 
 NSArray<NSString *> *RCTAppSetupUnstableModulesRequiringMainQueueSetup(id<RCTDependencyProvider> dependencyProvider)
 {
   // For oss, insert core main queue setup modules here
-  return (dependencyProvider != nullptr) ? dependencyProvider.unstableModulesRequiringMainQueueSetup : @[];
+  return dependencyProvider ? dependencyProvider.unstableModulesRequiringMainQueueSetup : @[];
 }
 
 id<RCTTurboModule> RCTAppSetupDefaultModuleFromClass(Class moduleClass, id<RCTDependencyProvider> dependencyProvider)
@@ -65,11 +70,11 @@
         NSArray<NSString *> *classNames = @[];
 
         if (protocol == @protocol(RCTImageURLLoader)) {
-          classNames = (dependencyProvider != nullptr) ? dependencyProvider.imageURLLoaderClassNames : @[];
+          classNames = dependencyProvider ? dependencyProvider.imageURLLoaderClassNames : @[];
         } else if (protocol == @protocol(RCTImageDataDecoder)) {
-          classNames = (dependencyProvider != nullptr) ? dependencyProvider.imageDataDecoderClassNames : @[];
+          classNames = dependencyProvider ? dependencyProvider.imageDataDecoderClassNames : @[];
         } else if (protocol == @protocol(RCTURLRequestHandler)) {
-          classNames = (dependencyProvider != nullptr) ? dependencyProvider.URLRequestHandlerClassNames : @[];
+          classNames = dependencyProvider ? dependencyProvider.URLRequestHandlerClassNames : @[];
         }
 
         NSMutableArray *modules = [NSMutableArray new];
@@ -120,7 +125,6 @@
     RCTTurboModuleManager *turboModuleManager,
     const std::shared_ptr<facebook::react::RuntimeScheduler> &runtimeScheduler)
 {
-#ifndef RCT_REMOVE_LEGACY_ARCH
   // Necessary to allow NativeModules to lookup TurboModules
   [bridge setRCTTurboModuleRegistry:turboModuleManager];
 
@@ -147,18 +151,12 @@
   return std::make_unique<facebook::react::HermesExecutorFactory>(
       facebook::react::RCTJSIExecutorRuntimeInstaller(runtimeInstallerLambda));
 #endif
-#else
-  // This method should not be invoked in the New Arch. So when Legacy Arch is removed, we can
-  // safly return a nullptr.
-  return nullptr;
-#endif
 }
 
 std::unique_ptr<facebook::react::JSExecutorFactory> RCTAppSetupJsExecutorFactoryForOldArch(
     RCTBridge *bridge,
     const std::shared_ptr<facebook::react::RuntimeScheduler> &runtimeScheduler)
 {
-#ifndef RCT_REMOVE_LEGACY_ARCH
   auto runtimeInstallerLambda = [bridge, runtimeScheduler](facebook::jsi::Runtime &runtime) {
     if (!bridge) {
       return;
@@ -171,9 +169,4 @@
   return std::make_unique<facebook::react::HermesExecutorFactory>(
       facebook::react::RCTJSIExecutorRuntimeInstaller(runtimeInstallerLambda));
 #endif
-#else
-  // This method should not be invoked in the New Arch. So when Legacy Arch is removed, we can
-  // safly return a nullptr.
-  return nullptr;
-#endif
 }
