--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/AppDelegate/RCTDefaultReactNativeFactoryDelegate.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/AppDelegate/RCTDefaultReactNativeFactoryDelegate.mm	2026-03-06 15:22:52
@@ -36,7 +36,7 @@
   return [[RCTBridge alloc] initWithDelegate:delegate launchOptions:launchOptions];
 }
 
-- (void)setRootView:(UIView *)rootView toRootViewController:(UIViewController *)rootViewController
+- (void)setRootView:(RCTPlatformView *)rootView toRootViewController:(RCTPlatformViewController *)rootViewController
 {
   rootViewController.view = rootView;
 }
@@ -53,13 +53,18 @@
   // Override point for customization after application launch.
 }
 
-- (UIView *)createRootViewWithBridge:(RCTBridge *)bridge
+- (RCTPlatformView *)createRootViewWithBridge:(RCTBridge *)bridge
                           moduleName:(NSString *)moduleName
                            initProps:(NSDictionary *)initProps
 {
-  UIView *rootView = RCTAppSetupDefaultRootView(bridge, moduleName, initProps, YES);
+  BOOL enableFabric = self.fabricEnabled;
+  RCTUIView *rootView = RCTAppSetupDefaultRootView(bridge, moduleName, initProps, enableFabric);
 
+#if !TARGET_OS_OSX // [macOS]
   rootView.backgroundColor = [UIColor systemBackgroundColor];
+#else // [macOS
+  rootView.backgroundColor = [NSColor windowBackgroundColor];
+#endif // macOS]
 
   return rootView;
 }
@@ -78,7 +83,7 @@
 
 - (NSDictionary<NSString *, Class<RCTComponentViewProtocol>> *)thirdPartyFabricComponents
 {
-  return (self.dependencyProvider != nullptr) ? self.dependencyProvider.thirdPartyFabricComponents : @{};
+  return self.dependencyProvider ? self.dependencyProvider.thirdPartyFabricComponents : @{};
 }
 
 - (void)hostDidStart:(RCTHost *)host
@@ -87,15 +92,13 @@
 
 - (NSArray<NSString *> *)unstableModulesRequiringMainQueueSetup
 {
-  return (self.dependencyProvider != nullptr)
-      ? RCTAppSetupUnstableModulesRequiringMainQueueSetup(self.dependencyProvider)
-      : @[];
+  return self.dependencyProvider ? RCTAppSetupUnstableModulesRequiringMainQueueSetup(self.dependencyProvider) : @[];
 }
 
 - (nullable id<RCTModuleProvider>)getModuleProvider:(const char *)name
 {
   NSString *providerName = [NSString stringWithCString:name encoding:NSUTF8StringEncoding];
-  return (self.dependencyProvider != nullptr) ? self.dependencyProvider.moduleProviders[providerName] : nullptr;
+  return self.dependencyProvider ? self.dependencyProvider.moduleProviders[providerName] : nullptr;
 }
 
 - (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const std::string &)name
@@ -108,22 +111,22 @@
 
 - (BOOL)newArchEnabled
 {
-  return YES;
+  return RCTIsNewArchEnabled();
 }
 
 - (BOOL)bridgelessEnabled
 {
-  return YES;
+  return self.newArchEnabled;
 }
 
 - (BOOL)fabricEnabled
 {
-  return YES;
+  return self.newArchEnabled;
 }
 
 - (BOOL)turboModuleEnabled
 {
-  return YES;
+  return self.newArchEnabled;
 }
 
 - (Class)getModuleClassFromName:(const char *)name
