--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/AppDelegate/RCTDefaultReactNativeFactoryDelegate.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/AppDelegate/RCTDefaultReactNativeFactoryDelegate.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 #import "RCTReactNativeFactory.h"
 
 NS_ASSUME_NONNULL_BEGIN
@@ -15,7 +15,11 @@
  * Contains default implementation of RCTReactNativeFactoryDelegate methods.
  */
 
+#if !TARGET_OS_OSX // [macOS]
 @interface RCTDefaultReactNativeFactoryDelegate : UIResponder <RCTReactNativeFactoryDelegate>
+#else // [macOS
+@interface RCTDefaultReactNativeFactoryDelegate : NSResponder <RCTReactNativeFactoryDelegate>
+#endif // macOS]
 @end
 
 NS_ASSUME_NONNULL_END
