--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.mm	2026-03-06 15:22:52
@@ -7,7 +7,6 @@
 
 #import "RCTRootViewFactory.h"
 #import <React/RCTCxxBridgeDelegate.h>
-#import <React/RCTDevMenu.h>
 #import <React/RCTLog.h>
 #import <React/RCTRootView.h>
 #import <React/RCTSurfacePresenterBridgeAdapter.h>
@@ -21,6 +20,7 @@
 #else
 #import <React/CoreModulesPlugins.h>
 #endif
+#import <React/RCTBundleURLProvider.h>
 #import <React/RCTComponentViewFactory.h>
 #import <React/RCTComponentViewProtocol.h>
 #import <React/RCTFabricSurface.h>
@@ -34,16 +34,26 @@
 #import <react/runtime/JSRuntimeFactory.h>
 #import <react/runtime/JSRuntimeFactoryCAPI.h>
 
+#if RCT_DEV_MENU // [macOS
+#import "RCTDevMenu.h"
+#endif // macOS]
+
 @implementation RCTRootViewFactoryConfiguration
 
 - (instancetype)initWithBundleURL:(NSURL *)bundleURL newArchEnabled:(BOOL)newArchEnabled
 {
-  return [self initWithBundleURL:bundleURL];
+  return [self initWithBundleURL:bundleURL
+                  newArchEnabled:newArchEnabled
+              turboModuleEnabled:newArchEnabled
+               bridgelessEnabled:newArchEnabled];
 }
 
 - (instancetype)initWithBundleURLBlock:(RCTBundleURLBlock)bundleURLBlock newArchEnabled:(BOOL)newArchEnabled
 {
-  return [self initWithBundleURLBlock:bundleURLBlock];
+  return [self initWithBundleURLBlock:bundleURLBlock
+                       newArchEnabled:newArchEnabled
+                   turboModuleEnabled:newArchEnabled
+                    bridgelessEnabled:newArchEnabled];
 }
 
 - (instancetype)initWithBundleURL:(NSURL *)bundleURL
@@ -51,9 +61,13 @@
                turboModuleEnabled:(BOOL)turboModuleEnabled
                 bridgelessEnabled:(BOOL)bridgelessEnabled
 {
-  return [self initWithBundleURLBlock:^{
-    return bundleURL;
-  }];
+  return [self
+      initWithBundleURLBlock:^{
+        return bundleURL;
+      }
+              newArchEnabled:newArchEnabled
+          turboModuleEnabled:turboModuleEnabled
+           bridgelessEnabled:bridgelessEnabled];
 }
 
 - (instancetype)initWithBundleURLBlock:(RCTBundleURLBlock)bundleURLBlock
@@ -63,35 +77,17 @@
 {
   if (self = [super init]) {
     _bundleURLBlock = bundleURLBlock;
-    _fabricEnabled = YES;
-    _turboModuleEnabled = YES;
-    _bridgelessEnabled = YES;
+    _fabricEnabled = newArchEnabled;
+    _turboModuleEnabled = turboModuleEnabled;
+    _bridgelessEnabled = bridgelessEnabled;
   }
   return self;
 }
 
-- (instancetype)initWithBundleURLBlock:(RCTBundleURLBlock)bundleURLBlock
-{
-  if (self = [super init]) {
-    _bundleURLBlock = bundleURLBlock;
-    _fabricEnabled = YES;
-    _turboModuleEnabled = YES;
-    _bridgelessEnabled = YES;
-  }
-  return self;
-}
-
-- (instancetype)initWithBundleURL:(NSURL *)bundleURL
-{
-  return [self initWithBundleURLBlock:^{
-    return bundleURL;
-  }];
-}
-
 @end
 
 @interface RCTRootViewFactory () <RCTCxxBridgeDelegate> {
-  std::shared_ptr<const facebook::react::ContextContainer> _contextContainer;
+  facebook::react::ContextContainer::Shared _contextContainer;
   std::shared_ptr<facebook::react::RuntimeScheduler> _runtimeScheduler;
 }
 @end
@@ -131,70 +127,70 @@
   return [self initWithConfiguration:configuration andTurboModuleManagerDelegate:nil];
 }
 
-- (UIView *)viewWithModuleName:(NSString *)moduleName initialProperties:(NSDictionary *)initialProperties
+- (RCTPlatformView *)viewWithModuleName:(NSString *)moduleName initialProperties:(NSDictionary *)initialProperties // [macOS]
 {
-  return [self viewWithModuleName:moduleName
-                initialProperties:initialProperties
-                    launchOptions:nil
-              bundleConfiguration:[RCTBundleConfiguration defaultConfiguration]
-             devMenuConfiguration:[RCTDevMenuConfiguration defaultConfiguration]];
+  return [self viewWithModuleName:moduleName initialProperties:initialProperties launchOptions:nil];
 }
 
-- (UIView *)viewWithModuleName:(NSString *)moduleName
+- (RCTPlatformView *)viewWithModuleName:(NSString *)moduleName // [macOS]
 {
-  return [self viewWithModuleName:moduleName
-                initialProperties:nil
-                    launchOptions:nil
-              bundleConfiguration:[RCTBundleConfiguration defaultConfiguration]
-             devMenuConfiguration:[RCTDevMenuConfiguration defaultConfiguration]];
+  return [self viewWithModuleName:moduleName initialProperties:nil launchOptions:nil];
 }
 
 - (void)initializeReactHostWithLaunchOptions:(NSDictionary *)launchOptions
-                         bundleConfiguration:(RCTBundleConfiguration *)bundleConfiguration
-                        devMenuConfiguration:(RCTDevMenuConfiguration *)devMenuConfiguration
 {
-  // Enable TurboModule interop by default in Bridgeless mode
-  RCTEnableTurboModuleInterop(YES);
-  RCTEnableTurboModuleInteropBridgeProxy(YES);
+  if (_configuration.bridgelessEnabled) {
+    // Enable TurboModule interop by default in Bridgeless mode
+    RCTEnableTurboModuleInterop(YES);
+    RCTEnableTurboModuleInteropBridgeProxy(YES);
 
-  [self createReactHostIfNeeded:launchOptions
-            bundleConfiguration:bundleConfiguration
-           devMenuConfiguration:devMenuConfiguration];
-  return;
-}
+    [self createReactHostIfNeeded:launchOptions];
+    return;
+  }
 
-- (UIView *)viewWithModuleName:(NSString *)moduleName
-             initialProperties:(NSDictionary *)initialProperties
-                 launchOptions:(NSDictionary *)launchOptions
-{
-  return [self viewWithModuleName:moduleName
-                initialProperties:initialProperties
-                    launchOptions:launchOptions
-              bundleConfiguration:[RCTBundleConfiguration defaultConfiguration]
-             devMenuConfiguration:[RCTDevMenuConfiguration defaultConfiguration]];
+  [self createBridgeIfNeeded:launchOptions];
+  [self createBridgeAdapterIfNeeded];
 }
 
-- (UIView *)viewWithModuleName:(NSString *)moduleName
-             initialProperties:(NSDictionary *)initProps
-                 launchOptions:(NSDictionary *)launchOptions
-           bundleConfiguration:(RCTBundleConfiguration *)bundleConfiguration
-          devMenuConfiguration:(RCTDevMenuConfiguration *)devMenuConfiguration
+- (RCTPlatformView *)viewWithModuleName:(NSString *)moduleName // [macOS]
+                      initialProperties:(NSDictionary *)initProps
+                          launchOptions:(NSDictionary *)launchOptions
 {
-  [self initializeReactHostWithLaunchOptions:launchOptions
-                         bundleConfiguration:bundleConfiguration
-                        devMenuConfiguration:devMenuConfiguration];
+  [self initializeReactHostWithLaunchOptions:launchOptions];
 
-  RCTFabricSurface *surface = [self.reactHost createSurfaceWithModuleName:moduleName
-                                                        initialProperties:initProps ? initProps : @{}];
+  if (_configuration.bridgelessEnabled) {
+    RCTFabricSurface *surface = [self.reactHost createSurfaceWithModuleName:moduleName initialProperties:initProps];
 
-  RCTSurfaceHostingProxyRootView *surfaceHostingProxyRootView =
-      [[RCTSurfaceHostingProxyRootView alloc] initWithSurface:surface];
+    RCTSurfaceHostingProxyRootView *surfaceHostingProxyRootView =
+        [[RCTSurfaceHostingProxyRootView alloc] initWithSurface:surface];
 
-  surfaceHostingProxyRootView.backgroundColor = [UIColor systemBackgroundColor];
+#if !TARGET_OS_OSX // [macOS]
+    surfaceHostingProxyRootView.backgroundColor = [UIColor systemBackgroundColor];
+#endif // [macOS]
+
+#if RCT_DEV_MENU // [macOS
+    RCTDevMenu *devMenu = [self.reactHost.moduleRegistry moduleForClass:[RCTDevMenu class]];
+    if (devMenu) {
+      surfaceHostingProxyRootView.devMenu = devMenu;
+    }
+#endif // macOS]
+
+    if (_configuration.customizeRootView != nil) {
+      _configuration.customizeRootView(surfaceHostingProxyRootView);
+    }
+    return surfaceHostingProxyRootView;
+  }
+
+  RCTPlatformView *rootView; // [macOS]
+  if (_configuration.createRootViewWithBridge != nil) {
+    rootView = _configuration.createRootViewWithBridge(self.bridge, moduleName, initProps);
+  } else {
+    rootView = [self createRootViewWithBridge:self.bridge moduleName:moduleName initProps:initProps];
+  }
   if (_configuration.customizeRootView != nil) {
-    _configuration.customizeRootView(surfaceHostingProxyRootView);
+    _configuration.customizeRootView(rootView);
   }
-  return surfaceHostingProxyRootView;
+  return rootView;
 }
 
 - (RCTBridge *)createBridgeWithDelegate:(id<RCTBridgeDelegate>)delegate launchOptions:(NSDictionary *)launchOptions
@@ -202,12 +198,25 @@
   return [[RCTBridge alloc] initWithDelegate:delegate launchOptions:launchOptions];
 }
 
-- (UIView *)createRootViewWithBridge:(RCTBridge *)bridge
-                          moduleName:(NSString *)moduleName
-                           initProps:(NSDictionary *)initProps
+- (RCTPlatformView *)createRootViewWithBridge:(RCTBridge *)bridge
+                                   moduleName:(NSString *)moduleName
+                                    initProps:(NSDictionary *)initProps
 {
-  UIView *rootView = RCTAppSetupDefaultRootView(bridge, moduleName, initProps, YES);
+  BOOL enableFabric = _configuration.fabricEnabled;
+  RCTPlatformView *rootView = RCTAppSetupDefaultRootView(bridge, moduleName, initProps, enableFabric); // [macOS]
+  
+#if RCT_DEV_MENU // [macOS
+  if (enableFabric && [rootView isKindOfClass:[RCTSurfaceHostingView class]]) {
+    RCTDevMenu *devMenu = [bridge moduleForClass:[RCTDevMenu class]];
+    if (devMenu) {
+      [(RCTSurfaceHostingView *)rootView setDevMenu:devMenu];
+    }
+  }
+#endif // macOS]
+  
+#if !TARGET_OS_OSX // [macOS]
   rootView.backgroundColor = [UIColor systemBackgroundColor];
+#endif // [macOS]
   return rootView;
 }
 
@@ -215,15 +224,19 @@
 - (std::unique_ptr<facebook::react::JSExecutorFactory>)jsExecutorFactoryForBridge:(RCTBridge *)bridge
 {
   _runtimeScheduler = std::make_shared<facebook::react::RuntimeScheduler>(RCTRuntimeExecutorFromBridge(bridge));
-
-  std::shared_ptr<facebook::react::CallInvoker> callInvoker =
-      std::make_shared<facebook::react::RuntimeSchedulerCallInvoker>(_runtimeScheduler);
-  RCTTurboModuleManager *turboModuleManager = [[RCTTurboModuleManager alloc] initWithBridge:bridge
-                                                                                   delegate:_turboModuleManagerDelegate
-                                                                                  jsInvoker:callInvoker];
-  _contextContainer->erase(facebook::react::RuntimeSchedulerKey);
-  _contextContainer->insert(facebook::react::RuntimeSchedulerKey, _runtimeScheduler);
-  return RCTAppSetupDefaultJsExecutorFactory(bridge, turboModuleManager, _runtimeScheduler);
+  if (RCTIsNewArchEnabled()) {
+    std::shared_ptr<facebook::react::CallInvoker> callInvoker =
+        std::make_shared<facebook::react::RuntimeSchedulerCallInvoker>(_runtimeScheduler);
+    RCTTurboModuleManager *turboModuleManager =
+        [[RCTTurboModuleManager alloc] initWithBridge:bridge
+                                             delegate:_turboModuleManagerDelegate
+                                            jsInvoker:callInvoker];
+    _contextContainer->erase("RuntimeScheduler");
+    _contextContainer->insert("RuntimeScheduler", _runtimeScheduler);
+    return RCTAppSetupDefaultJsExecutorFactory(bridge, turboModuleManager, _runtimeScheduler);
+  } else {
+    return RCTAppSetupJsExecutorFactoryForOldArch(bridge, _runtimeScheduler);
+  }
 }
 
 - (void)createBridgeIfNeeded:(NSDictionary *)launchOptions
@@ -241,7 +254,7 @@
 
 - (void)createBridgeAdapterIfNeeded
 {
-  if (self.bridgeAdapter != nullptr) {
+  if (!self->_configuration.fabricEnabled || self.bridgeAdapter) {
     return;
   }
 
@@ -253,29 +266,15 @@
 #pragma mark - New Arch Utilities
 
 - (void)createReactHostIfNeeded:(NSDictionary *)launchOptions
-            bundleConfiguration:(RCTBundleConfiguration *)bundleConfiguration
-           devMenuConfiguration:(RCTDevMenuConfiguration *)devMenuConfiguration
 {
   if (self.reactHost) {
     return;
   }
-
-  self.reactHost = [self createReactHost:launchOptions
-                     bundleConfiguration:bundleConfiguration
-                    devMenuConfiguration:devMenuConfiguration];
+  self.reactHost = [self createReactHost:launchOptions];
 }
 
 - (RCTHost *)createReactHost:(NSDictionary *)launchOptions
 {
-  return [self createReactHost:launchOptions
-           bundleConfiguration:[RCTBundleConfiguration defaultConfiguration]
-          devMenuConfiguration:[RCTDevMenuConfiguration defaultConfiguration]];
-}
-
-- (RCTHost *)createReactHost:(NSDictionary *)launchOptions
-         bundleConfiguration:(RCTBundleConfiguration *)bundleConfiguration
-        devMenuConfiguration:(RCTDevMenuConfiguration *)devMenuConfiguration
-{
   __weak __typeof(self) weakSelf = self;
   RCTHost *reactHost =
       [[RCTHost alloc] initWithBundleURLProvider:self->_configuration.bundleURLBlock
@@ -284,9 +283,7 @@
                                 jsEngineProvider:^std::shared_ptr<facebook::react::JSRuntimeFactory>() {
                                   return [weakSelf createJSRuntimeFactory];
                                 }
-                                   launchOptions:launchOptions
-                             bundleConfiguration:bundleConfiguration
-                            devMenuConfiguration:devMenuConfiguration];
+                                   launchOptions:launchOptions];
   [reactHost setBundleURLProvider:^NSURL *() {
     return [weakSelf bundleURL];
   }];
