--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/AppDelegate/RCTUIConfiguratorProtocol.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/AppDelegate/RCTUIConfiguratorProtocol.h	2026-03-06 15:22:52
@@ -6,7 +6,7 @@
  */
 
 #import <React/RCTConvert.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 @class RCTRootView;
 
@@ -54,7 +54,7 @@
  * If you are not using a simple UIViewController, then there could be other methods to use to setup the rootView.
  * For example: UISplitViewController requires `setViewController(_:for:)`
  */
-- (void)setRootView:(UIView *)rootView toRootViewController:(UIViewController *)rootViewController;
+- (void)setRootView:(RCTPlatformView *)rootView toRootViewController:(RCTPlatformViewController *)rootViewController; // [macOS]
 @end
 
 NS_ASSUME_NONNULL_END
