--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Utilities/ReactNativeTestTools.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Utilities/ReactNativeTestTools.js	2026-03-06 15:22:52
@@ -20,7 +20,8 @@
 const TextInput = require('../Components/TextInput/TextInput').default;
 const View = require('../Components/View/View').default;
 const Text = require('../Text/Text').default;
-const {VirtualizedList} = require('@react-native/virtualized-lists').default;
+const {VirtualizedList} =
+  require('@react-native-macos/virtualized-lists').default; // [macOS]
 
 export type ReactTestInstance = ReactTestRendererType['root'];
 export type Predicate = (node: ReactTestInstance) => boolean;
@@ -38,21 +39,13 @@
     node =>
       // note: <Text /> lazy-mounts press handlers after the first press,
       //       so this is a workaround for targeting text nodes.
-      /* $FlowFixMe[invalid-compare] Error discovered during Constant Condition
-       * roll out. See https://fburl.com/workplace/4oq3zi07. */
       (node.type === Text &&
         node.props &&
         typeof node.props.onPress === 'function') ||
       // note: Special casing <Switch /> since it doesn't use touchable
-      /* $FlowFixMe[invalid-compare] Error discovered during Constant Condition
-       * roll out. See https://fburl.com/workplace/4oq3zi07. */
       (node.type === Switch && node.props && node.props.disabled !== true) ||
-      /* $FlowFixMe[invalid-compare] Error discovered during Constant Condition
-       * roll out. See https://fburl.com/workplace/4oq3zi07. */
       (node.type === View &&
         node?.props?.onStartShouldSetResponder?.testOnly_pressabilityConfig) ||
-      /* $FlowFixMe[invalid-compare] Error discovered during Constant Condition
-       * roll out. See https://fburl.com/workplace/4oq3zi07. */
       (node.type === TouchableWithoutFeedback &&
         node.props &&
         typeof node.props.onPress === 'function') ||
@@ -124,7 +117,7 @@
 async function expectRendersMatchingSnapshot(
   name: string,
   ComponentProvider: () => React.MixedElement,
-  unmockComponent: () => unknown,
+  unmockComponent: () => mixed,
 ) {
   let instance;
 
@@ -182,12 +175,8 @@
 
 function tap(instance: ReactTestInstance) {
   const touchable = instance.find(byClickable());
-  /* $FlowFixMe[invalid-compare] Error discovered during Constant Condition
-   * roll out. See https://fburl.com/workplace/4oq3zi07. */
   if (touchable.type === Text && touchable.props && touchable.props.onPress) {
     touchable.props.onPress();
-    /* $FlowFixMe[invalid-compare] Error discovered during Constant Condition
-     * roll out. See https://fburl.com/workplace/4oq3zi07. */
   } else if (touchable.type === Switch && touchable.props) {
     const value = !touchable.props.value;
     const {onChange, onValueChange} = touchable.props;
