--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/PushNotificationIOS/RCTPushNotificationManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/PushNotificationIOS/RCTPushNotificationManager.mm	2026-03-06 15:22:52
@@ -81,6 +81,7 @@
 
 @end
 
+#if !TARGET_OS_OSX // [macOS]
 @implementation RCTConvert (UIBackgroundFetchResult)
 
 RCT_ENUM_CONVERTER(
@@ -94,6 +95,7 @@
     integerValue)
 
 @end
+#endif // [macOS]
 
 @implementation RCTPushNotificationManager
 
@@ -231,6 +233,7 @@
   }
 }
 
+#if !TARGET_OS_OSX // [macOS]
 + (void)didReceiveRemoteNotification:(NSDictionary *)notification
               fetchCompletionHandler:(RCTRemoteNotificationCallback)completionHandler
 {
@@ -241,6 +244,7 @@
                                                       object:self
                                                     userInfo:userInfo];
 }
+#endif // [macOS]
 
 + (void)setInitialNotification:(UNNotification *)notification
 {
@@ -253,6 +257,7 @@
 
   kInitialNotification = nil;
 }
+
 
 - (void)dealloc
 {
@@ -266,6 +271,7 @@
 
 - (void)handleRemoteNotificationReceived:(NSNotification *)notification
 {
+#if !TARGET_OS_OSX // [macOS]
   NSMutableDictionary *remoteNotification =
       [NSMutableDictionary dictionaryWithDictionary:notification.userInfo[@"notification"]];
   RCTRemoteNotificationCallback completionHandler = notification.userInfo[@"completionHandler"];
@@ -281,6 +287,7 @@
   }
 
   [self sendEventWithName:@"remoteNotificationReceived" body:remoteNotification];
+#endif // [macOS]
 }
 
 - (void)handleRemoteNotificationsRegistered:(NSNotification *)notification
@@ -301,6 +308,7 @@
 
 RCT_EXPORT_METHOD(onFinishRemoteNotification : (NSString *)notificationId fetchResult : (NSString *)fetchResult)
 {
+#if !TARGET_OS_OSX // [macOS]
   UIBackgroundFetchResult result = [RCTConvert UIBackgroundFetchResult:fetchResult];
   RCTRemoteNotificationCallback completionHandler = self.remoteNotificationCallbacks[notificationId];
   if (!completionHandler) {
@@ -309,6 +317,7 @@
   }
   completionHandler(result);
   [self.remoteNotificationCallbacks removeObjectForKey:notificationId];
+#endif // [macOS]
 }
 
 /**
@@ -316,7 +325,13 @@
  */
 RCT_EXPORT_METHOD(setApplicationIconBadgeNumber : (double)number)
 {
+#if !TARGET_OS_OSX // [macOS]
   RCTSharedApplication().applicationIconBadgeNumber = number;
+#else // [macOS
+  NSDockTile *tile = [NSApp dockTile];
+  tile.showsApplicationBadge = number > 0;
+  tile.badgeLabel = number > 0 ? [NSString stringWithFormat:@"%.0lf", number] : nil;
+#endif // macOS]
 }
 
 /**
@@ -324,13 +339,19 @@
  */
 RCT_EXPORT_METHOD(getApplicationIconBadgeNumber : (RCTResponseSenderBlock)callback)
 {
+#if !TARGET_OS_OSX // [macOS]
   callback(@[ @(RCTSharedApplication().applicationIconBadgeNumber) ]);
+#else // [macOS
+  callback(@[ @([NSApp dockTile].badgeLabel.integerValue) ]);
+#endif // macOS]
 }
 
-RCT_EXPORT_METHOD(
-    requestPermissions : (JS::NativePushNotificationManagerIOS::SpecRequestPermissionsPermission &)
-        permissions resolve : (RCTPromiseResolveBlock)resolve reject : (RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(requestPermissions
+                  : (JS::NativePushNotificationManagerIOS::SpecRequestPermissionsPermission &)permissions resolve
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (RCTPromiseRejectBlock)reject)
 {
+#if !TARGET_OS_OSX // [macOS]
   if (RCTRunningInAppExtension()) {
     reject(
         kErrorUnableToRequestPermissions,
@@ -338,6 +359,7 @@
         RCTErrorWithMessage(@"Requesting push notifications is currently unavailable in an app extension"));
     return;
   }
+#endif // [macOS]
 
   // Add a listener to make sure that startObserving has been called
   [self addListener:@"remoteNotificationsRegistered"];
@@ -353,7 +375,6 @@
   if (permissions.sound()) {
     options |= UNAuthorizationOptionSound;
   }
-
   [UNUserNotificationCenter.currentNotificationCenter
       requestAuthorizationWithOptions:options
                     completionHandler:^(BOOL granted, NSError *_Nullable error) {
@@ -379,10 +400,12 @@
 
 RCT_EXPORT_METHOD(checkPermissions : (RCTResponseSenderBlock)callback)
 {
+#if !TARGET_OS_OSX // [macOS]
   if (RCTRunningInAppExtension()) {
     callback(@[ RCTSettingsDictForUNNotificationSettings(NO, NO, NO, NO, NO, NO, UNAuthorizationStatusNotDetermined) ]);
     return;
   }
+#endif // [macOS]
 
   [UNUserNotificationCenter.currentNotificationCenter
       getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings *_Nonnull settings) {
@@ -505,8 +528,9 @@
   }];
 }
 
-RCT_EXPORT_METHOD(
-    getInitialNotification : (RCTPromiseResolveBlock)resolve reject : (__unused RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(getInitialNotification
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (__unused RCTPromiseRejectBlock)reject)
 {
   // The user actioned a local or remote notification to launch the app. Notification is represented by UNNotification.
   // Set this property in the implementation of
