--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/AppDelegate/RCTDefaultReactNativeFactoryDelegate.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.54uBpS4Pde	2026-03-06 18:50:41
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 #import "RCTReactNativeFactory.h"
 
 NS_ASSUME_NONNULL_BEGIN
@@ -15,7 +15,7 @@
  * Contains default implementation of RCTReactNativeFactoryDelegate methods.
  */
 
-@interface RCTDefaultReactNativeFactoryDelegate : UIResponder <RCTReactNativeFactoryDelegate>
-@end
-
-NS_ASSUME_NONNULL_END
+#if !TARGET_OS_OSX // [macOS]
+#else // [macOS
+@interface RCTDefaultReactNativeFactoryDelegate : NSResponder <RCTReactNativeFactoryDelegate>
+#endif // macOS]
