--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.qOCzRBNriV	2026-03-06 18:50:41
@@ -67,38 +67,39 @@
   return self;
 }
 
-- (void)startReactNativeWithModuleName:(NSString *)moduleName inWindow:(UIWindow *_Nullable)window
+- (void)startReactNativeWithModuleName:(NSString *)moduleName inWindow:(RCTPlatformWindow *_Nullable)window // [macOS]
 {
   [self startReactNativeWithModuleName:moduleName inWindow:window initialProperties:nil launchOptions:nil];
 }
 
 - (void)startReactNativeWithModuleName:(NSString *)moduleName
-                              inWindow:(UIWindow *_Nullable)window
+                              inWindow:(RCTPlatformWindow *_Nullable)window // [macOS]
                          launchOptions:(NSDictionary *_Nullable)launchOptions
 {
   [self startReactNativeWithModuleName:moduleName inWindow:window initialProperties:nil launchOptions:launchOptions];
 }
 
 - (void)startReactNativeWithModuleName:(NSString *)moduleName
-                              inWindow:(UIWindow *_Nullable)window
+                              inWindow:(RCTPlatformWindow *_Nullable)window
                      initialProperties:(NSDictionary *_Nullable)initialProperties
                          launchOptions:(NSDictionary *_Nullable)launchOptions
 {
-  UIView *rootView = [self.rootViewFactory viewWithModuleName:moduleName
+  RCTUIView *rootView = [self.rootViewFactory viewWithModuleName:moduleName  // [macOS]
                                             initialProperties:initialProperties
                                                 launchOptions:launchOptions
                                           bundleConfiguration:self.bundleConfiguration
                                          devMenuConfiguration:self.devMenuConfiguration];
   UIViewController *rootViewController = [_delegate createRootViewController];
   [_delegate setRootView:rootView toRootViewController:rootViewController];
+#if !TARGET_OS_OSX // [macOS]
   window.rootViewController = rootViewController;
   [window makeKeyAndVisible];
-}
-
-#pragma mark - RCTUIConfiguratorProtocol
-
-- (RCTColorSpace)defaultColorSpace
-{
+#else // [macOS
+  rootViewController.view.frame = window.frame;
+  window.contentViewController = rootViewController;
+  [window makeKeyAndOrderFront:self];
+  [window center];
+#endif // macOS]
   if ([_delegate respondsToSelector:@selector(defaultColorSpace)]) {
     return [_delegate defaultColorSpace];
   }
@@ -259,7 +260,7 @@
                                                    turboModuleEnabled:YES
                                                     bridgelessEnabled:YES];
 
-  configuration.createRootViewWithBridge = ^UIView *(RCTBridge *bridge, NSString *moduleName, NSDictionary *initProps) {
+  configuration.createRootViewWithBridge = ^RCTPlatformView *(RCTBridge *bridge, NSString *moduleName, NSDictionary *initProps) {
     return [weakSelf.delegate createRootViewWithBridge:bridge moduleName:moduleName initProps:initProps];
   };
 
@@ -267,8 +268,7 @@
     return [weakSelf.delegate createBridgeWithDelegate:delegate launchOptions:launchOptions];
   };
 
-  configuration.customizeRootView = ^(UIView *_Nonnull rootView) {
-    [weakSelf.delegate customizeRootView:(RCTRootView *)rootView];
+  configuration.customizeRootView = ^(RCTPlatformView *_Nonnull rootView) {
   };
 
   configuration.sourceURLForBridge = ^NSURL *_Nullable(RCTBridge *_Nonnull bridge)
