--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/nativemodule/samples/platform/ios/ReactCommon/RCTSampleTurboModule.mm	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/nativemodule/samples/platform/ios/ReactCommon/RCTSampleTurboModule.mm	2026-03-06 15:22:53
@@ -12,7 +12,7 @@
 #import <React/RCTInitializing.h>
 #import <React/RCTUtils.h>
 #import <ReactCommon/RCTTurboModuleWithJSIBindings.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 using namespace facebook::react;
 
@@ -20,7 +20,7 @@
 @end
 
 @implementation RCTSampleTurboModule {
-  facebook::react::ModuleConstants<JS::NativeSampleTurboModule::Constants> _constants;
+  NSDictionary *_constants;
 }
 
 // Backward-compatible export
@@ -34,14 +34,23 @@
 
 - (void)initialize
 {
-  UIScreen *mainScreen = UIScreen.mainScreen;
-  CGSize screenSize = mainScreen.bounds.size;
+#if !TARGET_OS_OSX // [macOS]
+#if !TARGET_OS_VISION // [visionOS]
+    UIScreen *mainScreen = UIScreen.mainScreen;
+    CGSize screenSize = mainScreen.bounds.size;
+#else // [visionOS
+	CGSize screenSize = CGSizeZero;
+#endif // visionOS]
+#else // [macOS
+    NSScreen *mainScreen = NSScreen.mainScreen;
+    CGSize screenSize = mainScreen.frame.size;
+#endif // macOS]
 
-  _constants = facebook::react::typedConstants<JS::NativeSampleTurboModule::Constants>({
-      .const1 = YES,
-      .const2 = screenSize.width,
-      .const3 = @"something",
-  });
+  _constants = @{
+    @"const1" : @YES,
+    @"const2" : @(screenSize.width),
+    @"const3" : @"something",
+  };
 }
 
 - (dispatch_queue_t)methodQueue
@@ -62,21 +71,17 @@
   NSLog(@"Invalidating RCTSampleTurboModule...");
 }
 
-- (facebook::react::ModuleConstants<JS::NativeSampleTurboModule::Constants>)constantsToExport
+- (NSDictionary *)getConstants
 {
   return _constants;
 }
 
-- (facebook::react::ModuleConstants<JS::NativeSampleTurboModule::Constants>)getConstants
+// TODO: Remove once fully migrated to TurboModule.
+- (NSDictionary *)constantsToExport
 {
-  return _constants;
+  return [self getConstants];
 }
 
-- (void)getImageUrl:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject
-{
-  reject(@"Exception", @"Not implemented", nil);
-}
-
 #pragma mark - RCTTurboModuleWithJSIBindings
 
 - (void)installJSIBindingsWithRuntime:(facebook::jsi::Runtime &)runtime
@@ -140,23 +145,25 @@
 {
   return @{
     @"x" : @(x),
-    @"y" : (y != nullptr) ? y : [NSNull null],
-    @"z" : (z != nullptr) ? z : [NSNull null],
+    @"y" : y ? y : [NSNull null],
+    @"z" : z ? z : [NSNull null],
   };
 }
 
 RCT_EXPORT_METHOD(getValueWithCallback : (RCTResponseSenderBlock)callback)
 {
-  if (callback == nullptr) {
+  if (!callback) {
     return;
   }
   callback(@[ @"value from callback!" ]);
 }
 
-RCT_EXPORT_METHOD(
-    getValueWithPromise : (BOOL)error resolve : (RCTPromiseResolveBlock)resolve reject : (RCTPromiseRejectBlock)reject)
+RCT_EXPORT_METHOD(getValueWithPromise
+                  : (BOOL)error resolve
+                  : (RCTPromiseResolveBlock)resolve reject
+                  : (RCTPromiseRejectBlock)reject)
 {
-  if ((resolve == nullptr) || (reject == nullptr)) {
+  if (!resolve || !reject) {
     return;
   }
 
@@ -172,7 +179,7 @@
 
 RCT_EXPORT_METHOD(voidFuncThrows)
 {
-  NSException *myException = [NSException exceptionWithName:@"Exception"
+  NSException *myException = [NSException exceptionWithName:@"Excepption"
                                                      reason:@"Intentional exception from ObjC voidFuncThrows"
                                                    userInfo:nil];
   @throw myException;
@@ -180,7 +187,7 @@
 
 RCT_EXPORT_SYNCHRONOUS_TYPED_METHOD(NSDictionary *, getObjectThrows : (NSDictionary *)arg)
 {
-  NSException *myException = [NSException exceptionWithName:@"Exception"
+  NSException *myException = [NSException exceptionWithName:@"Excepption"
                                                      reason:@"Intentional exception from ObjC getObjectThrows"
                                                    userInfo:nil];
   @throw myException;
@@ -188,7 +195,7 @@
 
 RCT_EXPORT_METHOD(promiseThrows : (RCTPromiseResolveBlock)resolve reject : (RCTPromiseRejectBlock)reject)
 {
-  NSException *myException = [NSException exceptionWithName:@"Exception"
+  NSException *myException = [NSException exceptionWithName:@"Excepption"
                                                      reason:@"Intentional exception from ObjC promiseThrows"
                                                    userInfo:nil];
   @throw myException;
