--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Modules/RCTLayoutAnimation.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Modules/RCTLayoutAnimation.m	2026-03-06 15:22:52
@@ -11,8 +11,11 @@
 
 @implementation RCTLayoutAnimation
 
+#if !TARGET_OS_OSX // [macOS]
 static UIViewAnimationCurve _currentKeyboardAnimationCurve;
+#endif // [macOS]
 
+#if !TARGET_OS_OSX // [macOS]
 static UIViewAnimationOptions UIViewAnimationOptionsFromRCTAnimationType(RCTAnimationType type)
 {
   switch (type) {
@@ -32,7 +35,26 @@
       return UIViewAnimationOptionCurveEaseInOut;
   }
 }
+#else // [macOS
+static NSString *CAMediaTimingFunctionNameFromRCTAnimationType(RCTAnimationType type)
+{
+  switch (type) {
+    case RCTAnimationTypeLinear:
+      return kCAMediaTimingFunctionLinear;
+    case RCTAnimationTypeEaseIn:
+      return kCAMediaTimingFunctionEaseIn;
+    case RCTAnimationTypeEaseOut:
+      return kCAMediaTimingFunctionEaseOut;
+    case RCTAnimationTypeEaseInEaseOut:
+      return kCAMediaTimingFunctionEaseInEaseOut;
+    default:
+      RCTLogError(@"Unsupported animation type %zd", type);
+      return kCAMediaTimingFunctionDefault;
+  }
+}
+#endif // macOS]
 
+#if !TARGET_OS_OSX // [macOS]
 // Use a custom initialization function rather than implementing `+initialize` so that we can control
 // when the initialization code runs. `+initialize` runs immediately before the first message is sent
 // to the class which may be too late for us. By this time, we may have missed some
@@ -53,6 +75,7 @@
   NSDictionary *userInfo = notification.userInfo;
   _currentKeyboardAnimationCurve = [userInfo[UIKeyboardAnimationCurveUserInfoKey] integerValue];
 }
+#endif // [macOS]
 
 - (instancetype)initWithDuration:(NSTimeInterval)duration
                            delay:(NSTimeInterval)delay
@@ -95,10 +118,12 @@
     }
 
     _animationType = [RCTConvert RCTAnimationType:config[@"type"]];
+#if !TARGET_OS_OSX // [macOS]
     if (_animationType == RCTAnimationTypeSpring) {
       _springDamping = [RCTConvert CGFloat:config[@"springDamping"]];
       _initialVelocity = [RCTConvert CGFloat:config[@"initialVelocity"]];
     }
+#endif // [macOS]
   }
 
   return self;
@@ -106,6 +131,7 @@
 
 - (void)performAnimations:(void (^)(void))animations withCompletionBlock:(void (^)(BOOL completed))completionBlock
 {
+#if !TARGET_OS_OSX // [macOS]
   if (_animationType == RCTAnimationTypeSpring) {
     [UIView animateWithDuration:_duration
                           delay:_delay
@@ -124,6 +150,36 @@
                      animations:animations
                      completion:completionBlock];
   }
+#else // [macOS
+  NSString *timingFunctionName = CAMediaTimingFunctionNameFromRCTAnimationType(_animationType);
+  
+  CAMediaTimingFunction *timingFunction = [CAMediaTimingFunction functionWithName:timingFunctionName];
+  NSTimeInterval duration = _duration;
+  
+  dispatch_block_t runAnimationGroup = ^{
+    [NSAnimationContext runAnimationGroup:^(NSAnimationContext *context) {
+      context.duration = duration;
+      context.timingFunction = timingFunction;
+      context.allowsImplicitAnimation = YES;
+      
+      if (animations != nil) {
+        animations();
+      }
+    }
+                        completionHandler:^{
+                          if (completionBlock != nil) {
+                            completionBlock(YES);
+                          }
+                        }];
+  };
+  
+  if (_delay == 0.0) {
+    runAnimationGroup();
+  } else {
+    dispatch_time_t time = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(_delay * NSEC_PER_SEC));
+    dispatch_after(time, dispatch_get_main_queue(), runAnimationGroup);
+  }
+#endif // macOS]
 }
 
 - (BOOL)isEqual:(RCTLayoutAnimation *)animation
