--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Modules/RCTRedBoxExtraDataViewController.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Modules/RCTRedBoxExtraDataViewController.h	2026-03-06 15:22:52
@@ -5,16 +5,21 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 @protocol RCTRedBoxExtraDataActionDelegate <NSObject>
 - (void)reload;
 @end
 
+#if !TARGET_OS_OSX // [macOS]
 @interface RCTRedBoxExtraDataViewController : UIViewController <UITableViewDelegate, UITableViewDataSource>
+#else // [macOS
+@interface RCTRedBoxExtraDataViewController : NSViewController <NSTableViewDelegate, NSTableViewDataSource>
+#endif // macOS]
 
 @property (nonatomic, weak) id<RCTRedBoxExtraDataActionDelegate> actionDelegate;
 
 - (void)addExtraData:(NSDictionary *)data forIdentifier:(NSString *)identifier;
 
 @end
+
