--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/React-RCTText.podspec	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.bh5pRQAPCm	2026-03-06 18:50:45
@@ -27,10 +27,12 @@
   s.platforms              = min_supported_versions
   s.source                 = source
   s.source_files           = podspec_sources("**/*.{h,m,mm}", "**/*.h")
-  s.preserve_paths         = "package.json", "LICENSE", "LICENSE-docs"
-  s.header_dir             = "RCTText"
-  s.framework              = ["MobileCoreServices"]
-  s.pod_target_xcconfig    = { "CLANG_CXX_LANGUAGE_STANDARD" => rct_cxx_language_standard() }
+  s.ios.exclude_files      = "**/macOS/*" # [macOS]
+  s.visionos.exclude_files = "**/macOS/*" # [visionOS]
+  # [macOS MobileCoreServices Not available on macOS
+  s.ios.frameworks         = "MobileCoreServices" 
+  s.visionos.frameworks    = "MobileCoreServices"
+  # macOS]
 
   s.dependency "Yoga"
   s.dependency "React-Core/RCTTextHeaders", version
