--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTEnhancedScrollView.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTEnhancedScrollView.mm	2026-03-06 15:22:52
@@ -9,11 +9,11 @@
 #import <React/RCTUtils.h>
 #import <react/utils/FloatComparison.h>
 
-@interface RCTEnhancedScrollView () <UIScrollViewDelegate>
+@interface RCTEnhancedScrollView () <RCTUIScrollViewDelegate> // [macOS]
 @end
 
 @implementation RCTEnhancedScrollView {
-  __weak id<UIScrollViewDelegate> _publicDelegate;
+  __weak id<RCTUIScrollViewDelegate> _publicDelegate; // [macOS]
   BOOL _isSetContentOffsetDisabled;
 }
 
@@ -31,6 +31,7 @@
 - (instancetype)initWithFrame:(CGRect)frame
 {
   if (self = [super initWithFrame:frame]) {
+#if !TARGET_OS_OSX // [macOS]
     // We set the default behavior to "never" so that iOS
     // doesn't do weird things to UIScrollView insets automatically
     // and keeps it as an opt-in behavior.
@@ -40,6 +41,7 @@
     // because this attribute affects a position of vertical scrollbar; we don't want this
     // scrollbar flip because we also flip it with whole `UIScrollView` flip.
     self.semanticContentAttribute = UISemanticContentAttributeForceLeftToRight;
+#endif // [macOS]
 
     __weak __typeof(self) weakSelf = self;
     _delegateSplitter = [[RCTGenericDelegateSplitter alloc] initWithDelegateUpdateBlock:^(id delegate) {
@@ -81,8 +83,7 @@
     return;
   }
 
-  CGFloat top = 0;
-  CGFloat left = 0;
+  CGFloat top = 0, left = 0;
   if (contentSize.width < boundsSize.width) {
     left = (boundsSize.width - contentSize.width) * 0.5f;
   }
@@ -108,12 +109,13 @@
   [self centerContentIfNeeded];
 }
 
-- (void)didAddSubview:(UIView *)subview
+- (void)didAddSubview:(RCTPlatformView *)subview // [macOS]
 {
   [super didAddSubview:subview];
   [self centerContentIfNeeded];
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (BOOL)touchesShouldCancelInContentView:(UIView *)view
 {
   if ([_overridingDelegate respondsToSelector:@selector(touchesShouldCancelInContentView:)]) {
@@ -122,20 +124,21 @@
 
   return [super touchesShouldCancelInContentView:view];
 }
+#endif // [macOS]
 
 #pragma mark - RCTGenericDelegateSplitter
 
-- (void)setPrivateDelegate:(id<UIScrollViewDelegate>)delegate
+- (void)setPrivateDelegate:(id<RCTUIScrollViewDelegate>)delegate // [macOS]
 {
   [super setDelegate:delegate];
 }
 
-- (id<UIScrollViewDelegate>)delegate
+- (id<RCTUIScrollViewDelegate>)delegate // [macOS]
 {
   return _publicDelegate;
 }
 
-- (void)setDelegate:(id<UIScrollViewDelegate>)delegate
+- (void)setDelegate:(id<RCTUIScrollViewDelegate>)delegate // [macOS]
 {
   if (_publicDelegate == delegate) {
     return;
@@ -156,15 +159,9 @@
 
 #pragma mark - UIScrollViewDelegate
 
-- (void)scrollViewDidScroll:(__unused UIScrollView *)scrollView
-{
-  // Empty implementation. This method exists to prevent crashes when the delegate splitter
-  // forwards scrollViewDidScroll: messages to RCTEnhancedScrollView.
-}
-
-- (void)scrollViewWillEndDragging:(UIScrollView *)scrollView
+- (void)scrollViewWillEndDragging:(RCTUIScrollView *)scrollView
                      withVelocity:(CGPoint)velocity
-              targetContentOffset:(inout CGPoint *)targetContentOffset
+              targetContentOffset:(inout CGPoint *)targetContentOffset // [macOS]
 {
   if (self.snapToOffsets && self.snapToOffsets.count > 0) {
     // An alternative to enablePaging and snapToInterval which allows setting custom
@@ -289,14 +286,14 @@
   }
 }
 
-- (void)scrollViewDidZoom:(__unused UIScrollView *)scrollView
+- (void)scrollViewDidZoom:(__unused RCTUIScrollView *)scrollView // [macOS]
 {
   [self centerContentIfNeeded];
 }
 
 #pragma mark -
 
-- (BOOL)isHorizontal:(UIScrollView *)scrollView
+- (BOOL)isHorizontal:(RCTUIScrollView *)scrollView // [macOS]
 {
   return !facebook::react::floatEquality(scrollView.contentSize.width, self.frame.size.width) &&
       scrollView.contentSize.width > self.frame.size.width;
