--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm	2026-03-06 15:22:52
@@ -24,13 +24,11 @@
 @end
 
 @implementation RCTPullToRefreshViewComponentView {
+  BOOL _isBeforeInitialLayout;
+#if !TARGET_OS_OSX // [macOS]
   UIRefreshControl *_refreshControl;
+#endif // [macOS]
   RCTScrollViewComponentView *__weak _scrollViewComponentView;
-  // This variable keeps track of whether the view is recycled or not. Once the view is recycled, the component
-  // creates a new instance of UIRefreshControl, resetting the native props to the default values.
-  // However, when recycling, we are keeping around the old _props. The flag is used to force the application
-  // of the current props to the newly created UIRefreshControl the first time that updateProps is called.
-  BOOL _recycled;
 }
 
 - (instancetype)initWithFrame:(CGRect)frame
@@ -40,8 +38,8 @@
     // attaching and detaching of a pull-to-refresh view to a scroll view.
     // The pull-to-refresh view is not a subview of this view.
     self.hidden = YES;
-    _props = PullToRefreshViewShadowNode::defaultSharedProps();
-    _recycled = NO;
+
+    _isBeforeInitialLayout = YES;
     [self _initializeUIRefreshControl];
   }
 
@@ -50,10 +48,12 @@
 
 - (void)_initializeUIRefreshControl
 {
+#if !TARGET_OS_OSX // [macOS]
   _refreshControl = [UIRefreshControl new];
   [_refreshControl addTarget:self
                       action:@selector(handleUIControlEventValueChanged)
             forControlEvents:UIControlEventValueChanged];
+#endif // [macOS]
 }
 
 #pragma mark - RCTComponentViewProtocol
@@ -67,53 +67,58 @@
 {
   [super prepareForRecycle];
   _scrollViewComponentView = nil;
+  _props = nil;
+  _isBeforeInitialLayout = YES;
   [self _initializeUIRefreshControl];
-  _recycled = YES;
 }
 
 - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &)oldProps
 {
+  // Prop updates are ignored by _refreshControl until after the initial layout, so just store them in _props until then
+  if (_isBeforeInitialLayout) {
+    _props = std::static_pointer_cast<const PullToRefreshViewProps>(props);
+    return;
+  }
+
   const auto &oldConcreteProps = static_cast<const PullToRefreshViewProps &>(*_props);
   const auto &newConcreteProps = static_cast<const PullToRefreshViewProps &>(*props);
 
-  if (_recycled || newConcreteProps.tintColor != oldConcreteProps.tintColor) {
+  if (newConcreteProps.tintColor != oldConcreteProps.tintColor) {
+#if !TARGET_OS_OSX // [macOS]
     _refreshControl.tintColor = RCTUIColorFromSharedColor(newConcreteProps.tintColor);
+#endif // [macOS]
   }
 
-  if (_recycled || newConcreteProps.progressViewOffset != oldConcreteProps.progressViewOffset) {
+  if (newConcreteProps.progressViewOffset != oldConcreteProps.progressViewOffset) {
     [self _updateProgressViewOffset:newConcreteProps.progressViewOffset];
   }
 
   BOOL needsUpdateTitle = NO;
 
-  if (_recycled || newConcreteProps.title != oldConcreteProps.title) {
+  if (newConcreteProps.title != oldConcreteProps.title) {
     needsUpdateTitle = YES;
   }
 
-  if (_recycled || newConcreteProps.titleColor != oldConcreteProps.titleColor) {
+  if (newConcreteProps.titleColor != oldConcreteProps.titleColor) {
     needsUpdateTitle = YES;
   }
 
   [super updateProps:props oldProps:oldProps];
 
-  if (_recycled || needsUpdateTitle) {
+  if (needsUpdateTitle) {
     [self _updateTitle];
   }
 
   // All prop updates must happen above the call to begin refreshing, or else _refreshControl will ignore the updates
-  if (_recycled || newConcreteProps.refreshing != oldConcreteProps.refreshing) {
+  if (newConcreteProps.refreshing != oldConcreteProps.refreshing) {
+#if !TARGET_OS_OSX // [macOS]
     if (newConcreteProps.refreshing) {
       [self beginRefreshingProgrammatically];
     } else {
       [_refreshControl endRefreshing];
     }
+#endif // [macOS]
   }
-
-  if (_recycled || newConcreteProps.zIndex != oldConcreteProps.zIndex) {
-    _refreshControl.layer.zPosition = newConcreteProps.zIndex.value_or(0);
-  }
-
-  _recycled = NO;
 }
 
 #pragma mark -
@@ -125,11 +130,13 @@
 
 - (void)_updateProgressViewOffset:(Float)progressViewOffset
 {
+#if !TARGET_OS_OSX // [macOS]
   _refreshControl.bounds = CGRectMake(
       _refreshControl.bounds.origin.x,
       -progressViewOffset,
       _refreshControl.bounds.size.width,
       _refreshControl.bounds.size.height);
+#endif // [macOS]
 }
 
 - (void)_updateTitle
@@ -137,7 +144,9 @@
   const auto &concreteProps = static_cast<const PullToRefreshViewProps &>(*_props);
 
   if (concreteProps.title.empty()) {
+#if !TARGET_OS_OSX // [macOS]
     _refreshControl.attributedTitle = nil;
+#endif // [macOS]
     return;
   }
 
@@ -146,8 +155,10 @@
     attributes[NSForegroundColorAttributeName] = RCTUIColorFromSharedColor(concreteProps.titleColor);
   }
 
+#if !TARGET_OS_OSX // [macOS]
   _refreshControl.attributedTitle =
       [[NSAttributedString alloc] initWithString:RCTNSStringFromString(concreteProps.title) attributes:attributes];
+#endif // [macOS]
 }
 
 #pragma mark - Attaching & Detaching
@@ -158,18 +169,22 @@
 
   // Attempts to begin refreshing before the initial layout are ignored by _refreshControl. So if the control is
   // refreshing when mounted, we need to call beginRefreshing in layoutSubviews or it won't work.
-  if (self.window) {
-    const auto &concreteProps = static_cast<const PullToRefreshViewProps &>(*_props);
+  if (_isBeforeInitialLayout) {
+    _isBeforeInitialLayout = NO;
 
-    if (concreteProps.refreshing) {
-      [self beginRefreshingProgrammatically];
-    }
+    [self updateProps:_props oldProps:PullToRefreshViewShadowNode::defaultSharedProps()];
   }
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)didMoveToSuperview
 {
   [super didMoveToSuperview];
+#else // macOS]
+- (void)viewDidMoveToSuperview
+{
+  [super viewDidMoveToSuperview];
+#endif // macOS]
   if (self.superview) {
     [self _attach];
   } else {
@@ -188,12 +203,14 @@
     return;
   }
 
+#if !TARGET_OS_OSX // [macOS]
   if (@available(macCatalyst 13.1, *)) {
     _scrollViewComponentView.scrollView.refreshControl = _refreshControl;
 
     // This ensures that layoutSubviews is called. Without this, recycled instances won't refresh on mount
     [self setNeedsLayout];
   }
+#endif // [macOS]
 }
 
 - (void)_detach
@@ -203,14 +220,17 @@
   }
 
   // iOS requires to end refreshing before unmounting.
+#if !TARGET_OS_OSX // [macOS]
   [_refreshControl endRefreshing];
 
   if (@available(macCatalyst 13.1, *)) {
     _scrollViewComponentView.scrollView.refreshControl = nil;
   }
+#endif // [macOS]
   _scrollViewComponentView = nil;
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)beginRefreshingProgrammatically
 {
   if (!_scrollViewComponentView) {
@@ -219,13 +239,13 @@
 
   // When refreshing programmatically (i.e. without pulling down), we must explicitly adjust the ScrollView content
   // offset, or else the _refreshControl won't be visible
-  if (!_refreshControl.isRefreshing) {
-    UIScrollView *scrollView = _scrollViewComponentView.scrollView;
-    CGPoint offset = {scrollView.contentOffset.x, scrollView.contentOffset.y - _refreshControl.frame.size.height};
-    [scrollView setContentOffset:offset];
-    [_refreshControl beginRefreshing];
-  }
+  UIScrollView *scrollView = _scrollViewComponentView.scrollView;
+  CGPoint offset = {scrollView.contentOffset.x, scrollView.contentOffset.y - _refreshControl.frame.size.height};
+  [scrollView setContentOffset:offset];
+
+  [_refreshControl beginRefreshing];
 }
+#endif // [macOS]
 
 #pragma mark - Native commands
 
@@ -236,11 +256,13 @@
 
 - (void)setNativeRefreshing:(BOOL)refreshing
 {
+#if !TARGET_OS_OSX // [macOS]
   if (refreshing) {
     [self beginRefreshingProgrammatically];
   } else {
     [_refreshControl endRefreshing];
   }
+#endif // [macOS]
 }
 
 #pragma mark - RCTRefreshableProtocol
