--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTScrollViewComponentView.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTScrollViewComponentView.mm	2026-03-06 15:22:52
@@ -24,7 +24,6 @@
 #import "RCTCustomPullToRefreshViewProtocol.h"
 #import "RCTEnhancedScrollView.h"
 #import "RCTFabricComponentsPlugins.h"
-#import "RCTVirtualViewContainerState.h"
 
 using namespace facebook::react;
 
@@ -34,6 +33,7 @@
 
 static const CGFloat kClippingLeeway = 44.0;
 
+#if TARGET_OS_IOS // [macOS] [visionOS]
 static UIScrollViewKeyboardDismissMode RCTUIKeyboardDismissModeFromProps(const ScrollViewProps &props)
 {
   switch (props.keyboardDismissMode) {
@@ -45,7 +45,9 @@
       return UIScrollViewKeyboardDismissModeInteractive;
   }
 }
+#endif // [macOS] [visionOS]
 
+#if !TARGET_OS_OSX // [macOS
 static UIScrollViewIndicatorStyle RCTUIScrollViewIndicatorStyleFromProps(const ScrollViewProps &props)
 {
   switch (props.indicatorStyle) {
@@ -57,13 +59,14 @@
       return UIScrollViewIndicatorStyleWhite;
   }
 }
+#endif // [macOS]
 
 // Once Fabric implements proper NativeAnimationDriver, this should be removed.
 // This is just a workaround to allow animations based on onScroll event.
 // This is only used to animate sticky headers in ScrollViews, and only the contentOffset and tag is used.
 // TODO: T116850910 [Fabric][iOS] Make Fabric not use legacy RCTEventDispatcher for native-driven AnimatedEvents
 static void
-RCTSendScrollEventForNativeAnimations_DEPRECATED(UIScrollView *scrollView, NSInteger tag, NSString *eventName)
+RCTSendScrollEventForNativeAnimations_DEPRECATED(RCTUIScrollView *scrollView, NSInteger tag, NSString *eventName) // [macOS]
 {
   if (ReactNativeFeatureFlags::cxxNativeAnimatedEnabled()) {
     return;
@@ -85,7 +88,9 @@
 }
 
 @interface RCTScrollViewComponentView () <
+#if !TARGET_OS_OSX // [macOS]
     UIScrollViewDelegate,
+#endif // [macOS]
     RCTScrollViewProtocol,
     RCTScrollableProtocol,
     RCTEnhancedScrollViewOverridingDelegate>
@@ -107,30 +112,18 @@
 
   CGPoint _contentOffsetWhenClipped;
 
-  __weak UIView *_contentView;
+  __weak RCTUIView *_contentView; // [macOS]
 
   CGRect _prevFirstVisibleFrame;
-  __weak UIView *_firstVisibleView;
-  NSInteger _firstVisibleViewTag;
+  __weak RCTPlatformView *_firstVisibleView; // [macOS]
 
   CGFloat _endDraggingSensitivityMultiplier;
-
-  // A flag indicating that accessibility API is used.
-  // It is not restored to the default value in prepareForRecycle.
-  // Once an accessibility API is used, view culling will be disabled for the entire session.
-  BOOL _isAccessibilityAPIUsed;
-
-  // Flag to temporarily disable maintainVisibleContentPosition adjustments during immediate state updates
-  // to prevent conflicts between immediate content offset updates and visible content position logic
-  BOOL _avoidAdjustmentForMaintainVisibleContentPosition;
-
-  RCTVirtualViewContainerState *_virtualViewContainerState;
 }
 
-+ (RCTScrollViewComponentView *_Nullable)findScrollViewComponentViewForView:(UIView *)view
++ (RCTScrollViewComponentView *_Nullable)findScrollViewComponentViewForView:(RCTPlatformView *)view // [macOS]
 {
   do {
-    view = view.superview;
+    view = (RCTPlatformView *)view.superview; // [macOS]
   } while (view != nil && ![view isKindOfClass:[RCTScrollViewComponentView class]]);
   return (RCTScrollViewComponentView *)view;
 }
@@ -142,18 +135,26 @@
     _scrollView = [[RCTEnhancedScrollView alloc] initWithFrame:self.bounds];
     _scrollView.clipsToBounds = _props->getClipsContentToBounds();
     _scrollView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+#if !TARGET_OS_OSX // [macOS]
     _scrollView.delaysContentTouches = NO;
+#endif // [macOS]
     ((RCTEnhancedScrollView *)_scrollView).overridingDelegate = self;
     _isUserTriggeredScrolling = NO;
     _shouldUpdateContentInsetAdjustmentBehavior = YES;
     _automaticallyAdjustKeyboardInsets = NO;
-    _isAccessibilityAPIUsed = NO;
     [self addSubview:_scrollView];
 
-    _containerView = [[UIView alloc] initWithFrame:CGRectZero];
+    _containerView = [[RCTUIView alloc] initWithFrame:CGRectZero]; // [macOS]
+#if !TARGET_OS_OSX // [macOS]
     [_scrollView addSubview:_containerView];
-
+#else // [macOS
+    _containerView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    [_scrollView setDocumentView:_containerView];
+#endif // macOS]
+    
+#if !TARGET_OS_OSX // [macOS]
     [self.scrollViewDelegateSplitter addDelegate:self];
+#endif // [macOS]
 #if TARGET_OS_IOS
     [self _registerKeyboardListener];
 #endif
@@ -169,7 +170,9 @@
 {
   // Removing all delegates from the splitter nils the actual delegate which prevents a crash on UIScrollView
   // deallocation.
+#if !TARGET_OS_OSX // [macOS]
   [self.scrollViewDelegateSplitter removeAllDelegates];
+#endif // [macOS]
 }
 
 #if TARGET_OS_IOS
@@ -276,7 +279,7 @@
 }
 #endif
 
-- (RCTGenericDelegateSplitter<id<UIScrollViewDelegate>> *)scrollViewDelegateSplitter
+- (RCTGenericDelegateSplitter<id<RCTUIScrollViewDelegate>> *)scrollViewDelegateSplitter
 {
   return ((RCTEnhancedScrollView *)_scrollView).delegateSplitter;
 }
@@ -313,7 +316,9 @@
         : CGAffineTransformMakeScale(-1, 1);
 
     _containerView.transform = transform;
+#if !TARGET_OS_OSX // [macOS]
     _scrollView.transform = transform;
+#endif // [macOS]
   }
 }
 
@@ -342,19 +347,25 @@
   // FIXME: Commented props are not supported yet.
   MAP_SCROLL_VIEW_PROP(alwaysBounceHorizontal);
   MAP_SCROLL_VIEW_PROP(alwaysBounceVertical);
+#if !TARGET_OS_OSX // [macOS]
   MAP_SCROLL_VIEW_PROP(bounces);
   MAP_SCROLL_VIEW_PROP(bouncesZoom);
   MAP_SCROLL_VIEW_PROP(canCancelContentTouches);
+#endif // [macOS]
   MAP_SCROLL_VIEW_PROP(centerContent);
   // MAP_SCROLL_VIEW_PROP(automaticallyAdjustContentInsets);
+#if !TARGET_OS_OSX // [macOS]
   MAP_SCROLL_VIEW_PROP(decelerationRate);
   MAP_SCROLL_VIEW_PROP(directionalLockEnabled);
   MAP_SCROLL_VIEW_PROP(maximumZoomScale);
   MAP_SCROLL_VIEW_PROP(minimumZoomScale);
+#endif // [macOS]
   MAP_SCROLL_VIEW_PROP(scrollEnabled);
+#if !TARGET_OS_OSX // [macOS]
   MAP_SCROLL_VIEW_PROP(pagingEnabled);
   MAP_SCROLL_VIEW_PROP(pinchGestureEnabled);
   MAP_SCROLL_VIEW_PROP(scrollsToTop);
+#endif // [macOS]
   MAP_SCROLL_VIEW_PROP(showsHorizontalScrollIndicator);
   MAP_SCROLL_VIEW_PROP(showsVerticalScrollIndicator);
 
@@ -367,7 +378,9 @@
   }
 
   if (oldScrollViewProps.indicatorStyle != newScrollViewProps.indicatorStyle) {
+#if !TARGET_OS_OSX // [macOS]
     _scrollView.indicatorStyle = RCTUIScrollViewIndicatorStyleFromProps(newScrollViewProps);
+#endif // [macOS]
   }
 
   _endDraggingSensitivityMultiplier = newScrollViewProps.endDraggingSensitivityMultiplier;
@@ -418,6 +431,7 @@
     scrollView.snapToOffsets = snapToOffsets;
   }
 
+#if !TARGET_OS_OSX // [macOS]
   if (oldScrollViewProps.automaticallyAdjustsScrollIndicatorInsets !=
       newScrollViewProps.automaticallyAdjustsScrollIndicatorInsets) {
     scrollView.automaticallyAdjustsScrollIndicatorInsets = newScrollViewProps.automaticallyAdjustsScrollIndicatorInsets;
@@ -437,12 +451,15 @@
     }
     _shouldUpdateContentInsetAdjustmentBehavior = NO;
   }
-
+#endif // [macOS]
+    
   MAP_SCROLL_VIEW_PROP(disableIntervalMomentum);
   MAP_SCROLL_VIEW_PROP(snapToInterval);
 
   if (oldScrollViewProps.keyboardDismissMode != newScrollViewProps.keyboardDismissMode) {
+#if TARGET_OS_IOS // [macOS] [visionOS]
     scrollView.keyboardDismissMode = RCTUIKeyboardDismissModeFromProps(newScrollViewProps);
+#endif // [macOS] [visionOS]
   }
 
   [super updateProps:props oldProps:oldProps];
@@ -493,16 +510,20 @@
   }];
 }
 
-- (UIView *)betterHitTest:(CGPoint)point withEvent:(UIEvent *)event
+- (RCTPlatformView *)betterHitTest:(CGPoint)point withEvent:(UIEvent *)event // [macOS]
 {
   // This is the same algorithm as in the RCTViewComponentView with the exception of
   // skipping the immediate child (_containerView) and checking grandchildren instead.
   // This prevents issues with touches outside of _containerView being ignored even
   // if they are within the bounds of the _containerView's children.
 
+#if !TARGET_OS_OSX // [macOS]
   if (!self.userInteractionEnabled || self.hidden || self.alpha < 0.01) {
-    return nil;
-  }
+#else // [macOS
+    if (!self.userInteractionEnabled || self.hidden || self.alphaValue < 0.01  ) {
+#endif // macOS]
+      return nil;
+    }
 
   BOOL isPointInside = [self pointInside:point withEvent:event];
 
@@ -514,8 +535,8 @@
     return nil;
   }
 
-  for (UIView *subview in [_containerView.subviews reverseObjectEnumerator]) {
-    UIView *hitView = [subview hitTest:[subview convertPoint:point fromView:self] withEvent:event];
+  for (RCTPlatformView *subview in [_containerView.subviews reverseObjectEnumerator]) { // [macOS]
+    RCTPlatformView *hitView = RCTUIViewHitTestWithEvent(subview, [subview convertPoint:point fromView:self], event); // [macOS]
     if (hitView) {
       return hitView;
     }
@@ -540,7 +561,7 @@
   [((RCTEnhancedScrollView *)_scrollView) preserveContentOffsetWithBlock:block];
 }
 
-- (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
+- (void)mountChildComponentView:(RCTUIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index // [macOS]
 {
   [_containerView insertSubview:childComponentView atIndex:index];
   if (![childComponentView conformsToProtocol:@protocol(RCTCustomPullToRefreshViewProtocol)]) {
@@ -548,7 +569,7 @@
   }
 }
 
-- (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
+- (void)unmountChildComponentView:(RCTUIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index // [macOS]
 {
   [childComponentView removeFromSuperview];
   if (![childComponentView conformsToProtocol:@protocol(RCTCustomPullToRefreshViewProtocol)] &&
@@ -563,17 +584,17 @@
  */
 - (BOOL)_shouldDisableScrollInteraction
 {
-  UIView *ancestorView = self.superview;
+  RCTUIView *ancestorView = (RCTUIView *)self.superview;  // [macOS]
 
   while (ancestorView) {
     if ([ancestorView respondsToSelector:@selector(isJSResponder)]) {
-      BOOL isJSResponder = ((UIView<RCTComponentViewProtocol> *)ancestorView).isJSResponder;
+      BOOL isJSResponder = ((RCTUIView<RCTComponentViewProtocol> *)ancestorView).isJSResponder; // [macOS]
       if (isJSResponder) {
         return YES;
       }
     }
 
-    ancestorView = ancestorView.superview;
+    ancestorView = (RCTUIView *)ancestorView.superview; // [macOS]
   }
 
   return NO;
@@ -607,68 +628,24 @@
   return metrics;
 }
 
-/**
- * When use of accessibility APIs is detected, view culling is disabled to make accessibility work correctly.
- */
-- (void)_disableViewCullingIfNecessary
-{
-  if (!_isAccessibilityAPIUsed) {
-    _isAccessibilityAPIUsed = YES;
-    [self _updateStateWithContentOffset];
-  }
-}
-
-- (NSInteger)accessibilityElementCount
-{
-  // From empirical testing, method `accessibilityElementCount` is called lazily only
-  // when accessibility is used.
-  // Why we don't use UIAccessibilitySwitchControlStatusDidChangeNotification and
-  // UIAccessibilityVoiceOverStatusDidChangeNotification? The notifications are not called when using Accessibility
-  // Inspector. We anticipate developers will want to debug accessbility with Accessibility Inspector and don't want to
-  // break that developer workflow with view culling. Therefore, we are using API use detection to disable view culling
-  // instead of the notifications.
-  [self _disableViewCullingIfNecessary];
-  return [super accessibilityElementCount];
-}
-
-- (NSArray<id<UIFocusItem>> *)focusItemsInRect:(CGRect)rect
-{
-  // From empirical testing, method `focusItemsInRect:` is called lazily only
-  // when keyboard navigation is used.
-  [self _disableViewCullingIfNecessary];
-  return [super focusItemsInRect:rect];
-}
-
 - (void)_updateStateWithContentOffset
 {
   if (!_state) {
     return;
   }
 
-  BOOL enableImmediateUpdateModeForContentOffsetChanges =
-      ReactNativeFeatureFlags::enableImmediateUpdateModeForContentOffsetChanges();
-
-  _avoidAdjustmentForMaintainVisibleContentPosition = enableImmediateUpdateModeForContentOffsetChanges;
-
   auto contentOffset = RCTPointFromCGPoint(_scrollView.contentOffset);
-  BOOL isAccessibilityAPIUsed = _isAccessibilityAPIUsed;
   _state->updateState(
-      [contentOffset, isAccessibilityAPIUsed](
+      [contentOffset](
           const ScrollViewShadowNode::ConcreteState::Data &oldData) -> ScrollViewShadowNode::ConcreteState::SharedData {
-        if (oldData.contentOffset == contentOffset && oldData.disableViewCulling == isAccessibilityAPIUsed) {
-          // avoid doing a state update if content offset and use of accessibility didn't change.
+        if (oldData.contentOffset == contentOffset) {
+          // avoid doing a state update if content offset didn't change.
           return nullptr;
         }
         auto newData = oldData;
         newData.contentOffset = contentOffset;
-        newData.disableViewCulling =
-            UIAccessibilityIsVoiceOverRunning() || UIAccessibilityIsSwitchControlRunning() || isAccessibilityAPIUsed;
         return std::make_shared<const ScrollViewShadowNode::ConcreteState::Data>(newData);
-      },
-      enableImmediateUpdateModeForContentOffsetChanges ? EventQueue::UpdateMode::unstable_Immediate
-                                                       : EventQueue::UpdateMode::Asynchronous);
-
-  _avoidAdjustmentForMaintainVisibleContentPosition = NO;
+      });
 }
 
 - (void)prepareForRecycle
@@ -680,12 +657,12 @@
 
   const auto &props = static_cast<const ScrollViewProps &>(*_props);
   _scrollView.contentOffset = RCTCGPointFromPoint(props.contentOffset);
-  // Reset zoom scale to default
-  _scrollView.zoomScale = 1.0;
   // We set the default behavior to "never" so that iOS
   // doesn't do weird things to UIScrollView insets automatically
   // and keeps it as an opt-in behavior.
+#if !TARGET_OS_OSX // [macOS]
   _scrollView.contentInsetAdjustmentBehavior = UIScrollViewContentInsetAdjustmentNever;
+#endif // [macOS]
   _shouldUpdateContentInsetAdjustmentBehavior = YES;
   _isUserTriggeredScrolling = NO;
   CGRect oldFrame = self.frame;
@@ -694,12 +671,11 @@
   _contentView = nil;
   _prevFirstVisibleFrame = CGRectZero;
   _firstVisibleView = nil;
-  _firstVisibleViewTag = 0;
-  _virtualViewContainerState = nil;
 }
 
 #pragma mark - UIScrollViewDelegate
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)scrollViewWillEndDragging:(UIScrollView *)scrollView
                      withVelocity:(CGPoint)velocity
               targetContentOffset:(inout CGPoint *)targetContentOffset
@@ -720,14 +696,15 @@
   static_cast<const ScrollViewEventEmitter &>(*_eventEmitter).onScrollEndDrag(metrics);
 }
 
-- (BOOL)touchesShouldCancelInContentView:(__unused UIView *)view
+- (BOOL)touchesShouldCancelInContentView:(__unused RCTPlatformView *)view // [macOS]
 {
   // Historically, `UIScrollView`s in React Native do not cancel touches
   // started on `UIControl`-based views (as normal iOS `UIScrollView`s do).
   return ![self _shouldDisableScrollInteraction];
 }
+#endif // [macOS]
 
-- (void)scrollViewDidScroll:(UIScrollView *)scrollView
+- (void)scrollViewDidScroll:(RCTUIScrollView *)scrollView // [macOS]
 {
   auto scrollMetrics = [self _scrollViewMetrics];
 
@@ -746,18 +723,18 @@
   [self _remountChildrenIfNeeded];
 }
 
-- (void)scrollViewDidZoom:(UIScrollView *)scrollView
+- (void)scrollViewDidZoom:(RCTUIScrollView *)scrollView // [macOS]
 {
   [self scrollViewDidScroll:scrollView];
 }
 
-- (BOOL)scrollViewShouldScrollToTop:(UIScrollView *)scrollView
+- (BOOL)scrollViewShouldScrollToTop:(RCTUIScrollView *)scrollView // [macOS]
 {
   _isUserTriggeredScrolling = YES;
   return YES;
 }
 
-- (void)scrollViewDidScrollToTop:(UIScrollView *)scrollView
+- (void)scrollViewDidScrollToTop:(RCTUIScrollView *)scrollView // [macOS]
 {
   if (!_eventEmitter) {
     return;
@@ -768,7 +745,7 @@
   [self _updateStateWithContentOffset];
 }
 
-- (void)scrollViewWillBeginDragging:(UIScrollView *)scrollView
+- (void)scrollViewWillBeginDragging:(RCTUIScrollView *)scrollView // [macOS]
 {
   [self _forceDispatchNextScrollEvent];
 
@@ -780,7 +757,7 @@
   _isUserTriggeredScrolling = YES;
 }
 
-- (void)scrollViewDidEndDragging:(UIScrollView *)scrollView willDecelerate:(BOOL)decelerate
+- (void)scrollViewDidEndDragging:(RCTUIScrollView *)scrollView willDecelerate:(BOOL)decelerate // [macOS]
 {
   [self _forceDispatchNextScrollEvent];
 
@@ -798,7 +775,7 @@
   }
 }
 
-- (void)scrollViewWillBeginDecelerating:(UIScrollView *)scrollView
+- (void)scrollViewWillBeginDecelerating:(RCTUIScrollView *)scrollView // [macOS]
 {
   [self _forceDispatchNextScrollEvent];
 
@@ -809,7 +786,7 @@
   static_cast<const ScrollViewEventEmitter &>(*_eventEmitter).onMomentumScrollBegin([self _scrollViewMetrics]);
 }
 
-- (void)scrollViewDidEndDecelerating:(UIScrollView *)scrollView
+- (void)scrollViewDidEndDecelerating:(RCTUIScrollView *)scrollView // [macOS]
 {
   [self _forceDispatchNextScrollEvent];
 
@@ -824,14 +801,20 @@
   RCTSendScrollEventForNativeAnimations_DEPRECATED(scrollView, self.tag, kOnScrollEndEvent);
 }
 
-- (void)scrollViewDidEndScrollingAnimation:(UIScrollView *)scrollView
+- (void)scrollViewDidEndScrollingAnimation:(RCTUIScrollView *)scrollView // [macOS]
 {
   [self _handleFinishedScrolling:scrollView];
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)didMoveToWindow
 {
   [super didMoveToWindow];
+#else // [macOS
+- (void)viewDidMoveToWindow // [macOS]
+{
+  [super viewDidMoveToWindow];
+#endif // [macOS]
 
   if (!self.window) {
     // The view is being removed, ensure that the scroll end event is dispatched
@@ -841,6 +824,7 @@
 
 - (void)_handleScrollEndIfNeeded
 {
+#if !TARGET_OS_OSX // [macOS]
   if (_scrollView.isDecelerating || !_scrollView.isTracking) {
     if (!_eventEmitter) {
       return;
@@ -850,9 +834,10 @@
     [self _updateStateWithContentOffset];
     _isUserTriggeredScrolling = NO;
   }
+#endif // [macOS]
 }
 
-- (void)_handleFinishedScrolling:(UIScrollView *)scrollView
+- (void)_handleFinishedScrolling:(RCTUIScrollView *)scrollView // [macOS]
 {
   [self _forceDispatchNextScrollEvent];
   [self scrollViewDidScroll:scrollView];
@@ -865,7 +850,7 @@
   [self _updateStateWithContentOffset];
 }
 
-- (void)scrollViewWillBeginZooming:(UIScrollView *)scrollView withView:(nullable UIView *)view
+- (void)scrollViewWillBeginZooming:(RCTUIScrollView *)scrollView withView:(nullable RCTUIView *)view // [macOS]
 {
   [self _forceDispatchNextScrollEvent];
 
@@ -876,7 +861,7 @@
   static_cast<const ScrollViewEventEmitter &>(*_eventEmitter).onScrollBeginDrag([self _scrollViewMetrics]);
 }
 
-- (void)scrollViewDidEndZooming:(UIScrollView *)scrollView withView:(nullable UIView *)view atScale:(CGFloat)scale
+- (void)scrollViewDidEndZooming:(RCTUIScrollView *)scrollView withView:(nullable RCTUIView *)view atScale:(CGFloat)scale // [macOS]
 {
   [self _forceDispatchNextScrollEvent];
 
@@ -890,7 +875,7 @@
   [self _updateStateWithContentOffset];
 }
 
-- (UIView *)viewForZoomingInScrollView:(__unused UIScrollView *)scrollView
+- (RCTPlatformView *)viewForZoomingInScrollView:(__unused RCTUIScrollView *)scrollView // [macOS]
 {
   return _containerView;
 }
@@ -911,7 +896,9 @@
 
 - (void)flashScrollIndicators
 {
+#if !TARGET_OS_OSX // [macOS]
   [_scrollView flashScrollIndicators];
+#endif // [macOS]
 }
 
 - (void)scrollTo:(double)x y:(double)y animated:(BOOL)animated
@@ -958,17 +945,13 @@
 
 #pragma mark - Child views mounting
 
-- (void)updateClippedSubviewsWithClipRect:(CGRect)clipRect relativeToView:(UIView *)clipView
+- (void)updateClippedSubviewsWithClipRect:(CGRect)clipRect relativeToView:(RCTUIView *)clipView // [macOS]
 {
   // Do nothing. ScrollView manages its subview clipping individually in `_remountChildren`.
 }
 
 - (void)_remountChildrenIfNeeded
 {
-  if (ReactNativeFeatureFlags::enableViewCulling()) {
-    return;
-  }
-
   CGPoint contentOffset = _scrollView.contentOffset;
 
   if (std::abs(_contentOffsetWhenClipped.x - contentOffset.x) < kClippingLeeway &&
@@ -983,12 +966,10 @@
 
 - (void)_remountChildren
 {
-  if (ReactNativeFeatureFlags::enableViewCulling()) {
-    return;
-  }
-
+#if !TARGET_OS_OSX // [macOS]
   [_scrollView updateClippedSubviewsWithClipRect:CGRectInset(_scrollView.bounds, -kClippingLeeway, -kClippingLeeway)
                                   relativeToView:_scrollView];
+#endif // [macOS]
 }
 
 #pragma mark - RCTScrollableProtocol
@@ -1027,15 +1008,17 @@
 
 - (void)zoomToRect:(CGRect)rect animated:(BOOL)animated
 {
+#if !TARGET_OS_OSX // [macOS]
   [_scrollView zoomToRect:rect animated:animated];
+#endif // [macOS]
 }
 
-- (void)addScrollListener:(NSObject<UIScrollViewDelegate> *)scrollListener
+- (void)addScrollListener:(NSObject<RCTUIScrollViewDelegate> *)scrollListener // [macOS]
 {
   [self.scrollViewDelegateSplitter addDelegate:scrollListener];
 }
 
-- (void)removeScrollListener:(NSObject<UIScrollViewDelegate> *)scrollListener
+- (void)removeScrollListener:(NSObject<RCTUIScrollViewDelegate> *)scrollListener // [macOS]
 {
   [self.scrollViewDelegateSplitter removeDelegate:scrollListener];
 }
@@ -1045,7 +1028,7 @@
 - (void)_prepareForMaintainVisibleScrollPosition
 {
   const auto &props = static_cast<const ScrollViewProps &>(*_props);
-  if (!props.maintainVisibleContentPosition || _avoidAdjustmentForMaintainVisibleContentPosition) {
+  if (!props.maintainVisibleContentPosition) {
     return;
   }
 
@@ -1053,7 +1036,7 @@
   int minIdx = props.maintainVisibleContentPosition.value().minIndexForVisible;
   for (NSUInteger ii = minIdx; ii < _contentView.subviews.count; ++ii) {
     // Find the first view that is partially or fully visible.
-    UIView *subview = _contentView.subviews[ii];
+    RCTPlatformView *subview = _contentView.subviews[ii]; // [macOS]
     BOOL hasNewView = NO;
     if (horizontal) {
       hasNewView = subview.frame.origin.x + subview.frame.size.width > _scrollView.contentOffset.x;
@@ -1063,7 +1046,6 @@
     if (hasNewView || ii == _contentView.subviews.count - 1) {
       _prevFirstVisibleFrame = subview.frame;
       _firstVisibleView = subview;
-      _firstVisibleViewTag = subview.tag;
       break;
     }
   }
@@ -1072,17 +1054,10 @@
 - (void)_adjustForMaintainVisibleContentPosition
 {
   const auto &props = static_cast<const ScrollViewProps &>(*_props);
-  if (!props.maintainVisibleContentPosition || _avoidAdjustmentForMaintainVisibleContentPosition) {
+  if (!props.maintainVisibleContentPosition) {
     return;
   }
 
-  if (ReactNativeFeatureFlags::enableViewCulling()) {
-    // Abort if the first visible view has changed (different tag)
-    if (_firstVisibleView && _firstVisibleView.tag != _firstVisibleViewTag) {
-      return;
-    }
-  }
-
   std::optional<int> autoscrollThreshold = props.maintainVisibleContentPosition.value().autoscrollToTopThreshold;
   BOOL horizontal = _scrollView.contentSize.width > self.frame.size.width;
   // TODO: detect and handle/ignore re-ordering
@@ -1116,16 +1091,6 @@
   }
 }
 
-#pragma mark - RCTVirtualViewContainerProtocol
-
-- (RCTVirtualViewContainerState *)virtualViewContainerState
-{
-  if (!_virtualViewContainerState) {
-    _virtualViewContainerState = [[RCTVirtualViewContainerState alloc] initWithScrollView:self];
-  }
-  return _virtualViewContainerState;
-}
-
 @end
 
 Class<RCTComponentViewProtocol> RCTScrollViewCls(void)
