--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTEnhancedScrollView.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.cfbaNDxhMW	2026-03-06 18:50:50
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTGenericDelegateSplitter.h>
 #import <React/RCTViewComponentView.h>
@@ -19,7 +19,7 @@
  */
 @protocol RCTEnhancedScrollViewOverridingDelegate <NSObject>
 
-- (BOOL)touchesShouldCancelInContentView:(UIView *)view;
+- (BOOL)touchesShouldCancelInContentView:(RCTPlatformView *)view; // [macOS]
 
 @end
 
@@ -27,7 +27,7 @@
  * `UIScrollView` subclass which has some improvements and tweaks
  * which are not directly related to React Native.
  */
-@interface RCTEnhancedScrollView : UIScrollView
+@interface RCTEnhancedScrollView : RCTUIScrollView // [macOS]
 
 /*
  * Returns a delegate splitter that can be used to create as many `UIScrollView` delegates as needed.
@@ -40,7 +40,7 @@
  * resilient to other code as possible: even if something else nil the delegate, other delegates that were subscribed
  * via the splitter will continue working.
  */
-@property (nonatomic, strong, readonly) RCTGenericDelegateSplitter<id<UIScrollViewDelegate>> *delegateSplitter;
+@property (nonatomic, strong, readonly) RCTGenericDelegateSplitter<id<RCTUIScrollViewDelegate>> *delegateSplitter; // [macOS]
 
 @property (nonatomic, weak) id<RCTEnhancedScrollViewOverridingDelegate> overridingDelegate;
 @property (nonatomic, assign) BOOL pinchGestureEnabled;
