--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/RCTBridgeProxy.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Base/RCTBridgeProxy.mm	2026-03-06 15:22:52
@@ -53,7 +53,7 @@
                        launchOptions:(nullable NSDictionary *)launchOptions
 {
   self = [super self];
-  if (self != nullptr) {
+  if (self) {
     _uiManagerProxy = [[RCTUIManagerProxy alloc] initWithViewRegistry:viewRegistry];
     _moduleRegistry = moduleRegistry;
     _bundleManager = bundleManager;
@@ -75,7 +75,7 @@
 
   if (queue == RCTJSThread) {
     _dispatchToJSThread(block);
-  } else if (queue != nullptr) {
+  } else if (queue) {
     dispatch_async(queue, block);
   }
 }
@@ -308,7 +308,7 @@
            cmd:_cmd];
 }
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
+#ifndef RCT_FIT_RM_OLD_RUNTIME
 - (void)registerModuleForFrameUpdates:(id<RCTBridgeModule>)module withModuleData:(RCTModuleData *)moduleData
 {
   [self logError:@"This method is not supported. Nooping" cmd:_cmd];
@@ -319,7 +319,7 @@
   [self logError:@"This method is not supported. Returning nil." cmd:_cmd];
   return nil;
 }
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // RCT_FIT_RM_OLD_RUNTIME
 
 - (void)registerAdditionalModuleClasses:(NSArray<Class> *)newModules
 {
@@ -418,7 +418,7 @@
 
 @implementation RCTUIManagerProxy {
   RCTViewRegistry *_viewRegistry;
-  NSMutableDictionary<NSNumber *, UIView *> *_legacyViewRegistry;
+  NSMutableDictionary<NSNumber *, RCTPlatformView *> *_legacyViewRegistry; // [macOS]
 }
 
 #pragma clang diagnostic push
@@ -427,7 +427,7 @@
 - (instancetype)initWithViewRegistry:(RCTViewRegistry *)viewRegistry
 {
   self = [super self];
-  if (self != nullptr) {
+  if (self) {
     _viewRegistry = viewRegistry;
     _legacyViewRegistry = [NSMutableDictionary new];
   }
@@ -439,13 +439,13 @@
 /**
  * RCTViewRegistry
  */
-- (UIView *)viewForReactTag:(NSNumber *)reactTag
+- (RCTPlatformView *)viewForReactTag:(NSNumber *)reactTag // [macOS]
 {
   [self logWarning:@"Please migrate to RCTViewRegistry: @synthesize viewRegistry_DEPRECATED = _viewRegistry_DEPRECATED."
                cmd:_cmd];
-  UIView *view = ([_viewRegistry viewForReactTag:reactTag] != nullptr) ? [_viewRegistry viewForReactTag:reactTag]
-                                                                       : [_legacyViewRegistry objectForKey:reactTag];
-  return RCTPaperViewOrCurrentView(view);
+  RCTPlatformView *view = [_viewRegistry viewForReactTag:reactTag] ? [_viewRegistry viewForReactTag:reactTag] // [macOS]
+                                                          : [_legacyViewRegistry objectForKey:reactTag];
+  return [RCTUIManager paperViewOrCurrentView:view];
 }
 
 - (void)addUIBlock:(RCTViewManagerUIBlock)block
@@ -457,7 +457,7 @@
   __weak __typeof(self) weakSelf = self;
   RCTExecuteOnMainQueue(^{
     __typeof(self) strongSelf = weakSelf;
-    if (strongSelf != nullptr) {
+    if (strongSelf) {
       RCTUIManager *proxiedManager = (RCTUIManager *)strongSelf;
       RCTComposedViewRegistry *composedViewRegistry =
           [[RCTComposedViewRegistry alloc] initWithUIManager:proxiedManager
