--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/RCTConstants.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Base/RCTConstants.h	2026-03-06 15:22:52
@@ -10,7 +10,11 @@
 RCT_EXTERN NSString *const RCTPlatformName;
 
 RCT_EXTERN NSString *const RCTUserInterfaceStyleDidChangeNotification;
+#if !TARGET_OS_OSX // [macOS]
 RCT_EXTERN NSString *const RCTUserInterfaceStyleDidChangeNotificationTraitCollectionKey;
+#else // [macOS
+RCT_EXTERN NSString *const RCTUserInterfaceStyleDidChangeNotificationAppearanceKey;
+#endif // macOS]
 
 RCT_EXTERN NSString *const RCTWindowFrameDidChangeNotification;
 
