--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/RCTConvert.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Base/RCTConvert.mm	2026-03-06 15:22:52
@@ -12,6 +12,7 @@
 #import <CoreText/CoreText.h>
 
 #import "RCTDefines.h"
+#import "RCTHandledKey.h" // [macOS]
 #import "RCTImageSource.h"
 #import "RCTParserUtils.h"
 #import "RCTUtils.h"
@@ -394,6 +395,7 @@
   }
 }
 
+#if !TARGET_OS_OSX // [macOS]
 RCT_ENUM_CONVERTER(
     UITextAutocapitalizationType,
     (@{
@@ -499,7 +501,6 @@
     UIUserInterfaceStyleUnspecified,
     integerValue)
 
-#if !TARGET_OS_TV
 RCT_ENUM_CONVERTER(
     UIInterfaceOrientationMask,
     (@{
@@ -511,7 +512,6 @@
     }),
     NSNotFound,
     unsignedIntegerValue)
-#endif
 
 RCT_ENUM_CONVERTER(
     UIModalPresentationStyle,
@@ -547,16 +547,71 @@
     }),
     UIViewContentModeScaleAspectFill,
     integerValue)
+#endif // [macOS]
 
 RCT_ENUM_CONVERTER(
     RCTCursor,
     (@{
       @"auto" : @(RCTCursorAuto),
+      @"alias" : @(RCTCursorAlias),
+      @"all-scroll" : @(RCTCursorAllScroll),
+      @"cell" : @(RCTCursorCell),
+      @"col-resize" : @(RCTCursorColResize),
+      @"context-menu" : @(RCTCursorContextMenu),
+      @"copy" : @(RCTCursorCopy),
+      @"crosshair" : @(RCTCursorCrosshair),
+      @"default" : @(RCTCursorDefault),
+      @"e-resize" : @(RCTCursorEResize),
+      @"ew-resize" : @(RCTCursorEWResize),
+      @"grab" : @(RCTCursorGrab),
+      @"grabbing" : @(RCTCursorGrabbing),
+      @"help" : @(RCTCursorHelp),
+      @"move" : @(RCTCursorMove),
+      @"ne-resize" : @(RCTCursorNEResize),
+      @"nesw-resize" : @(RCTCursorNESWResize),
+      @"n-resize" : @(RCTCursorNResize),
+      @"ns-resize" : @(RCTCursorNSResize),
+      @"nw-resize" : @(RCTCursorNWResize),
+      @"nwse-resize" : @(RCTCursorNWSEResize),
+      @"no-drop" : @(RCTCursorNoDrop),
+      @"none" : @(RCTCursorNone),
+      @"not-allowed" : @(RCTCursorNotAllowed),
       @"pointer" : @(RCTCursorPointer),
+      @"progress" : @(RCTCursorProgress),
+      @"row-resize" : @(RCTCursorRowResize),
+      @"s-resize" : @(RCTCursorSResize),
+      @"se-resize" : @(RCTCursorSEResize),
+      @"sw-resize" : @(RCTCursorSWResize),
+      @"text" : @(RCTCursorText),
+      @"url" : @(RCTCursorUrl),
+      @"vertical-text" : @(RCTCursorVerticalText),
+      @"w-resize" : @(RCTCursorWResize),
+      @"wait" : @(RCTCursorWait),
+      @"zoom-in" : @(RCTCursorZoomIn),
+      @"zoom-out" : @(RCTCursorZoomOut),
     }),
     RCTCursorAuto,
     integerValue)
+// macOS]
 
+#if TARGET_OS_OSX // [macOS
+RCT_MULTI_ENUM_CONVERTER(NSTextCheckingTypes, (@{
+  @"ortography": @(NSTextCheckingTypeOrthography),
+  @"spelling": @(NSTextCheckingTypeSpelling),
+  @"grammar": @(NSTextCheckingTypeGrammar),
+  @"calendarEvent": @(NSTextCheckingTypeDate),
+  @"address": @(NSTextCheckingTypeAddress),
+  @"link": @(NSTextCheckingTypeLink),
+  @"quote": @(NSTextCheckingTypeQuote),
+  @"dash": @(NSTextCheckingTypeDash),
+  @"replacement": @(NSTextCheckingTypeReplacement),
+  @"correction": @(NSTextCheckingTypeCorrection),
+  @"regularExpression": @(NSTextCheckingTypeRegularExpression),
+  @"phoneNumber": @(NSTextCheckingTypePhoneNumber),
+  @"transitInformation": @(NSTextCheckingTypeTransitInformation),
+}), NSTextCheckingTypeOrthography, unsignedLongLongValue)
+#endif // macOS]
+
 static void convertCGStruct(const char *type, NSArray *fields, CGFloat *result, id json)
 {
   NSUInteger count = fields.count;
@@ -664,6 +719,7 @@
   static NSDictionary<NSString *, NSDictionary *> *colorMap = nil;
   if (colorMap == nil) {
     NSMutableDictionary<NSString *, NSDictionary *> *map = [@{
+#if !TARGET_OS_OSX // [macOS]
       // https://developer.apple.com/documentation/uikit/uicolor/ui_element_colors
       // Label Colors
       @"labelColor" : @{
@@ -795,6 +851,88 @@
         // iOS 13.0
         RCTFallbackARGB : @(0x00000000)
       },
+#else // [macOS
+      // https://developer.apple.com/documentation/appkit/nscolor/ui_element_colors
+      // Label Colors
+      @"labelColor": @{}, // 10_10
+      @"secondaryLabelColor": @{}, // 10_10
+      @"tertiaryLabelColor": @{}, // 10_10
+      @"quaternaryLabelColor": @{}, // 10_10
+      // Text Colors
+      @"textColor": @{},
+      @"placeholderTextColor": @{}, // 10_10
+      @"selectedTextColor": @{},
+      @"textBackgroundColor": @{},
+      @"selectedTextBackgroundColor": @{},
+      @"keyboardFocusIndicatorColor": @{},
+      @"unemphasizedSelectedTextColor": @{ // 10_14
+        RCTFallback: @"selectedTextColor"
+      },
+      @"unemphasizedSelectedTextBackgroundColor": @{ // 10_14
+        RCTFallback: @"textBackgroundColor"
+      },
+      // Content Colors
+      @"linkColor": @{}, // 10_10
+      @"separatorColor": @{ // 10_14: Replacement for +controlHighlightColor, +controlLightHighlightColor, +controlShadowColor, +controlDarkShadowColor
+        RCTFallback: @"gridColor"
+      },
+      @"selectedContentBackgroundColor": @{ // 10_14: Alias for +alternateSelectedControlColor
+        RCTFallback: @"alternateSelectedControlColor"
+      },
+      @"unemphasizedSelectedContentBackgroundColor": @{ // 10_14: Alias for +secondarySelectedControlColor
+        RCTFallback: @"secondarySelectedControlColor"
+      },
+      // Menu Colors
+      @"selectedMenuItemTextColor": @{},
+      // Table Colors
+      @"gridColor": @{},
+      @"headerTextColor": @{},
+      @"alternatingEvenContentBackgroundColor": @{ // 10_14: Alias for +controlAlternatingRowBackgroundColors
+        RCTSelector: @"alternatingContentBackgroundColors",
+        RCTIndex: @0,
+        RCTFallback: @"controlAlternatingRowBackgroundColors"
+      },
+      @"alternatingOddContentBackgroundColor": @{ // 10_14: Alias for +controlAlternatingRowBackgroundColors
+        RCTSelector: @"alternatingContentBackgroundColors",
+        RCTIndex: @1,
+        RCTFallback: @"controlAlternatingRowBackgroundColors"
+      },
+      // Control Colors
+      @"controlAccentColor": @{ // 10_14
+        RCTFallback: @"controlColor"
+      },
+      @"controlColor": @{},
+      @"controlBackgroundColor": @{},
+      @"controlTextColor": @{},
+      @"disabledControlTextColor": @{},
+      @"selectedControlColor": @{},
+      @"selectedControlTextColor": @{},
+      @"alternateSelectedControlTextColor": @{},
+      @"scrubberTexturedBackgroundColor": @{}, // 10_12_2
+      // Window Colors
+      @"windowBackgroundColor": @{},
+      @"windowFrameTextColor": @{},
+      @"underPageBackgroundColor": @{}, // 10_8
+      // Highlights and Shadows
+      @"findHighlightColor": @{ // 10_13
+        RCTFallback: @"highlightColor"
+      },
+      @"highlightColor": @{},
+      @"shadowColor": @{},
+      // https://developer.apple.com/documentation/appkit/nscolor/standard_colors
+      // Standard Colors
+      @"systemBlueColor": @{},   // 10_10
+      @"systemBrownColor": @{},  // 10_10
+      @"systemGrayColor": @{},   // 10_10
+      @"systemGreenColor": @{},  // 10_10
+      @"systemOrangeColor": @{}, // 10_10
+      @"systemPinkColor": @{},   // 10_10
+      @"systemPurpleColor": @{}, // 10_10
+      @"systemRedColor": @{},    // 10_10
+      @"systemYellowColor": @{}, // 10_10
+      // Transparent Color
+      @"clearColor" : @{},
+#endif // macOS]
     } mutableCopy];
     // The color names are the Objective-C UIColor selector names,
     // but Swift selector names are valid as well, so make aliases.
@@ -804,9 +942,10 @@
       RCTAssert(
           [objcSelector hasSuffix:RCTColorSuffix], @"A selector in the color map did not end with the suffix Color.");
       NSMutableDictionary *entry = [map[objcSelector] mutableCopy];
-      RCTAssert([entry objectForKey:RCTSelector] == nil, @"Entry should not already have an RCTSelector");
+      if ([entry objectForKey:RCTSelector] == nil) {
+        entry[RCTSelector] = objcSelector;
+      }
       NSString *swiftSelector = [objcSelector substringToIndex:[objcSelector length] - [RCTColorSuffix length]];
-      entry[RCTSelector] = objcSelector;
       aliases[swiftSelector] = entry;
     }
     [map addEntriesFromDictionary:aliases];
@@ -833,13 +972,14 @@
   return colorMap;
 }
 
+// [macOS
 /** Returns a UIColor based on a semantic color name.
  *  Returns nil if the semantic color name is invalid.
  */
-static UIColor *RCTColorFromSemanticColorName(NSString *semanticColorName)
+static RCTUIColor *RCTColorFromSemanticColorName(NSString *semanticColorName)
 {
   NSDictionary<NSString *, NSDictionary *> *colorMap = RCTSemanticColorsMap();
-  UIColor *color = nil;
+  RCTUIColor *color = nil;
   NSDictionary<NSString *, id> *colorInfo = colorMap[semanticColorName];
   if (colorInfo) {
     NSString *semanticColorSelector = colorInfo[RCTSelector];
@@ -847,7 +987,7 @@
       semanticColorSelector = semanticColorName;
     }
     SEL selector = NSSelectorFromString(semanticColorSelector);
-    if (![UIColor respondsToSelector:selector]) {
+    if (![RCTUIColor respondsToSelector:selector]) {
       NSNumber *fallbackRGB = colorInfo[RCTFallbackARGB];
       if (fallbackRGB != nil) {
         RCTAssert([fallbackRGB isKindOfClass:[NSNumber class]], @"fallback ARGB is not a number");
@@ -856,12 +996,12 @@
       semanticColorSelector = colorInfo[RCTFallback];
       selector = NSSelectorFromString(semanticColorSelector);
     }
-    RCTAssert([UIColor respondsToSelector:selector], @"RCTUIColor does not respond to a semantic color selector.");
-    Class klass = [UIColor class];
+    RCTAssert ([RCTUIColor respondsToSelector:selector], @"RCTUIColor does not respond to a semantic color selector.");
+    Class klass = [RCTUIColor class];
     IMP imp = [klass methodForSelector:selector];
-    id (*getSemanticColorObject)(id, SEL) = (id (*)(id, SEL))imp;
+    id (*getSemanticColorObject)(id, SEL) = (id(*)(id, SEL))imp;
     id colorObject = getSemanticColorObject(klass, selector);
-    if ([colorObject isKindOfClass:[UIColor class]]) {
+    if ([colorObject isKindOfClass:[RCTUIColor class]]) {
       color = colorObject;
     } else if ([colorObject isKindOfClass:[NSArray class]]) {
       NSArray *colors = colorObject;
@@ -874,6 +1014,7 @@
   }
   return color;
 }
+// macOS]
 
 /** Returns an alphabetically sorted comma separated list of the valid semantic color names
  */
@@ -881,10 +1022,9 @@
 {
   NSMutableString *names = [NSMutableString new];
   NSDictionary<NSString *, NSDictionary *> *colorMap = RCTSemanticColorsMap();
-  NSArray *allKeys =
-      [[[colorMap allKeys] mutableCopy] sortedArrayUsingSelector:@selector(localizedCaseInsensitiveCompare:)];
+  NSArray *allKeys = [[[colorMap allKeys] mutableCopy] sortedArrayUsingSelector:@selector(localizedCaseInsensitiveCompare:)];
 
-  for (id key in allKeys) {
+  for(id key in allKeys) {
     if ([names length]) {
       [names appendString:@", "];
     }
@@ -908,21 +1048,21 @@
   _defaultColorSpace = colorSpace;
 }
 
-+ (UIColor *)UIColorWithRed:(CGFloat)red green:(CGFloat)green blue:(CGFloat)blue alpha:(CGFloat)alpha
++ (RCTUIColor *)UIColorWithRed:(CGFloat)red green:(CGFloat)green blue:(CGFloat)blue alpha:(CGFloat)alpha // [macOS]
 {
   RCTColorSpace space = RCTGetDefaultColorSpace();
   return [self UIColorWithRed:red green:green blue:blue alpha:alpha andColorSpace:space];
 }
-+ (UIColor *)UIColorWithRed:(CGFloat)red
-                      green:(CGFloat)green
-                       blue:(CGFloat)blue
-                      alpha:(CGFloat)alpha
-              andColorSpace:(RCTColorSpace)colorSpace
++ (RCTUIColor *)UIColorWithRed:(CGFloat)red // [macOS]
+                        green:(CGFloat)green
+                          blue:(CGFloat)blue
+                         alpha:(CGFloat)alpha
+                 andColorSpace:(RCTColorSpace)colorSpace
 {
   if (colorSpace == RCTColorSpaceDisplayP3) {
-    return [UIColor colorWithDisplayP3Red:red green:green blue:blue alpha:alpha];
+    return [RCTUIColor colorWithDisplayP3Red:red green:green blue:blue alpha:alpha]; // [macOS]
   }
-  return [UIColor colorWithRed:red green:green blue:blue alpha:alpha];
+  return [RCTUIColor colorWithRed:red green:green blue:blue alpha:alpha]; // [macOS]
 }
 
 + (RCTColorSpace)RCTColorSpaceFromString:(NSString *)colorSpace
@@ -935,7 +1075,34 @@
   return RCTGetDefaultColorSpace();
 }
 
-+ (UIColor *)UIColor:(id)json
+// [macOS
+#if TARGET_OS_OSX
++ (NSColor *)NSColor:(id)json
+{
+  return [RCTConvert UIColor:json];
+}
+
+static NSColor *RCTColorWithSystemEffect(NSColor* color, NSString *systemEffectString) {
+    NSColor *colorWithEffect = color;
+    if (systemEffectString != nil) {
+        if ([systemEffectString isEqualToString:@"none"]) {
+            colorWithEffect = [color colorWithSystemEffect:NSColorSystemEffectNone];
+        } else if ([systemEffectString isEqualToString:@"pressed"]) {
+            colorWithEffect = [color colorWithSystemEffect:NSColorSystemEffectPressed];
+        } else if ([systemEffectString isEqualToString:@"deepPressed"]) {
+            colorWithEffect = [color colorWithSystemEffect:NSColorSystemEffectDeepPressed];
+        } else if ([systemEffectString isEqualToString:@"disabled"]) {
+            colorWithEffect = [color colorWithSystemEffect:NSColorSystemEffectDisabled];
+        } else if ([systemEffectString isEqualToString:@"rollover"]) {
+            colorWithEffect = [color colorWithSystemEffect:NSColorSystemEffectRollover];
+        }
+    }
+    return colorWithEffect;
+}
+#endif // TARGET_OS_OSX
+// macOS]
+
++ (RCTUIColor *)UIColor:(id)json // [macOS]
 {
   if (!json) {
     return nil;
@@ -968,7 +1135,7 @@
     } else if ((value = [dictionary objectForKey:@"semantic"])) {
       if ([value isKindOfClass:[NSString class]]) {
         NSString *semanticName = value;
-        UIColor *color = [UIColor colorNamed:semanticName];
+        RCTUIColor *color = [RCTUIColor colorNamed:semanticName]; // [macOS]
         if (color != nil) {
           return color;
         }
@@ -981,7 +1148,7 @@
         return color;
       } else if ([value isKindOfClass:[NSArray class]]) {
         for (id name in value) {
-          UIColor *color = [UIColor colorNamed:name];
+          RCTUIColor *color = [RCTUIColor colorNamed:name]; // [macOS]
           if (color != nil) {
             return color;
           }
@@ -1002,14 +1169,15 @@
     } else if ((value = [dictionary objectForKey:@"dynamic"])) {
       NSDictionary *appearances = value;
       id light = [appearances objectForKey:@"light"];
-      UIColor *lightColor = [RCTConvert UIColor:light];
+      RCTUIColor *lightColor = [RCTConvert UIColor:light];
       id dark = [appearances objectForKey:@"dark"];
-      UIColor *darkColor = [RCTConvert UIColor:dark];
+      RCTUIColor *darkColor = [RCTConvert UIColor:dark]; // [macOS]
       id highContrastLight = [appearances objectForKey:@"highContrastLight"];
-      UIColor *highContrastLightColor = [RCTConvert UIColor:highContrastLight];
+      RCTUIColor *highContrastLightColor = [RCTConvert UIColor:highContrastLight]; // [macOS]
       id highContrastDark = [appearances objectForKey:@"highContrastDark"];
-      UIColor *highContrastDarkColor = [RCTConvert UIColor:highContrastDark];
+      RCTUIColor *highContrastDarkColor = [RCTConvert UIColor:highContrastDark]; // [macOS]
       if (lightColor != nil && darkColor != nil) {
+#if !TARGET_OS_OSX // [macOS]
         UIColor *color = [UIColor colorWithDynamicProvider:^UIColor *_Nonnull(UITraitCollection *_Nonnull collection) {
           if (collection.userInterfaceStyle == UIUserInterfaceStyleDark) {
             if (collection.accessibilityContrast == UIAccessibilityContrastHigh && highContrastDarkColor != nil) {
@@ -1026,13 +1194,52 @@
           }
         }];
         return color;
+#else // [macOS
+        NSColor *color = [NSColor colorWithName:nil dynamicProvider:^NSColor * _Nonnull(NSAppearance * _Nonnull appearance) {
+          NSMutableArray<NSAppearanceName> *appearances = [NSMutableArray arrayWithArray:@[NSAppearanceNameAqua,NSAppearanceNameDarkAqua]];
+          if (highContrastLightColor != nil) {
+            [appearances addObject:NSAppearanceNameAccessibilityHighContrastAqua];
+          }
+          if (highContrastDarkColor != nil) {
+            [appearances addObject:NSAppearanceNameAccessibilityHighContrastDarkAqua];
+          }
+          NSAppearanceName bestMatchingAppearance = [appearance bestMatchFromAppearancesWithNames:appearances];
+          if (bestMatchingAppearance == NSAppearanceNameAqua) {
+            return lightColor;
+          } else if (bestMatchingAppearance == NSAppearanceNameDarkAqua) {
+            return darkColor;
+          } else if (bestMatchingAppearance == NSAppearanceNameAccessibilityHighContrastAqua) {
+            return highContrastLightColor;
+          } else if (bestMatchingAppearance == NSAppearanceNameAccessibilityHighContrastDarkAqua) {
+            return highContrastDarkColor;
+          } else {
+            RCTLogWarn(@"DynamicColorMacOS: Could not resolve current appearance. Defaulting to light.");
+            return lightColor;
+          }
+        }];
+        return color;
+#endif // macOS]
 
       } else {
-        RCTLogConvertError(json, @"a UIColor. Expected an iOS dynamic appearance aware color.");
+        RCTLogConvertError(json, @"a UIColor. Expected an apple dynamic appearance aware color."); // [macOS]
         return nil;
       }
+#if TARGET_OS_OSX // [macOS
+    } else if((value = [dictionary objectForKey:@"colorWithSystemEffect"])) {
+        NSDictionary *colorWithSystemEffect = value;
+        id base = [colorWithSystemEffect objectForKey:@"baseColor"];
+        NSColor *baseColor = [RCTConvert UIColor:base];
+        NSString * systemEffectString = [colorWithSystemEffect objectForKey:@"systemEffect"];
+        if (baseColor != nil && systemEffectString != nil) {
+            return RCTColorWithSystemEffect(baseColor, systemEffectString);
+        } else {
+            RCTLogConvertError(
+                json, @"a UIColor.  Expected a color with a system effect string, but got something else");
+            return nil;
+        }
+#endif // macOS]
     } else {
-      RCTLogConvertError(json, @"a UIColor. Expected an iOS semantic color or dynamic appearance aware color.");
+      RCTLogConvertError(json, @"a UIColor. Expected an apple semantic color, dynamic appearance aware color, or color with system effect"); // [macOS]
       return nil;
     }
   } else {
@@ -1075,7 +1282,7 @@
   __block BOOL copy = NO;
   __block NSArray *values = json = [RCTConvert NSArray:json];
   [json enumerateObjectsUsingBlock:^(id jsonValue, NSUInteger idx, __unused BOOL *stop) {
-    id value = ((id (*)(Class, SEL, id))objc_msgSend)([RCTConvert class], type, jsonValue);
+    id value = ((id(*)(Class, SEL, id))objc_msgSend)([RCTConvert class], type, jsonValue);
     if (copy) {
       if (value) {
         [(NSMutableArray *)values addObject:value];
@@ -1097,7 +1304,7 @@
 
 RCT_ARRAY_CONVERTER(NSURL)
 RCT_ARRAY_CONVERTER(RCTFileURL)
-RCT_ARRAY_CONVERTER(UIColor)
+RCT_ARRAY_CONVERTER(RCTUIColor) // [macOS]
 
 /**
  * This macro is used for creating converter functions for directly
@@ -1130,6 +1337,39 @@
   return colors;
 }
 
+#if TARGET_OS_OSX // [macOS
++ (NSArray<NSPasteboardType> *)NSPasteboardType:(id)json
+{
+  NSString *type = [self NSString:json];
+  if (!type) {
+    return @[];
+  }
+  
+  if ([type isEqualToString:@"fileUrl"]) {
+    return @[NSFilenamesPboardType];
+  } else if ([type isEqualToString:@"image"]) {
+    return @[NSPasteboardTypePNG, NSPasteboardTypeTIFF];
+  } else if ([type isEqualToString:@"string"]) {
+    return @[NSPasteboardTypeString];
+  }
+  return @[];
+}
+
++ (NSArray<NSPasteboardType> *)NSPasteboardTypeArray:(id)json
+{
+  if ([json isKindOfClass:[NSString class]]) {
+    return [RCTConvert NSPasteboardType:json];
+  } else if ([json isKindOfClass:[NSArray class]]) {
+    NSMutableArray *mutablePasteboardTypes = [NSMutableArray new];
+    for (NSString *type in json) {
+      [mutablePasteboardTypes addObjectsFromArray:[RCTConvert NSPasteboardType:type]];
+    }
+    return mutablePasteboardTypes.copy;
+  }
+  return @[];
+}
+#endif // macOS]
+
 static id RCTConvertPropertyListValue(id json)
 {
   if (!json || json == (id)kCFNull) {
@@ -1279,16 +1519,189 @@
 RCT_ENUM_CONVERTER(
     RCTAnimationType,
     (@{
+#if !TARGET_OS_OSX // [macOS]
       @"spring" : @(RCTAnimationTypeSpring),
+#endif // [macOS]
       @"linear" : @(RCTAnimationTypeLinear),
       @"easeIn" : @(RCTAnimationTypeEaseIn),
       @"easeOut" : @(RCTAnimationTypeEaseOut),
       @"easeInEaseOut" : @(RCTAnimationTypeEaseInEaseOut),
+#if !TARGET_OS_OSX // [macOS]
       @"keyboard" : @(RCTAnimationTypeKeyboard),
+#endif // [macOS]
     }),
     RCTAnimationTypeEaseInEaseOut,
     integerValue)
+
+#if TARGET_OS_OSX // [macOS
+// This is for the role prop & has slightly different mappings than the
+// old accessibilityRole prop for back compatability. `role` matches ARIA Core
+// AAM spec and takes precedence. See https://www.w3.org/TR/core-aam-1.1/
++ (NSString*) accessibilityRoleFromAriaRole:(NSString*)ariaRole
+{
+  // rowgroup is explicitly not mapped
+  if ([ariaRole isEqualToString:@"rowgroup"]) {
+      return nil;
+  }
+  static NSDictionary<NSString *, NSAccessibilityRole> * ariaRoleToNSAccessibilityRole;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    ariaRoleToNSAccessibilityRole = @{
+      @"alert": NSAccessibilityGroupRole,
+      @"alertdialog": NSAccessibilityGroupRole,
+      @"application": NSAccessibilityGroupRole,
+      @"article": NSAccessibilityGroupRole,
+      @"banner": NSAccessibilityGroupRole,
+      @"button": NSAccessibilityButtonRole,
+      @"cell": NSAccessibilityCellRole,
+      @"checkbox": NSAccessibilityCheckBoxRole,
+      @"columnheader": NSAccessibilityCellRole,
+      @"combobox": NSAccessibilityComboBoxRole,
+      @"complementary": NSAccessibilityGroupRole,
+      @"contentinfo": NSAccessibilityGroupRole,
+      @"definition": NSAccessibilityGroupRole,
+      @"dialog": NSAccessibilityGroupRole,
+      @"directory": NSAccessibilityListRole,
+      @"document": NSAccessibilityGroupRole,
+      @"feed": NSAccessibilityGroupRole,
+      @"figure": NSAccessibilityGroupRole,
+      @"form": NSAccessibilityGroupRole,
+      @"grid": NSAccessibilityTableRole,
+      @"gridcell": NSAccessibilityCellRole,
+      @"group": NSAccessibilityGroupRole,
+      @"heading": NSAccessibilityStaticTextRole,
+      @"image": NSAccessibilityImageRole,
+      @"img": NSAccessibilityImageRole,
+      @"link": NSAccessibilityLinkRole,
+      @"list": NSAccessibilityListRole,
+      @"listbox": NSAccessibilityListRole,
+      @"listitem": NSAccessibilityGroupRole,
+      @"log": NSAccessibilityGroupRole,
+      @"main": NSAccessibilityGroupRole,
+      @"marquee": NSAccessibilityGroupRole,
+      @"math": NSAccessibilityGroupRole,
+      @"menu": NSAccessibilityMenuRole,
+      @"menubar": NSAccessibilityMenuBarRole,
+      @"menuitem": NSAccessibilityMenuItemRole,
+      @"menuitemcheckbox": NSAccessibilityMenuItemRole,
+      @"menuitemradio": NSAccessibilityMenuItemRole,
+      @"meter": NSAccessibilityLevelIndicatorRole,
+      @"navigation": NSAccessibilityGroupRole,
+      @"none": NSAccessibilityGroupRole,
+      @"note": NSAccessibilityGroupRole,
+      @"option": NSAccessibilityStaticTextRole,
+      @"presentation": NSAccessibilityGroupRole,
+      @"progressbar": NSAccessibilityProgressIndicatorRole,
+      @"radio": NSAccessibilityRadioButtonRole,
+      @"radiogroup": NSAccessibilityRadioGroupRole,
+      @"region": NSAccessibilityGroupRole,
+      @"row": NSAccessibilityRowRole,
+      @"rowheader": NSAccessibilityCellRole,
+      @"scrollbar": NSAccessibilityScrollBarRole,
+      @"search": NSAccessibilityGroupRole,
+      @"searchbox": NSAccessibilityTextFieldRole,
+      @"separator": NSAccessibilitySplitterRole,
+      @"slider": NSAccessibilitySliderRole,
+      @"spinbutton": NSAccessibilityIncrementorRole,
+      @"status": NSAccessibilityGroupRole,
+      @"switch": NSAccessibilityCheckBoxRole,
+      @"tab": NSAccessibilityRadioButtonRole,
+      @"table": NSAccessibilityTableRole,
+      @"tablist": NSAccessibilityTabGroupRole,
+      @"tabpanel": NSAccessibilityGroupRole,
+      @"term": NSAccessibilityGroupRole,
+      @"textbox": NSAccessibilityTextFieldRole,
+      @"timer": NSAccessibilityGroupRole,
+      @"toolbar": NSAccessibilityToolbarRole,
+      @"tooltip": NSAccessibilityGroupRole,
+      @"tree": NSAccessibilityOutlineRole,
+      @"treegrid": NSAccessibilityTableRole,
+      @"treeitem": NSAccessibilityRowRole,
+    };
+  });
+  NSAccessibilityRole nsRole = [ariaRoleToNSAccessibilityRole valueForKey: ariaRole];
+  if (nsRole == nil) {
+    // Fall back to legacy mappings if an aria mapping is not found. This would
+    // include macOS specific roles like disclosure and legacy accessibilityTrait
+    // based mappings like adjustable
+    nsRole = [RCTConvert accessibilityRoleFromTrait:ariaRole];
+  }
+  return nsRole;
+}
 
+// This function is for accessibilityRole & has slightly different mappings 
+// than the new role prop for back compatability. role matches ARIA spec and
+// takes precedence.
++ (NSString*)accessibilityRoleFromTrait:(NSString*)trait
+{
+  static NSDictionary<NSString *, NSString *> *traitOrRoleToAccessibilityRole;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    traitOrRoleToAccessibilityRole = @{
+      // from https://reactnative.dev/docs/accessibility#accessibilityrole
+      @"adjustable": NSAccessibilitySliderRole,
+      @"alert": NSAccessibilityStaticTextRole, // no exact match on macOS
+      @"button": NSAccessibilityButtonRole, // also a legacy iOS accessibilityTraits
+      @"checkbox": NSAccessibilityCheckBoxRole,
+      @"combobox": NSAccessibilityComboBoxRole,
+      @"header": NSAccessibilityStaticTextRole, // no exact match on macOS
+      @"image": NSAccessibilityImageRole, // also a legacy iOS accessibilityTraits
+      @"imagebutton": NSAccessibilityButtonRole, // no exact match on macOS
+      @"keyboardkey": NSAccessibilityButtonRole, // no exact match on macOS
+      @"link": NSAccessibilityLinkRole, // also a legacy iOS accessibilityTraits
+      @"menu": NSAccessibilityMenuRole,
+      @"menubar": NSAccessibilityMenuBarRole,
+      @"menuitem": NSAccessibilityMenuItemRole,
+      @"none": NSAccessibilityUnknownRole,
+      @"progressbar": NSAccessibilityProgressIndicatorRole,
+      @"radio": NSAccessibilityRadioButtonRole,
+      @"radiogroup": NSAccessibilityRadioGroupRole,
+      @"scrollbar": NSAccessibilityScrollBarRole,
+      @"search": NSAccessibilityTextFieldRole, // no exact match on macOS
+      @"spinbutton": NSAccessibilityIncrementorRole,
+      @"summary": NSAccessibilityStaticTextRole, // no exact match on macOS
+      @"switch": NSAccessibilityCheckBoxRole, // no exact match on macOS
+      @"tab": NSAccessibilityButtonRole, // no exact match on macOS
+      @"tablist": NSAccessibilityTabGroupRole,
+      @"text": NSAccessibilityStaticTextRole, // also a legacy iOS accessibilityTraits
+      @"timer": NSAccessibilityStaticTextRole, // no exact match on macOS
+      @"toolbar": NSAccessibilityToolbarRole,
+      // Roles/traits that are macOS specific and are used by some of the core components (Lists):
+      @"disclosure": NSAccessibilityDisclosureTriangleRole,
+      @"group": NSAccessibilityGroupRole,
+      @"list": NSAccessibilityListRole,
+      @"popupbutton": NSAccessibilityPopUpButtonRole,
+      @"menubutton": NSAccessibilityMenuButtonRole,
+      @"table": NSAccessibilityTableRole,
+    };
+  });
+
+  NSString *role = [traitOrRoleToAccessibilityRole valueForKey:trait];
+  if (role == nil) {
+    role = NSAccessibilityUnknownRole;
+  }
+  return role;
+}
+
++ (NSString *)accessibilityRoleFromTraits:(id)json useAriaMappings:(BOOL)useAriaMappings
+{
+  if ([json isKindOfClass:[NSString class]]) {
+    return useAriaMappings ? [RCTConvert accessibilityRoleFromAriaRole:json] : [RCTConvert accessibilityRoleFromTrait:json];
+  } else if ([json isKindOfClass:[NSArray class]]) {
+    for (NSString *trait in json) {
+      NSString *accessibilityRole = useAriaMappings ? [RCTConvert accessibilityRoleFromAriaRole:json] : [RCTConvert accessibilityRoleFromTrait:trait];
+      if (![accessibilityRole isEqualToString:NSAccessibilityUnknownRole]) {
+        return accessibilityRole;
+      }
+    }
+  }
+  return NSAccessibilityUnknownRole;
+}
+
+RCT_ARRAY_CONVERTER(RCTHandledKey);
+
+#endif // macOS]
+
 @end
 
 @interface RCTImageSource (Packager)
@@ -1300,7 +1713,7 @@
 @implementation RCTConvert (Deprecated)
 
 /* This method is only used when loading images synchronously, e.g. for tabbar icons */
-+ (UIImage *)UIImage:(id)json
++ (RCTPlatformImage *)UIImage:(id)json // [macOS]
 {
   if (!json) {
     return nil;
@@ -1311,7 +1724,7 @@
     return nil;
   }
 
-  __block UIImage *image;
+  __block RCTPlatformImage *image; // [macOS]
   if (!RCTIsMainQueue()) {
     // It seems that none of the UIImage loading methods can be guaranteed
     // thread safe, so we'll pick the lesser of two evils here and block rather
@@ -1341,23 +1754,29 @@
       RCTLogConvertError(json, @"an image. File not found.");
     }
   } else if ([scheme isEqualToString:@"data"]) {
-    image = [UIImage imageWithData:[NSData dataWithContentsOfURL:URL]];
+    image = UIImageWithData([NSData dataWithContentsOfURL:URL]); // [macOS]
   } else if ([scheme isEqualToString:@"http"] && imageSource.packagerAsset) {
-    image = [UIImage imageWithData:[NSData dataWithContentsOfURL:URL]];
+    image = UIImageWithData([NSData dataWithContentsOfURL:URL]); // [macOS]
   } else {
     RCTLogConvertError(json, @"an image. Only local files or data URIs are supported.");
     return nil;
   }
 
+  CGImageRef imageRef = UIImageGetCGImageRef(image); // [macOS]
+#if !TARGET_OS_OSX // [macOS]
   CGFloat scale = imageSource.scale;
   if (!scale && imageSource.size.width) {
     // If no scale provided, set scale to image width / source width
-    scale = CGImageGetWidth(image.CGImage) / imageSource.size.width;
+    scale = CGImageGetWidth(imageRef) / imageSource.size.width; // [macOS]
   }
-
   if (scale) {
-    image = [UIImage imageWithCGImage:image.CGImage scale:scale orientation:image.imageOrientation];
+    image = [UIImage imageWithCGImage:imageRef scale:scale orientation:image.imageOrientation];
   }
+#else // [macOS
+  if (!CGSizeEqualToSize(image.size, imageSource.size)) {
+    image = [[NSImage alloc] initWithCGImage:imageRef size:imageSource.size];
+  }
+#endif // macOS]
 
   if (!CGSizeEqualToSize(imageSource.size, CGSizeZero) && !CGSizeEqualToSize(imageSource.size, image.size)) {
     RCTLogInfo(
@@ -1372,7 +1791,7 @@
 
 + (CGImageRef)CGImage:(id)json
 {
-  return [self UIImage:json].CGImage;
+  return UIImageGetCGImageRef([self UIImage:json]); // [macOS]
 }
 
 @end
