--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/RCTRootContentView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Base/RCTRootContentView.h	2026-03-06 15:22:52
@@ -5,9 +5,9 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
+#ifndef RCT_FIT_RM_OLD_RUNTIME
 
 #import <React/RCTInvalidating.h>
 #import <React/RCTRootView.h>
@@ -18,26 +18,19 @@
 
 @interface RCTRootContentView : RCTView <RCTInvalidating>
 
-@property (nonatomic, readonly, weak)
-    RCTBridge *bridge __deprecated_msg("This API will be removed along with the legacy architecture.");
-@property (nonatomic, readonly, assign)
-    BOOL contentHasAppeared __deprecated_msg("This API will be removed along with the legacy architecture.");
-@property (nonatomic, readonly, strong)
-    RCTTouchHandler *touchHandler __deprecated_msg("This API will be removed along with the legacy architecture.");
-@property (nonatomic, readonly, assign)
-    CGSize availableSize __deprecated_msg("This API will be removed along with the legacy architecture.");
+@property (nonatomic, readonly, weak) RCTBridge *bridge;
+@property (nonatomic, readonly, assign) BOOL contentHasAppeared;
+@property (nonatomic, readonly, strong) RCTTouchHandler *touchHandler;
+@property (nonatomic, readonly, assign) CGSize availableSize;
 
-@property (nonatomic, assign)
-    BOOL passThroughTouches __deprecated_msg("This API will be removed along with the legacy architecture.");
-@property (nonatomic, assign) RCTRootViewSizeFlexibility sizeFlexibility __deprecated_msg(
-    "This API will be removed along with the legacy architecture.");
+@property (nonatomic, assign) BOOL passThroughTouches;
+@property (nonatomic, assign) RCTRootViewSizeFlexibility sizeFlexibility;
 
 - (instancetype)initWithFrame:(CGRect)frame
                        bridge:(RCTBridge *)bridge
                      reactTag:(NSNumber *)reactTag
-              sizeFlexibility:(RCTRootViewSizeFlexibility)sizeFlexibility NS_DESIGNATED_INITIALIZER
-    __deprecated_msg("This API will be removed along with the legacy architecture.");
+              sizeFlexibility:(RCTRootViewSizeFlexibility)sizeFlexibility NS_DESIGNATED_INITIALIZER;
 
 @end
 
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // RCT_FIT_RM_OLD_RUNTIME
