--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/RCTRootView.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Base/RCTRootView.m	2026-03-06 15:22:52
@@ -15,7 +15,9 @@
 #import "RCTBridge+Private.h"
 #import "RCTBridge.h"
 #import "RCTConstants.h"
-#import "RCTKeyCommands.h"
+#import "RCTDevSettings.h" // [macOS]
+#import "RCTFocusChangeEvent.h" // [macOS]
+// [macOS] remove #import "RCTKeyCommands.h"
 #import "RCTLog.h"
 #import "RCTPerformanceLogger.h"
 #import "RCTProfile.h"
@@ -28,9 +30,13 @@
 #import "RCTView.h"
 #import "UIView+React.h"
 
+#if __has_include("RCTDevMenu.h") // [macOS
+#import "RCTDevMenu.h"
+#endif // macOS]
+
 NSString *const RCTContentDidAppearNotification = @"RCTContentDidAppearNotification";
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
+#ifndef RCT_FIT_RM_OLD_RUNTIME
 
 @implementation RCTRootView {
   RCTBridge *_bridge;
@@ -55,7 +61,9 @@
   }
 
   if (self = [super initWithFrame:frame]) {
+    /* [TODO(OSS Candidate ISS#2710739): don't set the background color on mac or ios so that the view is invisible during initial render
     self.backgroundColor = [UIColor whiteColor];
+    ]TODO(OSS Candidate ISS#2710739) */
 
     _bridge = bridge;
     _moduleName = moduleName;
@@ -112,7 +120,7 @@
 RCT_NOT_IMPLEMENTED(-(instancetype)initWithFrame : (CGRect)frame)
 RCT_NOT_IMPLEMENTED(-(instancetype)initWithCoder : (NSCoder *)aDecoder)
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
   return self;
 }
@@ -152,7 +160,19 @@
 {
   [super layoutSubviews];
   _contentView.frame = self.bounds;
+#if !TARGET_OS_OSX // [macOS]
   _loadingView.center = (CGPoint){CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds)};
+#else // [macOS
+  CGFloat centerX = CGRectGetMidX(self.bounds);
+  CGFloat centerY = CGRectGetMidY(self.bounds);
+  NSRect newFrame = NSMakeRect(
+    centerX - _loadingView.frame.size.width / 2.0,
+    centerY - _loadingView.frame.size.height / 2.0,
+    _loadingView.frame.size.width,
+    _loadingView.frame.size.height
+  );
+  _loadingView.frame = newFrame;
+#endif // macOS]
 }
 
 - (void)setMinimumSize:(CGSize)minimumSize
@@ -179,10 +199,14 @@
 
 - (BOOL)canBecomeFirstResponder
 {
+#if !TARGET_OS_OSX // [macOS]
   return YES;
+#else // [macOS
+  return NO; // commit 01aba7e8: Merged PR 94656: Enable keyboard accessibility and support for focus ring drawing for button, textfields etc
+#endif // macOS]
 }
 
-- (void)setLoadingView:(UIView *)loadingView
+- (void)setLoadingView:(RCTUIView *)loadingView // [macOS]
 {
   _loadingView = loadingView;
   if (!_contentView.contentHasAppeared) {
@@ -206,6 +230,7 @@
           dispatch_time(DISPATCH_TIME_NOW, (int64_t)(_loadingViewFadeDelay * NSEC_PER_SEC)),
           dispatch_get_main_queue(),
           ^{
+#if !TARGET_OS_OSX // [macOS]
             [UIView transitionWithView:self
                 duration:self->_loadingViewFadeDuration
                 options:UIViewAnimationOptionTransitionCrossDissolve
@@ -215,7 +240,16 @@
                 completion:^(__unused BOOL finished) {
                   [self->_loadingView removeFromSuperview];
                 }];
-          });
+#else // [macOS
+                       [NSAnimationContext runAnimationGroup:^(__unused NSAnimationContext *context) {
+                         self->_loadingView.animator.alphaValue = 0.0;
+                       } completionHandler:^{
+                         [self->_loadingView removeFromSuperview];
+                         self->_loadingView.hidden = YES;
+                         self->_loadingView.alphaValue = 1.0;
+                       }];
+#endif // macOS]
+                     });
     } else {
       _loadingView.hidden = YES;
       [_loadingView removeFromSuperview];
@@ -303,10 +337,10 @@
   _contentView.sizeFlexibility = _sizeFlexibility;
 }
 
-- (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
+- (RCTPlatformView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event // [macOS]
 {
   // The root view itself should never receive touches
-  UIView *hitView = [super hitTest:point withEvent:event];
+  RCTPlatformView *hitView = [super hitTest:point withEvent:event]; // [macOS]
   if (self.passThroughTouches && hitView == self) {
     return nil;
   }
@@ -338,13 +372,24 @@
 
   _intrinsicContentSize = intrinsicContentSize;
 
+  [self invalidateIntrinsicContentSize];
+#if !TARGET_OS_OSX // [macOS]
+  [self.superview setNeedsLayout];
+#else // [macOS
+	[self.superview setNeedsLayout:YES];
+#endif // macOS]
+
   // Don't notify the delegate if the content remains invisible or its size has not changed
   if (bothSizesHaveAZeroDimension || sizesAreEqual) {
     return;
   }
 
   [self invalidateIntrinsicContentSize];
-  [self.superview setNeedsLayout];
+  #if !TARGET_OS_OSX // [macOS]
+	[self.superview setNeedsLayout];
+  #else // [macOS
+	  [self.superview setNeedsLayout:YES];
+  #endif // macOS]
 
   [_delegate rootViewDidChangeIntrinsicSize:self];
 }
@@ -361,6 +406,7 @@
   [self showLoadingView];
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
 {
   [super traitCollectionDidChange:previousTraitCollection];
@@ -374,13 +420,99 @@
                   userInfo:@{
                     RCTUserInterfaceStyleDidChangeNotificationTraitCollectionKey : self.traitCollection,
                   }];
+}
+#else // [macOS
+- (void)viewDidChangeEffectiveAppearance
+{
+  [super viewDidChangeEffectiveAppearance];
+  [[NSNotificationCenter defaultCenter]
+      postNotificationName:RCTUserInterfaceStyleDidChangeNotification
+                    object:self
+                  userInfo:@{
+                    RCTUserInterfaceStyleDidChangeNotificationAppearanceKey : self.effectiveAppearance,
+                  }];
+}
+#endif // macOS]
+
+
+#pragma mark - Key window blur/focus
+
+#if TARGET_OS_OSX // [macOS
+- (void)viewDidMoveToWindow {
+  [super viewDidMoveToWindow];
+
+  NSWindow *window = [self window];
+  if (window == nil) {
+    return;
+  }
+
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(containingWindowDidBecomeKey)
+                                               name:NSWindowDidBecomeKeyNotification
+                                             object:window];
+
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(containingWindowDidResignKey)
+                                               name:NSWindowDidResignKeyNotification
+                                             object:window];
 }
 
++ (NSNumber *)reactTagClosestToView:(RCTPlatformView *)view {
+  // The first responder may not necessarily have a React tag. For example, if we're focused on
+  // a multiline <TextInput>, the RN component whose blur/focus events we need to touch is an
+  // RCTMultilineTextInputView, but the first responder will be the underlying RCTUITextView.
+  for (RCTPlatformView *testView = view; testView != nil; testView = [testView superview]) {
+    NSNumber *reactTag = [testView reactTag];
+    if (reactTag != nil) {
+      return reactTag;
+    }
+  }
+  return nil;
+}
+
+- (void)containingWindowDidBecomeKey {
+  NSResponder *firstResponder = [[self window] firstResponder];
+  if ([firstResponder isKindOfClass:[RCTPlatformView class]]) {
+    NSNumber *reactTag = [RCTRootView reactTagClosestToView:(RCTPlatformView *)firstResponder];
+    if (reactTag != nil) {
+      [[[self bridge] eventDispatcher] sendEvent:[RCTFocusChangeEvent focusEventWithReactTag:reactTag]];
+    }
+  }
+}
+
+- (void)containingWindowDidResignKey {
+  NSResponder *firstResponder = [[self window] firstResponder];
+  if ([firstResponder isKindOfClass:[RCTPlatformView class]]) {
+    NSNumber *reactTag = [RCTRootView reactTagClosestToView:(RCTPlatformView *)firstResponder];
+    if (reactTag != nil) {
+      [[[self bridge] eventDispatcher] sendEvent:[RCTFocusChangeEvent blurEventWithReactTag:reactTag]];
+    }
+  }
+}
+#endif // macOS]
+
+
+#if TARGET_OS_OSX // [macOS
+- (NSMenu *)menuForEvent:(NSEvent *)event
+{
+  NSMenu *menu = nil;
+#if RCT_DEV_MENU
+  menu = [[_bridge devMenu] menu];
+#endif // RCT_DEV_MENU
+  if (menu == nil) {
+    menu = [super menuForEvent:event];
+  }
+  if (menu) {
+    [[_contentView touchHandler] willShowMenuWithEvent:event];
+  }
+  return menu;
+}
+#endif // macOS]
+
 - (void)dealloc
 {
   [_contentView invalidate];
 }
-
 @end
 
 @implementation RCTRootView (Deprecated)
@@ -399,7 +531,7 @@
 
 @end
 
-#else // RCT_REMOVE_LEGACY_ARCH
+#else // RCT_FIT_RM_OLD_RUNTIME
 
 @implementation RCTRootView
 - (nonnull instancetype)initWithFrame:(CGRect)frame
@@ -427,4 +559,4 @@
 
 @end
 
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // RCT_FIT_RM_OLD_RUNTIME
