--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/RCTRootContentView.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Base/RCTRootContentView.m	2026-03-06 15:22:52
@@ -7,9 +7,10 @@
 
 #import "RCTRootContentView.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
+#ifndef RCT_FIT_RM_OLD_RUNTIME
 
 #import "RCTBridge.h"
+#import "RCTConstants.h" // [macOS]
 #import "RCTPerformanceLogger.h"
 #import "RCTRootView.h"
 #import "RCTRootViewInternal.h"
@@ -18,6 +19,11 @@
 #import "UIView+React.h"
 
 @implementation RCTRootContentView
+{ // [macOS
+#if TARGET_OS_OSX
+  BOOL _subscribedToWindowNotifications;
+#endif
+} // macOS]
 
 - (instancetype)initWithFrame:(CGRect)frame
                        bridge:(RCTBridge *)bridge
@@ -31,6 +37,10 @@
     _touchHandler = [[RCTTouchHandler alloc] initWithBridge:_bridge];
     [_touchHandler attachToView:self];
     [_bridge.uiManager registerRootView:self];
+#if TARGET_OS_OSX // [macOS
+    self.postsFrameChangedNotifications = YES;
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sendFrameChangedEvent:) name:NSViewFrameDidChangeNotification object:self];
+#endif // macOS]
   }
   return self;
 }
@@ -38,13 +48,50 @@
 RCT_NOT_IMPLEMENTED(-(instancetype)initWithFrame : (CGRect)frame)
 RCT_NOT_IMPLEMENTED(-(instancetype)initWithCoder : (nonnull NSCoder *)aDecoder)
 
+#if TARGET_OS_OSX // [macOS
+- (void)dealloc
+{
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)viewWillMoveToWindow:(nullable NSWindow *)newWindow
+{
+  if (_subscribedToWindowNotifications &&
+      self.window != nil &&
+      self.window != newWindow) {
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                    name:NSWindowDidChangeBackingPropertiesNotification
+                                                  object:self.window];
+    _subscribedToWindowNotifications = NO;
+  }
+}
+
+- (void)viewDidMoveToWindow
+{
+  if (!_subscribedToWindowNotifications &&
+      self.window != nil) {
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(sendFrameChangedEvent:)
+                                                 name:NSWindowDidChangeBackingPropertiesNotification
+                                               object:self.window];
+    _subscribedToWindowNotifications = YES;
+  }
+}
+
+- (void)sendFrameChangedEvent:(__unused NSNotification *)notification
+{
+  [[NSNotificationCenter defaultCenter] postNotificationName:RCTWindowFrameDidChangeNotification object:self];
+}
+
+#endif // macOS]
+
 - (void)layoutSubviews
 {
   [super layoutSubviews];
   [self updateAvailableSize];
 }
 
-- (void)insertReactSubview:(UIView *)subview atIndex:(NSInteger)atIndex
+- (void)insertReactSubview:(RCTUIView *)subview atIndex:(NSInteger)atIndex // [macOS]
 {
   [super insertReactSubview:subview atIndex:atIndex];
   [_bridge.performanceLogger markStopForTag:RCTPLTTI];
@@ -83,10 +130,10 @@
   [_bridge.uiManager setAvailableSize:self.availableSize forRootView:self];
 }
 
-- (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
+- (RCTPlatformView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event // [macOS]
 {
   // The root content view itself should never receive touches
-  UIView *hitView = [super hitTest:point withEvent:event];
+  RCTPlatformView *hitView = [super hitTest:point withEvent:event]; // [macOS]
   if (_passThroughTouches && hitView == self) {
     return nil;
   }
@@ -108,4 +155,4 @@
 
 @end
 
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // RCT_FIT_RM_OLD_RUNTIME
