--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/RCTReloadCommand.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Base/RCTReloadCommand.m	2026-03-06 15:22:52
@@ -8,7 +8,9 @@
 #import "RCTReloadCommand.h"
 
 #import "RCTAssert.h"
+#if !TARGET_OS_OSX // [macOS]
 #import "RCTKeyCommands.h"
+#endif // [macOS]
 #import "RCTUtils.h"
 
 static NSHashTable<id<RCTReloadListener>> *listeners;
@@ -28,7 +30,7 @@
   if (!listeners) {
     listeners = [NSHashTable weakObjectsHashTable];
   }
-#if RCT_DEV
+#if RCT_DEV && !TARGET_OS_OSX // [macOS]
   RCTAssertMainQueue(); // because registerKeyCommandWithInput: must be called on the main thread
   static dispatch_once_t onceToken;
   dispatch_once(&onceToken, ^{
@@ -38,7 +40,7 @@
                                                             RCTTriggerReloadCommandListeners(@"Command + R");
                                                           }];
   });
-#endif
+#endif // [macOS]
   [listeners addObject:listener];
   [listenersLock unlock];
 }
