--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/RCTUtils.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Base/RCTUtils.h	2026-03-06 15:22:52
@@ -7,7 +7,7 @@
 
 #import <CoreGraphics/CoreGraphics.h>
 #import <Foundation/Foundation.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTAssert.h>
 #import <React/RCTDefines.h>
@@ -47,13 +47,13 @@
 // Legacy function to execute the specified block on the main queue synchronously.
 // Please do not use this unless you know what you're doing.
 RCT_EXTERN void RCTUnsafeExecuteOnMainQueueSync(dispatch_block_t block);
+RCT_EXTERN void RCTUnsafeExecuteOnMainQueueSyncWithError(dispatch_block_t block, NSString *context);
 
 // Get screen metrics in a thread-safe way
 RCT_EXTERN CGFloat RCTScreenScale(void);
 RCT_EXTERN CGFloat RCTFontSizeMultiplier(void);
 RCT_EXTERN CGSize RCTScreenSize(void);
 RCT_EXTERN CGSize RCTViewportSize(void);
-RCT_EXTERN CGSize RCTSwitchSize(void);
 
 // Round float coordinates to nearest whole screen pixel (not point)
 RCT_EXTERN CGFloat RCTRoundPixelValue(CGFloat value);
@@ -63,8 +63,8 @@
 RCT_EXTERN CGSize RCTSizeInPixels(CGSize pointSize, CGFloat scale);
 
 // Method swizzling
-RCT_EXTERN void RCTSwapClassMethods(Class cls, SEL original, SEL replacement);
-RCT_EXTERN void RCTSwapInstanceMethods(Class cls, SEL original, SEL replacement);
+RCT_EXTERN IMP RCTSwapClassMethods(Class cls, SEL original, SEL replacement); // [macOS]
+RCT_EXTERN IMP RCTSwapInstanceMethods(Class cls, SEL original, SEL replacement); // [macOS]
 RCT_EXTERN void RCTSwapInstanceMethodWithBlock(Class cls, SEL original, id replacementBlock, SEL replacementSelector);
 
 // Module subclass support
@@ -86,24 +86,31 @@
 // Returns YES if React is running in a test environment
 RCT_EXTERN BOOL RCTRunningInTestEnvironment(void);
 
+#if !TARGET_OS_OSX // [macOS]
 // Returns YES if React is running in an iOS App Extension
 RCT_EXTERN BOOL RCTRunningInAppExtension(void);
+#endif // [macOS]
 
 // Returns the shared UIApplication instance, or nil if running in an App Extension
-RCT_EXTERN UIApplication *__nullable RCTSharedApplication(void);
+RCT_EXTERN RCTUIApplication *__nullable RCTSharedApplication(void); // [macOS]
 
 // Returns the current main window, useful if you need to access the root view
 // or view controller
-RCT_EXTERN UIWindow *__nullable RCTKeyWindow(void);
+RCT_EXTERN RCTPlatformWindow *__nullable RCTKeyWindow(void); // [macOS]
 
+#if !TARGET_OS_OSX // [macOS]
+// Returns UIStatusBarManager to get it's configuration info.
+RCT_EXTERN UIStatusBarManager *__nullable RCTUIStatusBarManager(void);
+#endif // [macOS]
+
+#if !TARGET_OS_OSX // [macOS]
 // Returns the presented view controller, useful if you need
 // e.g. to present a modal view controller or alert over it
 RCT_EXTERN UIViewController *__nullable RCTPresentedViewController(void);
 
 // Retrieve current window UIStatusBarManager
-#if !TARGET_OS_TV
 RCT_EXTERN UIStatusBarManager *__nullable RCTUIStatusBarManager(void) API_AVAILABLE(ios(13));
-#endif
+#endif // [macOS]
 
 // Does this device support force touch (aka 3D Touch)?
 RCT_EXTERN BOOL RCTForceTouchAvailable(void);
@@ -134,10 +141,6 @@
 // Gzip functionality - compression level in range 0 - 1 (-1 for default)
 RCT_EXTERN NSData *__nullable RCTGzipData(NSData *__nullable data, float level);
 
-// Gzip decompression - maxDecompressedSize of 0 means no limit, returns nil if
-// limit exceeded or decompression fails
-RCT_EXTERN NSData *__nullable RCTDecompressGzipData(NSData *__nullable data, NSUInteger maxDecompressedSize);
-
 // Returns the relative path within the main bundle for an absolute URL
 // (or nil, if the URL does not specify a path within the main bundle)
 RCT_EXTERN NSString *__nullable RCTBundlePathForURL(NSURL *__nullable URL);
@@ -160,12 +163,12 @@
 
 // Returns an UIImage for a local image asset. Returns nil if the URL
 // does not correspond to a local asset.
-RCT_EXTERN UIImage *__nullable RCTImageFromLocalAssetURL(NSURL *imageURL);
+RCT_EXTERN RCTPlatformImage *__nullable RCTImageFromLocalAssetURL(NSURL *imageURL); // [macOS]
 
 // Only used in case when RCTImageFromLocalAssetURL fails to get an image
 // This method basically checks for the image in the bundle location, instead
 // of the CodePush location
-RCT_EXTERN UIImage *__nullable RCTImageFromLocalBundleAssetURL(NSURL *imageURL);
+RCT_EXTERN RCTPlatformImage *__nullable RCTImageFromLocalBundleAssetURL(NSURL *imageURL); // [macOS]
 
 // Creates a new, unique temporary file path with the specified extension
 RCT_EXTERN NSString *__nullable RCTTempFilePath(NSString *__nullable extension, NSError **error);
@@ -176,12 +179,17 @@
 // Converts a CGColor to a hex string
 RCT_EXTERN NSString *RCTColorToHexString(CGColorRef color);
 
+#if !TARGET_OS_OSX // [macOS]
 // Get standard localized string (if it exists)
 RCT_EXTERN NSString *RCTUIKitLocalizedString(NSString *string);
+#endif // [macOS]
 
 // Get a human readable type string from an NSObject. For example NSString becomes string
 RCT_EXTERN NSString *RCTHumanReadableType(NSObject *obj);
 
+// Get a human readable type string from an NSObject. For example NSString becomes string
+RCT_EXTERN NSString *RCTHumanReadableType(NSObject *obj);
+
 // URL manipulation
 RCT_EXTERN NSString *__nullable RCTGetURLQueryParam(NSURL *__nullable URL, NSString *param);
 RCT_EXTERN NSURL *__nullable
