--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/RCTViewRegistry.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Base/RCTViewRegistry.m	2026-03-06 15:22:52
@@ -13,33 +13,33 @@
 
 @implementation RCTViewRegistry {
   RCTBridgelessComponentViewProvider _bridgelessComponentViewProvider;
-#ifndef RCT_REMOVE_LEGACY_ARCH
+#ifndef RCT_FIT_RM_OLD_RUNTIME
   __weak RCTBridge *_bridge;
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // RCT_FIT_RM_OLD_RUNTIME
 }
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
+#ifndef RCT_FIT_RM_OLD_RUNTIME
 - (void)setBridge:(RCTBridge *)bridge
 {
   _bridge = bridge;
 }
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // RCT_FIT_RM_OLD_RUNTIME
 
 - (void)setBridgelessComponentViewProvider:(RCTBridgelessComponentViewProvider)bridgelessComponentViewProvider
 {
   _bridgelessComponentViewProvider = bridgelessComponentViewProvider;
 }
 
-- (UIView *)viewForReactTag:(NSNumber *)reactTag
+- (RCTPlatformView *)viewForReactTag:(NSNumber *)reactTag // [macOS]
 {
-  UIView *view = nil;
+  RCTPlatformView *view = nil; // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
+#ifndef RCT_FIT_RM_OLD_RUNTIME
   RCTBridge *bridge = _bridge;
   if (bridge) {
     view = [bridge.uiManager viewForReactTag:reactTag];
   }
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // RCT_FIT_RM_OLD_RUNTIME
 
   if (view == nil && _bridgelessComponentViewProvider) {
     view = _bridgelessComponentViewProvider(reactTag);
@@ -55,9 +55,9 @@
   }
 
   __weak __typeof(self) weakSelf = self;
-#ifndef RCT_REMOVE_LEGACY_ARCH
+#ifndef RCT_FIT_RM_OLD_RUNTIME
   if (_bridge) {
-    [_bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry) {
+    [_bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTPlatformView *> *viewRegistry) {
       __typeof(self) strongSelf = weakSelf;
       if (strongSelf) {
         block(strongSelf);
@@ -72,7 +72,7 @@
     if (strongSelf) {
       block(strongSelf);
     }
-  }); // RCT_REMOVE_LEGACY_ARCH
+  }); // RCT_FIT_RM_OLD_RUNTIME
 }
 
 @end
