--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Lists/FillRateHelper.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Lists/FillRateHelper.js	2026-03-06 15:22:52
@@ -10,10 +10,10 @@
 
 'use strict';
 
-import VirtualizedLists from '@react-native/virtualized-lists';
+import VirtualizedLists from '@react-native-macos/virtualized-lists'; // [macOS]
 
 type FillRateHelperType = typeof VirtualizedLists.FillRateHelper;
 const FillRateHelper: FillRateHelperType = VirtualizedLists.FillRateHelper;
 
-export type {FillRateInfo} from '@react-native/virtualized-lists';
+export type {FillRateInfo} from '@react-native-macos/virtualized-lists'; // [macOS]
 export default FillRateHelper;
