--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Lists/FlatList.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Lists/FlatList.js	2026-03-06 15:22:52
@@ -13,15 +13,15 @@
 import type {
   ListRenderItem,
   ListRenderItemInfo,
-  ListViewToken,
   ViewabilityConfigCallbackPair,
+  ViewToken,
   VirtualizedListProps,
-} from '@react-native/virtualized-lists';
+} from '@react-native-macos/virtualized-lists'; // [macOS]
 
 import * as ReactNativeFeatureFlags from '../../src/private/featureflags/ReactNativeFeatureFlags';
 import {type ScrollResponderType} from '../Components/ScrollView/ScrollView';
 import View from '../Components/View/View';
-import VirtualizedLists from '@react-native/virtualized-lists';
+import VirtualizedLists from '@react-native-macos/virtualized-lists'; // [macOS]
 import memoizeOne from 'memoize-one';
 import * as React from 'react';
 
@@ -73,7 +73,25 @@
    * Optional custom style for multi-item rows generated when numColumns > 1.
    */
   columnWrapperStyle?: ViewStyleProp,
+  // [macOS
   /**
+   * Allows you to 'select' a row using arrow keys. The selected row will have the prop `isSelected`
+   * passed in as true to it's renderItem / ListItemComponent. You can also imperatively select a row
+   * using the `selectRowAtIndex` method. You can set the initially selected row using the
+   * `initialSelectedIndex` prop.
+   * Keyboard Behavior:
+   * - ArrowUp: Select row above current selected row
+   * - ArrowDown: Select row below current selected row
+   * - Option+ArrowUp: Select the first row
+   * - Opton+ArrowDown: Select the last 'realized' row
+   * - Home: Scroll to top of list
+   * - End: Scroll to end of list
+   *
+   * @platform macos
+   */
+  enableSelectionOnKeyPress?: ?boolean,
+  // macOS]
+  /**
    * A marker property for telling the list to re-render (since it implements `PureComponent`). If
    * any of your `renderItem`, Header, Footer, etc. functions depend on anything outside of the
    * `data` prop, stick it here and treat it immutably.
@@ -118,8 +136,14 @@
    * `getItemLayout` to be implemented.
    */
   initialScrollIndex?: ?number,
+  // [macOS
   /**
-   * Reverses the direction of scroll. Uses scale transforms of -1.
+   * The initially selected row, if `enableSelectionOnKeyPress` is set.
+   */
+  initialSelectedIndex?: ?number,
+  // macOS]
+  /**
+   * Reverses the direction of scroll. Uses native inversion on macOS and scale transforms of -1 elsewhere
    */
   inverted?: ?boolean,
   /**
@@ -172,7 +196,7 @@
   return numColumns ?? 1;
 }
 
-function isArrayLike(data: unknown): boolean {
+function isArrayLike(data: mixed): boolean {
   // $FlowExpectedError[incompatible-use]
   return typeof Object(data).length === 'number';
 }
@@ -385,7 +409,20 @@
     }
   }
 
+  // [macOS
   /**
+   * Move selection to the specified index
+   *
+   * @platform macos
+   */
+  selectRowAtIndex(index: number) {
+    if (this._listRef) {
+      this._listRef.selectRowAtIndex(index);
+    }
+  }
+  // macOS]
+
+  /**
    * Provides a handle to the underlying scroll responder.
    */
   getScrollResponder(): ?ScrollResponderType {
@@ -413,7 +450,7 @@
     }
   }
 
-  setNativeProps(props: {[string]: unknown, ...}) {
+  setNativeProps(props: {[string]: mixed, ...}) {
     if (this._listRef) {
       this._listRef.setNativeProps(props);
     }
@@ -432,7 +469,7 @@
         }));
     } else if (this.props.onViewableItemsChanged) {
       this._virtualizedListPairs.push({
-        /* $FlowFixMe[incompatible-type] (>=0.63.0 site=react_native_fb) This
+        /* $FlowFixMe[incompatible-call] (>=0.63.0 site=react_native_fb) This
          * comment suppresses an error found when Flow v0.63 was deployed. To
          * see the error delete this comment and run Flow. */
         viewabilityConfig: this.props.viewabilityConfig,
@@ -569,11 +606,11 @@
         .join(':');
     }
 
-    // $FlowFixMe[incompatible-type] Can't call keyExtractor with an array
+    // $FlowFixMe[incompatible-call] Can't call keyExtractor with an array
     return keyExtractor(items, index);
   };
 
-  _pushMultiColumnViewable(arr: Array<ListViewToken>, v: ListViewToken): void {
+  _pushMultiColumnViewable(arr: Array<ViewToken>, v: ViewToken): void {
     const numColumns = numColumnsOrDefault(this.props.numColumns);
     const keyExtractor = this.props.keyExtractor ?? defaultKeyExtractor;
     v.item.forEach((item, ii) => {
@@ -585,22 +622,22 @@
 
   _createOnViewableItemsChanged(
     onViewableItemsChanged: ?(info: {
-      viewableItems: Array<ListViewToken>,
-      changed: Array<ListViewToken>,
+      viewableItems: Array<ViewToken>,
+      changed: Array<ViewToken>,
       ...
     }) => void,
     // $FlowFixMe[missing-local-annot]
   ) {
     return (info: {
-      viewableItems: Array<ListViewToken>,
-      changed: Array<ListViewToken>,
+      viewableItems: Array<ViewToken>,
+      changed: Array<ViewToken>,
       ...
     }) => {
       const numColumns = numColumnsOrDefault(this.props.numColumns);
       if (onViewableItemsChanged) {
         if (numColumns > 1) {
-          const changed: Array<ListViewToken> = [];
-          const viewableItems: Array<ListViewToken> = [];
+          const changed: Array<ViewToken> = [];
+          const viewableItems: Array<ViewToken> = [];
           info.viewableItems.forEach(v =>
             this._pushMultiColumnViewable(viewableItems, v),
           );
@@ -626,10 +663,11 @@
     const render = (props: ListRenderItemInfo<ItemT>): React.Node => {
       if (ListItemComponent) {
         // $FlowFixMe[not-a-component] Component isn't valid
-        // $FlowFixMe[incompatible-type] Component isn't valid
+        // $FlowFixMe[incompatible-type-arg] Component isn't valid
+        // $FlowFixMe[incompatible-return] Component isn't valid
         return <ListItemComponent {...props} />;
       } else if (renderItem) {
-        // $FlowFixMe[incompatible-type]
+        // $FlowFixMe[incompatible-call]
         return renderItem(props);
       } else {
         return null;
@@ -647,9 +685,10 @@
           <View style={StyleSheet.compose(styles.row, columnWrapperStyle)}>
             {item.map((it, kk) => {
               const element = render({
-                // $FlowFixMe[incompatible-type]
+                // $FlowFixMe[incompatible-call]
                 item: it,
                 index: index * cols + kk,
+                isSelected: info.isSelected, // [macOS]
                 separators: info.separators,
               });
               return element != null ? (
