--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Lists/ViewabilityHelper.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Lists/ViewabilityHelper.js	2026-03-06 15:22:52
@@ -11,12 +11,12 @@
 'use strict';
 
 export type {
-  ListViewToken as ViewToken,
+  ViewToken,
   ViewabilityConfig,
   ViewabilityConfigCallbackPair,
-} from '@react-native/virtualized-lists';
+} from '@react-native-macos/virtualized-lists'; // [macOS]
 
-import VirtualizedLists from '@react-native/virtualized-lists';
+import VirtualizedLists from '@react-native-macos/virtualized-lists'; // [macOS]
 
 type ViewabilityHelperType = typeof VirtualizedLists.ViewabilityHelper;
 const ViewabilityHelper: ViewabilityHelperType =
