--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Lists/VirtualizedList.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Lists/VirtualizedList.js	2026-03-06 15:22:52
@@ -10,7 +10,7 @@
 
 'use strict';
 
-import VirtualizedLists from '@react-native/virtualized-lists';
+import VirtualizedLists from '@react-native-macos/virtualized-lists'; // [macOS]
 
 type VirtualizedListType = typeof VirtualizedLists.VirtualizedList;
 const VirtualizedListComponent: VirtualizedListType =
@@ -19,8 +19,7 @@
 export type {
   ListRenderItemInfo,
   ListRenderItem,
-  ListViewToken,
   Separators,
   VirtualizedListProps,
-} from '@react-native/virtualized-lists';
+} from '@react-native-macos/virtualized-lists'; // [macOS]
 export default VirtualizedListComponent;
