--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTActivityIndicatorViewManager.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTActivityIndicatorViewManager.m	2026-03-06 15:22:52
@@ -7,8 +7,6 @@
 
 #import "RCTActivityIndicatorViewManager.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import "RCTActivityIndicatorView.h"
 #import "RCTConvert.h"
 
@@ -29,26 +27,26 @@
 
 RCT_EXPORT_MODULE()
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
   return [RCTActivityIndicatorView new];
 }
 
 RCT_EXPORT_VIEW_PROPERTY(color, UIColor)
 RCT_EXPORT_VIEW_PROPERTY(hidesWhenStopped, BOOL)
-RCT_CUSTOM_VIEW_PROPERTY(size, UIActivityIndicatorViewStyle, UIActivityIndicatorView)
+RCT_CUSTOM_VIEW_PROPERTY(size, UIActivityIndicatorViewStyle, RCTActivityIndicatorView) // [macOS]
 {
   /*
     Setting activityIndicatorViewStyle overrides the color, so restore the original color
     after setting the indicator style.
   */
-  UIColor *oldColor = view.color;
+  RCTUIColor *oldColor = view.color; // [macOS]
   view.activityIndicatorViewStyle =
       json ? [RCTConvert UIActivityIndicatorViewStyle:json] : defaultView.activityIndicatorViewStyle;
   view.color = oldColor;
 }
 
-RCT_CUSTOM_VIEW_PROPERTY(animating, BOOL, UIActivityIndicatorView)
+RCT_CUSTOM_VIEW_PROPERTY(animating, BOOL, RCTActivityIndicatorView) // [macOS]
 {
   BOOL animating = json ? [RCTConvert BOOL:json] : [defaultView isAnimating];
   if (animating != [view isAnimating]) {
@@ -61,5 +59,3 @@
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
