--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTAnimationType.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTAnimationType.h	2026-03-06 15:22:52
@@ -8,10 +8,14 @@
 #import <Foundation/Foundation.h>
 
 typedef NS_ENUM(NSInteger, RCTAnimationType) {
+#if !TARGET_OS_OSX // [macOS]
   RCTAnimationTypeSpring = 0,
+#endif // [macOS]
   RCTAnimationTypeLinear,
   RCTAnimationTypeEaseIn,
   RCTAnimationTypeEaseOut,
   RCTAnimationTypeEaseInEaseOut,
+#if !TARGET_OS_OSX // [macOS]
   RCTAnimationTypeKeyboard,
+#endif // [macOS]
 };
