--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTBorderDrawing.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTBorderDrawing.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTBorderStyle.h>
 #import <React/RCTDefines.h>
@@ -29,10 +29,10 @@
 } RCTCornerInsets;
 
 typedef struct {
-  UIColor *top;
-  UIColor *left;
-  UIColor *bottom;
-  UIColor *right;
+  RCTUIColor *top; // [macOS]
+  RCTUIColor *left; // [macOS]
+  RCTUIColor *bottom; // [macOS]
+  RCTUIColor *right; // [macOS]
 } RCTBorderColors;
 
 /**
@@ -61,19 +61,20 @@
     CGRect bounds,
     RCTCornerInsets cornerInsets,
     const CGAffineTransform *transform,
-    BOOL inverted);
+    const BOOL inverted);
 
 /**
  * Draw a CSS-compliant border as an image. You can determine if it's scalable
  * by inspecting the image's `capInsets`.
  *
  * `borderInsets` defines the border widths for each edge.
+ * `scaleFactor` defines the backing scale factor of the device for supporting high-resolution drawing. // [macOS]
  */
-RCT_EXTERN UIImage *RCTGetBorderImage(
+RCT_EXTERN RCTUIImage *RCTGetBorderImage( // [macOS]
     RCTBorderStyle borderStyle,
     CGSize viewSize,
     RCTCornerRadii cornerRadii,
     UIEdgeInsets borderInsets,
     RCTBorderColors borderColors,
-    UIColor *backgroundColor,
+    RCTUIColor *backgroundColor, // [macOS]
     BOOL drawToEdge);
