--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTComponentData.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTComponentData.mm	2026-03-06 15:22:52
@@ -50,7 +50,7 @@
                               bridge:(RCTBridge *)bridge
                      eventDispatcher:(id<RCTEventDispatcherProtocol>)eventDispatcher
 {
-  if ((self = [super init]) != nullptr) {
+  if ((self = [super init])) {
     _bridge = bridge;
     _eventDispatcher = eventDispatcher;
     _managerClass = managerClass;
@@ -71,26 +71,28 @@
 
 - (RCTViewManager *)manager
 {
-  if ((_manager == nullptr) && [self isBridgeMode]) {
+  if (!_manager && [self isBridgeMode]) {
     _manager = [_bridge moduleForClass:_managerClass];
-  } else if ((_manager == nullptr) && (_bridgelessViewManager == nullptr)) {
+  } else if (!_manager && !_bridgelessViewManager) {
     _bridgelessViewManager = [_bridge moduleForClass:_managerClass];
   }
-  return (_manager != nullptr) ? _manager : _bridgelessViewManager;
+  return _manager ? _manager : _bridgelessViewManager;
 }
 
 RCT_NOT_IMPLEMENTED(-(instancetype)init)
 
-- (UIView *)createViewWithTag:(nullable NSNumber *)tag rootTag:(nullable NSNumber *)rootTag
+- (RCTPlatformView *)createViewWithTag:(nullable NSNumber *)tag rootTag:(nullable NSNumber *)rootTag // [macOS]
 {
   RCTAssertMainQueue();
 
-  UIView *view = [self.manager view];
+  RCTPlatformView *view = [self.manager view]; // [macOS]
   view.reactTag = tag;
   view.rootTag = rootTag;
+#if !TARGET_OS_OSX // [macOS]
   view.multipleTouchEnabled = YES;
   view.userInteractionEnabled = YES; // required for touch handling
   view.layer.allowsGroupOpacity = YES; // required for touch handling
+#endif // [macOS]
   return view;
 }
 
@@ -106,7 +108,7 @@
 {
   json = RCTNilIfNull(json);
   if (!isShadowView) {
-    if ((json == nullptr) && (_defaultView == nullptr)) {
+    if (!json && !_defaultView) {
       // Only create default view if json is null
       _defaultView = [self createViewWithTag:nil rootTag:nil];
     }
@@ -130,11 +132,11 @@
       eventHandler = ^(NSDictionary *event) {
         // The component no longer exists, we shouldn't send the event
         id<RCTComponent> strongTarget = weakTarget;
-        if (strongTarget == nullptr) {
+        if (!strongTarget) {
           return;
         }
 
-        if (eventInterceptor != nullptr) {
+        if (eventInterceptor) {
           eventInterceptor(propName, event, strongTarget.reactTag);
         } else {
           RCTComponentEvent *componentEvent = [[RCTComponentEvent alloc] initWithName:propName
@@ -158,13 +160,13 @@
   __block NSMutableData *defaultValue = nil;
 
   return ^(id target, id json) {
-    if (target == nullptr) {
+    if (!target) {
       return;
     }
 
     // Get default value
-    if (defaultValue == nullptr) {
-      if (json == nullptr) {
+    if (!defaultValue) {
+      if (!json) {
         // We only set the defaultValue when we first pass a non-null
         // value, so if the first value sent for a prop is null, it's
         // a no-op (we'd be resetting it to its default when its
@@ -186,10 +188,10 @@
     // Get value
     BOOL freeValueOnCompletion = NO;
     void *value = defaultValue.mutableBytes;
-    if (json != nullptr) {
+    if (json) {
       freeValueOnCompletion = YES;
       value = malloc(typeSignature.methodReturnLength);
-      if (value == nullptr) {
+      if (!value) {
         // CWE - 391 : Unchecked error condition
         // https://www.cvedetails.com/cwe-details/391/Unchecked-Error-Condition.html
         // https://eli.thegreenplace.net/2009/10/30/handling-out-of-memory-conditions-in-c
@@ -201,7 +203,7 @@
     }
 
     // Set value
-    if (targetInvocation == nullptr) {
+    if (!targetInvocation) {
       NSMethodSignature *signature = [target methodSignatureForSelector:setter];
       targetInvocation = [NSInvocation invocationWithMethodSignature:signature];
       targetInvocation.selector = setter;
@@ -252,7 +254,7 @@
     // Disect keypath
     NSString *key = name;
     NSArray<NSString *> *parts = [keyPath componentsSeparatedByString:@"."];
-    if (parts != nullptr) {
+    if (parts) {
       key = parts.lastObject;
       parts = [parts subarrayWithRange:(NSRange){0, parts.count - 1}];
     }
@@ -275,7 +277,7 @@
     } else {
       // Ordinary property handlers
       NSMethodSignature *typeSignature = [[RCTConvert class] methodSignatureForSelector:type];
-      if (typeSignature == nullptr) {
+      if (!typeSignature) {
         RCTLogError(@"No +[RCTConvert %@] function found.", NSStringFromSelector(type));
         return ^(__unused id<RCTComponent> view, __unused id json) {
         };
@@ -347,7 +349,7 @@
 {
   RCTPropBlockDictionary *propBlocks = isShadowView ? _shadowPropBlocks : _viewPropBlocks;
   RCTPropBlock propBlock = propBlocks[name];
-  if (propBlock == nullptr) {
+  if (!propBlock) {
     propBlock = [self createPropBlock:name isShadowView:isShadowView];
 
 #if RCT_DEBUG
@@ -381,7 +383,7 @@
 
 - (void)setProps:(NSDictionary<NSString *, id> *)props forView:(id<RCTComponent>)view isShadowView:(BOOL)isShadowView
 {
-  if (view == nullptr) {
+  if (!view) {
     return;
   }
 
@@ -467,13 +469,13 @@
 
     // We need to handle both propConfig_* and propConfigShadow_* methods
     const char *underscorePos = strchr(selectorName + strlen("propConfig"), '_');
-    if (underscorePos == nullptr) {
+    if (!underscorePos) {
       continue;
     }
 
     NSString *name = @(underscorePos + 1);
     NSString *type = ((NSArray<NSString *> * (*)(id, SEL)) objc_msgSend)(managerClass, selector)[0];
-    if (RCT_DEBUG && (propTypes[name] != nullptr) && ![propTypes[name] isEqualToString:type]) {
+    if (RCT_DEBUG && propTypes[name] && ![propTypes[name] isEqualToString:type]) {
       RCTLogError(
           @"Property '%@' of component '%@' redefined from '%@' "
            "to '%@'",
