--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTDebuggingOverlayManager.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTDebuggingOverlayManager.m	2026-03-06 15:22:52
@@ -8,8 +8,6 @@
 #import "RCTDebuggingOverlayManager.h"
 #import "RCTDebuggingOverlay.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTLog.h>
 #import <React/RCTUIManager.h>
 
@@ -19,15 +17,15 @@
 
 RCT_EXPORT_MODULE(DebuggingOverlay)
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
   return [RCTDebuggingOverlay new];
 }
 
 RCT_EXPORT_METHOD(highlightTraceUpdates : (nonnull NSNumber *)viewTag nodes : (NSArray *)updates)
 {
-  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry) {
-    UIView *view = viewRegistry[viewTag];
+  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTPlatformView *> *viewRegistry) { // [macOS]
+    RCTPlatformView *view = viewRegistry[viewTag]; // [macOS]
 
     if ([view isKindOfClass:[RCTDebuggingOverlay class]]) {
       [(RCTDebuggingOverlay *)view highlightTraceUpdates:updates];
@@ -39,8 +37,8 @@
 
 RCT_EXPORT_METHOD(highlightElements : (nonnull NSNumber *)viewTag elements : (NSArray *)elements)
 {
-  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry) {
-    UIView *view = viewRegistry[viewTag];
+  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTPlatformView *> *viewRegistry) { // [macOS]
+    RCTPlatformView *view = viewRegistry[viewTag]; // [macOS]
 
     if ([view isKindOfClass:[RCTDebuggingOverlay class]]) {
       [(RCTDebuggingOverlay *)view highlightElements:elements];
@@ -52,8 +50,8 @@
 
 RCT_EXPORT_METHOD(clearElementsHighlights : (nonnull NSNumber *)viewTag)
 {
-  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry) {
-    UIView *view = viewRegistry[viewTag];
+  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTPlatformView *> *viewRegistry) { // [macOS[
+    RCTPlatformView *view = viewRegistry[viewTag]; // [macOS]
 
     if ([view isKindOfClass:[RCTDebuggingOverlay class]]) {
       [(RCTDebuggingOverlay *)view clearElementsHighlights];
@@ -64,5 +62,3 @@
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
