--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTFont.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTFont.mm	2026-03-06 15:22:52
@@ -6,8 +6,6 @@
  */
 
 #import "RCTFont.h"
-#import "RCTFont+Private.h"
-
 #import "RCTAssert.h"
 #import "RCTLog.h"
 
@@ -124,15 +122,6 @@
   return @"regular";
 }
 
-UIFont *RCTGetLegacyDefaultFont(CGFloat size, UIFontWeight fontWeight)
-{
-  if (defaultFontHandler != nil) {
-    return defaultFontHandler(size, FontWeightDescriptionFromUIFontWeight(fontWeight));
-  } else {
-    return nil;
-  }
-}
-
 static UIFont *cachedSystemFont(CGFloat size, RCTFontWeight weight)
 {
   static NSCache<NSValue *, UIFont *> *fontCache = [NSCache new];
@@ -142,14 +131,20 @@
     RCTFontWeight weight;
   };
 
-  CacheKey key{.size = size, .weight = weight};
+  CacheKey key{size, weight};
   NSValue *cacheKey = [[NSValue alloc] initWithBytes:&key objCType:@encode(CacheKey)];
   UIFont *font = [fontCache objectForKey:cacheKey];
 
-  if (font == nil) {
-    if (defaultFontHandler != nil) {
+  if (!font) {
+    if (defaultFontHandler) {
       NSString *fontWeightDescription = FontWeightDescriptionFromUIFontWeight(weight);
       font = defaultFontHandler(size, fontWeightDescription);
+#pragma clang diagnostic push // [macOS]
+#pragma clang diagnostic ignored "-Wunguarded-availability" // [macOS]
+    } else if ([UIFont respondsToSelector:@selector(systemFontOfSize:weight:)]) {
+      // Only supported on iOS8.2/macOS10.11 and above
+      font = [UIFont systemFontOfSize:size weight:weight];
+#pragma clang diagnostic pop // [macOS]
     } else {
       font = [UIFont systemFontOfSize:size weight:weight];
     }
@@ -178,7 +173,15 @@
 
   auto names = [cache objectForKey:familyName];
   if (!names) {
+#if !TARGET_OS_OSX // [macOS]
     names = [UIFont fontNamesForFamilyName:familyName] ?: [NSArray new];
+#else // [macOS
+    NSMutableArray<NSString *> *fontNames = [NSMutableArray array];
+    for (NSArray *fontSettings in [[NSFontManager sharedFontManager] availableMembersOfFontFamily:familyName]) {
+      [fontNames addObject:fontSettings[0]];
+    }
+    names = fontNames;
+#endif // macOS]
     [cache setObject:names forKey:familyName];
   }
   return names;
@@ -216,7 +219,7 @@
     UIFontWeightRegular,
     doubleValue)
 
-using RCTFontStyle = BOOL;
+typedef BOOL RCTFontStyle;
 RCT_ENUM_CONVERTER(
     RCTFontStyle,
     (@{
@@ -227,7 +230,7 @@
     NO,
     boolValue)
 
-using RCTFontVariantDescriptor = NSDictionary;
+typedef NSDictionary RCTFontVariantDescriptor;
 + (RCTFontVariantDescriptor *)RCTFontVariantDescriptor:(id)json
 {
   static NSDictionary *mapping;
@@ -465,8 +468,11 @@
     } else {
       // Not a valid font or family
       RCTLogInfo(@"Unrecognized font family '%@'", familyName);
+#pragma clang diagnostic push // [macOS]
+#pragma clang diagnostic ignored "-Wunguarded-availability" // [macOS]
       if ([UIFont respondsToSelector:@selector(systemFontOfSize:weight:)]) {
         font = [UIFont systemFontOfSize:fontSize weight:fontWeight];
+#pragma clang diagnostic pop // [macOS]
       } else if (fontWeight > UIFontWeightRegular) {
         font = [UIFont boldSystemFontOfSize:fontSize];
       } else {
