--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTModalHostViewManager.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTModalHostViewManager.h	2026-03-06 15:22:52
@@ -9,17 +9,14 @@
 #import <React/RCTInvalidating.h>
 #import <React/RCTViewManager.h>
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
+#if !TARGET_OS_OSX // [macOS]
 typedef void (^RCTModalViewInteractionBlock)(
     UIViewController *reactViewController,
     UIViewController *viewController,
     BOOL animated,
-    dispatch_block_t completionBlock)
-    __attribute__((deprecated("This API will be removed along with the legacy architecture.")));
+    dispatch_block_t completionBlock);
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTModalHostViewManager : RCTViewManager<RCTInvalidating>
+@interface RCTModalHostViewManager : RCTViewManager <RCTInvalidating>
 
 /**
  * `presentationBlock` and `dismissalBlock` allow you to control how a Modal interacts with your case,
@@ -30,5 +27,4 @@
 @property (nonatomic, strong) RCTModalViewInteractionBlock dismissalBlock;
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // [macOS]
