--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTModalHostView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTModalHostView.h	2026-03-06 15:22:52
@@ -5,10 +5,9 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#if !TARGET_OS_OSX // [macOS]
 #import <UIKit/UIKit.h>
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTInvalidating.h>
 #import <React/RCTModalHostViewManager.h>
 #import <React/RCTView.h>
@@ -18,8 +17,7 @@
 
 @protocol RCTModalHostViewInteractor;
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTModalHostView : UIView<RCTInvalidating, UIAdaptivePresentationControllerDelegate>
+@interface RCTModalHostView : RCTUIView <RCTInvalidating, UIAdaptivePresentationControllerDelegate> // [macOS]
 
 @property (nonatomic, copy) NSString *animationType;
 @property (nonatomic, assign) UIModalPresentationStyle presentationStyle;
@@ -38,10 +36,8 @@
 
 @property (nonatomic, weak) id<RCTModalHostViewInteractor> delegate;
 
-#if !TARGET_OS_TV
 @property (nonatomic, copy) NSArray<NSString *> *supportedOrientations;
 @property (nonatomic, copy) RCTDirectEventBlock onOrientationChange;
-#endif
 
 // Fabric only
 @property (nonatomic, copy) RCTDirectEventBlock onDismiss;
@@ -60,5 +56,4 @@
                     animated:(BOOL)animated;
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // [macOS]
