--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTModalHostView.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTModalHostView.m	2026-03-06 15:22:52
@@ -5,10 +5,9 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#if !TARGET_OS_OSX // [macOS]
 #import "RCTModalHostView.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <UIKit/UIKit.h>
 
 #import "RCTAssert.h"
@@ -25,9 +24,7 @@
   RCTModalHostViewController *_modalViewController;
   RCTTouchHandler *_touchHandler;
   UIView *_reactSubview;
-#if !TARGET_OS_TV
   UIInterfaceOrientation _lastKnownOrientation;
-#endif
   RCTDirectEventBlock _onRequestClose;
 }
 
@@ -92,7 +89,7 @@
 
 - (void)notifyForOrientationChange
 {
-#if !TARGET_OS_TV
+#if TARGET_OS_IOS // [visionOS]
   if (!_onOrientationChange) {
     return;
   }
@@ -109,7 +106,7 @@
     @"orientation" : isPortrait ? @"portrait" : @"landscape",
   };
   _onOrientationChange(eventPayload);
-#endif
+#endif // [visionOS]
 }
 
 - (void)insertReactSubview:(UIView *)subview atIndex:(NSInteger)atIndex
@@ -195,9 +192,7 @@
   if (shouldBePresented) {
     RCTAssert(self.reactViewController, @"Can't present modal view controller without a presenting view controller");
 
-#if !TARGET_OS_TV
     _modalViewController.supportedInterfaceOrientations = [self supportedOrientationsMask];
-#endif
 
     if ([self.animationType isEqualToString:@"fade"]) {
       _modalViewController.modalTransitionStyle = UIModalTransitionStyleCrossDissolve;
@@ -230,7 +225,6 @@
       transparent ? UIModalPresentationOverFullScreen : UIModalPresentationFullScreen;
 }
 
-#if !TARGET_OS_TV
 - (UIInterfaceOrientationMask)supportedOrientationsMask
 {
   if (_supportedOrientations.count == 0) {
@@ -257,8 +251,6 @@
   }
   return supportedOrientations;
 }
-#endif
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // [macOS]
