--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTModalHostViewController.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTModalHostViewController.m	2026-03-06 15:22:52
@@ -5,10 +5,9 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#if !TARGET_OS_OSX // [macOS]
 #import "RCTModalHostViewController.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import "RCTLog.h"
 #import "RCTModalHostView.h"
 
@@ -24,8 +23,10 @@
     return nil;
   }
 
+#if !TARGET_OS_OSX // [macOS]
   _preferredStatusBarStyle = [RCTUIStatusBarManager() statusBarStyle];
   _preferredStatusBarHidden = [RCTUIStatusBarManager() isStatusBarHidden];
+#endif // [macOS]
 
   return self;
 }
@@ -40,6 +41,7 @@
   }
 }
 
+#if TARGET_OS_IOS // [visionOS]
 - (UIStatusBarStyle)preferredStatusBarStyle
 {
   return _preferredStatusBarStyle;
@@ -49,6 +51,7 @@
 {
   return _preferredStatusBarHidden;
 }
+#endif // [visionOS]
 
 #if RCT_DEV && TARGET_OS_IOS
 - (UIInterfaceOrientationMask)supportedInterfaceOrientations
@@ -70,5 +73,4 @@
 #endif // RCT_DEV
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // [macOS]
