--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTModalHostViewManager.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTModalHostViewManager.m	2026-03-06 15:22:52
@@ -5,10 +5,9 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#if !TARGET_OS_OSX // [macOS]
 #import "RCTModalHostViewManager.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import "RCTBridge.h"
 #import "RCTModalHostView.h"
 #import "RCTModalHostViewController.h"
@@ -127,5 +126,4 @@
 RCT_EXPORT_VIEW_PROPERTY(onDismiss, RCTDirectEventBlock)
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // [macOS]
