--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTSwitchManager.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTSwitchManager.m	2026-03-06 15:22:52
@@ -7,8 +7,6 @@
 
 #import "RCTSwitchManager.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTUIManager.h>
 #import "RCTBridge.h"
 #import "RCTSwitch.h"
@@ -18,10 +16,15 @@
 
 RCT_EXPORT_MODULE()
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
   RCTSwitch *switcher = [RCTSwitch new];
+#if !TARGET_OS_OSX // [macOS]
   [switcher addTarget:self action:@selector(onChange:) forControlEvents:UIControlEventValueChanged];
+#else // [macOS
+  [switcher setTarget:self];
+  [switcher setAction:@selector(onChange:)];
+#endif // macOS]
   return switcher;
 }
 
@@ -37,20 +40,22 @@
 
 RCT_EXPORT_METHOD(setValue : (nonnull NSNumber *)viewTag toValue : (BOOL)value)
 {
-  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry) {
-    UIView *view = viewRegistry[viewTag];
+  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTUIView *> *viewRegistry) { // [macOS]
+    RCTUIView *view = viewRegistry[viewTag]; // [macOS]
 
-    if ([view isKindOfClass:[UISwitch class]]) {
-      [(UISwitch *)view setOn:value animated:NO];
+    if ([view isKindOfClass:[RCTSwitch class]]) {
+      [(RCTSwitch *)view setOn:value animated:NO];
     } else {
-      RCTLogError(@"view type must be UISwitch");
+      RCTLogError(@"view type must be RCTUISwitch"); // [macOS]
     }
   }];
 }
 
+#if !TARGET_OS_OSX // [macOS]
 RCT_EXPORT_VIEW_PROPERTY(onTintColor, UIColor);
 RCT_EXPORT_VIEW_PROPERTY(tintColor, UIColor);
 RCT_EXPORT_VIEW_PROPERTY(thumbTintColor, UIColor);
+#endif // [macOS]
 RCT_REMAP_VIEW_PROPERTY(value, on, BOOL);
 RCT_EXPORT_VIEW_PROPERTY(onChange, RCTBubblingEventBlock);
 RCT_CUSTOM_VIEW_PROPERTY(disabled, BOOL, RCTSwitch)
@@ -66,5 +71,3 @@
 RCT_REMAP_VIEW_PROPERTY(trackColorForTrue, onTintColor, UIColor);
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
