--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTSwitch.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTSwitch.h	2026-03-06 15:22:52
@@ -5,18 +5,13 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTComponent.h>
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTSwitch : UISwitch
+@interface RCTSwitch : RCTUISwitch
 
 @property (nonatomic, assign) BOOL wasOn;
 @property (nonatomic, copy) RCTBubblingEventBlock onChange;
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
