--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTView.h	2026-03-06 15:22:52
@@ -5,26 +5,45 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTBorderCurve.h>
 #import <React/RCTBorderStyle.h>
 #import <React/RCTComponent.h>
 #import <React/RCTCursor.h>
+#import <React/RCTEventDispatcherProtocol.h> // [macOS]
 #import <React/RCTPointerEvents.h>
 
+#if !TARGET_OS_OSX // [macOS]
 extern const UIAccessibilityTraits SwitchAccessibilityTrait;
+#endif // [macOS]
 
 @protocol RCTAutoInsetsProtocol;
 
-@interface RCTView : UIView
+@class RCTHandledKey; // [macOS]
 
+@interface RCTView : RCTUIView // [macOS]
+
+// [macOS
+- (instancetype)initWithEventDispatcher:(id<RCTEventDispatcherProtocol>)eventDispatcher;
+
+- (BOOL)becomeFirstResponder;
+- (BOOL)resignFirstResponder;
+
+#if TARGET_OS_OSX
+- (NSDictionary*)dataTransferInfoFromPasteboard:(NSPasteboard*)pasteboard;
+- (BOOL)handleKeyboardEvent:(NSEvent *)event;
+#endif
+// macOS]
+
 /**
  * Accessibility event handlers
  */
 @property (nonatomic, copy) RCTDirectEventBlock onAccessibilityAction;
 @property (nonatomic, copy) RCTDirectEventBlock onAccessibilityTap;
+#if !TARGET_OS_OSX // [macOS]
 @property (nonatomic, copy) RCTDirectEventBlock onMagicTap;
+#endif // [macOS]
 @property (nonatomic, copy) RCTDirectEventBlock onAccessibilityEscape;
 
 /**
@@ -32,8 +51,8 @@
  */
 @property (nonatomic, assign) RCTPointerEvents pointerEvents;
 
-+ (void)autoAdjustInsetsForView:(UIView<RCTAutoInsetsProtocol> *)parentView
-                 withScrollView:(UIScrollView *)scrollView
++ (void)autoAdjustInsetsForView:(RCTUIView<RCTAutoInsetsProtocol> *)parentView // [macOS]
+                 withScrollView:(RCTUIScrollView *)scrollView // [macOS]
                    updateOffset:(BOOL)updateOffset;
 
 /**
@@ -80,16 +99,16 @@
 /**
  * Border colors (actually retained).
  */
-@property (nonatomic, strong) UIColor *borderTopColor;
-@property (nonatomic, strong) UIColor *borderRightColor;
-@property (nonatomic, strong) UIColor *borderBottomColor;
-@property (nonatomic, strong) UIColor *borderLeftColor;
-@property (nonatomic, strong) UIColor *borderStartColor;
-@property (nonatomic, strong) UIColor *borderEndColor;
-@property (nonatomic, strong) UIColor *borderColor;
-@property (nonatomic, strong) UIColor *borderBlockColor;
-@property (nonatomic, strong) UIColor *borderBlockEndColor;
-@property (nonatomic, strong) UIColor *borderBlockStartColor;
+@property (nonatomic, strong) RCTUIColor *borderTopColor;
+@property (nonatomic, strong) RCTUIColor *borderRightColor;
+@property (nonatomic, strong) RCTUIColor *borderBottomColor;
+@property (nonatomic, strong) RCTUIColor *borderLeftColor;
+@property (nonatomic, strong) RCTUIColor *borderStartColor;
+@property (nonatomic, strong) RCTUIColor *borderEndColor;
+@property (nonatomic, strong) RCTUIColor *borderColor;
+@property (nonatomic, strong) RCTUIColor *borderBlockColor;
+@property (nonatomic, strong) RCTUIColor *borderBlockEndColor;
+@property (nonatomic, strong) RCTUIColor *borderBlockStartColor;
 
 /**
  * Border widths.
@@ -138,4 +157,39 @@
 @property (nonatomic, assign) RCTBubblingEventBlock onGotPointerCapture;
 @property (nonatomic, assign) RCTBubblingEventBlock onLostPointerCapture;
 
+#if TARGET_OS_OSX // [macOS
+/**
+ * macOS Properties
+ */
+
+@property (nonatomic, copy) RCTBubblingEventBlock onFocus;
+@property (nonatomic, copy) RCTBubblingEventBlock onBlur;
+
+// `allowsVibrancy` is readonly on NSView, so let's create a new property to make it assignable
+// that we can set through JS and the getter for `allowsVibrancy` can read in RCTView.
+@property (nonatomic, assign) BOOL allowsVibrancyInternal;
+
+@property (nonatomic, copy) RCTDirectEventBlock onMouseEnter;
+@property (nonatomic, copy) RCTDirectEventBlock onMouseLeave;
+@property (nonatomic, copy) RCTDirectEventBlock onDragEnter;
+@property (nonatomic, copy) RCTDirectEventBlock onDragLeave;
+@property (nonatomic, copy) RCTDirectEventBlock onDrop;
+@property (nonatomic, copy) RCTDirectEventBlock onDoubleClick;
+
+// Keyboarding events
+// NOTE does not properly work with single line text inputs (most key downs). This is because those are
+// presumably handled by the window's field editor. To make it work, we'd need to look into providing
+// a custom field editor for NSTextField controls.
+@property (nonatomic, copy) RCTBubblingEventBlock onKeyDown;
+@property (nonatomic, copy) RCTBubblingEventBlock onKeyUp;
+@property (nonatomic, copy) NSArray<RCTHandledKey*> *keyDownEvents;
+@property (nonatomic, copy) NSArray<RCTHandledKey*> *keyUpEvents;
+
+// Shadow Properties
+@property (nonatomic, assign) CGColorRef shadowColor;
+@property (nonatomic, assign) CGFloat shadowOpacity;
+@property (nonatomic, assign) CGFloat shadowRadius;
+@property (nonatomic, assign) CGSize shadowOffset;
+#endif // macOS]
+
 @end
