--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTViewManager.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTViewManager.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTBridgeModule.h>
 #import <React/RCTConvert.h>
@@ -19,7 +19,7 @@
 @class RCTSparseArray;
 @class RCTUIManager;
 
-typedef void (^RCTViewManagerUIBlock)(RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry);
+typedef void (^RCTViewManagerUIBlock)(RCTUIManager *uiManager, NSDictionary<NSNumber *, __kindof RCTPlatformView *> *viewRegistry); // [macOS]
 
 @interface RCTViewManager : NSObject <RCTBridgeModule>
 
@@ -37,7 +37,7 @@
  * return a fresh instance each time. The view module MUST NOT cache the returned
  * view and return the same instance for subsequent calls.
  */
-- (UIView *)view;
+- (RCTPlatformView *)view; // [macOS]
 
 /**
  * This method instantiates a shadow view to be managed by the module. If omitted,
@@ -79,7 +79,26 @@
     return @[ @ #type, @ #keyPath ];                    \
   }
 
+#if TARGET_OS_OSX // [macOS
 /**
+ * These macros allow properties to only be mapped in OSX
+ */
+#define RCT_EXPORT_NOT_OSX_VIEW_PROPERTY(name, type)
+#define RCT_EXPORT_OSX_VIEW_PROPERTY(name, type) \
+RCT_EXPORT_VIEW_PROPERTY(name, type)
+#define RCT_REMAP_NOT_OSX_VIEW_PROPERTY(name, keyPath, type)
+#define RCT_REMAP_OSX_VIEW_PROPERTY(name, keyPath, type) \
+RCT_REMAP_VIEW_PROPERTY(name, keyPath, type)
+#else
+#define RCT_EXPORT_NOT_OSX_VIEW_PROPERTY(name, type) \
+RCT_EXPORT_VIEW_PROPERTY(name, type)
+#define RCT_EXPORT_OSX_VIEW_PROPERTY(name, type)
+#define RCT_REMAP_NOT_OSX_VIEW_PROPERTY(name, keyPath, type) \
+RCT_REMAP_VIEW_PROPERTY(name, keyPath, type)
+#define RCT_REMAP_OSX_VIEW_PROPERTY(name, keyPath, type)
+#endif // macOS]
+
+/**
  * This macro can be used when you need to provide custom logic for setting
  * view properties. The macro should be followed by a method body, which can
  * refer to "json", "view" and "defaultView" to implement the required logic.
