--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTViewManager.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RCTViewManager.m	2026-03-06 15:22:52
@@ -24,6 +24,7 @@
 
 @implementation RCTConvert (UIAccessibilityTraits)
 
+#if !TARGET_OS_OSX // [macOS]
 RCT_MULTI_ENUM_CONVERTER(
     UIAccessibilityTraits,
     (@{
@@ -118,9 +119,15 @@
       @"treeitem" : @(UIAccessibilityTraitNone),
       @"viewgroup" : @(UIAccessibilityTraitNone),
       @"webview" : @(UIAccessibilityTraitNone),
+      // [macOS a set of RN accessibilityTraits are macOS specific accessiblity roles and map to nothing on iOS
+      @"disclosure" : @(UIAccessibilityTraitNone),
+      @"group" : @(UIAccessibilityTraitNone),
+      @"table": @(UIAccessibilityTraitNone),
+      // macOS]
     }),
     UIAccessibilityTraitNone,
     unsignedLongLongValue)
+#endif // [macOS]
 
 @end
 
@@ -137,12 +144,14 @@
 
 - (void)setBridge:(RCTBridge *)bridge
 {
+  RCTErrorNewArchitectureValidation(
+      RCTNotAllowedInBridgeless, self, @"RCTViewManager must not be initialized for the new architecture");
   _bridge = bridge;
 }
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
-  return [RCTView new];
+  return [[RCTView alloc] initWithEventDispatcher:self.bridge.eventDispatcher]; // [macOS]
 }
 
 - (RCTShadowView *)shadowView
@@ -171,35 +180,86 @@
   ];
 }
 
+#if TARGET_OS_OSX // [macOS
+RCT_EXPORT_METHOD(focus : (nonnull NSNumber *)viewTag)
+{
+  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTUIView *> *viewRegistry) {
+    RCTUIView *view = viewRegistry[viewTag];
+    [view reactFocus];
+  }];
+}
+
+RCT_EXPORT_METHOD(blur : (nonnull NSNumber *)viewTag)
+{
+  [self.bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTUIView *> *viewRegistry) {
+    RCTUIView *view = viewRegistry[viewTag];
+    [view reactBlur];
+  }];
+}
+#endif // macOS]
+
 #pragma mark - View properties
 
 // Accessibility related properties
+#if !TARGET_OS_OSX // [macOS]
 RCT_REMAP_VIEW_PROPERTY(accessible, reactAccessibilityElement.isAccessibilityElement, BOOL)
+#else // [macOS
+RCT_REMAP_VIEW_PROPERTY(accessible, reactAccessibilityElement.accessibilityElement, BOOL)
+#endif // macOS]
 RCT_REMAP_VIEW_PROPERTY(accessibilityActions, reactAccessibilityElement.accessibilityActions, NSDictionaryArray)
 RCT_REMAP_VIEW_PROPERTY(accessibilityLabel, reactAccessibilityElement.accessibilityLabel, NSString)
+#if !TARGET_OS_OSX // [macOS]
 RCT_REMAP_VIEW_PROPERTY(accessibilityHint, reactAccessibilityElement.accessibilityHint, NSString)
+#else // [macOS
+RCT_REMAP_VIEW_PROPERTY(accessibilityHint, reactAccessibilityElement.accessibilityHelp, NSString)
+#endif // macOS]
 RCT_REMAP_VIEW_PROPERTY(accessibilityLanguage, reactAccessibilityElement.accessibilityLanguage, NSString)
 RCT_REMAP_VIEW_PROPERTY(accessibilityValue, reactAccessibilityElement.accessibilityValueInternal, NSDictionary)
+#if !TARGET_OS_OSX // [macOS]
 RCT_REMAP_VIEW_PROPERTY(accessibilityViewIsModal, reactAccessibilityElement.accessibilityViewIsModal, BOOL)
 RCT_REMAP_VIEW_PROPERTY(accessibilityElementsHidden, reactAccessibilityElement.accessibilityElementsHidden, BOOL)
 RCT_REMAP_VIEW_PROPERTY(
     accessibilityIgnoresInvertColors,
     reactAccessibilityElement.shouldAccessibilityIgnoresInvertColors,
     BOOL)
+#endif // [macOS]
 RCT_REMAP_VIEW_PROPERTY(onAccessibilityAction, reactAccessibilityElement.onAccessibilityAction, RCTDirectEventBlock)
 RCT_REMAP_VIEW_PROPERTY(onAccessibilityTap, reactAccessibilityElement.onAccessibilityTap, RCTDirectEventBlock)
+#if !TARGET_OS_OSX // [macOS]
 RCT_REMAP_VIEW_PROPERTY(onMagicTap, reactAccessibilityElement.onMagicTap, RCTDirectEventBlock)
+#else // [macOS accessibilityTraits is gone in react-native and deprecated in react-native-macos, use accessibilityRole instead
+RCT_CUSTOM_VIEW_PROPERTY(accessibilityTraits, NSString, RCTView)
+{
+  view.reactAccessibilityElement.accessibilityRoleInternal = json ? [RCTConvert accessibilityRoleFromTraits:json useAriaMappings:NO] : nil;
+  if (view.reactAccessibilityElement.accessibilityRole != view.reactAccessibilityElement.accessibilityRoleInternal) {
+    [self updateAccessibilityRole:view withDefaultView:defaultView];
+  }
+}
+#endif // macOS]
 RCT_REMAP_VIEW_PROPERTY(onAccessibilityEscape, reactAccessibilityElement.onAccessibilityEscape, RCTDirectEventBlock)
 RCT_REMAP_VIEW_PROPERTY(testID, reactAccessibilityElement.accessibilityIdentifier, NSString)
 
 RCT_EXPORT_VIEW_PROPERTY(backgroundColor, UIColor)
 RCT_REMAP_VIEW_PROPERTY(backfaceVisibility, layer.doubleSided, css_backface_visibility_t)
 RCT_EXPORT_VIEW_PROPERTY(cursor, RCTCursor)
+#if !TARGET_OS_OSX // [macOS]
 RCT_REMAP_VIEW_PROPERTY(opacity, alpha, CGFloat)
+#else // [macOS
+RCT_REMAP_VIEW_PROPERTY(opacity, alphaValue, CGFloat)
+#endif // macOS]
+
+#if !TARGET_OS_OSX // [macOS]
 RCT_REMAP_VIEW_PROPERTY(shadowColor, layer.shadowColor, CGColor)
 RCT_REMAP_VIEW_PROPERTY(shadowOffset, layer.shadowOffset, CGSize)
 RCT_REMAP_VIEW_PROPERTY(shadowOpacity, layer.shadowOpacity, float)
 RCT_REMAP_VIEW_PROPERTY(shadowRadius, layer.shadowRadius, CGFloat)
+#else // [macOS
+RCT_EXPORT_VIEW_PROPERTY(shadowColor, CGColor)
+RCT_EXPORT_VIEW_PROPERTY(shadowOffset, CGSize)
+RCT_EXPORT_VIEW_PROPERTY(shadowOpacity, CGFloat)
+RCT_EXPORT_VIEW_PROPERTY(shadowRadius, CGFloat)
+#endif // macOS]
+
 RCT_REMAP_VIEW_PROPERTY(needsOffscreenAlphaCompositing, layer.allowsGroupOpacity, BOOL)
 RCT_CUSTOM_VIEW_PROPERTY(overflow, YGOverflow, RCTView)
 {
@@ -209,18 +269,21 @@
     view.clipsToBounds = defaultView.clipsToBounds;
   }
 }
+#if !TARGET_OS_OSX // [macOS]
 RCT_CUSTOM_VIEW_PROPERTY(shouldRasterizeIOS, BOOL, RCTView)
 {
   view.layer.shouldRasterize = json ? [RCTConvert BOOL:json] : defaultView.layer.shouldRasterize;
   view.layer.rasterizationScale =
       view.layer.shouldRasterize ? view.traitCollection.displayScale : defaultView.layer.rasterizationScale;
 }
+#endif // [macOS]
 
 RCT_REMAP_VIEW_PROPERTY(transform, reactTransform, CATransform3D)
 RCT_REMAP_VIEW_PROPERTY(transformOrigin, reactTransformOrigin, RCTTransformOrigin)
 
 RCT_CUSTOM_VIEW_PROPERTY(accessibilityRole, UIAccessibilityTraits, RCTView)
 {
+  #if !TARGET_OS_OSX // [macOS]
   UIAccessibilityTraits accessibilityRoleTraits =
       json ? [RCTConvert UIAccessibilityTraits:json] : UIAccessibilityTraitNone;
   if (view.reactAccessibilityElement.accessibilityRoleTraits != accessibilityRoleTraits) {
@@ -228,18 +291,36 @@
     view.reactAccessibilityElement.accessibilityRole = json ? [RCTConvert NSString:json] : nil;
     [self updateAccessibilityTraitsForRole:view withDefaultView:defaultView];
   }
+  #else // [macOS
+    // accessibilityRoleInternal is used to cache the converted value from the prop
+  view.reactAccessibilityElement.accessibilityRoleInternal = json ? [RCTConvert accessibilityRoleFromTraits:json useAriaMappings:NO] : nil;
+  // update the actual NSAccessibilityRole if it doesn't match
+  if (view.reactAccessibilityElement.accessibilityRole != view.reactAccessibilityElement.accessibilityRoleInternal) {
+    [self updateAccessibilityRole:view withDefaultView:defaultView];
+  }
+  #endif // macOS]
 }
 
 RCT_CUSTOM_VIEW_PROPERTY(role, UIAccessibilityTraits, RCTView)
 {
+#if !TARGET_OS_OSX // [macOS]
   UIAccessibilityTraits roleTraits = json ? [RCTConvert UIAccessibilityTraits:json] : UIAccessibilityTraitNone;
   if (view.reactAccessibilityElement.roleTraits != roleTraits) {
     view.roleTraits = roleTraits;
     view.reactAccessibilityElement.role = json ? [RCTConvert NSString:json] : nil;
     [self updateAccessibilityTraitsForRole:view withDefaultView:defaultView];
   }
+#else // [macOS
+  // role is used to cache the converted value from the prop
+  view.reactAccessibilityElement.role = json ? [RCTConvert accessibilityRoleFromTraits:json useAriaMappings:YES] : nil;
+  // update the actual NSAccessibilityRole if it doesn't match
+  if (view.reactAccessibilityElement.accessibilityRole != view.reactAccessibilityElement.role) {
+    [self updateAccessibilityRole:view withDefaultView:defaultView];
+  }
+#endif // macOS]
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)updateAccessibilityTraitsForRole:(RCTView *)view withDefaultView:(RCTView *)defaultView
 {
   const UIAccessibilityTraits AccessibilityRolesMask = UIAccessibilityTraitNone | UIAccessibilityTraitButton |
@@ -254,8 +335,19 @@
   view.reactAccessibilityElement.accessibilityTraits |= view.reactAccessibilityElement.role
       ? view.reactAccessibilityElement.roleTraits
       : view.reactAccessibilityElement.accessibilityRole ? view.reactAccessibilityElement.accessibilityRoleTraits
-                                                         : (defaultView.accessibilityTraits & AccessibilityRolesMask);
+                                                         : (defaultView.accessibilityTraits & AccessibilityRolesMask);                        
+}
+#else // [macOS
+- (void) updateAccessibilityRole:(RCTView *)view withDefaultView:(RCTView *)defaultView
+{
+  // First check the value from `role`
+  view.reactAccessibilityElement.accessibilityRole = view.reactAccessibilityElement.role ? view.reactAccessibilityElement.role : nil;
+  // Fallback to `accessibilityRole` if nil, or the defaultView's NSAccessibilityRole
+  if (view.reactAccessibilityElement.accessibilityRole == nil) {
+    view.reactAccessibilityElement.accessibilityRole = view.reactAccessibilityElement.accessibilityRoleInternal ? view.reactAccessibilityElement.accessibilityRoleInternal : defaultView.accessibilityRole;
+  }
 }
+#endif // macOS]
 
 RCT_CUSTOM_VIEW_PROPERTY(accessibilityState, NSDictionary, RCTView)
 {
@@ -266,6 +358,7 @@
     return;
   }
 
+#if !TARGET_OS_OSX // [macOS]
   const UIAccessibilityTraits AccessibilityStatesMask = UIAccessibilityTraitNotEnabled | UIAccessibilityTraitSelected;
   view.reactAccessibilityElement.accessibilityTraits =
       view.reactAccessibilityElement.accessibilityTraits & ~AccessibilityStatesMask;
@@ -283,16 +376,23 @@
       newState[s] = val;
     }
   }
+#else // [macOS
+  for (NSString *s in state) {
+    id val = [state objectForKey:s];
+    if (val == nil) {
+      continue;
+    }
+    newState[s] = val;
+  }
+#endif // macOS]
   if (newState.count > 0) {
     view.reactAccessibilityElement.accessibilityState = newState;
-    // Post a layout change notification to make sure VoiceOver get notified for the state
-    // changes that don't happen upon users' click.
-    UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, nil);
   } else {
     view.reactAccessibilityElement.accessibilityState = nil;
   }
 }
 
+#if !TARGET_OS_OSX // [macOS]
 RCT_CUSTOM_VIEW_PROPERTY(accessibilityShowsLargeContentViewer, BOOL, RCTView)
 {
   if (@available(iOS 13.0, *)) {
@@ -314,6 +414,7 @@
     view.largeContentTitle = json ? [RCTConvert NSString:json] : defaultView.largeContentTitle;
   }
 }
+#endif // [macOS]
 
 RCT_CUSTOM_VIEW_PROPERTY(nativeID, NSString *, RCTView)
 {
@@ -341,13 +442,11 @@
       // Unspecified values do not.
       // This wouldn't override a container view's `userInteractionEnabled = NO`
       view.userInteractionEnabled = YES;
-      break;
     case RCTPointerEventsNone:
       view.userInteractionEnabled = NO;
       break;
     default:
       RCTLogInfo(@"UIView base class does not support pointerEvent value: %@", json);
-      break;
   }
 }
 RCT_CUSTOM_VIEW_PROPERTY(removeClippedSubviews, BOOL, RCTView)
@@ -410,6 +509,47 @@
   }
 }
 
+#if TARGET_OS_OSX // [macOS
+// macOS properties
+RCT_CUSTOM_VIEW_PROPERTY(acceptsFirstMouse, BOOL, RCTView)
+{
+  if ([view respondsToSelector:@selector(setAcceptsFirstMouse:)]) {
+    view.acceptsFirstMouse = json ? [RCTConvert BOOL:json] : defaultView.acceptsFirstMouse;
+  }
+}
+
+RCT_EXPORT_VIEW_PROPERTY(mouseDownCanMoveWindow, BOOL)
+
+RCT_REMAP_VIEW_PROPERTY(allowsVibrancy, allowsVibrancyInternal, BOOL)
+
+RCT_CUSTOM_VIEW_PROPERTY(focusable, BOOL, RCTView)
+{
+  if ([view respondsToSelector:@selector(setFocusable:)]) {
+    view.focusable = json ? [RCTConvert BOOL:json] : defaultView.focusable;
+  }
+}
+
+RCT_CUSTOM_VIEW_PROPERTY(enableFocusRing, BOOL, RCTView)
+{
+  if ([view respondsToSelector:@selector(setEnableFocusRing:)]) {
+    view.enableFocusRing = json ? [RCTConvert BOOL:json] : defaultView.enableFocusRing;
+  }
+}
+
+RCT_REMAP_VIEW_PROPERTY(tooltip, toolTip, NSString)
+
+RCT_CUSTOM_VIEW_PROPERTY(draggedTypes, NSArray<NSPasteboardType>*, RCTView)
+{
+  NSArray<NSPasteboardType> *currentTypes = view.registeredDraggedTypes;
+  NSArray<NSPasteboardType> *types = json ? [RCTConvert NSPasteboardTypeArray:json] : defaultView.registeredDraggedTypes;
+  if (![currentTypes isEqualToArray:types]) {
+    [view unregisterDraggedTypes];
+  }
+  [view registerForDraggedTypes:types];
+}
+
+#endif // macOS]
+
 RCT_CUSTOM_VIEW_PROPERTY(collapsable, BOOL, RCTView)
 {
   // Property is only to be used in the new renderer.
@@ -494,6 +634,26 @@
 RCT_REMAP_VIEW_PROPERTY(display, reactDisplay, YGDisplay)
 RCT_REMAP_VIEW_PROPERTY(zIndex, reactZIndex, NSInteger)
 
+#if TARGET_OS_OSX // [macOS
+
+#pragma mark - macOS properties
+
+RCT_EXPORT_VIEW_PROPERTY(onFocus, RCTBubblingEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onBlur, RCTBubblingEventBlock)
+
+RCT_EXPORT_VIEW_PROPERTY(onDoubleClick, RCTDirectEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onMouseEnter, RCTDirectEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onMouseLeave, RCTDirectEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onDragEnter, RCTDirectEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onDragLeave, RCTDirectEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onDrop, RCTDirectEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onKeyDown, RCTBubblingEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onKeyUp, RCTBubblingEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(keyDownEvents, NSArray<RCTHandledKey *>)
+RCT_EXPORT_VIEW_PROPERTY(keyUpEvents, NSArray<RCTHandledKey *>)
+
+#endif // macOS]
+
 #pragma mark - ShadowView properties
 
 RCT_EXPORT_SHADOW_PROPERTY(top, YGValue)
