--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/UIView+Private.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/UIView+Private.h	2026-03-06 15:22:52
@@ -5,13 +5,13 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
-@interface UIView (Private)
+@interface RCTPlatformView (Private) // [macOS]
 
 // remove clipped subviews implementation
 - (void)react_remountAllSubviews;
-- (void)react_updateClippedSubviewsWithClipRect:(CGRect)clipRect relativeToView:(UIView *)clipView;
-- (UIView *)react_findClipView;
+- (void)react_updateClippedSubviewsWithClipRect:(CGRect)clipRect relativeToView:(RCTPlatformView *)clipView; // [macOS]
+- (RCTPlatformView *)react_findClipView; // [macOS]
 
 @end
