--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/UIView+React.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/UIView+React.m	2026-03-06 15:22:52
@@ -13,7 +13,7 @@
 #import "RCTLog.h"
 #import "RCTShadowView.h"
 
-@implementation UIView (React)
+@implementation RCTPlatformView (React) // [macOS]
 
 - (NSNumber *)reactTag
 {
@@ -47,12 +47,18 @@
 
 - (BOOL)shouldAccessibilityIgnoresInvertColors
 {
+#if !TARGET_OS_OSX // [macOS]
   return self.accessibilityIgnoresInvertColors;
+#else // [macOS
+  return NO;
+#endif // macOS]
 }
 
 - (void)setShouldAccessibilityIgnoresInvertColors:(BOOL)shouldAccessibilityIgnoresInvertColors
 {
+#if !TARGET_OS_OSX // [macOS]
   self.accessibilityIgnoresInvertColors = shouldAccessibilityIgnoresInvertColors;
+#endif // [macOS]
 }
 
 - (BOOL)isReactRootView
@@ -62,24 +68,24 @@
 
 - (NSNumber *)reactTagAtPoint:(CGPoint)point
 {
-  UIView *view = [self hitTest:point withEvent:nil];
+  RCTPlatformView *view = RCTUIViewHitTestWithEvent(self, point, nil); // [macOS]
   while (view && !view.reactTag) {
     view = view.superview;
   }
   return view.reactTag;
 }
 
-- (NSArray<UIView *> *)reactSubviews
+- (NSArray<RCTPlatformView *> *)reactSubviews // [macOS]
 {
   return objc_getAssociatedObject(self, _cmd);
 }
 
-- (UIView *)reactSuperview
+- (RCTPlatformView *)reactSuperview // [macOS]
 {
   return self.superview;
 }
 
-- (void)insertReactSubview:(UIView *)subview atIndex:(NSInteger)atIndex
+- (void)insertReactSubview:(RCTPlatformView *)subview atIndex:(NSInteger)atIndex // [macOS]
 {
   // We access the associated object directly here in case someone overrides
   // the `reactSubviews` getter method and returns an immutable array.
@@ -91,7 +97,7 @@
   [subviews insertObject:subview atIndex:atIndex];
 }
 
-- (void)removeReactSubview:(UIView *)subview
+- (void)removeReactSubview:(RCTPlatformView *)subview // [macOS]
 {
   // We access the associated object directly here in case someone overrides
   // the `reactSubviews` getter method and returns an immutable array.
@@ -116,23 +122,37 @@
 
 - (UIUserInterfaceLayoutDirection)reactLayoutDirection
 {
+#if !TARGET_OS_OSX // [macOS]
   if ([self respondsToSelector:@selector(semanticContentAttribute)]) {
+#pragma clang diagnostic push // [macOS]
+#pragma clang diagnostic ignored "-Wunguarded-availability" // [macOS]
     return [UIView userInterfaceLayoutDirectionForSemanticContentAttribute:self.semanticContentAttribute];
+#pragma clang diagnostic pop // [macOS]
   } else {
     return [objc_getAssociatedObject(self, @selector(reactLayoutDirection)) integerValue];
   }
+#else // [macOS
+	return self.userInterfaceLayoutDirection;
+#endif // macOS]
 }
 
 - (void)setReactLayoutDirection:(UIUserInterfaceLayoutDirection)layoutDirection
 {
+#if !TARGET_OS_OSX // [macOS]
   if ([self respondsToSelector:@selector(setSemanticContentAttribute:)]) {
+#pragma clang diagnostic push // [macOS]
+#pragma clang diagnostic ignored "-Wunguarded-availability" // [macOS]
     self.semanticContentAttribute = layoutDirection == UIUserInterfaceLayoutDirectionLeftToRight
         ? UISemanticContentAttributeForceLeftToRight
         : UISemanticContentAttributeForceRightToLeft;
+#pragma clang diagnostic pop // [macOS]
   } else {
     objc_setAssociatedObject(
         self, @selector(reactLayoutDirection), @(layoutDirection), OBJC_ASSOCIATION_RETAIN_NONATOMIC);
   }
+#else // [macOS
+	self.userInterfaceLayoutDirection	= layoutDirection;
+#endif // macOS]
 }
 
 #pragma mark - zIndex
@@ -147,17 +167,17 @@
   self.layer.zPosition = reactZIndex;
 }
 
-- (NSArray<UIView *> *)reactZIndexSortedSubviews
+- (NSArray<RCTPlatformView *> *)reactZIndexSortedSubviews // [macOS]
 {
   // Check if sorting is required - in most cases it won't be.
   BOOL sortingRequired = NO;
-  for (UIView *subview in self.subviews) {
+  for (RCTUIView *subview in self.subviews) { // [macOS]
     if (subview.reactZIndex != 0) {
       sortingRequired = YES;
       break;
     }
   }
-  return sortingRequired ? [self.reactSubviews sortedArrayUsingComparator:^NSComparisonResult(UIView *a, UIView *b) {
+  return sortingRequired ? [self.reactSubviews sortedArrayUsingComparator:^NSComparisonResult(RCTUIView *a, RCTUIView *b) { // [macOS]
     if (a.reactZIndex > b.reactZIndex) {
       return NSOrderedDescending;
     } else {
@@ -171,7 +191,7 @@
 
 - (void)didUpdateReactSubviews
 {
-  for (UIView *subview in self.reactSubviews) {
+  for (RCTPlatformView *subview in self.reactSubviews) { // [macOS]
     [self addSubview:subview];
   }
 }
@@ -183,6 +203,7 @@
 
 - (void)reactSetFrame:(CGRect)frame
 {
+#if !TARGET_OS_OSX // [macOS]
   // These frames are in terms of anchorPoint = topLeft, but internally the
   // views are anchorPoint = center for easier scale and rotation animations.
   // Convert the frame so it works with anchorPoint = center.
@@ -203,7 +224,18 @@
 
   self.center = position;
   self.bounds = bounds;
+#else // [macOS
+  // Avoid crashes due to nan coords
+  if (isnan(frame.origin.x) || isnan(frame.origin.y) ||
+      isnan(frame.size.width) || isnan(frame.size.height)) {
+    RCTLogError(@"Invalid layout for (%@)%@. frame: %@",
+                self.reactTag, self, NSStringFromCGRect(frame));
+    return;
+  }
 
+	self.frame = frame;
+#endif // macOS]
+
   id transformOrigin = objc_getAssociatedObject(self, @selector(reactTransformOrigin));
   if (transformOrigin) {
     updateTransform(self);
@@ -243,7 +275,7 @@
   updateTransform(self);
 }
 
-static void updateTransform(UIView *view)
+static void updateTransform(RCTPlatformView *view) // [macOS]
 {
   CATransform3D transform;
   id rawTansformOrigin = objc_getAssociatedObject(view, @selector(reactTransformOrigin));
@@ -281,6 +313,7 @@
 
 - (UIViewController *)reactViewController
 {
+#if !TARGET_OS_OSX // [macOS]
   id responder = [self nextResponder];
   while (responder) {
     if ([responder isKindOfClass:[UIViewController class]]) {
@@ -288,9 +321,20 @@
     }
     responder = [responder nextResponder];
   }
+#else // [macOS
+  NSView *view = self;
+  while (view) {
+    if ([view.nextResponder isKindOfClass:[NSViewController class]]) {
+      return (NSViewController *)view.nextResponder;
+    }
+    view = view.superview;
+    }
+#endif // macOS]
+
   return nil;
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)reactAddControllerToClosestParent:(UIViewController *)controller
 {
   if (!controller.parentViewController) {
@@ -306,39 +350,63 @@
     return;
   }
 }
+#endif // [macOS]
 
-/**
- * Focus manipulation.
- */
-- (BOOL)reactIsFocusNeeded
+// [macOS Github#1412
+- (void)reactViewDidMoveToWindow
 {
-  return [(NSNumber *)objc_getAssociatedObject(self, @selector(reactIsFocusNeeded)) boolValue];
+	[self reactFocusIfNeeded];
 }
+// macOS]
 
-- (void)setReactIsFocusNeeded:(BOOL)isFocusNeeded
-{
-  objc_setAssociatedObject(self, @selector(reactIsFocusNeeded), @(isFocusNeeded), OBJC_ASSOCIATION_RETAIN_NONATOMIC);
-}
+/**
+ * Focus manipulation.
+ */
+static __weak RCTPlatformView *_pendingFocusView; // [macOS]
 
 - (void)reactFocus
 {
-  if (![self becomeFirstResponder]) {
-    self.reactIsFocusNeeded = YES;
+#if !TARGET_OS_OSX // [macOS]
+	if (![self becomeFirstResponder]) {
+#else // [macOS
+	if (![[self window] makeFirstResponder:self]) {
+#endif // macOS]
+// [macOS Github#1412
+    _pendingFocusView = self;
+  } else {
+    _pendingFocusView = nil;
   }
+// macOS]
 }
 
 - (void)reactFocusIfNeeded
 {
-  if (self.reactIsFocusNeeded) {
-    if ([self becomeFirstResponder]) {
-      self.reactIsFocusNeeded = NO;
-    }
-  }
+	if ([self isEqual:_pendingFocusView]) { // [macOS]
+#if TARGET_OS_OSX // [macOS
+		if ([[self window] makeFirstResponder:self]) {
+#else
+		if ([self becomeFirstResponder]) {
+#endif // macOS]
+			_pendingFocusView = nil; // [macOS] Github#1412
+		}
+	}
 }
 
 - (void)reactBlur
 {
+#if !TARGET_OS_OSX // [macOS]
   [self resignFirstResponder];
+#else // [macOS
+  if (self == [[self window] firstResponder]) {
+    [[self window] makeFirstResponder:[[self window] nextResponder]];
+  }
+#endif // macOS]
+
+// [macOS Github#1412
+  if ([self isEqual:_pendingFocusView]) {
+    _pendingFocusView = nil;
+  }
+// macOS]
 }
 
 #pragma mark - Layout
@@ -373,7 +441,7 @@
 
 #pragma mark - Accessibility
 
-- (UIView *)reactAccessibilityElement
+- (RCTPlatformView *)reactAccessibilityElement // [macOS]
 {
   return self;
 }
@@ -400,14 +468,26 @@
       self, @selector(accessibilityLanguage), accessibilityLanguage, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (NSString *)accessibilityRole
+#else // [macOS renamed so it doesn't conflict with -[NSAccessibility accessibilityRole].
+- (NSString *)accessibilityRoleInternal
+#endif // macOS]
 {
   return objc_getAssociatedObject(self, _cmd);
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)setAccessibilityRole:(NSString *)accessibilityRole
+#else // [macOS renamed so it doesn't conflict with -[NSAccessibility accessibilityRole].
+- (void)setAccessibilityRoleInternal:(NSString *)accessibilityRole
+#endif // macOS]
 {
+#if !TARGET_OS_OSX // [macOS]
   objc_setAssociatedObject(self, @selector(accessibilityRole), accessibilityRole, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+#else // [macOS
+  objc_setAssociatedObject(self, @selector(accessibilityRoleInternal), accessibilityRole, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+#endif // macOS]
 }
 
 - (NSString *)role
@@ -440,6 +520,7 @@
       self, @selector(accessibilityValueInternal), accessibilityValue, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (UIAccessibilityTraits)accessibilityRoleTraits
 {
   NSNumber *traitsAsNumber = objc_getAssociatedObject(self, _cmd);
@@ -466,6 +547,7 @@
   objc_setAssociatedObject(
       self, @selector(roleTraits), [NSNumber numberWithUnsignedLongLong:roleTraits], OBJC_ASSOCIATION_RETAIN_NONATOMIC);
 }
+#endif // [macOS]
 
 #pragma mark - Debug
 - (void)react_addRecursiveDescriptionToString:(NSMutableString *)string atLevel:(NSUInteger)level
@@ -477,7 +559,7 @@
   [string appendString:self.description];
   [string appendString:@"\n"];
 
-  for (UIView *subview in self.subviews) {
+  for (RCTPlatformView *subview in self.subviews) { // [macOS]
     [subview react_addRecursiveDescriptionToString:string atLevel:level + 1];
   }
 }
@@ -489,4 +571,16 @@
   return description;
 }
 
+// [macOS
+#pragma mark - Hit testing
+#if TARGET_OS_OSX
+// IMPORTANT -- point is in local coordinate space, unlike the hitTest: version from OSX which is in super's coordinate space
+- (RCTPlatformView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
+{
+  NSView *superview = [self superview];
+  NSPoint pointInSuper = superview != nil ? [self convertPoint:point toView:superview] : point;
+  return [self hitTest:pointInSuper];
+}
+#endif // macOS]
+
 @end
