--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/InputAccessory/RCTInputAccessoryContentView.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/ComponentViews/InputAccessory/RCTInputAccessoryContentView.mm	2026-03-06 15:22:52
@@ -8,7 +8,7 @@
 #import "RCTInputAccessoryContentView.h"
 
 @implementation RCTInputAccessoryContentView {
-  UIView *_safeAreaContainer;
+  RCTUIView *_safeAreaContainer; // [macOS]
   NSLayoutConstraint *_heightConstraint;
 }
 
@@ -17,7 +17,7 @@
   if (self = [super init]) {
     self.autoresizingMask = UIViewAutoresizingFlexibleHeight;
 
-    _safeAreaContainer = [UIView new];
+    _safeAreaContainer = [RCTUIView new]; // [macOS]
     _safeAreaContainer.translatesAutoresizingMaskIntoConstraints = NO;
     [self addSubview:_safeAreaContainer];
 
@@ -40,7 +40,7 @@
   return CGSizeZero;
 }
 
-- (void)insertSubview:(UIView *)view atIndex:(NSInteger)index
+- (void)insertSubview:(RCTPlatformView *)view atIndex:(NSInteger)index // [macOS]
 {
   [_safeAreaContainer insertSubview:view atIndex:index];
 }
