--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/nativeImageSource.js	2026-03-06 18:19:27
+++ /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/nativeImageSource.js.patched	2026-03-06 18:50:43
@@ -15,6 +15,7 @@
 type NativeImageSourceSpec = $ReadOnly<{
   android?: string,
   ios?: string,
+  macos?: string, // [macOS]
   default?: string,

   // For more details on width and height, see
@@ -44,6 +45,7 @@
     android: spec.android,
     default: spec.default,
     ios: spec.ios,
+    macos: spec.macos, // [macOS]
   });
   if (uri == null) {
     console.warn(
