--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/RCTDisplayWeakRefreshable.mm	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.ygQILyZnPY	2026-03-06 18:50:42
@@ -9,10 +9,10 @@
 
 @implementation RCTDisplayWeakRefreshable
 
-+ (CADisplayLink *)displayLinkWithWeakRefreshable:(id<RCTDisplayRefreshable>)refreshable
++ (RCTPlatformDisplayLink *)displayLinkWithWeakRefreshable:(id<RCTDisplayRefreshable>)refreshable // [macOS]
 {
   RCTDisplayWeakRefreshable *target = [[RCTDisplayWeakRefreshable alloc] initWithRefreshable:refreshable];
-  return [CADisplayLink displayLinkWithTarget:target selector:@selector(displayDidRefresh:)];
+  return [RCTPlatformDisplayLink displayLinkWithTarget:target selector:@selector(displayDidRefresh:)]; // [macOS]
 }
 
 - (instancetype)initWithRefreshable:(id<RCTDisplayRefreshable>)refreshable
@@ -23,7 +23,7 @@
   return self;
 }
 
-- (void)displayDidRefresh:(CADisplayLink *)displayLink
+- (void)displayDidRefresh:(RCTPlatformDisplayLink *)displayLink // [macOS]
 {
   [_refreshable displayDidRefresh:displayLink];
 }
