--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Image/RCTImageLoaderProtocol.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.oJotT7GNJ4	2026-03-06 18:50:43
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTBridge.h>
 #import <React/RCTImageCache.h>
@@ -129,7 +129,6 @@
  * protocol. This method should be called in bridgeDidInitializeModule.
  */
 - (void)setImageCache:(id<RCTImageCache>)cache;
-
 @end
 
 NS_ASSUME_NONNULL_END
