--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Modal/Modal.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Modal/Modal.js	2026-03-06 15:22:52
@@ -18,7 +18,7 @@
 import {type EventSubscription} from '../vendor/emitter/EventEmitter';
 import NativeModalManager from './NativeModalManager';
 import RCTModalHostView from './RCTModalHostViewNativeComponent';
-import VirtualizedLists from '@react-native/virtualized-lists';
+import VirtualizedLists from '@react-native-macos/virtualized-lists'; // [macOS]
 import * as React from 'react';
 
 const ScrollView = require('../Components/ScrollView/ScrollView').default;
@@ -292,7 +292,7 @@
       backgroundColor:
         this.props.transparent === true
           ? 'transparent'
-          : (this.props.backdropColor ?? 'white'),
+          : this.props.backdropColor ?? 'white',
     };
 
     let animationType = this.props.animationType || 'none';
@@ -370,13 +370,13 @@
   modal: {
     position: 'absolute',
   },
-  /* $FlowFixMe[incompatible-type] Natural Inference rollout. See
+  /* $FlowFixMe[incompatible-call] Natural Inference rollout. See
    * https://fburl.com/workplace/6291gfvu */
   container: {
     /* $FlowFixMe[invalid-computed-prop] (>=0.111.0 site=react_native_fb) This
      * comment suppresses an error found when Flow v0.111 was deployed. To see
      * the error, delete this comment and run Flow. */
-    // $FlowFixMe[incompatible-type]
+    // $FlowFixMe[incompatible-call]
     [side]: 0,
     top: 0,
     flex: 1,
