--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/NativeAnimation/Nodes/RCTInterpolationAnimatedNode.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/NativeAnimation/Nodes/RCTInterpolationAnimatedNode.mm	2026-03-06 15:22:52
@@ -86,7 +86,7 @@
 
 - (instancetype)initWithTag:(NSNumber *)tag config:(NSDictionary<NSString *, id> *)config
 {
-  if ((self = [super initWithTag:tag config:config]) != nullptr) {
+  if ((self = [super initWithTag:tag config:config])) {
     _inputRange = config[@"inputRange"];
 
     NSArray *outputRangeConfig = config[@"outputRange"];
@@ -103,8 +103,8 @@
     for (id value in outputRangeConfig) {
       switch (_outputType) {
         case RCTInterpolationOutputColor: {
-          UIColor *color = [RCTConvert UIColor:value];
-          [outputRange addObject:(color != nullptr) ? color : [UIColor whiteColor]];
+          RCTUIColor *color = [RCTConvert UIColor:value]; // [macOS]
+          [outputRange addObject:color ? color : [RCTUIColor whiteColor]]; // [macOS]
           break;
         }
         case RCTInterpolationOutputString:
@@ -141,7 +141,7 @@
 - (void)performUpdate
 {
   [super performUpdate];
-  if (_parentNode == nullptr) {
+  if (!_parentNode) {
     return;
   }
 
