--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/TextInput/Multiline/RCTMultilineTextInputView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/TextInput/Multiline/RCTMultilineTextInputView.h	2026-03-06 15:22:52
@@ -7,15 +7,18 @@
 
 #import "RCTBaseTextInputView.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 NS_ASSUME_NONNULL_BEGIN
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
 @interface RCTMultilineTextInputView : RCTBaseTextInputView
 
+#if TARGET_OS_OSX // [macOS
+
+@property (nonatomic, assign) BOOL scrollEnabled;
+
+- (void)setReadablePasteBoardTypes:(NSArray<NSPasteboardType> *)readablePasteboardTypes;
+@property (nonatomic, assign) BOOL hideVerticalScrollIndicator;
+#endif // macOS]
+
 @end
 
 NS_ASSUME_NONNULL_END
-
-#endif // RCT_REMOVE_LEGACY_ARCH
