--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/TextInput/Multiline/RCTMultilineTextInputViewManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/TextInput/Multiline/RCTMultilineTextInputViewManager.mm	2026-03-06 15:22:52
@@ -7,22 +7,32 @@
 
 #import <React/RCTMultilineTextInputView.h>
 #import <React/RCTMultilineTextInputViewManager.h>
+#import <React/RCTUITextView.h> // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 @implementation RCTMultilineTextInputViewManager
 
 RCT_EXPORT_MODULE()
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
   return [[RCTMultilineTextInputView alloc] initWithBridge:self.bridge];
 }
 
 #pragma mark - Multiline <TextInput> (aka TextView) specific properties
 
-RCT_REMAP_VIEW_PROPERTY(dataDetectorTypes, backedTextInputView.dataDetectorTypes, UIDataDetectorTypes)
+RCT_REMAP_NOT_OSX_VIEW_PROPERTY(dataDetectorTypes, backedTextInputView.dataDetectorTypes, UIDataDetectorTypes) // [macOS]
+RCT_REMAP_OSX_VIEW_PROPERTY(dataDetectorTypes, backedTextInputView.enabledTextCheckingTypes, NSTextCheckingTypes) // [macOS]
 
-@end
+#if TARGET_OS_OSX // [macOS
+RCT_EXPORT_VIEW_PROPERTY(scrollEnabled, BOOL)
+RCT_EXPORT_VIEW_PROPERTY(hideVerticalScrollIndicator, BOOL)
+RCT_CUSTOM_VIEW_PROPERTY(pastedTypes, NSArray<NSPasteboardType>*, RCTUITextView)
+{
+  NSArray<NSPasteboardType> *types = json ? [RCTConvert NSPasteboardTypeArray:json] : nil;
+  if ([view respondsToSelector:@selector(setReadablePasteBoardTypes:)]) {
+    [view setReadablePasteBoardTypes: types];
+  }
+}
+#endif // macOS]
 
-#endif // RCT_REMOVE_LEGACY_ARCH
+@end
