--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Lists/VirtualizedList.js	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.LwOosLik5r	2026-03-06 18:50:44
@@ -10,7 +10,6 @@
 
 'use strict';
 
-import VirtualizedLists from '@react-native/virtualized-lists';
 
 type VirtualizedListType = typeof VirtualizedLists.VirtualizedList;
 const VirtualizedListComponent: VirtualizedListType =
@@ -22,5 +21,3 @@
   ListViewToken,
   Separators,
   VirtualizedListProps,
-} from '@react-native/virtualized-lists';
-export default VirtualizedListComponent;
