--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Lists/VirtualizedSectionList.js	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.7H3VpAs8yO	2026-03-06 18:50:44
@@ -10,7 +10,6 @@
 
 'use strict';
 
-import VirtualizedLists from '@react-native/virtualized-lists';
 
 type VirtualizedSectionListType =
   typeof VirtualizedLists.VirtualizedSectionList;
@@ -21,5 +20,4 @@
   SectionBase,
   ScrollToLocationParamsType,
   VirtualizedSectionListProps,
-} from '@react-native/virtualized-lists';
 export default VirtualizedSectionList;
