--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTWebSocketModule.mm	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.8ujYlbAZgy	2026-03-06 18:50:49
@@ -66,38 +66,40 @@
     connect : (NSURL *)URL protocols : (NSArray *)protocols options : (JS::NativeWebSocketModule::SpecConnectOptions &)
         options socketID : (double)socketID)
 {
-  NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
+  RCTAssertParam(URL); // [macOS] prevent crashes when URL is erroneously nil
+  if (URL != nil) { // [macOS] prevent crashes when URL is erroneously nil
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
 
-  // We load cookies from sharedHTTPCookieStorage (shared with XHR and
-  // fetch). To get secure cookies for wss URLs, replace wss with https
-  // in the URL.
-  NSURLComponents *components = [NSURLComponents componentsWithURL:URL resolvingAgainstBaseURL:true];
-  if ([components.scheme.lowercaseString isEqualToString:@"wss"]) {
-    components.scheme = @"https";
-  }
+    // We load cookies from sharedHTTPCookieStorage (shared with XHR and
+    // fetch). To get secure cookies for wss URLs, replace wss with https
+    // in the URL.
+    NSURLComponents *components = [NSURLComponents componentsWithURL:URL resolvingAgainstBaseURL:true];
+    if ([components.scheme.lowercaseString isEqualToString:@"wss"]) {
+      components.scheme = @"https";
+    }
 
-  // Load and set the cookie header.
-  NSArray<NSHTTPCookie *> *cookies = [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookiesForURL:components.URL];
-  request.allHTTPHeaderFields = [NSHTTPCookie requestHeaderFieldsWithCookies:cookies];
+    // Load and set the cookie header.
+    NSArray<NSHTTPCookie *> *cookies = [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookiesForURL:components.URL];
+    request.allHTTPHeaderFields = [NSHTTPCookie requestHeaderFieldsWithCookies:cookies];
 
-  // Load supplied headers
-  if ([options.headers() isKindOfClass:NSDictionary.class]) {
-    NSDictionary *headers = (NSDictionary *)options.headers();
-    [headers enumerateKeysAndObjectsUsingBlock:^(NSString *key, id value, BOOL *stop) {
-      [request addValue:[RCTConvert NSString:value] forHTTPHeaderField:key];
-    }];
-  }
+    // Load supplied headers
+    if ([options.headers() isKindOfClass:NSDictionary.class]) {
+      NSDictionary *headers = (NSDictionary *)options.headers();
+      [headers enumerateKeysAndObjectsUsingBlock:^(NSString *key, id value, BOOL *stop) {
+        [request addValue:[RCTConvert NSString:value] forHTTPHeaderField:key];
+      }];
+    }
 
-  SRWebSocket *webSocket = [[SRWebSocket alloc] initWithURLRequest:request protocols:protocols];
-  [webSocket setDelegateDispatchQueue:[self methodQueue]];
-  webSocket.delegate = self;
-  webSocket.reactTag = @(socketID);
-  if (!_sockets) {
-    _sockets = [NSMutableDictionary new];
-  }
-  _sockets[@(socketID)] = webSocket;
-  [webSocket open];
-}
+    SRWebSocket *webSocket = [[SRWebSocket alloc] initWithURLRequest:request protocols:protocols];
+    [webSocket setDelegateDispatchQueue:[self methodQueue]];
+    webSocket.delegate = self;
+    webSocket.reactTag = @(socketID);
+    if (!_sockets) {
+      _sockets = [NSMutableDictionary new];
+    }
+    _sockets[@(socketID)] = webSocket;
+    [webSocket open];
+  } // [macOS] prevent crashes when URL is erroneously nil
 
 RCT_EXPORT_METHOD(send : (NSString *)message forSocketID : (double)socketID)
 {
