--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/React-Fabric.podspec	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.BcEMFiRIrz	2026-03-06 18:50:53
@@ -32,7 +32,10 @@
                             "CLANG_CXX_LANGUAGE_STANDARD" => rct_cxx_language_standard(),
                             "DEFINES_MODULE" => "YES" }
 
-  resolve_use_frameworks(s, header_mappings_dir: "./", module_name: "React_Fabric")
+  if ENV['USE_FRAMEWORKS']
+    s.header_mappings_dir     = './'
+    s.module_name             = 'React_Fabric'
+  end
 
   s.dependency "React-jsiexecutor"
   s.dependency "RCTRequired"
@@ -53,7 +56,6 @@
 
   depend_on_js_engine(s)
   add_rn_third_party_dependencies(s)
-  add_rncore_dependency(s)
 
   s.subspec "animated" do |ss|
     ss.source_files         = podspec_sources("react/renderer/animated/**/*.{m,mm,cpp,h}", "react/renderer/animated/**/*.{h}")
@@ -129,8 +131,8 @@
     ss.subspec "view" do |sss|
       sss.dependency             "React-renderercss"
       sss.dependency             "Yoga"
-      sss.source_files         = podspec_sources(["react/renderer/components/view/*.{m,mm,cpp,h}", "react/renderer/components/view/platform/cxx/**/*.{m,mm,cpp,h}"], ["react/renderer/components/view/*.{h}", "react/renderer/components/view/platform/cxx/**/*.{h}"])
-      sss.header_dir           = "react/renderer/components/view"
+      sss.source_files         = "react/renderer/components/view/**/*.{m,mm,cpp,h}" # [macOS]
+      sss.exclude_files        = "react/renderer/components/view/tests", "react/renderer/components/view/platform/android", "react/renderer/components/view/platform/windows" # [macOS]
     end
 
     ss.subspec "scrollview" do |sss|
