--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/CoreModulesPlugins.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/CoreModulesPlugins.mm	2026-03-06 15:22:52
@@ -13,105 +13,46 @@
 
 #import "CoreModulesPlugins.h"
 
-#import <string_view>
+#import <string>
+#import <unordered_map>
 
-using namespace std::literals;
+Class RCTCoreModulesClassProvider(const char *name) {
+  // Intentionally leak to avoid crashing after static destructors are run.
+  static const auto sCoreModuleClassMap = new const std::unordered_map<std::string, Class (*)(void)>{
+#if !TARGET_OS_OSX // [macOS] Do we need these?
+    {"AccessibilityManager", RCTAccessibilityManagerCls},
+#endif // [macOS]
+    {"ActionSheetManager", RCTActionSheetManagerCls},
+    {"AlertManager", RCTAlertManagerCls},
+    {"AppState", RCTAppStateCls},
+    {"Appearance", RCTAppearanceCls},
+    {"Clipboard", RCTClipboardCls},
+    {"DevLoadingView", RCTDevLoadingViewCls},
+    {"DevMenu", RCTDevMenuCls},
+    {"DevSettings", RCTDevSettingsCls},
+    {"DeviceInfo", RCTDeviceInfoCls},
+    {"EventDispatcher", RCTEventDispatcherCls},
+    {"ExceptionsManager", RCTExceptionsManagerCls},
+    {"I18nManager", RCTI18nManagerCls},
+    {"KeyboardObserver", RCTKeyboardObserverCls},
+    {"LogBox", RCTLogBoxCls},
+#if !TARGET_OS_OSX // [macOS] Do we need these?
+    {"PerfMonitor", RCTPerfMonitorCls},
+#endif // [macOS]
+    {"PlatformConstants", RCTPlatformCls},
+    {"RedBox", RCTRedBoxCls},
+    {"SourceCode", RCTSourceCodeCls},
+    {"StatusBarManager", RCTStatusBarManagerCls},
+    {"Timing", RCTTimingCls},
+    {"WebSocketModule", RCTWebSocketModuleCls},
+    {"BlobModule", RCTBlobManagerCls},
+  };
 
-Class RCTCoreModulesClassProvider(const char *name)
-{
-
-  if (name == "AccessibilityManager"sv) {
-    return RCTAccessibilityManagerCls();
+  auto p = sCoreModuleClassMap->find(name);
+  if (p != sCoreModuleClassMap->end()) {
+    auto classFunc = p->second;
+    return classFunc();
   }
-
-  if (name == "ActionSheetManager"sv) {
-    return RCTActionSheetManagerCls();
-  }
-
-  if (name == "AlertManager"sv) {
-    return RCTAlertManagerCls();
-  }
-
-  if (name == "AppState"sv) {
-    return RCTAppStateCls();
-  }
-
-  if (name == "Appearance"sv) {
-    return RCTAppearanceCls();
-  }
-
-  if (name == "Clipboard"sv) {
-    return RCTClipboardCls();
-  }
-
-  if (name == "DevLoadingView"sv) {
-    return RCTDevLoadingViewCls();
-  }
-
-  if (name == "DevMenu"sv) {
-    return RCTDevMenuCls();
-  }
-
-  if (name == "DevSettings"sv) {
-    return RCTDevSettingsCls();
-  }
-
-  if (name == "DeviceInfo"sv) {
-    return RCTDeviceInfoCls();
-  }
-
-  if (name == "EventDispatcher"sv) {
-    return RCTEventDispatcherCls();
-  }
-
-  if (name == "ExceptionsManager"sv) {
-    return RCTExceptionsManagerCls();
-  }
-
-  if (name == "I18nManager"sv) {
-    return RCTI18nManagerCls();
-  }
-
-  if (name == "KeyboardObserver"sv) {
-    return RCTKeyboardObserverCls();
-  }
-
-  if (name == "LogBox"sv) {
-    return RCTLogBoxCls();
-  }
-
-  if (name == "PerfMonitor"sv) {
-    return RCTPerfMonitorCls();
-  }
-
-  if (name == "PlatformConstants"sv) {
-    return RCTPlatformCls();
-  }
-
-  if (name == "RedBox"sv) {
-    return RCTRedBoxCls();
-  }
-
-  if (name == "SourceCode"sv) {
-    return RCTSourceCodeCls();
-  }
-
-  if (name == "StatusBarManager"sv) {
-    return RCTStatusBarManagerCls();
-  }
-
-  if (name == "Timing"sv) {
-    return RCTTimingCls();
-  }
-
-  if (name == "WebSocketModule"sv) {
-    return RCTWebSocketModuleCls();
-  }
-
-  if (name == "BlobModule"sv) {
-    return RCTBlobManagerCls();
-  }
-
   return nil;
 }
 
