--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTAppearance.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/RCTAppearance.mm	2026-03-06 15:22:52
@@ -13,6 +13,10 @@
 
 #import "CoreModulesPlugins.h"
 
+#if TARGET_OS_OSX // [macOS
+#import <React/RCTUtils.h>
+#endif // macOS]
+
 using namespace facebook::react;
 
 NSString *const RCTAppearanceColorSchemeLight = @"light";
@@ -41,6 +45,7 @@
   return sColorSchemeOverride;
 }
 
+#if !TARGET_OS_OSX // [macOS]
 NSString *RCTColorSchemePreference(UITraitCollection *traitCollection)
 {
   static NSDictionary *appearances;
@@ -71,7 +76,31 @@
 
   return appearances[@(systemStyle)] ?: RCTAppearanceColorSchemeLight;
 }
+#else // [macOS
+NSString *RCTColorSchemePreference(NSAppearance *appearance)
+{
+  static NSDictionary *appearances;
+  static dispatch_once_t onceToken;
 
+  dispatch_once(&onceToken, ^{
+    appearances = @{
+                    NSAppearanceNameAqua: RCTAppearanceColorSchemeLight,
+                    NSAppearanceNameDarkAqua: RCTAppearanceColorSchemeDark
+                    };
+  });
+
+  if (!sAppearancePreferenceEnabled) {
+    // Return the default if the app doesn't allow different color schemes.
+    return RCTAppearanceColorSchemeLight;
+  }
+
+  appearance = appearance ?: [NSApp effectiveAppearance];
+
+  NSAppearanceName appearanceName = [appearance bestMatchFromAppearancesWithNames:@[NSAppearanceNameAqua, NSAppearanceNameDarkAqua]];
+  return appearances[appearanceName] ?: RCTAppearanceColorSchemeLight;
+}
+#endif // macOS]
+
 @interface RCTAppearance () <NativeAppearanceSpec>
 @end
 
@@ -82,8 +111,13 @@
 - (instancetype)init
 {
   if ((self = [super init])) {
+#if !TARGET_OS_OSX // [macOS]
     UITraitCollection *traitCollection = RCTKeyWindow().traitCollection;
     _currentColorScheme = RCTColorSchemePreference(traitCollection);
+#else // [macOS
+  NSAppearance *appearance = RCTKeyWindow().effectiveAppearance;
+  _currentColorScheme = RCTColorSchemePreference(appearance);
+#endif // macOS]
     [[NSNotificationCenter defaultCenter] addObserver:self
                                              selector:@selector(appearanceChanged:)
                                                  name:RCTUserInterfaceStyleDidChangeNotification
@@ -111,6 +145,7 @@
 
 RCT_EXPORT_METHOD(setColorScheme : (NSString *)style)
 {
+#if !TARGET_OS_OSX // [macOS]
   UIUserInterfaceStyle userInterfaceStyle = [RCTConvert UIUserInterfaceStyle:style];
   NSMutableArray<UIWindow *> *windows = [NSMutableArray new];
   for (UIWindowScene *scene in RCTSharedApplication().connectedScenes) {
@@ -120,6 +155,17 @@
   for (UIWindow *window in windows) {
     window.overrideUserInterfaceStyle = userInterfaceStyle;
   }
+#else // [macOS
+  RCTExecuteOnMainQueue(^{
+    NSAppearance *appearance = nil;
+    if ([style isEqualToString:@"light"]) {
+      appearance = [NSAppearance appearanceNamed:NSAppearanceNameAqua];
+    } else if ([style isEqualToString:@"dark"]) {
+      appearance = [NSAppearance appearanceNamed:NSAppearanceNameDarkAqua];
+    }
+    RCTSharedApplication().appearance = appearance;
+  });
+#endif // macOS]
 }
 
 RCT_EXPORT_SYNCHRONOUS_TYPED_METHOD(NSString *, getColorScheme)
@@ -127,17 +173,26 @@
   return _currentColorScheme;
 }
 
+
 - (void)appearanceChanged:(NSNotification *)notification
 {
   NSDictionary *userInfo = [notification userInfo];
+#if !TARGET_OS_OSX // [macOS
   UITraitCollection *traitCollection = nil;
   if (userInfo) {
     traitCollection = userInfo[RCTUserInterfaceStyleDidChangeNotificationTraitCollectionKey];
   }
   NSString *newColorScheme = RCTColorSchemePreference(traitCollection);
+#else // [macOS
+  NSAppearance *appearance = nil;
+  if (userInfo) {
+    appearance = userInfo[RCTUserInterfaceStyleDidChangeNotificationAppearanceKey];
+  }
+  NSString *newColorScheme = RCTColorSchemePreference(appearance);
+#endif // macOS]
   if (![_currentColorScheme isEqualToString:newColorScheme]) {
     _currentColorScheme = newColorScheme;
-    [self sendEventWithName:@"appearanceChanged" body:@{@"colorScheme" : newColorScheme}];
+    [self sendEventWithName:@"appearanceChanged" body:@{ @"colorScheme" : newColorScheme }];
   }
 }
 
