--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTAccessibilityManager.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/RCTAccessibilityManager.h	2026-03-06 15:22:52
@@ -22,6 +22,7 @@
 
 @property (nonatomic, assign) BOOL isBoldTextEnabled;
 @property (nonatomic, assign) BOOL isGrayscaleEnabled;
+@property (nonatomic, assign) BOOL isHighContrastEnabled; // [macOS] maps to shouldIncreaseContrast on macOS
 @property (nonatomic, assign) BOOL isInvertColorsEnabled;
 @property (nonatomic, assign) BOOL isReduceMotionEnabled;
 @property (nonatomic, assign) BOOL isDarkerSystemColorsEnabled;
