--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTAlertController.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/RCTAlertController.mm	2026-03-06 15:22:52
@@ -11,23 +11,21 @@
 
 @interface RCTAlertController ()
 
+#if !TARGET_OS_OSX // [macOS]
 @property (nonatomic, strong) UIWindow *alertWindow;
+#endif // [macOS]
 
 @end
 
 @implementation RCTAlertController
 
+#if !TARGET_OS_OSX // [macOS]
 - (UIWindow *)alertWindow
 {
   if (_alertWindow == nil) {
-    UIWindowScene *scene = RCTKeyWindow().windowScene;
-    if (scene != nil) {
-      _alertWindow = [[UIWindow alloc] initWithWindowScene:scene];
-    } else {
-      _alertWindow = [[UIWindow alloc] initWithFrame:UIScreen.mainScreen.bounds];
-    }
+    _alertWindow = [[UIWindow alloc] initWithWindowScene:RCTKeyWindow().windowScene];
 
-    if (_alertWindow != nullptr) {
+    if (_alertWindow) {
       _alertWindow.rootViewController = [UIViewController new];
       _alertWindow.windowLevel = UIWindowLevelAlert + 1;
     }
@@ -41,7 +39,7 @@
   UIUserInterfaceStyle style = self.overrideUserInterfaceStyle;
   if (style == UIUserInterfaceStyleUnspecified) {
     UIUserInterfaceStyle overriddenStyle = RCTKeyWindow().overrideUserInterfaceStyle;
-    style = (overriddenStyle != 0) ? overriddenStyle : UIUserInterfaceStyleUnspecified;
+    style = overriddenStyle ? overriddenStyle : UIUserInterfaceStyleUnspecified;
   }
 
   self.overrideUserInterfaceStyle = style;
@@ -58,5 +56,5 @@
 
   _alertWindow = nil;
 }
-
+#endif // [macOS]
 @end
