--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTDeviceInfo.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/RCTDeviceInfo.h	2026-03-06 15:22:52
@@ -4,13 +4,11 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  */
+ 
+#import <React/RCTUIKit.h> // [macOS]
 
-#import <UIKit/UIKit.h>
-
 #import <React/RCTBridge.h>
 
 @interface RCTDeviceInfo : NSObject <RCTBridgeModule>
 
-- (instancetype)initWithDimensionsProvider:(NSDictionary * (^)(void))dimensionsProvider;
-
 @end
