--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTDevMenu.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/RCTDevMenu.mm	2026-03-06 15:22:52
@@ -12,7 +12,9 @@
 #import <React/RCTBundleURLProvider.h>
 #import <React/RCTDefines.h>
 #import <React/RCTDevSettings.h>
+#if !TARGET_OS_OSX // [macOS]
 #import <React/RCTKeyCommands.h>
+#endif // [macOS]
 #import <React/RCTLog.h>
 #import <React/RCTReloadCommand.h>
 #import <React/RCTUtils.h>
@@ -29,31 +31,18 @@
 
 NSString *const RCTShowDevMenuNotification = @"RCTShowDevMenuNotification";
 
-@implementation RCTDevMenuConfiguration
-- (instancetype)initWithDevMenuEnabled:(BOOL)devMenuEnabled
-                   shakeGestureEnabled:(BOOL)shakeGestureEnabled
-              keyboardShortcutsEnabled:(BOOL)keyboardShortcutsEnabled
-{
-  if (self = [super init]) {
-    _devMenuEnabled = devMenuEnabled;
-    _shakeGestureEnabled = shakeGestureEnabled;
-    _keyboardShortcutsEnabled = keyboardShortcutsEnabled;
-  }
-  return self;
-}
+#if !TARGET_OS_OSX // [macOS]
 
-+ (instancetype)defaultConfiguration
-{
-  return [[self alloc] initWithDevMenuEnabled:RCT_DEV_MENU
-                          shakeGestureEnabled:RCT_DEV_MENU
-                     keyboardShortcutsEnabled:RCT_DEV_MENU];
-}
-@end
+// [macOS
+typedef void (*MotionEndedWithEventImpType)(id self, SEL selector, UIEventSubtype motion, UIEvent *event);
+static MotionEndedWithEventImpType RCTOriginalUIWindowMotionEndedWithEventImp = nil;
+// macOS]
 
 @implementation UIWindow (RCTDevMenu)
 
 - (void)RCT_motionEnded:(__unused UIEventSubtype)motion withEvent:(UIEvent *)event
 {
+  RCTOriginalUIWindowMotionEndedWithEventImp(self, @selector(motionEnded:withEvent:), motion, event); // [macOS]
   if (event.subtype == UIEventSubtypeMotionShake) {
     [[NSNotificationCenter defaultCenter] postNotificationName:RCTShowDevMenuNotification object:nil];
   }
@@ -61,6 +50,8 @@
 
 @end
 
+#endif // [macOS]
+
 @implementation RCTDevMenuItem {
   RCTDevMenuItemTitleBlock _titleBlock;
   dispatch_block_t _handler;
@@ -68,7 +59,7 @@
 
 - (instancetype)initWithTitleBlock:(RCTDevMenuItemTitleBlock)titleBlock handler:(dispatch_block_t)handler
 {
-  if ((self = [super init]) != nullptr) {
+  if ((self = [super init])) {
     _titleBlock = [titleBlock copy];
     _handler = [handler copy];
   }
@@ -93,14 +84,14 @@
 
 - (void)callHandler
 {
-  if (_handler != nullptr) {
+  if (_handler) {
     _handler();
   }
 }
 
 - (NSString *)title
 {
-  if (_titleBlock != nullptr) {
+  if (_titleBlock) {
     return _titleBlock();
   }
   return nil;
@@ -108,14 +99,22 @@
 
 @end
 
+#if !TARGET_OS_OSX // [macOS]
 typedef void (^RCTDevMenuAlertActionHandler)(UIAlertAction *action);
+#else // [macOS
+typedef void (^RCTDevMenuAlertActionHandler)(NSModalResponse response);
+#endif // macOS]
 
 @interface RCTDevMenu () <RCTBridgeModule, RCTInvalidating, NativeDevMenuSpec>
 
 @end
 
 @implementation RCTDevMenu {
+#if !TARGET_OS_OSX // [macOS]
   UIAlertController *_actionSheet;
+#else // [macOS
+  NSAlert *_alert;
+#endif // [macOS]
   NSMutableArray<RCTDevMenuItem *> *_extraMenuItems;
 }
 
@@ -128,10 +127,10 @@
 
 + (void)initialize
 {
+#if !TARGET_OS_OSX // [macOS]
   // We're swizzling here because it's poor form to override methods in a category,
-  // however UIWindow doesn't actually implement motionEnded:withEvent:, so there's
-  // no need to call the original implementation.
-  RCTSwapInstanceMethods([UIWindow class], @selector(motionEnded:withEvent:), @selector(RCT_motionEnded:withEvent:));
+  RCTOriginalUIWindowMotionEndedWithEventImp = (MotionEndedWithEventImpType) RCTSwapInstanceMethods([UIWindow class], @selector(motionEnded:withEvent:), @selector(RCT_motionEnded:withEvent:)); // [macOS]
+#endif // [macOS]
 }
 
 + (BOOL)requiresMainQueueSetup
@@ -141,15 +140,13 @@
 
 - (instancetype)init
 {
-  if ((self = [super init]) != nullptr) {
+  if ((self = [super init])) {
     [[NSNotificationCenter defaultCenter] addObserver:self
                                              selector:@selector(showOnShake)
                                                  name:RCTShowDevMenuNotification
                                                object:nil];
     _extraMenuItems = [NSMutableArray new];
 
-    _keyboardShortcutsEnabled = true;
-    _devMenuEnabled = true;
     [self registerHotkeys];
   }
   return self;
@@ -185,6 +182,7 @@
 
   [commands unregisterKeyCommandWithInput:@"d" modifierFlags:UIKeyModifierCommand];
   [commands unregisterKeyCommandWithInput:@"i" modifierFlags:UIKeyModifierCommand];
+  [commands unregisterKeyCommandWithInput:@"n" modifierFlags:UIKeyModifierCommand];
 #endif
 }
 
@@ -194,30 +192,13 @@
   RCTKeyCommands *commands = [RCTKeyCommands sharedInstance];
 
   return [commands isKeyCommandRegisteredForInput:@"d" modifierFlags:UIKeyModifierCommand] &&
-      [commands isKeyCommandRegisteredForInput:@"i" modifierFlags:UIKeyModifierCommand];
+      [commands isKeyCommandRegisteredForInput:@"i" modifierFlags:UIKeyModifierCommand] &&
+      [commands isKeyCommandRegisteredForInput:@"n" modifierFlags:UIKeyModifierCommand];
 #else
   return NO;
 #endif
 }
 
-- (BOOL)isReloadCommandRegistered
-{
-#if TARGET_OS_SIMULATOR || TARGET_OS_MACCATALYST
-  RCTKeyCommands *commands = [RCTKeyCommands sharedInstance];
-  return [commands isKeyCommandRegisteredForInput:@"r" modifierFlags:UIKeyModifierCommand];
-#else
-  return NO;
-#endif
-}
-
-- (void)unregisterReloadCommand
-{
-#if TARGET_OS_SIMULATOR || TARGET_OS_MACCATALYST
-  RCTKeyCommands *commands = [RCTKeyCommands sharedInstance];
-  [commands unregisterKeyCommandWithInput:@"r" modifierFlags:UIKeyModifierCommand];
-#endif
-}
-
 - (dispatch_queue_t)methodQueue
 {
   return dispatch_get_main_queue();
@@ -226,19 +207,23 @@
 - (void)invalidate
 {
   _presentedItems = nil;
+#if !TARGET_OS_OSX // [macOS]
   [_actionSheet dismissViewControllerAnimated:YES
                                    completion:^(void){
                                    }];
+#endif // [macOS]
 }
 
 - (void)showOnShake
 {
   if ([((RCTDevSettings *)[_moduleRegistry moduleForName:"DevSettings"]) isShakeToShowDevMenuEnabled]) {
-    NSMutableArray<UIWindow *> *windows = [NSMutableArray new];
+    NSMutableArray<RCTPlatformWindow *> *windows = [NSMutableArray new]; // [macOS]
+#if !TARGET_OS_OSX // [macOS]
     for (UIWindowScene *scene in RCTSharedApplication().connectedScenes) {
       [windows addObjectsFromArray:scene.windows];
     }
-    for (UIWindow *window in windows) {
+#endif // [macOS]
+    for (RCTPlatformWindow *window in windows) { // [macOS]
       NSString *recursiveDescription = [window valueForKey:@"recursiveDescription"];
       if ([recursiveDescription containsString:@"RCTView"]) {
         [self show];
@@ -248,12 +233,13 @@
   }
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)toggle
 {
   if (_actionSheet.isBeingPresented || _actionSheet.beingDismissed) {
     return;
   }
-  if (_actionSheet != nullptr) {
+  if (_actionSheet) {
     [_actionSheet dismissViewControllerAnimated:YES
                                      completion:^(void) {
                                        self->_actionSheet = nil;
@@ -263,10 +249,15 @@
     [self show];
   }
 }
+#endif // macOS]
 
 - (BOOL)isActionSheetShown
 {
+#if !TARGET_OS_OSX // [macOS]
   return _actionSheet != nil;
+#else // [macOS
+  return _alert != nil;
+#endif // macOS]
 }
 
 - (void)addItem:(NSString *)title handler:(void (^)(void))handler
@@ -279,17 +270,6 @@
   [_extraMenuItems addObject:item];
 }
 
-- (void)setKeyboardShortcutsEnabled:(BOOL)keyboardShortcutsEnabled
-{
-  if (_keyboardShortcutsEnabled != keyboardShortcutsEnabled) {
-    [self setHotkeysEnabled:keyboardShortcutsEnabled];
-
-    if (!keyboardShortcutsEnabled) {
-      [self disableReloadCommand];
-    }
-  }
-}
-
 - (void)setDefaultJSBundle
 {
   [[RCTBundleURLProvider sharedSettings] resetToDefaults];
@@ -365,6 +345,7 @@
                       return @"Configure Bundler";
                     }
                     handler:^{
+#if !TARGET_OS_OSX // [macOS]
                       UIAlertController *alertController = [UIAlertController
                           alertControllerWithTitle:@"Configure Bundler"
                                            message:@"Provide a custom bundler address, port, and entrypoint."
@@ -415,15 +396,84 @@
                                                                         handler:^(__unused UIAlertAction *action) {
                                                                           [weakSelf setDefaultJSBundle];
                                                                         }]];
-                      UIAlertAction *configCancelAction =
-                          [UIAlertAction actionWithTitle:@"Cancel"
-                                                   style:UIAlertActionStyleCancel
-                                                 handler:^(__unused UIAlertAction *action) {
-                                                   return;
-                                                 }];
-                      [configCancelAction setValue:[UIColor systemRedColor] forKey:@"titleTextColor"];
-                      [alertController addAction:configCancelAction];
+                      [alertController addAction:[UIAlertAction actionWithTitle:@"Cancel"
+                                                                          style:UIAlertActionStyleCancel
+                                                                        handler:^(__unused UIAlertAction *action) {
+                                                                          return;
+                                                                        }]];
                       [RCTPresentedViewController() presentViewController:alertController animated:YES completion:NULL];
+#else // [macOS
+                      NSAlert *alert = [NSAlert new];
+                      [alert setMessageText:@"Change packager location"];
+                      [alert setInformativeText:@"Input packager IP, port and entrypoint"];
+
+                      // Create accessory view with text fields
+                      NSView *accessoryView = [[NSView alloc] initWithFrame:NSMakeRect(0.0, 0.0, 300.0, 90.0)];
+
+                      // IP Address text field
+                      NSTextField *ipTextField = [[NSTextField alloc] initWithFrame:NSMakeRect(0.0, 60.0, 300.0, 22.0)];
+                      ipTextField.placeholderString = @"0.0.0.0";
+                      ipTextField.cell.scrollable = YES;
+                      [accessoryView addSubview:ipTextField];
+
+                      // Port text field
+                      NSTextField *portTextField =
+                          [[NSTextField alloc] initWithFrame:NSMakeRect(0.0, 30.0, 300.0, 22.0)];
+                      portTextField.placeholderString = @"8081";
+                      portTextField.cell.scrollable = YES;
+                      [accessoryView addSubview:portTextField];
+
+                      // Entrypoint text field
+                      NSTextField *entrypointTextField =
+                          [[NSTextField alloc] initWithFrame:NSMakeRect(0.0, 0.0, 300.0, 22.0)];
+                      entrypointTextField.placeholderString = @"index";
+                      entrypointTextField.cell.scrollable = YES;
+                      [accessoryView addSubview:entrypointTextField];
+
+                      alert.accessoryView = accessoryView;
+
+                      [alert addButtonWithTitle:@"Apply Changes"];
+                      [alert addButtonWithTitle:@"Reset to Default"];
+                      [alert addButtonWithTitle:@"Cancel"];
+                      [alert setAlertStyle:NSAlertStyleWarning];
+
+                      NSModalResponse response = [alert runModal];
+                      
+                      if (response == NSAlertFirstButtonReturn) {
+                        // Apply Changes
+                        NSString *ipAddress = ipTextField.stringValue;
+                        NSString *port = portTextField.stringValue;
+                        NSString *bundleRoot = entrypointTextField.stringValue;
+
+                        if (ipAddress.length == 0 && port.length == 0) {
+                          [weakSelf setDefaultJSBundle];
+                          return;
+                        }
+
+                        NSNumberFormatter *formatter = [NSNumberFormatter new];
+                        formatter.numberStyle = NSNumberFormatterDecimalStyle;
+                        NSNumber *portNumber = [formatter numberFromString:port];
+                        if (portNumber == nil) {
+                          portNumber = [NSNumber numberWithInt:RCT_METRO_PORT];
+                        }
+
+                        [RCTBundleURLProvider sharedSettings].jsLocation =
+                            [NSString stringWithFormat:@"%@:%d", ipAddress, portNumber.intValue];
+
+                        if (bundleRoot.length == 0) {
+                          [bundleManager resetBundleURL];
+                        } else {
+                          bundleManager.bundleURL = [[RCTBundleURLProvider sharedSettings]
+                              jsBundleURLForBundleRoot:bundleRoot];
+                        }
+
+                        RCTTriggerReloadCommandListeners(@"Dev menu - apply changes");
+                      } else if (response == NSAlertSecondButtonReturn) {
+                        // Reset to Default
+                        [weakSelf setDefaultJSBundle];
+                      }
+                      // Cancel - do nothing
+#endif // macOS]
                     }]];
 
   [items addObjectsFromArray:_extraMenuItems];
@@ -432,21 +482,34 @@
 
 RCT_EXPORT_METHOD(show)
 {
-  if ((_actionSheet != nullptr) || RCTRunningInAppExtension() || !_devMenuEnabled) {
+#if !TARGET_OS_OSX // [macOS]
+  if (_actionSheet || RCTRunningInAppExtension()) {
     return;
   }
+#else // [macOS
+  if (_alert) {
+    return;
+  }
+#endif // [macOS]
 
+  NSString *bridgeDescription = _bridge.bridgeDescription;
+  NSString *description =
+      bridgeDescription.length > 0 ? [NSString stringWithFormat:@"Running %@", bridgeDescription] : nil;
+
+#if !TARGET_OS_OSX // [macOS]
   // On larger devices we don't have an anchor point for the action sheet
   UIAlertControllerStyle style = [[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPhone
       ? UIAlertControllerStyleActionSheet
       : UIAlertControllerStyleAlert;
+#else // [macOS
+  NSAlertStyle style = NSAlertStyleInformational;
+#endif // macOS]
 
-  _actionSheet = [UIAlertController alertControllerWithTitle:@"React Native Dev Menu" message:nil preferredStyle:style];
+  NSString *devMenuType = [self.bridge isKindOfClass:RCTBridge.class] ? @"Bridge" : @"Bridgeless";
+  NSString *devMenuTitle = [NSString stringWithFormat:@"React Native Dev Menu (%@)", devMenuType];
 
-  NSAttributedString *title =
-      [[NSAttributedString alloc] initWithString:@"React Native Dev Menu"
-                                      attributes:@{NSFontAttributeName : [UIFont boldSystemFontOfSize:17]}];
-  [_actionSheet setValue:title forKey:@"_attributedTitle"];
+#if !TARGET_OS_OSX // [macOS]
+  _actionSheet = [UIAlertController alertControllerWithTitle:devMenuTitle message:description preferredStyle:style];
 
   NSArray<RCTDevMenuItem *> *items = [self _menuItemsToPresent];
   for (RCTDevMenuItem *item in items) {
@@ -457,29 +520,118 @@
     [_actionSheet addAction:action];
   }
 
-  UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"Cancel"
-                                                         style:UIAlertActionStyleCancel
-                                                       handler:[self alertActionHandlerForDevItem:nil]];
-  [cancelAction setValue:[UIColor systemRedColor] forKey:@"titleTextColor"];
-  [_actionSheet addAction:cancelAction];
+  [_actionSheet addAction:[UIAlertAction actionWithTitle:@"Cancel"
+                                                   style:UIAlertActionStyleCancel
+                                                 handler:[self alertActionHandlerForDevItem:nil]]];
 
   _presentedItems = items;
   [RCTPresentedViewController() presentViewController:_actionSheet animated:YES completion:nil];
+#else // [macOS
+  _alert = [NSAlert new];
+  [_alert setMessageText:devMenuTitle];
+  [_alert setInformativeText:description];
+  [_alert setAlertStyle:NSAlertStyleInformational];
 
+  NSArray<RCTDevMenuItem *> *items = [self _menuItemsToPresent];
+  for (RCTDevMenuItem *item in items) {
+    [_alert addButtonWithTitle:item.title];
+  }
+
+  [_alert addButtonWithTitle:@"Cancel"];
+
+  _presentedItems = items;
+  
+  // If Invoked from Metro, both the key window and main window may be nil, so we fallback to the first window in that case
+  NSWindow *window = RCTKeyWindow() ?: [NSApp mainWindow] ?: [[NSApp windows] firstObject];
+
+
+  [_alert beginSheetModalForWindow:window completionHandler:^(NSModalResponse response) {
+    // Button responses are NSAlertFirstButtonReturn, NSAlertSecondButtonReturn, etc.
+    // The last button (Cancel) will have response = NSAlertFirstButtonReturn + menuItems.count
+    NSInteger buttonIndex = response - NSAlertFirstButtonReturn;
+
+    RCTDevMenuItem *selectedItem = nil;
+    if (buttonIndex >= 0 && buttonIndex < self->_presentedItems.count) {
+      // Execute the corresponding menu item
+      selectedItem = self->_presentedItems[buttonIndex];
+    }
+    RCTDevMenuAlertActionHandler handler = [self alertActionHandlerForDevItem:selectedItem];
+    handler(response);
+  }];
+#endif // macOS]
+
   [_callableJSModules invokeModule:@"RCTNativeAppEventEmitter" method:@"emit" withArgs:@[ @"RCTDevMenuShown" ]];
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (RCTDevMenuAlertActionHandler)alertActionHandlerForDevItem:(RCTDevMenuItem *__nullable)item
 {
   return ^(__unused UIAlertAction *action) {
-    if (item != nullptr) {
+    if (item) {
       [item callHandler];
     }
 
     self->_actionSheet = nil;
   };
 }
+#else // [macOS
+- (RCTDevMenuAlertActionHandler)alertActionHandlerForDevItem:(RCTDevMenuItem *__nullable)item
+{
+  return ^(NSModalResponse response) {
+    if (item) {
+      [item callHandler];
+    }
+    
+    self->_alert = nil;
+  };
+}
+#endif // [macOS]
 
+#if TARGET_OS_OSX // [macOS
+- (NSMenu *)menu
+{
+  if ([((RCTDevSettings *)[_moduleRegistry moduleForName:"DevSettings"]) isSecondaryClickToShowDevMenuEnabled]) {
+    NSMenu *menu = [NSMenu new];
+    
+    NSString *devMenuType = [self.bridge isKindOfClass:RCTBridge.class] ? @"Bridge" : @"Bridgeless";
+    NSString *devMenuTitle = [NSString stringWithFormat:@"React Native Dev Menu (%@)", devMenuType];
+    
+    NSMenuItem *titleItem = [NSMenuItem sectionHeaderWithTitle:devMenuTitle];
+    if (@available(macOS 14.4, *)) {
+      NSString *bridgeDescription = _bridge.bridgeDescription;
+      NSString *description =
+      bridgeDescription.length > 0 ? [NSString stringWithFormat:@"Running %@", bridgeDescription] : nil;
+      [titleItem setSubtitle:description];
+    }
+    [menu addItem:titleItem];
+    
+    NSArray<RCTDevMenuItem *> *items = [self _menuItemsToPresent];
+    for (RCTDevMenuItem *item in items) {
+      NSMenuItem *menuItem = [[NSMenuItem alloc] initWithTitle:[item title]
+                                                        action:@selector(menuItemSelected:)
+                                                 keyEquivalent:@""];
+      [menuItem setTarget:self];
+      [menuItem setRepresentedObject:item];
+      [menu addItem:menuItem];
+    }
+    return menu;
+  }
+  return nil;
+}
+
+- (void)menuItemSelected:(id)sender
+{
+  NSMenuItem *menuItem = (NSMenuItem *)sender;
+  RCTDevMenuItem *item = (RCTDevMenuItem *)[menuItem representedObject];
+  [item callHandler];
+}
+
+- (void)setSecondaryClickToShow:(BOOL)secondaryClickToShow
+{
+  _bridge.devSettings.isSecondaryClickToShowDevMenuEnabled = secondaryClickToShow;
+}
+#endif // macOS]
+
 #pragma mark - deprecated methods and properties
 
 #define WARN_DEPRECATED_DEV_MENU_EXPORT() \
@@ -487,7 +639,7 @@
 
 - (void)setShakeToShow:(BOOL)shakeToShow
 {
-  [[_moduleRegistry moduleForName:"DevSettings"] setIsShakeToShowDevMenuEnabled:shakeToShow];
+  ((RCTDevSettings *)[_moduleRegistry moduleForName:"DevSettings"]).isShakeToShowDevMenuEnabled = shakeToShow;
 }
 
 - (BOOL)shakeToShow
@@ -537,13 +689,6 @@
   return [self isHotkeysRegistered];
 }
 
-- (void)disableReloadCommand
-{
-  if ([self isReloadCommandRegistered]) {
-    [self unregisterReloadCommand];
-  }
-}
-
 - (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
     (const facebook::react::ObjCTurboModule::InitParams &)params
 {
@@ -557,15 +702,6 @@
 @interface RCTDevMenu () <NativeDevMenuSpec>
 @end
 
-@implementation RCTDevMenuConfiguration
-
-+ (instancetype)defaultConfiguration
-{
-  return nil;
-}
-
-@end
-
 @implementation RCTDevMenu
 
 - (void)show
@@ -581,10 +717,6 @@
 {
 }
 
-- (void)disableReloadCommand
-{
-}
-
 - (BOOL)isActionSheetShown
 {
   return NO;
