--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTKeyboardObserver.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/RCTKeyboardObserver.mm	2026-03-06 15:22:52
@@ -24,6 +24,8 @@
 
 - (void)startObserving
 {
+#if !TARGET_OS_OSX // [macOS]
+
   NSNotificationCenter *nc = [NSNotificationCenter defaultCenter];
 
 #define ADD_KEYBOARD_HANDLER(NAME, SELECTOR) [nc addObserver:self selector:@selector(SELECTOR:) name:NAME object:nil]
@@ -36,6 +38,7 @@
   ADD_KEYBOARD_HANDLER(UIKeyboardDidChangeFrameNotification, keyboardDidChangeFrame);
 
 #undef ADD_KEYBOARD_HANDLER
+#endif // [macOS]
 }
 
 - (NSArray<NSString *> *)supportedEvents
@@ -91,7 +94,7 @@
     @"height" : @(rect.size.height),
   };
 }
-
+#if !TARGET_OS_OSX // [macOS]
 static NSString *RCTAnimationNameForCurve(UIViewAnimationCurve curve)
 {
   switch (curve) {
@@ -107,9 +110,11 @@
       return @"keyboard";
   }
 }
+#endif // [macOS]
 
 static NSDictionary *RCTParseKeyboardNotification(NSNotification *notification)
 {
+#if !TARGET_OS_OSX // [macOS]
   NSDictionary *userInfo = notification.userInfo;
   CGRect beginFrame = [userInfo[UIKeyboardFrameBeginUserInfoKey] CGRectValue];
   CGRect endFrame = [userInfo[UIKeyboardFrameEndUserInfoKey] CGRectValue];
@@ -119,8 +124,13 @@
   NSInteger isLocalUserInfoKey = [userInfo[UIKeyboardIsLocalUserInfoKey] integerValue];
 
   UIWindow *window = RCTKeyWindow();
+#if !TARGET_OS_VISION // [visionOS]
   beginFrame = [window convertRect:beginFrame fromCoordinateSpace:window.screen.coordinateSpace];
   endFrame = [window convertRect:endFrame fromCoordinateSpace:window.screen.coordinateSpace];
+#else // [visionOS
+  beginFrame = [window convertRect:beginFrame fromCoordinateSpace:window.coordinateSpace];
+  endFrame = [window convertRect:endFrame fromCoordinateSpace:window.coordinateSpace];
+#endif // visionOS]
 
   return @{
     @"startCoordinates" : RCTRectDictionaryValue(beginFrame),
@@ -129,6 +139,9 @@
     @"easing" : RCTAnimationNameForCurve(curve),
     @"isEventFromThisApp" : isLocalUserInfoKey == 1 ? @YES : @NO,
   };
+#else // [macOS
+  return @{};
+#endif // macOS]
 }
 
 Class RCTKeyboardObserverCls(void)
