--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTLogBoxView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/RCTLogBoxView.h	2026-03-06 15:22:52
@@ -8,19 +8,30 @@
 #import <React/RCTBridge.h>
 #import <React/RCTSurfacePresenterStub.h>
 #import <React/RCTSurfaceView.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
+#if !TARGET_OS_OSX // [macOS]
 @interface RCTLogBoxView : UIWindow
+#else // [macOS
+@interface RCTLogBoxView : NSWindow
+#endif // macOS]
 
+#if !TARGET_OS_OSX // [macOS]
 - (instancetype)initWithFrame:(CGRect)frame;
+#endif // [macOS]
 
-- (void)createRootViewController:(UIView *)view;
+- (void)createRootViewController:(RCTPlatformView *)view; // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-- (instancetype)initWithWindow:(UIWindow *)window bridge:(RCTBridge *)bridge;
-#endif // RCT_REMOVE_LEGACY_ARCH
-- (instancetype)initWithWindow:(UIWindow *)window surfacePresenter:(id<RCTSurfacePresenterStub>)surfacePresenter;
+#ifndef RCT_FIT_RM_OLD_RUNTIME
+- (instancetype)initWithWindow:(RCTPlatformWindow *)window bridge:(RCTBridge *)bridge; // [macOS]
+#endif // RCT_FIT_RM_OLD_RUNTIME
+- (instancetype)initWithWindow:(RCTPlatformWindow *)window surfacePresenter:(id<RCTSurfacePresenterStub>)surfacePresenter; // [macOS]
 
 - (void)show;
+#if TARGET_OS_OSX // [macOS
+- (void)setHidden:(BOOL)hidden;
+#endif // macOS]
 
 @end
+
+
