--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTPlatform.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/RCTPlatform.mm	2026-03-06 15:22:52
@@ -7,7 +7,7 @@
 
 #import "RCTPlatform.h"
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <FBReactNativeSpec/FBReactNativeSpec.h>
 #import <React/RCTInitializing.h>
@@ -20,6 +20,7 @@
 
 using namespace facebook::react;
 
+#if !TARGET_OS_OSX // [macOS]
 static NSString *interfaceIdiom(UIUserInterfaceIdiom idiom)
 {
   switch (idiom) {
@@ -31,15 +32,19 @@
       return @"tv";
     case UIUserInterfaceIdiomCarPlay:
       return @"carplay";
+#if TARGET_OS_VISION
     case UIUserInterfaceIdiomVision:
       return @"vision";
-    case UIUserInterfaceIdiomMac:
-      return @"mac";
-    case UIUserInterfaceIdiomUnspecified:
+#endif
     default:
       return @"unknown";
   }
 }
+#else // [macOS
+static NSString *interfaceIdiom() {
+  return @"mac";
+}
+#endif // macOS]
 
 @interface RCTPlatform () <NativePlatformConstantsIOSSpec, RCTInitializing>
 @end
@@ -57,13 +62,24 @@
 
 - (void)initialize
 {
+#if !TARGET_OS_OSX // [macOS]
   UIDevice *device = [UIDevice currentDevice];
+#else // [macOS]
+  NSProcessInfo *processInfo = [NSProcessInfo processInfo];
+  NSOperatingSystemVersion osVersion = [[NSProcessInfo processInfo] operatingSystemVersion];
+#endif // [macOS]
   auto versions = RCTGetReactNativeVersion();
   _constants = typedConstants<JS::NativePlatformConstantsIOS::Constants>({
       .forceTouchAvailable = RCTForceTouchAvailable() ? true : false,
+#if !TARGET_OS_OSX // [macOS]
       .osVersion = [device systemVersion],
       .systemName = [device systemName],
       .interfaceIdiom = interfaceIdiom([device userInterfaceIdiom]),
+#else // [macOS
+      .osVersion = [NSString stringWithFormat:@"%ld.%ld.%ld", osVersion.majorVersion, osVersion.minorVersion, osVersion.patchVersion],
+      .systemName = [processInfo operatingSystemVersionString],
+      .interfaceIdiom = interfaceIdiom(),
+#endif // macOS]
       .isTesting = RCTRunningInTestEnvironment() ? true : false,
       .reactNativeVersion = JS::NativePlatformConstantsIOS::ConstantsReactNativeVersion::Builder(
           {.minor = [versions[@"minor"] doubleValue],
