--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTRedBox.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/RCTRedBox.mm	2026-03-06 15:22:52
@@ -25,15 +25,27 @@
 
 @class RCTRedBoxController;
 
+#if !TARGET_OS_OSX // [macOS]
 @interface UIButton (RCTRedBox)
+#else // [macOS
+@interface NSButton (RCTRedBox)
+#endif // macOS]
 
 @property (nonatomic) RCTRedBoxButtonPressHandler rct_handler;
 
+#if !TARGET_OS_OSX // [macOS]
 - (void)rct_addBlock:(RCTRedBoxButtonPressHandler)handler forControlEvents:(UIControlEvents)controlEvents;
+#else // [macOS
+- (void)rct_addBlock:(RCTRedBoxButtonPressHandler)handler;
+#endif // macOS]
 
 @end
 
+#if !TARGET_OS_OSX // [macOS]
 @implementation UIButton (RCTRedBox)
+#else // [macOS
+@implementation NSButton (RCTRedBox)
+#endif // macOS]
 
 - (RCTRedBoxButtonPressHandler)rct_handler
 {
@@ -52,10 +64,19 @@
   }
 }
 
-- (void)rct_addBlock:(RCTRedBoxButtonPressHandler)handler forControlEvents:(UIControlEvents)controlEvents
+#if !TARGET_OS_OSX // [macOS]
+- (void)rct_addBlock:(RCTRedBoxButtonPressHandler)handler forControlEvents:(UIControlEvents)controlEvents;
+#else // [macOS
+- (void)rct_addBlock:(RCTRedBoxButtonPressHandler)handler
+#endif // macOS]
 {
   self.rct_handler = handler;
+#if !TARGET_OS_OSX // [macOS]
   [self addTarget:self action:@selector(rct_callBlock) forControlEvents:controlEvents];
+#else // [macOS
+  [self setTarget:self];
+  [self setAction:@selector(rct_callBlock)];
+#endif // macOS]
 }
 
 @end
@@ -68,12 +89,20 @@
 
 @end
 
+#if !TARGET_OS_OSX // [macOS]
 @interface RCTRedBoxController : UIViewController <UITableViewDelegate, UITableViewDataSource>
+#else // [macOS
+@interface RCTRedBoxController : NSViewController <NSTableViewDelegate, NSTableViewDataSource>
+#endif // macOS]
 @property (nonatomic, weak) id<RCTRedBoxControllerActionDelegate> actionDelegate;
 @end
 
 @implementation RCTRedBoxController {
+#if !TARGET_OS_OSX // [macOS]
   UITableView *_stackTraceTableView;
+#else // [macOS
+  NSTableView *_stackTraceTableView;
+#endif //  macOS]
   NSString *_lastErrorMessage;
   NSArray<RCTJSStackFrame *> *_lastStackTrace;
   NSArray<NSString *> *_customButtonTitles;
@@ -96,13 +125,19 @@
 - (void)viewDidLoad
 {
   [super viewDidLoad];
+#if !TARGET_OS_OSX // [macOS]
   self.view.backgroundColor = [UIColor blackColor];
+#else // [macOS
+  self.view.wantsLayer = YES;
+  self.view.layer.backgroundColor = [[NSColor blackColor] CGColor];
+#endif // macOS]
 
   const CGFloat buttonHeight = 60;
 
   CGRect detailsFrame = self.view.bounds;
   detailsFrame.size.height -= buttonHeight + (double)[self bottomSafeViewHeight];
 
+#if !TARGET_OS_OSX // [macOS]
   _stackTraceTableView = [[UITableView alloc] initWithFrame:detailsFrame style:UITableViewStylePlain];
   _stackTraceTableView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
   _stackTraceTableView.delegate = self;
@@ -112,8 +147,31 @@
   _stackTraceTableView.separatorStyle = UITableViewCellSeparatorStyleNone;
   _stackTraceTableView.indicatorStyle = UIScrollViewIndicatorStyleWhite;
   [self.view addSubview:_stackTraceTableView];
+#else // [macOS
+  NSScrollView *scrollView = [[NSScrollView alloc] initWithFrame:NSZeroRect];
+  scrollView.translatesAutoresizingMaskIntoConstraints = NO;
+  scrollView.autoresizesSubviews = YES;
+  scrollView.drawsBackground = NO;
 
-#if TARGET_OS_SIMULATOR || TARGET_OS_MACCATALYST
+  _stackTraceTableView = [[NSTableView alloc] initWithFrame:NSZeroRect];
+  _stackTraceTableView.translatesAutoresizingMaskIntoConstraints = NO;
+  _stackTraceTableView.dataSource = self;
+  _stackTraceTableView.delegate = self;
+  _stackTraceTableView.headerView = nil;
+  _stackTraceTableView.allowsColumnReordering = NO;
+  _stackTraceTableView.allowsColumnResizing = NO;
+  _stackTraceTableView.columnAutoresizingStyle = NSTableViewFirstColumnOnlyAutoresizingStyle;
+  _stackTraceTableView.backgroundColor = [NSColor clearColor];
+  _stackTraceTableView.allowsTypeSelect = NO;
+  
+  NSTableColumn *tableColumn = [[NSTableColumn alloc] initWithIdentifier:@"info"];
+  [_stackTraceTableView addTableColumn:tableColumn];
+
+  scrollView.documentView = _stackTraceTableView;
+  [self.view addSubview:scrollView];
+#endif // macOS]
+
+#if TARGET_OS_SIMULATOR || TARGET_OS_MACCATALYST || TARGET_OS_OSX // [macOS]
   NSString *reloadText = @"Reload\n(\u2318R)";
   NSString *dismissText = @"Dismiss\n(ESC)";
   NSString *copyText = @"Copy\n(\u2325\u2318C)";
@@ -125,6 +183,7 @@
   NSString *extraText = @"Extra Info";
 #endif
 
+#if !TARGET_OS_OSX // [macOS]
   UIButton *dismissButton = [self redBoxButton:dismissText
                        accessibilityIdentifier:@"redbox-dismiss"
                                       selector:@selector(dismiss)
@@ -141,6 +200,29 @@
                      accessibilityIdentifier:@"redbox-extra"
                                     selector:@selector(showExtraDataViewController)
                                        block:nil];
+#else // [macOS
+  NSButton *dismissButton = [self redBoxButton:dismissText
+                       accessibilityIdentifier:@"redbox-dismiss"
+                                      selector:@selector(dismiss)
+                                         block:nil];
+  [dismissButton setKeyEquivalent:@"\e"];
+  NSButton *reloadButton = [self redBoxButton:reloadText
+                      accessibilityIdentifier:@"redbox-reload"
+                                     selector:@selector(reload)
+                                        block:nil];
+  [reloadButton setKeyEquivalent:@"r"];
+  [reloadButton setKeyEquivalentModifierMask:NSEventModifierFlagCommand];
+  NSButton *copyButton = [self redBoxButton:copyText
+                    accessibilityIdentifier:@"redbox-copy"
+                                   selector:@selector(copyStack)
+                                      block:nil];
+  [copyButton setKeyEquivalent:@"c"];
+  [copyButton setKeyEquivalentModifierMask:NSEventModifierFlagOption | NSEventModifierFlagCommand];
+  NSButton *extraButton = [self redBoxButton:extraText
+                     accessibilityIdentifier:@"redbox-extra"
+                                    selector:@selector(showExtraDataViewController)
+                                       block:nil];
+#endif // macOS]
 
   [NSLayoutConstraint activateConstraints:@[
     [dismissButton.heightAnchor constraintEqualToConstant:buttonHeight],
@@ -149,12 +231,22 @@
     [extraButton.heightAnchor constraintEqualToConstant:buttonHeight]
   ]];
 
+#if !TARGET_OS_OSX // [macOS]
   UIStackView *buttonStackView = [[UIStackView alloc] init];
   buttonStackView.translatesAutoresizingMaskIntoConstraints = NO;
   buttonStackView.axis = UILayoutConstraintAxisHorizontal;
   buttonStackView.distribution = UIStackViewDistributionFillEqually;
   buttonStackView.alignment = UIStackViewAlignmentTop;
   buttonStackView.backgroundColor = [UIColor colorWithRed:0.1 green:0.1 blue:0.1 alpha:1];
+#else // [macOS
+  NSStackView *buttonStackView = [[NSStackView alloc] init];
+  buttonStackView.translatesAutoresizingMaskIntoConstraints = NO;
+  buttonStackView.orientation = NSUserInterfaceLayoutOrientationHorizontal;
+  buttonStackView.distribution = NSStackViewDistributionFillEqually;
+  buttonStackView.alignment = NSLayoutAttributeTop;
+  buttonStackView.wantsLayer = YES;
+  buttonStackView.layer.backgroundColor = [NSColor colorWithRed:0.1 green:0.1 blue:0.1 alpha:1].CGColor;
+#endif // macOS]
 
   [buttonStackView addArrangedSubview:dismissButton];
   [buttonStackView addArrangedSubview:reloadButton];
@@ -170,18 +262,31 @@
     [buttonStackView.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor]
   ]];
 
+
   for (NSUInteger i = 0; i < [_customButtonTitles count]; i++) {
+#if !TARGET_OS_OSX // [macOS]
     UIButton *button = [self redBoxButton:_customButtonTitles[i]
                   accessibilityIdentifier:@""
                                  selector:nil
                                     block:_customButtonHandlers[i]];
+#else // [macOS
+  NSButton *button = [self redBoxButton:_customButtonTitles[i]
+                  accessibilityIdentifier:@""
+                                 selector:nil
+                                    block:_customButtonHandlers[i]];
+#endif // macOS]
     [button.heightAnchor constraintEqualToConstant:buttonHeight].active = YES;
     [buttonStackView addArrangedSubview:button];
   }
 
-  UIView *topBorder = [[UIView alloc] init];
+  RCTPlatformView *topBorder = [[RCTPlatformView alloc] init]; // [macOS]
   topBorder.translatesAutoresizingMaskIntoConstraints = NO;
+#if !TARGET_OS_OSX // [macOS]
   topBorder.backgroundColor = [UIColor colorWithRed:0.70 green:0.70 blue:0.70 alpha:1.0];
+#else // [macOS
+  topBorder.wantsLayer = true;
+  topBorder.layer.backgroundColor = [NSColor colorWithRed:0.70 green:0.70 blue:0.70 alpha:1.0].CGColor;
+#endif // macOS]
   [topBorder.heightAnchor constraintEqualToConstant:1].active = YES;
 
   [self.view addSubview:topBorder];
@@ -189,10 +294,20 @@
   [NSLayoutConstraint activateConstraints:@[
     [topBorder.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
     [topBorder.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
-    [topBorder.bottomAnchor constraintEqualToAnchor:buttonStackView.topAnchor],
+    [topBorder.bottomAnchor constraintEqualToAnchor:buttonStackView.topAnchor]
   ]];
+
+#if TARGET_OS_OSX // [macOS
+  [NSLayoutConstraint activateConstraints:@[
+    [[scrollView leadingAnchor] constraintEqualToAnchor:[[self view] leadingAnchor]],
+    [[scrollView topAnchor] constraintEqualToAnchor:[[self view] topAnchor]],
+    [[scrollView trailingAnchor] constraintEqualToAnchor:[[self view] trailingAnchor]],
+    [[scrollView bottomAnchor] constraintEqualToAnchor:[topBorder topAnchor]],
+  ]];
+#endif // macOS]
 }
 
+#if !TARGET_OS_OSX // [macOS]
 - (UIButton *)redBoxButton:(NSString *)title
     accessibilityIdentifier:(NSString *)accessibilityIdentifier
                    selector:(SEL)selector
@@ -216,10 +331,34 @@
   }
   return button;
 }
+#else // [macOS
+- (NSButton *)redBoxButton:(NSString *)title
+   accessibilityIdentifier:(NSString *)accessibilityIdentifier
+                  selector:(SEL)selector
+                     block:(RCTRedBoxButtonPressHandler)block
+{
+  NSButton *button = [[NSButton alloc] initWithFrame:NSZeroRect];
+  button.translatesAutoresizingMaskIntoConstraints = NO;
+  button.accessibilityIdentifier = @"accessibilityIdentifier";
+  button.bordered = NO;
+  NSAttributedString *attributedTitle = [[NSAttributedString alloc]
+                                         initWithString:title
+                                         attributes:@{NSForegroundColorAttributeName : [ NSColor whiteColor] }];
+  button.attributedTitle = attributedTitle;
+  [button setButtonType:NSButtonTypeMomentaryPushIn];
+  if (selector) {
+    button.target = self;
+    button.action = selector;
+  } else if (block) {
+    [button rct_addBlock:block];
+  }
+  return button;
+}
+#endif // macOS]
 
 - (NSInteger)bottomSafeViewHeight
 {
-#if TARGET_OS_MACCATALYST
+#if TARGET_OS_MACCATALYST || TARGET_OS_OSX // [macOS]
   return 0;
 #else
   return RCTKeyWindow().safeAreaInsets.bottom;
@@ -262,17 +401,28 @@
     [_stackTraceTableView reloadData];
 
     if (!isRootViewControllerPresented) {
+#if !TARGET_OS_OSX // [macOS]
       [_stackTraceTableView scrollToRowAtIndexPath:[NSIndexPath indexPathForRow:0 inSection:0]
                                   atScrollPosition:UITableViewScrollPositionTop
                                           animated:NO];
       [RCTKeyWindow().rootViewController presentViewController:self animated:YES completion:nil];
+#else // [macOS
+	  [_stackTraceTableView scrollRowToVisible:0];
+    [[RCTKeyWindow() contentViewController] presentViewControllerAsSheet:self];
+#endif // macOS]
     }
   }
 }
 
 - (void)dismiss
 {
+#if !TARGET_OS_OSX // [macOS]
   [self dismissViewControllerAnimated:YES completion:nil];
+#else // [macOS]
+  if (self.presentingViewController) {
+    [[RCTKeyWindow() contentViewController] dismissViewController:self];
+  }
+#endif // macOS]
 }
 
 - (void)reload
@@ -294,22 +444,28 @@
 - (void)copyStack
 {
   NSMutableString *fullStackTrace;
-
+  
   if (_lastErrorMessage != nil) {
     fullStackTrace = [_lastErrorMessage mutableCopy];
     [fullStackTrace appendString:@"\n\n"];
   } else {
     fullStackTrace = [NSMutableString string];
   }
-
+  
   for (RCTJSStackFrame *stackFrame in _lastStackTrace) {
     [fullStackTrace appendString:[NSString stringWithFormat:@"%@\n", stackFrame.methodName]];
     if (stackFrame.file) {
       [fullStackTrace appendFormat:@"    %@\n", [self formatFrameSource:stackFrame]];
     }
   }
+#if !TARGET_OS_OSX // [macOS]
   UIPasteboard *pb = [UIPasteboard generalPasteboard];
   [pb setString:fullStackTrace];
+#else // [macOS
+  NSPasteboard *pasteboard = [NSPasteboard generalPasteboard];
+  [pasteboard clearContents];
+  [pasteboard setString:fullStackTrace forType:NSPasteboardTypeString];
+#endif // macOS]
 }
 
 - (NSString *)formatFrameSource:(RCTJSStackFrame *)stackFrame
@@ -325,6 +481,7 @@
 
 #pragma mark - TableView
 
+#if !TARGET_OS_OSX // [macOS]
 - (NSInteger)numberOfSectionsInTableView:(__unused UITableView *)tableView
 {
   return 2;
@@ -334,7 +491,14 @@
 {
   return section == 0 ? 1 : _lastStackTrace.count;
 }
+#else // [macOS
+- (NSInteger)numberOfRowsInTableView:(__unused NSTableView *)tableView
+{
+  return (_lastErrorMessage != nil) + _lastStackTrace.count;
+}
+#endif // macOS]
 
+#if !TARGET_OS_OSX // [macOS]
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
 {
   if (indexPath.section == 0) {
@@ -346,7 +510,22 @@
   RCTJSStackFrame *stackFrame = _lastStackTrace[index];
   return [self reuseCell:cell forStackFrame:stackFrame];
 }
+#else // [macOS
+- (nullable NSView *)tableView:(NSTableView *)tableView viewForTableColumn:(nullable NSTableColumn *)tableColumn row:(NSInteger)row
+{
+  if (row == 0) {
+    NSTableCellView *cell = [tableView makeViewWithIdentifier:@"msg-cell" owner:nil];
+    return [self reuseCell:cell forErrorMessage:_lastErrorMessage];
+  }
+  NSTableCellView *cell = [tableView makeViewWithIdentifier:@"cell" owner:nil];
+  NSUInteger index = row - 1;
+  RCTJSStackFrame *stackFrame = _lastStackTrace[index];
+  return [self reuseCell:cell forStackFrame:stackFrame];
 
+}
+#endif // macOS]
+
+#if !TARGET_OS_OSX
 - (UITableViewCell *)reuseCell:(UITableViewCell *)cell forErrorMessage:(NSString *)message
 {
   if (!cell) {
@@ -369,7 +548,56 @@
 
   return cell;
 }
+#else // [macOS
+- (NSTableCellView *)reuseCell:(NSTableCellView *)cell forErrorMessage:(NSString *)message
+{
+  if (!cell) {
+    cell = [[NSTableCellView alloc] initWithFrame:NSZeroRect];
+    cell.rowSizeStyle = NSTableViewRowSizeStyleCustom;
+    cell.textField.accessibilityIdentifier = @"red box-error";
 
+    NSTextField *label = [[NSTextField alloc] initWithFrame:NSZeroRect];
+    label.translatesAutoresizingMaskIntoConstraints = NO;
+    label.drawsBackground = NO;
+    label.bezeled = NO;
+    label.editable = NO;
+    
+    [cell addSubview:label];
+    cell.textField = label;
+    
+    [NSLayoutConstraint activateConstraints:@[
+      [[label leadingAnchor] constraintEqualToAnchor:[cell leadingAnchor] constant:5],
+      [[label topAnchor] constraintEqualToAnchor:[cell topAnchor] constant:5],
+      [[label trailingAnchor] constraintEqualToAnchor:[cell trailingAnchor] constant:-5],
+      [[label bottomAnchor] constraintEqualToAnchor:[cell bottomAnchor] constant:-5],
+    ]];
+
+    // Prefer a monofont for formatting messages that were designed
+    // to be displayed in a terminal.
+    cell.textField.font = [NSFont monospacedSystemFontOfSize:14 weight:NSFontWeightBold];
+
+    cell.textField.lineBreakMode = NSLineBreakByWordWrapping;
+    cell.textField.maximumNumberOfLines = 0;
+    cell.wantsLayer = true;
+    cell.layer.cornerRadius = 8.0;
+    cell.layer.cornerCurve = kCACornerCurveContinuous;
+    
+    cell.layer.backgroundColor = [NSColor colorWithRed:0.82 green:0.10 blue:0.15 alpha:1.0].CGColor;
+  }
+
+  NSDictionary<NSString *, id> *attributes = @{
+    NSForegroundColorAttributeName : [NSColor whiteColor],
+    NSFontAttributeName : [NSFont systemFontOfSize:16],
+  };
+  NSAttributedString *title = [[NSAttributedString alloc] initWithString:message attributes:attributes];
+  
+  cell.textField.attributedStringValue = title;
+
+  return cell;
+}
+#endif // [macOS]
+
+#if !TARGET_OS_OSX // [macOS]
 - (UITableViewCell *)reuseCell:(UITableViewCell *)cell forStackFrame:(RCTJSStackFrame *)stackFrame
 {
   if (!cell) {
@@ -396,15 +624,96 @@
                                                        : [UIColor colorWithRed:0.70 green:0.70 blue:0.70 alpha:1.0];
   return cell;
 }
+#else // [macOS
+- (NSTableCellView *)reuseCell:(NSTableCellView *)cell forStackFrame:(RCTJSStackFrame *)stackFrame
+{
+  if (!cell) {
+    cell = [[NSTableCellView alloc] initWithFrame:NSZeroRect];
 
+    NSTextField *label = [[NSTextField alloc] initWithFrame:NSZeroRect];
+    label.translatesAutoresizingMaskIntoConstraints = NO;
+    label.backgroundColor = [NSColor clearColor];
+    label.bezeled = NO;
+    label.editable = NO;
+    
+    label.maximumNumberOfLines = 2;
+
+    [cell addSubview:label];
+    cell.textField = label;
+
+    [NSLayoutConstraint activateConstraints:@[
+      [[label leadingAnchor] constraintEqualToAnchor:[cell leadingAnchor] constant:5],
+      [[label topAnchor] constraintEqualToAnchor:[cell topAnchor]],
+      [[label trailingAnchor] constraintEqualToAnchor:[cell trailingAnchor] constant:-5],
+      [[label bottomAnchor] constraintEqualToAnchor:[cell bottomAnchor]],
+    ]];
+  }
+  
+  NSString *text = stackFrame.methodName ?: @"(unnamed method)";
+  
+  NSMutableParagraphStyle *textParagraphStyle = [NSMutableParagraphStyle new];
+  textParagraphStyle.lineBreakMode = NSLineBreakByCharWrapping;
+  
+  NSDictionary<NSString *, id> *textAttributes = @{
+    NSForegroundColorAttributeName : stackFrame.collapse ? [NSColor lightGrayColor] : [NSColor whiteColor],
+    NSFontAttributeName : [NSFont fontWithName:@"Menlo-Regular" size:14],
+    NSParagraphStyleAttributeName : textParagraphStyle,
+  };
+  
+  NSAttributedString *attributedText = [[NSAttributedString alloc] initWithString:text attributes:textAttributes];
+  
+  
+  NSMutableAttributedString *title = [attributedText mutableCopy];
+
+  // NSTableCellView doesn't contain a subtitle text field. Rather than define our own custom row view,
+  // let's append the detail text with a new line if it is needed.
+  if (stackFrame.file) {
+    cell.textField.maximumNumberOfLines = 3;
+    
+    NSString *detailText = [self formatFrameSource:stackFrame];
+    
+    NSMutableParagraphStyle *detailTextParagraphStyle = [NSMutableParagraphStyle new];
+    detailTextParagraphStyle.lineBreakMode = NSLineBreakByTruncatingMiddle;
+    
+    NSDictionary<NSString *, id> *detailTextAttributes = @{
+      NSForegroundColorAttributeName : stackFrame.collapse ?
+        [NSColor colorWithRed:0.50 green:0.50 blue:0.50 alpha:1.0] :
+        [NSColor colorWithRed:0.70 green:0.70 blue:0.70 alpha:1.0],
+      NSFontAttributeName : [NSFont fontWithName:@"Menlo-Regular" size:11],
+      NSParagraphStyleAttributeName : detailTextParagraphStyle,
+    };
+    NSAttributedString *attributedDetailText = [[NSAttributedString alloc] initWithString:detailText attributes:detailTextAttributes];
+    
+    [title appendAttributedString:[[NSAttributedString alloc] initWithString:@"\n"]];
+    [title appendAttributedString:attributedDetailText];
+  }
+  
+  cell.textField.attributedStringValue = title;
+
+  return cell;
+}
+#endif // macOS]
+
+#if !TARGET_OS_OSX // [macOS]
 - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
+#else // [macOS
+- (CGFloat)tableView:(NSTableView *)tableView heightOfRow:(NSInteger)row
+#endif // macOS]
 {
+#if !TARGET_OS_OSX // [macOS]
   if (indexPath.section == 0) {
+#else // [macOS
+  if (row == 0) {
+#endif // macOS]
     NSMutableParagraphStyle *paragraphStyle = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
     paragraphStyle.lineBreakMode = NSLineBreakByWordWrapping;
 
     NSDictionary *attributes =
+#if !TARGET_OS_OSX // [macOS]
         @{NSFontAttributeName : [UIFont boldSystemFontOfSize:16], NSParagraphStyleAttributeName : paragraphStyle};
+#else // [macOS
+        @{NSFontAttributeName : [NSFont boldSystemFontOfSize:16], NSParagraphStyleAttributeName : paragraphStyle};
+#endif // macOS]
     CGRect boundingRect =
         [_lastErrorMessage boundingRectWithSize:CGSizeMake(tableView.frame.size.width - 30, CGFLOAT_MAX)
                                         options:NSStringDrawingUsesLineFragmentOrigin
@@ -416,6 +725,7 @@
   }
 }
 
+#if !TARGET_OS_OSX // [macOS
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
 {
   if (indexPath.section == 1) {
@@ -425,9 +735,21 @@
   }
   [tableView deselectRowAtIndexPath:indexPath animated:YES];
 }
+#else // [macOS
+- (BOOL)tableView:(__unused NSTableView *)tableView shouldSelectRow:(__unused NSInteger)row
+{
+  if (row != 0) {
+    NSUInteger index = row - 1;
+    RCTJSStackFrame *stackFrame = _lastStackTrace[index];
+    [_actionDelegate redBoxController:self openStackFrameInEditor:stackFrame];
+  }
+  return NO;
+}
+#endif // macOS]
 
 #pragma mark - Key commands
 
+#if !TARGET_OS_OSX // [macOS]
 - (NSArray<UIKeyCommand *> *)keyCommands
 {
   // NOTE: We could use RCTKeyCommands for this, but since
@@ -458,6 +780,7 @@
 {
   return YES;
 }
+#endif // [macOS]
 
 @end
 
@@ -632,19 +955,21 @@
   });
 }
 
-- (void)loadExtraDataViewController
-{
+- (void)loadExtraDataViewController {
   dispatch_async(dispatch_get_main_queue(), ^{
+#if !TARGET_OS_OSX // [macOS]
     // Make sure the CMD+E shortcut doesn't call this twice
     if (self->_extraDataViewController != nil && ![self->_controller presentedViewController]) {
       [self->_controller presentViewController:self->_extraDataViewController animated:YES completion:nil];
     }
+#else // [macOS
+    // Do nothing, as we haven't implemented `RCTRedBoxExtraDataViewController` on macOS yet
+#endif // [macOS]
   });
 }
 
-RCT_EXPORT_METHOD(setExtraData : (NSDictionary *)extraData forIdentifier : (NSString *)identifier)
-{
-  [_extraDataViewController addExtraData:extraData forIdentifier:identifier];
+RCT_EXPORT_METHOD(setExtraData:(NSDictionary *)extraData forIdentifier:(NSString *)identifier) {
+    [_extraDataViewController addExtraData:extraData forIdentifier:identifier];
 }
 
 RCT_EXPORT_METHOD(dismiss)
