--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/CoreModules/RCTStatusBarManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/CoreModules/RCTStatusBarManager.mm	2026-03-06 15:22:52
@@ -18,6 +18,7 @@
 static NSString *const kStatusBarFrameDidChange = @"statusBarFrameDidChange";
 static NSString *const kStatusBarFrameWillChange = @"statusBarFrameWillChange";
 
+#if !TARGET_OS_OSX // [macOS]
 @implementation RCTConvert (UIStatusBar)
 
 + (UIStatusBarStyle)UIStatusBarStyle:(id)json RCT_DYNAMIC
@@ -47,6 +48,7 @@
     integerValue);
 
 @end
+#endif // [macOS]
 
 @interface RCTStatusBarManager () <NativeStatusBarManagerIOSSpec, RCTInitializing>
 @end
@@ -55,6 +57,7 @@
   facebook::react::ModuleConstants<JS::NativeStatusBarManagerIOS::Constants> _constants;
 }
 
+#if !TARGET_OS_OSX // [macOS]
 static BOOL RCTViewControllerBasedStatusBarAppearance()
 {
   static BOOL value;
@@ -67,6 +70,7 @@
 
   return value;
 }
+#endif // [macOS]
 
 RCT_EXPORT_MODULE()
 
@@ -78,7 +82,11 @@
 - (void)initialize
 {
   _constants = facebook::react::typedConstants<JS::NativeStatusBarManagerIOS::Constants>({
+#if !TARGET_OS_OSX // [macOS]
       .HEIGHT = RCTUIStatusBarManager().statusBarFrame.size.height,
+#else // [macOS
+      .HEIGHT = 0,
+#endif // macOS]
       .DEFAULT_BACKGROUND_COLOR = std::nullopt,
   });
 }
@@ -90,6 +98,7 @@
 
 - (void)startObserving
 {
+#if TARGET_OS_IOS // [visionOS]
   NSNotificationCenter *nc = [NSNotificationCenter defaultCenter];
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
@@ -101,6 +110,7 @@
          selector:@selector(applicationWillChangeStatusBarFrame:)
              name:UIApplicationWillChangeStatusBarFrameNotification
            object:nil];
+#endif // [visionOS]
 #pragma clang diagnostic pop
 }
 
@@ -111,6 +121,7 @@
 
 - (void)emitEvent:(NSString *)eventName forNotification:(NSNotification *)notification
 {
+#if TARGET_OS_IOS // [visionOS]
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
   CGRect frame = [notification.userInfo[UIApplicationStatusBarFrameUserInfoKey] CGRectValue];
@@ -124,6 +135,7 @@
     },
   };
   [self sendEventWithName:eventName body:event];
+#endif // [visionOS]
 }
 
 - (void)applicationDidChangeStatusBarFrame:(NSNotification *)notification
@@ -138,13 +150,26 @@
 
 RCT_EXPORT_METHOD(getHeight : (RCTResponseSenderBlock)callback)
 {
+#if !TARGET_OS_OSX // [macOS]
+#if !TARGET_OS_VISION // [visionOS]
   callback(@[ @{
     @"height" : @(RCTUIStatusBarManager().statusBarFrame.size.height),
   } ]);
+#else // [visionOS
+  callback(@[ @{
+    @"height" : @(RCTUIStatusBarManager().statusBarFrame.size),
+  } ]);
+#endif // visionOS]
+#else // [macOS
+  callback(@[ @{
+    @"height" : @(0),
+  } ]);
+#endif // macOS]
 }
 
 RCT_EXPORT_METHOD(setStyle : (NSString *)style animated : (BOOL)animated)
 {
+#if TARGET_OS_IOS // [macOS] [visionOS]
   dispatch_async(dispatch_get_main_queue(), ^{
     UIStatusBarStyle statusBarStyle = [RCTConvert UIStatusBarStyle:style];
     if (RCTViewControllerBasedStatusBarAppearance()) {
@@ -157,10 +182,12 @@
     }
 #pragma clang diagnostic pop
   });
+#endif // [macOS] [visionOS]
 }
 
 RCT_EXPORT_METHOD(setHidden : (BOOL)hidden withAnimation : (NSString *)withAnimation)
 {
+#if TARGET_OS_IOS // [macOS] [visionOS]
   dispatch_async(dispatch_get_main_queue(), ^{
     UIStatusBarAnimation animation = [RCTConvert UIStatusBarAnimation:withAnimation];
     if (RCTViewControllerBasedStatusBarAppearance()) {
@@ -173,10 +200,12 @@
 #pragma clang diagnostic pop
     }
   });
+#endif // [macOS] [visionOS]
 }
 
 RCT_EXPORT_METHOD(setNetworkActivityIndicatorVisible : (BOOL)visible)
 {
+#if TARGET_OS_IOS // [visionOS]
   dispatch_async(dispatch_get_main_queue(), ^{
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
@@ -184,6 +213,7 @@
     RCTSharedApplication().networkActivityIndicatorVisible = visible;
 #pragma clang diagnostic pop
   });
+#endif // [visionOS]
 }
 
 - (facebook::react::ModuleConstants<JS::NativeStatusBarManagerIOS::Constants>)getConstants
@@ -193,7 +223,7 @@
 
 - (facebook::react::ModuleConstants<JS::NativeStatusBarManagerIOS::Constants>)constantsToExport
 {
-  return [self getConstants];
+  return (facebook::react::ModuleConstants<JS::NativeStatusBarManagerIOS::Constants>)[self getConstants];
 }
 
 - (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
