--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/components/view/AccessibilityPrimitives.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/components/view/AccessibilityPrimitives.h	2026-03-06 15:22:53
@@ -34,15 +34,32 @@
   Header = (1 << 15),
   Switch = (1 << 16),
   TabBar = (1 << 17),
+#if TARGET_OS_OSX  // [macOS
+  ComboBox = (1 << 18),
+  Menu = (1 << 19),
+  PopUp = (1 << 20),
+  Bar = (1 << 21),
+  Item = (1 << 22),
+  Group = (1 << 23),
+  List = (1 << 24),
+  Tab = (1 << 25),
+  Table = (1 << 26),
+  Disclosure = (1 << 27),
+  Radio = (1 << 28),
+  ScrollBar = (1 << 29),
+  SpinButton = (1 << 30),
+#endif // macOS]
 };
 
-constexpr enum AccessibilityTraits operator|(const enum AccessibilityTraits lhs, const enum AccessibilityTraits rhs)
-{
+constexpr enum AccessibilityTraits operator|(
+    const enum AccessibilityTraits lhs,
+    const enum AccessibilityTraits rhs) {
   return (enum AccessibilityTraits)((uint32_t)lhs | (uint32_t)rhs);
 }
 
-constexpr enum AccessibilityTraits operator&(const enum AccessibilityTraits lhs, const enum AccessibilityTraits rhs)
-{
+constexpr enum AccessibilityTraits operator&(
+    const enum AccessibilityTraits lhs,
+    const enum AccessibilityTraits rhs) {
   return (enum AccessibilityTraits)((uint32_t)lhs & (uint32_t)rhs);
 }
 
@@ -51,13 +68,36 @@
   std::optional<std::string> label{};
 };
 
-inline static bool operator==(const AccessibilityAction &lhs, const AccessibilityAction &rhs)
-{
+inline std::string toString(const AccessibilityAction& accessibilityAction) {
+  std::string result = accessibilityAction.name;
+  if (accessibilityAction.label.has_value()) {
+    result += ": '" + accessibilityAction.label.value() + "'";
+  }
+  return result;
+}
+
+inline std::string toString(
+    std::vector<AccessibilityAction> accessibilityActions) {
+  std::string result = "[";
+  for (size_t i = 0; i < accessibilityActions.size(); i++) {
+    result += toString(accessibilityActions[i]);
+    if (i < accessibilityActions.size() - 1) {
+      result += ", ";
+    }
+  }
+  result += "]";
+  return result;
+}
+
+inline static bool operator==(
+    const AccessibilityAction& lhs,
+    const AccessibilityAction& rhs) {
   return lhs.name == rhs.name && lhs.label == rhs.label;
 }
 
-inline static bool operator!=(const AccessibilityAction &lhs, const AccessibilityAction &rhs)
-{
+inline static bool operator!=(
+    const AccessibilityAction& lhs,
+    const AccessibilityAction& rhs) {
   return !(rhs == lhs);
 }
 
@@ -66,17 +106,20 @@
   bool selected{false};
   bool busy{false};
   std::optional<bool> expanded{std::nullopt};
-  enum CheckedState { Unchecked, Checked, Mixed, None } checked{None};
+  enum { Unchecked, Checked, Mixed, None } checked{None};
 };
 
-constexpr bool operator==(const AccessibilityState &lhs, const AccessibilityState &rhs)
-{
-  return lhs.disabled == rhs.disabled && lhs.selected == rhs.selected && lhs.checked == rhs.checked &&
-      lhs.busy == rhs.busy && lhs.expanded == rhs.expanded;
+constexpr bool operator==(
+    const AccessibilityState& lhs,
+    const AccessibilityState& rhs) {
+  return lhs.disabled == rhs.disabled && lhs.selected == rhs.selected &&
+      lhs.checked == rhs.checked && lhs.busy == rhs.busy &&
+      lhs.expanded == rhs.expanded;
 }
 
-constexpr bool operator!=(const AccessibilityState &lhs, const AccessibilityState &rhs)
-{
+constexpr bool operator!=(
+    const AccessibilityState& lhs,
+    const AccessibilityState& rhs) {
   return !(rhs == lhs);
 }
 
@@ -84,13 +127,15 @@
   std::vector<std::string> value{};
 };
 
-inline static bool operator==(const AccessibilityLabelledBy &lhs, const AccessibilityLabelledBy &rhs)
-{
+inline static bool operator==(
+    const AccessibilityLabelledBy& lhs,
+    const AccessibilityLabelledBy& rhs) {
   return lhs.value == rhs.value;
 }
 
-inline static bool operator!=(const AccessibilityLabelledBy &lhs, const AccessibilityLabelledBy &rhs)
-{
+inline static bool operator!=(
+    const AccessibilityLabelledBy& lhs,
+    const AccessibilityLabelledBy& rhs) {
   return !(lhs == rhs);
 }
 
@@ -101,13 +146,16 @@
   std::optional<std::string> text{};
 };
 
-constexpr bool operator==(const AccessibilityValue &lhs, const AccessibilityValue &rhs)
-{
-  return lhs.min == rhs.min && lhs.max == rhs.max && lhs.now == rhs.now && lhs.text == rhs.text;
+constexpr bool operator==(
+    const AccessibilityValue& lhs,
+    const AccessibilityValue& rhs) {
+  return lhs.min == rhs.min && lhs.max == rhs.max && lhs.now == rhs.now &&
+      lhs.text == rhs.text;
 }
 
-constexpr bool operator!=(const AccessibilityValue &lhs, const AccessibilityValue &rhs)
-{
+constexpr bool operator!=(
+    const AccessibilityValue& lhs,
+    const AccessibilityValue& rhs) {
   return !(rhs == lhs);
 }
 
@@ -124,7 +172,19 @@
   Assertive,
 };
 
-enum class AccessibilityRole : uint8_t {
+inline std::string toString(
+    const AccessibilityLiveRegion& accessibilityLiveRegion) {
+  switch (accessibilityLiveRegion) {
+    case AccessibilityLiveRegion::None:
+      return "none";
+    case AccessibilityLiveRegion::Polite:
+      return "polite";
+    case AccessibilityLiveRegion::Assertive:
+      return "assertive";
+  }
+}
+
+enum class AccessibilityRole {
   None,
   Button,
   Dropdownlist,
@@ -167,7 +227,7 @@
   Iconmenu,
 };
 
-enum class Role : uint8_t {
+enum class Role {
   Alert,
   Alertdialog,
   Application,
