--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/components/view/BaseTouch.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/components/view/BaseTouch.h	2026-03-06 15:22:53
@@ -55,33 +55,62 @@
    * The time in seconds when the touch occurred or when it was last mutated.
    */
   Float timestamp;
+#if TARGET_OS_OSX // [macOS
+  /*
+   * The button indicating which pointer is used.
+   */
+  int button;
 
   /*
+   * A flag indicating if the alt key is pressed.
+   */
+  bool altKey;
+
+  /*
+   * A flag indicating if the control key is pressed.
+   */
+  bool ctrlKey;
+
+  /*
+   * A flag indicating if the shift key is pressed.
+   */
+  bool shiftKey;
+
+  /*
+   * A flag indicating if the meta key is pressed.
+   */
+  bool metaKey;
+#endif // macOS]
+
+  /*
    * The particular implementation of `Hasher` and (especially) `Comparator`
    * make sense only when `Touch` object is used as a *key* in indexed
    * collections. Because of that they are expressed as separate classes.
    */
   struct Hasher {
-    size_t operator()(const BaseTouch &touch) const
-    {
+    size_t operator()(const BaseTouch& touch) const {
       return std::hash<decltype(touch.identifier)>()(touch.identifier);
     }
   };
 
   struct Comparator {
-    bool operator()(const BaseTouch &lhs, const BaseTouch &rhs) const
-    {
+    bool operator()(const BaseTouch& lhs, const BaseTouch& rhs) const {
       return lhs.identifier == rhs.identifier;
     }
   };
 };
 
-void setTouchPayloadOnObject(jsi::Object &object, jsi::Runtime &runtime, const BaseTouch &touch);
+void setTouchPayloadOnObject(
+    jsi::Object& object,
+    jsi::Runtime& runtime,
+    const BaseTouch& touch);
 
 #if RN_DEBUG_STRING_CONVERTIBLE
 
-std::string getDebugName(const BaseTouch &touch);
-std::vector<DebugStringConvertibleObject> getDebugProps(const BaseTouch &touch, DebugStringConvertibleOptions options);
+std::string getDebugName(const BaseTouch& touch);
+std::vector<DebugStringConvertibleObject> getDebugProps(
+    const BaseTouch& touch,
+    DebugStringConvertibleOptions options);
 
 #endif
 
