--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/components/view/primitives.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/components/view/primitives.h	2026-03-06 15:22:53
@@ -69,24 +69,20 @@
     PointerUpCapture = 37,
   };
 
-  constexpr bool operator[](const Offset offset) const
-  {
+  constexpr bool operator[](const Offset offset) const {
     return bits[static_cast<std::size_t>(offset)];
   }
 
-  std::bitset<64>::reference operator[](const Offset offset)
-  {
+  std::bitset<64>::reference operator[](const Offset offset) {
     return bits[static_cast<std::size_t>(offset)];
   }
 };
 
-inline static bool operator==(const ViewEvents &lhs, const ViewEvents &rhs)
-{
+inline static bool operator==(const ViewEvents& lhs, const ViewEvents& rhs) {
   return lhs.bits == rhs.bits;
 }
 
-inline static bool operator!=(const ViewEvents &lhs, const ViewEvents &rhs)
-{
+inline static bool operator!=(const ViewEvents& lhs, const ViewEvents& rhs) {
   return lhs.bits != rhs.bits;
 }
 
@@ -102,7 +98,7 @@
   float vertical{0.0f};
   float horizontal{0.0f};
 
-  bool operator==(const CornerRadii &other) const = default;
+  bool operator==(const CornerRadii& other) const = default;
 };
 
 enum class Cursor : uint8_t {
@@ -138,6 +134,7 @@
   SWResize,
   Text,
   Url,
+  VerticalText, // [macOS]
   WResize,
   Wait,
   ZoomIn,
@@ -164,27 +161,29 @@
   OptionalT blockStart{};
   OptionalT blockEnd{};
 
-  Counterpart resolve(bool isRTL, T defaults) const
-  {
+  Counterpart resolve(bool isRTL, T defaults) const {
     const auto leadingEdge = isRTL ? end : start;
     const auto trailingEdge = isRTL ? start : end;
-    const auto horizontalOrAllOrDefault = horizontal.value_or(all.value_or(defaults));
-    const auto verticalOrAllOrDefault = vertical.value_or(all.value_or(defaults));
+    const auto horizontalOrAllOrDefault =
+        horizontal.value_or(all.value_or(defaults));
+    const auto verticalOrAllOrDefault =
+        vertical.value_or(all.value_or(defaults));
 
     return {
         /* .left = */
         left.value_or(leadingEdge.value_or(horizontalOrAllOrDefault)),
         /* .top = */
-        blockStart.value_or(block.value_or(top.value_or(verticalOrAllOrDefault))),
+        blockStart.value_or(
+            block.value_or(top.value_or(verticalOrAllOrDefault))),
         /* .right = */
         right.value_or(trailingEdge.value_or(horizontalOrAllOrDefault)),
         /* .bottom = */
-        blockEnd.value_or(block.value_or(bottom.value_or(verticalOrAllOrDefault))),
+        blockEnd.value_or(
+            block.value_or(bottom.value_or(verticalOrAllOrDefault))),
     };
   }
 
-  bool operator==(const CascadedRectangleEdges<T> &rhs) const
-  {
+  bool operator==(const CascadedRectangleEdges<T>& rhs) const {
     return std::tie(
                this->left,
                this->top,
@@ -213,8 +212,7 @@
                rhs.blockEnd);
   }
 
-  bool operator!=(const CascadedRectangleEdges<T> &rhs) const
-  {
+  bool operator!=(const CascadedRectangleEdges<T>& rhs) const {
     return !(*this == rhs);
   }
 };
@@ -238,8 +236,7 @@
   OptionalT startEnd{};
   OptionalT startStart{};
 
-  Counterpart resolve(bool isRTL, T defaults) const
-  {
+  Counterpart resolve(bool isRTL, T defaults) const {
     const auto logicalTopStart = topStart ? topStart : startStart;
     const auto logicalTopEnd = topEnd ? topEnd : startEnd;
     const auto logicalBottomStart = bottomStart ? bottomStart : endStart;
@@ -251,7 +248,8 @@
     const auto bottomTrailing = isRTL ? logicalBottomStart : logicalBottomEnd;
 
     return {
-        /* .topLeft = */ topLeft.value_or(topLeading.value_or(all.value_or(defaults))),
+        /* .topLeft = */ topLeft.value_or(
+            topLeading.value_or(all.value_or(defaults))),
         /* .topRight = */
         topRight.value_or(topTrailing.value_or(all.value_or(defaults))),
         /* .bottomLeft = */
@@ -261,8 +259,7 @@
     };
   }
 
-  bool operator==(const CascadedRectangleCorners<T> &rhs) const
-  {
+  bool operator==(const CascadedRectangleCorners<T>& rhs) const {
     return std::tie(
                this->topLeft,
                this->topRight,
@@ -293,8 +290,7 @@
                rhs.startStart);
   }
 
-  bool operator!=(const CascadedRectangleCorners<T> &rhs) const
-  {
+  bool operator!=(const CascadedRectangleCorners<T>& rhs) const {
     return !(*this == rhs);
   }
 };
@@ -318,15 +314,22 @@
   BorderCurves borderCurves{};
   BorderStyles borderStyles{};
 
-  bool operator==(const BorderMetrics &rhs) const
-  {
+  bool operator==(const BorderMetrics& rhs) const {
     return std::tie(
-               this->borderColors, this->borderWidths, this->borderRadii, this->borderCurves, this->borderStyles) ==
-        std::tie(rhs.borderColors, rhs.borderWidths, rhs.borderRadii, rhs.borderCurves, rhs.borderStyles);
+               this->borderColors,
+               this->borderWidths,
+               this->borderRadii,
+               this->borderCurves,
+               this->borderStyles) ==
+        std::tie(
+               rhs.borderColors,
+               rhs.borderWidths,
+               rhs.borderRadii,
+               rhs.borderCurves,
+               rhs.borderStyles);
   }
 
-  bool operator!=(const BorderMetrics &rhs) const
-  {
+  bool operator!=(const BorderMetrics& rhs) const {
     return !(*this == rhs);
   }
 };
