--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/imagemanager/platform/ios/react/renderer/imagemanager/RCTImagePrimitivesConversions.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/imagemanager/platform/ios/react/renderer/imagemanager/RCTImagePrimitivesConversions.h	2026-03-06 15:22:53
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTConvert.h>
 #import <React/RCTImageLoader.h>
@@ -81,7 +81,7 @@
 
     NSURL *url = [[NSURL alloc] initWithString:urlString];
 
-    if (url.scheme != nullptr) {
+    if (url.scheme) {
       // Well-formed absolute URL.
       return url;
     }
@@ -119,7 +119,7 @@
 {
   NSURL *url = NSURLFromImageSource(imageSource);
 
-  if (url == nullptr) {
+  if (!url) {
     RCTLogError(@"URI parsing error.");
     return nil;
   }
