--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/imagemanager/platform/ios/react/renderer/imagemanager/RCTSyncImageManager.mm	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/imagemanager/platform/ios/react/renderer/imagemanager/RCTSyncImageManager.mm	2026-03-06 15:22:53
@@ -51,7 +51,7 @@
 
   NSURLRequest *request = NSURLRequestFromImageSource(imageSource);
 
-  auto completionBlock = ^(NSError *error, UIImage *image, id metadata) {
+  auto completionBlock = ^(NSError *error, RCTPlatformImage *image, id metadata) { // [macOS]
     auto observerCoordinator = weakObserverCoordinator.lock();
     if (!observerCoordinator) {
       return;
@@ -85,12 +85,12 @@
                                          priority:RCTImageLoaderPriorityImmediate
                                       attribution:{
                                                       .surfaceId = surfaceId,
-  }
+                                                  }
                                     progressBlock:progressBlock
                                  partialLoadBlock:nil
                                   completionBlock:completionBlock];
 
-  auto result = dispatch_group_wait(imageWaitGroup, dispatch_time(DISPATCH_TIME_NOW, 5 * NSEC_PER_SEC));
+  auto result = dispatch_group_wait(imageWaitGroup, dispatch_time(DISPATCH_TIME_NOW, 2 * NSEC_PER_SEC));
   if (result != 0) {
     RCTLogError(@"Image timed out in test environment for url: %@", loaderRequest.imageURL);
   }
