--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/RCTComponentViewDescriptor.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/RCTComponentViewDescriptor.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTComponentViewProtocol.h>
 
@@ -21,15 +21,14 @@
   /*
    * Associated (and owned) native view instance.
    */
-  __strong UIView<RCTComponentViewProtocol> *const view = nil;
-
+  __strong RCTUIView<RCTComponentViewProtocol> *view = nil; // [macOS]
   /*
    * Indicates a requirement to call on the view methods from
    * `RCTMountingTransactionObserving` protocol.
    */
-  const bool observesMountingTransactionWillMount{false};
-  const bool observesMountingTransactionDidMount{false};
-  const bool shouldBeRecycled{true};
+  bool observesMountingTransactionWillMount{false};
+  bool observesMountingTransactionDidMount{false};
+  bool shouldBeRecycled{true};
 };
 
 inline bool operator==(const RCTComponentViewDescriptor &lhs, const RCTComponentViewDescriptor &rhs)
