--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/RCTComponentViewRegistry.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/RCTComponentViewRegistry.mm	2026-03-06 15:22:52
@@ -31,10 +31,12 @@
   if (self = [super init]) {
     _componentViewFactory = [RCTComponentViewFactory currentComponentViewFactory];
 
+#if !TARGET_OS_OSX // [macOS]
     [[NSNotificationCenter defaultCenter] addObserver:self
                                              selector:@selector(handleApplicationDidReceiveMemoryWarningNotification)
                                                  name:UIApplicationDidReceiveMemoryWarningNotification
                                                object:nil];
+#endif // [macOS]
   }
 
   return self;
@@ -50,7 +52,11 @@
       @"RCTComponentViewRegistry: Attempt to dequeue already registered component.");
 
   auto componentViewDescriptor = [self _dequeueComponentViewWithComponentHandle:componentHandle];
+#if !TARGET_OS_OSX // [macOS]
   componentViewDescriptor.view.tag = tag;
+#else // [macOS
+  componentViewDescriptor.view.reactTag = @(tag);
+#endif // macOS]
   auto it = _registry.insert({tag, componentViewDescriptor});
   return it.first->second;
 }
@@ -65,7 +71,11 @@
       _registry.find(tag) != _registry.end(), @"RCTComponentViewRegistry: Attempt to enqueue unregistered component.");
 
   _registry.erase(tag);
+#if !TARGET_OS_OSX // [macOS]
   componentViewDescriptor.view.tag = 0;
+#else // [macOS
+  componentViewDescriptor.view.reactTag = @0;
+#endif // macOS]
   [self _enqueueComponentViewWithComponentHandle:componentHandle componentViewDescriptor:componentViewDescriptor];
 }
 
@@ -77,7 +87,7 @@
   return iterator->second;
 }
 
-- (nullable UIView<RCTComponentViewProtocol> *)findComponentViewWithTag:(Tag)tag
+- (nullable RCTUIView<RCTComponentViewProtocol> *)findComponentViewWithTag:(Tag)tag // [macOS]
 {
   RCTAssertMainQueue();
   auto iterator = _registry.find(tag);
@@ -108,7 +118,6 @@
   auto &recycledViews = _recyclePool[componentHandle];
 
   if (recycledViews.size() > RCTComponentViewRegistryRecyclePoolMaxSize || !componentViewDescriptor.shouldBeRecycled) {
-    [componentViewDescriptor.view invalidate];
     return;
   }
 
