--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/RCTMountingManager.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/RCTMountingManager.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTMountingManagerDelegate.h>
 #import <React/RCTPrimitives.h>
@@ -28,7 +28,7 @@
 @property (nonatomic, weak) id<RCTMountingManagerDelegate> delegate;
 @property (nonatomic, strong) RCTComponentViewRegistry *componentViewRegistry;
 
-- (void)setContextContainer:(std::shared_ptr<const facebook::react::ContextContainer>)contextContainer;
+- (void)setContextContainer:(facebook::react::ContextContainer::Shared)contextContainer;
 
 /**
  * Designates the view as a rendering viewport of a React Native surface.
@@ -37,12 +37,12 @@
  * influence the intrinsic size of the view and cannot be measured using UIView/UIKit layout API.
  * Must be called on the main thead.
  */
-- (void)attachSurfaceToView:(UIView *)view surfaceId:(facebook::react::SurfaceId)surfaceId;
+- (void)attachSurfaceToView:(RCTPlatformView *)view surfaceId:(facebook::react::SurfaceId)surfaceId; // [macOS]
 
 /**
  * Stops designating the view as a rendering viewport of a React Native surface.
  */
-- (void)detachSurfaceFromView:(UIView *)view surfaceId:(facebook::react::SurfaceId)surfaceId;
+- (void)detachSurfaceFromView:(RCTPlatformView *)view surfaceId:(facebook::react::SurfaceId)surfaceId; // [macOS]
 
 /**
  * Schedule a mounting transaction to be performed on the main thread.
