--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.QnnaHSpWR8	2026-03-06 18:50:50
@@ -10,7 +10,9 @@
 /**
  * UIView class for root <ModalHostView> component.
  */
-@interface RCTModalHostViewComponentView : RCTViewComponentView <UIAdaptivePresentationControllerDelegate>
+@interface RCTModalHostViewComponentView : RCTViewComponentView
+#if !TARGET_OS_OSX // [macOS]
+<UIAdaptivePresentationControllerDelegate>
 
 /**
  * Subclasses may override this method and present the modal on different view controller.
@@ -27,5 +29,4 @@
 - (void)dismissViewController:(UIViewController *)modalViewController
                      animated:(BOOL)animated
                    completion:(void (^)(void))completion;
-
-@end
+#endif // [macOS]
