--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/SafeAreaView/RCTSafeAreaView.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/SafeAreaView/RCTSafeAreaView.m	2026-03-06 15:22:52
@@ -5,10 +5,9 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#if !TARGET_OS_OSX // [macOS]
 #import "RCTSafeAreaView.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTBridge.h>
 #import <React/RCTUIManager.h>
 
@@ -31,6 +30,7 @@
 RCT_NOT_IMPLEMENTED(-(instancetype)initWithCoder : (NSCoder *)decoder)
 RCT_NOT_IMPLEMENTED(-(instancetype)initWithFrame : (CGRect)frame)
 
+#if DEBUG // [macOS] description is a debug-only feature
 - (NSString *)description
 {
   NSString *superDescription = [super description];
@@ -45,6 +45,7 @@
                                     NSStringFromUIEdgeInsets(self.safeAreaInsets),
                                     NSStringFromUIEdgeInsets(_currentSafeAreaInsets)];
 }
+#endif // [macOS]
 
 static BOOL UIEdgeInsetsEqualToEdgeInsetsWithThreshold(UIEdgeInsets insets1, UIEdgeInsets insets2, CGFloat threshold)
 {
@@ -75,5 +76,4 @@
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // [macOS]
