--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/SafeAreaView/RCTSafeAreaViewLocalData.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/SafeAreaView/RCTSafeAreaViewLocalData.h	2026-03-06 15:22:52
@@ -5,13 +5,10 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 NS_ASSUME_NONNULL_BEGIN
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
 @interface RCTSafeAreaViewLocalData : NSObject
 
 - (instancetype)initWithInsets:(UIEdgeInsets)insets;
@@ -21,5 +18,3 @@
 @end
 
 NS_ASSUME_NONNULL_END
-
-#endif // RCT_REMOVE_LEGACY_ARCH
