diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTActivityIndicatorView.h upstream-rn-0.84.1/React/Views/RCTActivityIndicatorView.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTActivityIndicatorView.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTActivityIndicatorView.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTAutoInsetsProtocol.h upstream-rn-0.84.1/React/Views/RCTAutoInsetsProtocol.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTAutoInsetsProtocol.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTAutoInsetsProtocol.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 /**
  * Defines a View that wants to support auto insets adjustment
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTBorderDrawing.h upstream-rn-0.84.1/React/Views/RCTBorderDrawing.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTBorderDrawing.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTBorderDrawing.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #import <React/RCTBorderStyle.h>
 #import <React/RCTDefines.h>
@@ -29,10 +29,10 @@
 } RCTCornerInsets;
 
 typedef struct {
-  UIColor *top;
-  UIColor *left;
-  UIColor *bottom;
-  UIColor *right;
+  RCTUIColor *top;
+  RCTUIColor *left;
+  RCTUIColor *bottom;
+  RCTUIColor *right;
 } RCTBorderColors;
 
 /**
@@ -75,5 +75,5 @@
     RCTCornerRadii cornerRadii,
     UIEdgeInsets borderInsets,
     RCTBorderColors borderColors,
-    UIColor *backgroundColor,
+    RCTUIColor *backgroundColor,
     BOOL drawToEdge);
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTComponentData.h upstream-rn-0.84.1/React/Views/RCTComponentData.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTComponentData.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTComponentData.h	2026-04-14 21:48:12
@@ -34,7 +34,7 @@
                               bridge:(RCTBridge *)bridge
                      eventDispatcher:(id<RCTEventDispatcherProtocol>)eventDispatcher NS_DESIGNATED_INITIALIZER;
 
-- (UIView *)createViewWithTag:(nullable NSNumber *)tag rootTag:(nullable NSNumber *)rootTag;
+- (RCTPlatformView *)createViewWithTag:(nullable NSNumber *)tag rootTag:(nullable NSNumber *)rootTag;
 - (RCTShadowView *)createShadowViewWithTag:(NSNumber *)tag;
 - (void)setProps:(NSDictionary<NSString *, id> *)props forView:(id<RCTComponent>)view;
 - (void)setProps:(NSDictionary<NSString *, id> *)props forShadowView:(RCTShadowView *)shadowView;
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTDebuggingOverlay.h upstream-rn-0.84.1/React/Views/RCTDebuggingOverlay.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTDebuggingOverlay.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTDebuggingOverlay.h	2026-04-14 21:48:12
@@ -5,16 +5,16 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #import <React/RCTView.h>
 
 @interface TraceUpdateTuple : NSObject
 
-@property (nonatomic, strong, readonly) UIView *view;
+@property (nonatomic, strong, readonly) RCTPlatformView *view;
 @property (nonatomic, copy, readonly) dispatch_block_t cleanupBlock;
 
-- (instancetype)initWithView:(UIView *)view cleanupBlock:(dispatch_block_t)cleanupBlock;
+- (instancetype)initWithView:(RCTPlatformView *)view cleanupBlock:(dispatch_block_t)cleanupBlock;
 
 @end
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTFont+Private.h upstream-rn-0.84.1/React/Views/RCTFont+Private.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTFont+Private.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTFont+Private.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #import <React/RCTDefines.h>
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTLayout.h upstream-rn-0.84.1/React/Views/RCTLayout.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTLayout.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTLayout.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #import <React/RCTDefines.h>
 #import <yoga/Yoga.h>
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTModalHostView.h upstream-rn-0.84.1/React/Views/RCTModalHostView.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTModalHostView.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTModalHostView.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
@@ -19,7 +19,7 @@
 @protocol RCTModalHostViewInteractor;
 
 __attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTModalHostView : UIView<RCTInvalidating, UIAdaptivePresentationControllerDelegate>
+@interface RCTModalHostView : RCTUIView<RCTInvalidating, UIAdaptivePresentationControllerDelegate>
 
 @property (nonatomic, copy) NSString *animationType;
 @property (nonatomic, assign) UIModalPresentationStyle presentationStyle;
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTModalHostViewController.h upstream-rn-0.84.1/React/Views/RCTModalHostViewController.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTModalHostViewController.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTModalHostViewController.h	2026-04-14 21:48:12
@@ -5,12 +5,12 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
 __attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTModalHostViewController : UIViewController
+@interface RCTModalHostViewController : RCTPlatformViewController
 
 @property (nonatomic, copy) void (^boundsDidChangeBlock)(CGRect newBounds);
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTModalHostViewManager.h upstream-rn-0.84.1/React/Views/RCTModalHostViewManager.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTModalHostViewManager.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTModalHostViewManager.h	2026-04-14 21:48:12
@@ -12,8 +12,8 @@
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
 typedef void (^RCTModalViewInteractionBlock)(
-    UIViewController *reactViewController,
-    UIViewController *viewController,
+    RCTPlatformViewController *reactViewController,
+    RCTPlatformViewController *viewController,
     BOOL animated,
     dispatch_block_t completionBlock)
     __attribute__((deprecated("This API will be removed along with the legacy architecture.")));
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTModalManager.h upstream-rn-0.84.1/React/Views/RCTModalManager.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTModalManager.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTModalManager.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTShadowView+Internal.h upstream-rn-0.84.1/React/Views/RCTShadowView+Internal.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTShadowView+Internal.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTShadowView+Internal.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #import <React/RCTShadowView.h>
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTShadowView+Layout.h upstream-rn-0.84.1/React/Views/RCTShadowView+Layout.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTShadowView+Layout.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTShadowView+Layout.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #import <React/RCTShadowView.h>
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTShadowView.h upstream-rn-0.84.1/React/Views/RCTShadowView.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTShadowView.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTShadowView.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #import <React/RCTComponent.h>
 // Keeps RCTConvert.h here before yoga for clang module to generate correct header imports.
@@ -17,7 +17,7 @@
 @class RCTRootShadowView;
 @class RCTSparseArray;
 
-typedef void (^RCTApplierBlock)(NSDictionary<NSNumber *, UIView *> *viewRegistry);
+typedef void (^RCTApplierBlock)(NSDictionary<NSNumber *, RCTPlatformView *> *viewRegistry);
 
 /**
  * ShadowView tree mirrors RCT view tree. Every node is highly stateful.
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTSwitch.h upstream-rn-0.84.1/React/Views/RCTSwitch.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTSwitch.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTSwitch.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTView.h upstream-rn-0.84.1/React/Views/RCTView.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTView.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTView.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #import <React/RCTBorderCurve.h>
 #import <React/RCTBorderStyle.h>
@@ -13,11 +13,11 @@
 #import <React/RCTCursor.h>
 #import <React/RCTPointerEvents.h>
 
-extern const UIAccessibilityTraits SwitchAccessibilityTrait;
+extern const RCTUIAccessibilityTraits SwitchAccessibilityTrait;
 
 @protocol RCTAutoInsetsProtocol;
 
-@interface RCTView : UIView
+@interface RCTView : RCTUIView
 
 /**
  * Accessibility event handlers
@@ -32,8 +32,8 @@
  */
 @property (nonatomic, assign) RCTPointerEvents pointerEvents;
 
-+ (void)autoAdjustInsetsForView:(UIView<RCTAutoInsetsProtocol> *)parentView
-                 withScrollView:(UIScrollView *)scrollView
++ (void)autoAdjustInsetsForView:(RCTPlatformView<RCTAutoInsetsProtocol> *)parentView
+                 withScrollView:(RCTUIScrollView *)scrollView
                    updateOffset:(BOOL)updateOffset;
 
 /**
@@ -80,16 +80,16 @@
 /**
  * Border colors (actually retained).
  */
-@property (nonatomic, strong) UIColor *borderTopColor;
-@property (nonatomic, strong) UIColor *borderRightColor;
-@property (nonatomic, strong) UIColor *borderBottomColor;
-@property (nonatomic, strong) UIColor *borderLeftColor;
-@property (nonatomic, strong) UIColor *borderStartColor;
-@property (nonatomic, strong) UIColor *borderEndColor;
-@property (nonatomic, strong) UIColor *borderColor;
-@property (nonatomic, strong) UIColor *borderBlockColor;
-@property (nonatomic, strong) UIColor *borderBlockEndColor;
-@property (nonatomic, strong) UIColor *borderBlockStartColor;
+@property (nonatomic, strong) RCTUIColor *borderTopColor;
+@property (nonatomic, strong) RCTUIColor *borderRightColor;
+@property (nonatomic, strong) RCTUIColor *borderBottomColor;
+@property (nonatomic, strong) RCTUIColor *borderLeftColor;
+@property (nonatomic, strong) RCTUIColor *borderStartColor;
+@property (nonatomic, strong) RCTUIColor *borderEndColor;
+@property (nonatomic, strong) RCTUIColor *borderColor;
+@property (nonatomic, strong) RCTUIColor *borderBlockColor;
+@property (nonatomic, strong) RCTUIColor *borderBlockEndColor;
+@property (nonatomic, strong) RCTUIColor *borderBlockStartColor;
 
 /**
  * Border widths.
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTViewManager.h upstream-rn-0.84.1/React/Views/RCTViewManager.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTViewManager.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTViewManager.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #import <React/RCTBridgeModule.h>
 #import <React/RCTConvert.h>
@@ -19,7 +19,7 @@
 @class RCTSparseArray;
 @class RCTUIManager;
 
-typedef void (^RCTViewManagerUIBlock)(RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry);
+typedef void (^RCTViewManagerUIBlock)(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTPlatformView *> *viewRegistry);
 
 @interface RCTViewManager : NSObject <RCTBridgeModule>
 
@@ -37,7 +37,7 @@
  * return a fresh instance each time. The view module MUST NOT cache the returned
  * view and return the same instance for subsequent calls.
  */
-- (UIView *)view;
+- (RCTPlatformView *)view;
 
 /**
  * This method instantiates a shadow view to be managed by the module. If omitted,
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTViewUtils.h upstream-rn-0.84.1/React/Views/RCTViewUtils.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTViewUtils.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTViewUtils.h	2026-04-14 21:48:12
@@ -6,13 +6,13 @@
  */
 
 #import <React/RCTDefines.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
 RCT_EXTERN_C_BEGIN
 
-UIEdgeInsets RCTContentInsets(UIView *view);
+UIEdgeInsets RCTContentInsets(RCTPlatformView *view);
 
 RCT_EXTERN_C_END
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTWrapperViewController.h upstream-rn-0.84.1/React/Views/RCTWrapperViewController.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RCTWrapperViewController.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RCTWrapperViewController.h	2026-04-14 21:48:12
@@ -5,12 +5,12 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 @class RCTWrapperViewController;
 
-@interface RCTWrapperViewController : UIViewController
+@interface RCTWrapperViewController : RCTPlatformViewController
 
-- (instancetype)initWithContentView:(UIView *)contentView NS_DESIGNATED_INITIALIZER;
+- (instancetype)initWithContentView:(RCTPlatformView *)contentView NS_DESIGNATED_INITIALIZER;
 
 @end
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RefreshControl/RCTRefreshControl.h upstream-rn-0.84.1/React/Views/RefreshControl/RCTRefreshControl.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RefreshControl/RCTRefreshControl.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RefreshControl/RCTRefreshControl.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
@@ -17,7 +17,7 @@
 
 @property (nonatomic, copy) NSString *title;
 @property (nonatomic, copy) RCTDirectEventBlock onRefresh;
-@property (nonatomic, weak) UIScrollView *scrollView;
+@property (nonatomic, weak) RCTUIScrollView *scrollView;
 
 @end
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/RefreshControl/RCTRefreshableProtocol.h upstream-rn-0.84.1/React/Views/RefreshControl/RCTRefreshableProtocol.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/RefreshControl/RCTRefreshableProtocol.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/RefreshControl/RCTRefreshableProtocol.h	2026-04-14 21:48:12
@@ -6,7 +6,7 @@
  */
 
 #import <React/RCTComponent.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 /**
  * Protocol used to dispatch commands in `RCTRefreshControlManager.h`.
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/SafeAreaView/RCTSafeAreaView.h upstream-rn-0.84.1/React/Views/SafeAreaView/RCTSafeAreaView.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/SafeAreaView/RCTSafeAreaView.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/SafeAreaView/RCTSafeAreaView.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/SafeAreaView/RCTSafeAreaViewLocalData.h upstream-rn-0.84.1/React/Views/SafeAreaView/RCTSafeAreaViewLocalData.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/SafeAreaView/RCTSafeAreaViewLocalData.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/SafeAreaView/RCTSafeAreaViewLocalData.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/ScrollView/RCTScrollContentShadowView.h upstream-rn-0.84.1/React/Views/ScrollView/RCTScrollContentShadowView.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/ScrollView/RCTScrollContentShadowView.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/ScrollView/RCTScrollContentShadowView.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/ScrollView/RCTScrollContentView.h upstream-rn-0.84.1/React/Views/ScrollView/RCTScrollContentView.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/ScrollView/RCTScrollContentView.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/ScrollView/RCTScrollContentView.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/ScrollView/RCTScrollView.h upstream-rn-0.84.1/React/Views/ScrollView/RCTScrollView.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/ScrollView/RCTScrollView.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/ScrollView/RCTScrollView.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIScrollView.h>
+#import <React/RCTUIKit.h>
 
 #ifndef RCT_REMOVE_LEGACY_ARCH
 
@@ -15,10 +15,10 @@
 #import <React/RCTScrollableProtocol.h>
 #import <React/RCTView.h>
 
-@protocol UIScrollViewDelegate;
+@protocol RCTUIScrollViewDelegate;
 
 __attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTScrollView : RCTView<UIScrollViewDelegate, RCTScrollableProtocol, RCTAutoInsetsProtocol>
+@interface RCTScrollView : RCTView<RCTUIScrollViewDelegate, RCTScrollableProtocol, RCTAutoInsetsProtocol>
 
 - (instancetype)initWithEventDispatcher:(id<RCTEventDispatcherProtocol>)eventDispatcher NS_DESIGNATED_INITIALIZER;
 
@@ -29,12 +29,12 @@
  * efficiently since it will have already been computed by the off-main-thread
  * layout system.
  */
-@property (nonatomic, readonly) UIView *contentView;
+@property (nonatomic, readonly) RCTPlatformView *contentView;
 
 /**
  * The underlying scrollView (TODO: can we remove this?)
  */
-@property (nonatomic, readonly) UIScrollView *scrollView;
+@property (nonatomic, readonly) RCTUIScrollView *scrollView;
 
 @property (nonatomic, assign) UIEdgeInsets contentInset;
 @property (nonatomic, assign) BOOL automaticallyAdjustContentInsets;
@@ -54,7 +54,7 @@
 /** Focus area of newly-activated text input relative to the window to compare against UIKeyboardFrameBegin/End */
 @property (nonatomic, assign) CGRect firstResponderFocus;
 /** newly-activated text input outside of the scroll view */
-@property (nonatomic, weak) UIView *firstResponderViewOutsideScrollView;
+@property (nonatomic, weak) RCTPlatformView *firstResponderViewOutsideScrollView;
 
 // NOTE: currently these event props are only declared so we can export the
 // event names to JS - we don't call the blocks directly because scroll events
@@ -68,7 +68,7 @@
 
 @end
 
-@interface UIView (RCTScrollView)
+@interface RCTPlatformView (RCTScrollView)
 
 - (void)reactUpdateResponderOffsetForScrollView:(RCTScrollView *)scrollView;
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/ScrollView/RCTScrollableProtocol.h upstream-rn-0.84.1/React/Views/ScrollView/RCTScrollableProtocol.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/ScrollView/RCTScrollableProtocol.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/ScrollView/RCTScrollableProtocol.h	2026-04-14 21:48:12
@@ -6,7 +6,7 @@
  */
 
 #import <React/RCTComponent.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 /**
  * Contains any methods related to scrolling. Any `RCTView` that has scrolling
@@ -25,8 +25,8 @@
 - (void)scrollToEnd:(BOOL)animated;
 - (void)zoomToRect:(CGRect)rect animated:(BOOL)animated;
 
-- (void)addScrollListener:(NSObject<UIScrollViewDelegate> *)scrollListener;
-- (void)removeScrollListener:(NSObject<UIScrollViewDelegate> *)scrollListener;
+- (void)addScrollListener:(NSObject<RCTUIScrollViewDelegate> *)scrollListener;
+- (void)removeScrollListener:(NSObject<RCTUIScrollViewDelegate> *)scrollListener;
 
 @end
 
@@ -39,7 +39,7 @@
 @property (nonatomic, readonly, getter=isRefreshing) BOOL refreshing;
 
 @optional
-@property (nonatomic, weak) UIScrollView *scrollView;
+@property (nonatomic, weak) RCTUIScrollView *scrollView;
 
 @end
 
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/UIView+Private.h upstream-rn-0.84.1/React/Views/UIView+Private.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/UIView+Private.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/UIView+Private.h	2026-04-14 21:48:12
@@ -5,13 +5,13 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
-@interface UIView (Private)
+@interface RCTPlatformView (Private)
 
 // remove clipped subviews implementation
 - (void)react_remountAllSubviews;
-- (void)react_updateClippedSubviewsWithClipRect:(CGRect)clipRect relativeToView:(UIView *)clipView;
-- (UIView *)react_findClipView;
+- (void)react_updateClippedSubviewsWithClipRect:(CGRect)clipRect relativeToView:(RCTPlatformView *)clipView;
+- (RCTPlatformView *)react_findClipView;
 
 @end
diff -ruN /tmp/rn-diff-work/react-native-0.84.1/React/Views/UIView+React.h upstream-rn-0.84.1/React/Views/UIView+React.h
--- /tmp/rn-diff-work/react-native-0.84.1/React/Views/UIView+React.h	1985-10-26 01:15:00
+++ upstream-rn-0.84.1/React/Views/UIView+React.h	2026-04-14 21:48:12
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 
 #import <React/RCTComponent.h>
 #import <React/RCTConvert.h>
@@ -19,15 +19,15 @@
   CGFloat z;
 } RCTTransformOrigin;
 
-@interface UIView (React) <RCTComponent>
+@interface RCTPlatformView (React) <RCTComponent>
 
 /**
  * RCTComponent interface.
  */
-- (NSArray<UIView *> *)reactSubviews NS_REQUIRES_SUPER;
-- (UIView *)reactSuperview NS_REQUIRES_SUPER;
-- (void)insertReactSubview:(UIView *)subview atIndex:(NSInteger)atIndex NS_REQUIRES_SUPER;
-- (void)removeReactSubview:(UIView *)subview NS_REQUIRES_SUPER;
+- (NSArray<RCTPlatformView *> *)reactSubviews NS_REQUIRES_SUPER;
+- (RCTPlatformView *)reactSuperview NS_REQUIRES_SUPER;
+- (void)insertReactSubview:(RCTPlatformView *)subview atIndex:(NSInteger)atIndex NS_REQUIRES_SUPER;
+- (void)removeReactSubview:(RCTPlatformView *)subview NS_REQUIRES_SUPER;
 
 /**
  * The native id of the view, used to locate view from native codes
@@ -63,7 +63,7 @@
  * Subviews sorted by z-index. Note that this method doesn't do any caching (yet)
  * and sorts all the views each call.
  */
-- (NSArray<UIView *> *)reactZIndexSortedSubviews;
+- (NSArray<RCTPlatformView *> *)reactZIndexSortedSubviews;
 
 /**
  * Updates the subviews array based on the reactSubviews. Default behavior is
@@ -86,7 +86,7 @@
 /**
  * This method finds and returns the containing view controller for the view.
  */
-- (UIViewController *)reactViewController;
+- (RCTPlatformViewController *)reactViewController;
 
 /**
  * This method attaches the specified controller as a child of the
@@ -94,7 +94,7 @@
  * controller is found (which may happen if the view is not currently
  * attached to the view hierarchy).
  */
-- (void)reactAddControllerToClosestParent:(UIViewController *)controller;
+- (void)reactAddControllerToClosestParent:(RCTPlatformViewController *)controller;
 
 /**
  * Focus manipulation.
@@ -125,7 +125,7 @@
  * transparent in favour of some subview.
  * Defaults to `self`.
  */
-@property (nonatomic, readonly) UIView *reactAccessibilityElement;
+@property (nonatomic, readonly) RCTPlatformView *reactAccessibilityElement;
 
 /**
  * Accessibility properties
@@ -136,8 +136,8 @@
 @property (nonatomic, copy) NSArray<NSDictionary *> *accessibilityActions;
 @property (nonatomic, copy) NSDictionary *accessibilityValueInternal;
 @property (nonatomic, copy) NSString *accessibilityLanguage;
-@property (nonatomic) UIAccessibilityTraits accessibilityRoleTraits;
-@property (nonatomic) UIAccessibilityTraits roleTraits;
+@property (nonatomic) RCTUIAccessibilityTraits accessibilityRoleTraits;
+@property (nonatomic) RCTUIAccessibilityTraits roleTraits;
 
 /**
  * Used in debugging to get a description of the view hierarchy rooted at
