--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/Keyboard/Keyboard.js	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.lMH3TuoicM	2026-03-06 18:50:41
@@ -110,8 +110,10 @@
     new NativeEventEmitter(
       // T88715063: NativeEventEmitter only used this parameter on iOS. Now it uses it on all platforms, so this code was modified automatically to preserve its behavior
       // If you want to use the native module on other platforms, please remove this condition and test its behavior
-      Platform.OS !== 'ios' ? null : NativeKeyboardObserver,
-    );
+      Platform.OS !== 'ios' && Platform.OS !== 'macos' // [macOS] Also use this parameter on macOS
+        ? null
+        : NativeKeyboardObserver,
+    );
 
   constructor() {
     this.addListener('keyboardDidShow', ev => {
