--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Pressability/Pressability.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Pressability/Pressability.js	2026-03-06 15:22:52
@@ -15,9 +15,10 @@
   GestureResponderEvent,
   MouseEvent,
 } from '../Types/CoreEventTypes';
+import type {KeyEvent} from '../Types/CoreEventTypes'; // [macOS]
 
-import * as ReactNativeFeatureFlags from '../../src/private/featureflags/ReactNativeFeatureFlags';
 import SoundManager from '../Components/Sound/SoundManager';
+import ReactNativeFeatureFlags from '../ReactNative/ReactNativeFeatureFlags';
 import UIManager from '../ReactNative/UIManager';
 import {type RectOrSize, normalizeRect} from '../StyleSheet/Rect';
 import {type PointerEvent} from '../Types/CoreEventTypes';
@@ -88,47 +89,63 @@
   /**
    * Called after the element loses focus.
    */
-  onBlur?: ?(event: BlurEvent) => unknown,
+  onBlur?: ?(event: BlurEvent) => void,
 
   /**
    * Called after the element is focused.
    */
-  onFocus?: ?(event: FocusEvent) => unknown,
+  onFocus?: ?(event: FocusEvent) => void,
+
+  // [macOS
+  /*
+   * Called after a key down event is detected.
+   *
+   * @platform macos
+   */
+  onKeyDown?: ?(event: KeyEvent) => void,
+
+  /*
+   * Called after a key up event is detected.
+   *
+   * @platform macos
+   */
+  onKeyUp?: ?(event: KeyEvent) => void,
+  // macOS]
 
   /**
    * Called when the hover is activated to provide visual feedback.
    */
-  onHoverIn?: ?(event: MouseEvent) => unknown,
+  onHoverIn?: ?(event: MouseEvent) => mixed,
 
   /**
    * Called when the hover is deactivated to undo visual feedback.
    */
-  onHoverOut?: ?(event: MouseEvent) => unknown,
+  onHoverOut?: ?(event: MouseEvent) => mixed,
 
   /**
    * Called when a long press gesture has been triggered.
    */
-  onLongPress?: ?(event: GestureResponderEvent) => unknown,
+  onLongPress?: ?(event: GestureResponderEvent) => mixed,
 
   /**
    * Called when a press gesture has been triggered.
    */
-  onPress?: ?(event: GestureResponderEvent) => unknown,
+  onPress?: ?(event: GestureResponderEvent) => mixed,
 
   /**
    * Called when the press is activated to provide visual feedback.
    */
-  onPressIn?: ?(event: GestureResponderEvent) => unknown,
+  onPressIn?: ?(event: GestureResponderEvent) => mixed,
 
   /**
    * Called when the press location moves. (This should rarely be used.)
    */
-  onPressMove?: ?(event: GestureResponderEvent) => unknown,
+  onPressMove?: ?(event: GestureResponderEvent) => mixed,
 
   /**
    * Called when the press is deactivated to undo visual feedback.
    */
-  onPressOut?: ?(event: GestureResponderEvent) => unknown,
+  onPressOut?: ?(event: GestureResponderEvent) => mixed,
 
   /**
    * Whether to prevent any other native components from becoming responder
@@ -141,6 +158,8 @@
   onBlur: (event: BlurEvent) => void,
   onClick: (event: GestureResponderEvent) => void,
   onFocus: (event: FocusEvent) => void,
+  onKeyDown?: (event: KeyEvent) => void,
+  onKeyUp?: (event: KeyEvent) => void,
   onMouseEnter?: (event: MouseEvent) => void,
   onMouseLeave?: (event: MouseEvent) => void,
   onPointerEnter?: (event: PointerEvent) => void,
@@ -556,6 +575,49 @@
         () => this._config;
     }
 
+    // [macOS
+    const keyboardEventHandlers =
+      Platform.OS !== 'macos'
+        ? null
+        : {
+            onKeyDown: (event: KeyEvent): void => {
+              const {onKeyDown} = this._config;
+              if (onKeyDown != null) {
+                onKeyDown(event);
+              }
+              // Pressables on macOS should respond to the enter/return and spacebar keys.
+              // The keyDown event triggers a press event as well as the pressIn effect mimicking a native control behavior.
+              if (
+                (event.nativeEvent.key === 'Enter' ||
+                  event.nativeEvent.key === ' ') &&
+                event.defaultPrevented !== true
+              ) {
+                const {onPress, onPressIn} = this._config;
+                // $FlowFixMe: PressEvents don't mesh with keyboarding APIs. Keep legacy behavior of passing KeyEvents instead
+                onPressIn && onPressIn(event);
+                // $FlowFixMe: PressEvents don't mesh with keyboarding APIs. Keep legacy behavior of passing KeyEvents instead
+                onPress && onPress(event);
+              }
+            },
+            onKeyUp: (event: KeyEvent): void => {
+              const {onKeyUp} = this._config;
+              if (onKeyUp != null) {
+                onKeyUp(event);
+              }
+              // The keyUp event triggers the pressOut effect.
+              if (
+                (event.nativeEvent.key === 'Enter' ||
+                  event.nativeEvent.key === ' ') &&
+                event.defaultPrevented !== true
+              ) {
+                const {onPressOut} = this._config;
+                // $FlowFixMe: PressEvents don't mesh with keyboarding APIs. Keep legacy behavior of passing KeyEvents instead
+                onPressOut && onPressOut(event);
+              }
+            },
+          };
+    // macOS]
+
     if (
       ReactNativeFeatureFlags.shouldPressibilityUseW3CPointerEventsForHover()
     ) {
@@ -604,6 +666,7 @@
         ...focusEventHandlers,
         ...responderEventHandlers,
         ...hoverPointerEvents,
+        ...keyboardEventHandlers, // [macOS]
       };
     } else {
       const mouseEventHandlers =
@@ -655,6 +718,8 @@
       return {
         ...focusEventHandlers,
         ...responderEventHandlers,
+        // $FlowFixMe[exponential-spread] // [macOS]
+        ...keyboardEventHandlers, // [macOS]
         ...mouseEventHandlers,
       };
     }
