--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RefreshControl/RCTRefreshableProtocol.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RefreshControl/RCTRefreshableProtocol.h	2026-03-06 15:22:52
@@ -6,7 +6,7 @@
  */
 
 #import <React/RCTComponent.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 /**
  * Protocol used to dispatch commands in `RCTRefreshControlManager.h`.
