--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RefreshControl/RCTRefreshControl.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RefreshControl/RCTRefreshControl.h	2026-03-06 15:22:52
@@ -5,20 +5,17 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#if !TARGET_OS_OSX // [macOS]
 #import <UIKit/UIKit.h>
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTComponent.h>
 #import <React/RCTScrollableProtocol.h>
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTRefreshControl : UIRefreshControl<RCTCustomRefreshControlProtocol>
+@interface RCTRefreshControl : UIRefreshControl <RCTCustomRefreshControlProtocol>
 
 @property (nonatomic, copy) NSString *title;
 @property (nonatomic, copy) RCTDirectEventBlock onRefresh;
-@property (nonatomic, weak) UIScrollView *scrollView;
+@property (nonatomic, weak) RCTUIScrollView *scrollView; // [macOS]
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // [macOS]
