--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RefreshControl/RCTRefreshControlManager.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RefreshControl/RCTRefreshControlManager.m	2026-03-06 15:22:52
@@ -5,10 +5,9 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#if !TARGET_OS_OSX // [macOS]
 #import <React/RCTUIManager.h>
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import "RCTRefreshControl.h"
 #import "RCTRefreshControlManager.h"
 #import "RCTRefreshableProtocol.h"
@@ -43,5 +42,4 @@
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // [macOS]
