--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RefreshControl/RCTRefreshControl.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Views/RefreshControl/RCTRefreshControl.m	2026-03-06 15:22:52
@@ -5,11 +5,10 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#if !TARGET_OS_OSX // [macOS]
 #import "RCTRefreshControl.h"
 #import "RCTRefreshableProtocol.h"
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import "RCTUtils.h"
 
 @interface RCTRefreshControl () <RCTRefreshableProtocol>
@@ -73,9 +72,8 @@
 
 - (void)beginRefreshingProgrammatically
 {
-  if (!_hasMovedToWindow) {
+  if (!_hasMovedToWindow)
     return;
-  }
 
   UInt64 beginRefreshingTimestamp = _currentRefreshingStateTimestamp;
   _refreshingProgrammatically = YES;
@@ -111,9 +109,8 @@
 
 - (void)endRefreshingProgrammatically
 {
-  if (!_hasMovedToWindow) {
+  if (!_hasMovedToWindow)
     return;
-  }
   // The contentOffset of the scrollview MUST be greater than the contentInset before calling
   // endRefreshing otherwise the next pull to refresh will not work properly.
   UIScrollView *scrollView = self.scrollView;
@@ -226,5 +223,4 @@
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
+#endif // [macOS]
