--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTAttributedTextUtils.mm	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTAttributedTextUtils.mm	2026-03-06 15:22:53
@@ -14,7 +14,6 @@
 #include <react/renderer/textlayoutmanager/RCTFontUtils.h>
 #include <react/renderer/textlayoutmanager/RCTTextPrimitivesConversions.h>
 #include <react/utils/ManagedObjectWrapper.h>
-#include <array>
 
 using namespace facebook::react;
 
@@ -23,19 +22,21 @@
   assert(fontWeight > 50);
   assert(fontWeight < 950);
 
-  static auto weights = std::to_array<UIFontWeight>({/* ~100 */ UIFontWeightUltraLight,
-                                                     /* ~200 */ UIFontWeightThin,
-                                                     /* ~300 */ UIFontWeightLight,
-                                                     /* ~400 */ UIFontWeightRegular,
-                                                     /* ~500 */ UIFontWeightMedium,
-                                                     /* ~600 */ UIFontWeightSemibold,
-                                                     /* ~700 */ UIFontWeightBold,
-                                                     /* ~800 */ UIFontWeightHeavy,
-                                                     /* ~900 */ UIFontWeightBlack});
+  static UIFontWeight weights[] = {
+      /* ~100 */ UIFontWeightUltraLight,
+      /* ~200 */ UIFontWeightThin,
+      /* ~300 */ UIFontWeightLight,
+      /* ~400 */ UIFontWeightRegular,
+      /* ~500 */ UIFontWeightMedium,
+      /* ~600 */ UIFontWeightSemibold,
+      /* ~700 */ UIFontWeightBold,
+      /* ~800 */ UIFontWeightHeavy,
+      /* ~900 */ UIFontWeightBlack};
   // The expression is designed to convert something like 760 or 830 to 7.
   return weights[(fontWeight + 50) / 100 - 1];
 }
 
+#if !TARGET_OS_OSX // [macOS]
 inline static UIFontTextStyle RCTUIFontTextStyleForDynamicTypeRamp(const DynamicTypeRamp &dynamicTypeRamp)
 {
   switch (dynamicTypeRamp) {
@@ -63,6 +64,7 @@
       return UIFontTextStyleLargeTitle;
   }
 }
+#endif // [macOS]
 
 inline static CGFloat RCTBaseSizeForDynamicTypeRamp(const DynamicTypeRamp &dynamicTypeRamp)
 {
@@ -97,6 +99,7 @@
 inline static CGFloat RCTEffectiveFontSizeMultiplierFromTextAttributes(const TextAttributes &textAttributes)
 {
   if (textAttributes.allowFontScaling.value_or(true)) {
+#if !TARGET_OS_OSX // [macOS]
     CGFloat fontSizeMultiplier = !isnan(textAttributes.fontSizeMultiplier) ? textAttributes.fontSizeMultiplier : 1.0;
     if (textAttributes.dynamicTypeRamp.has_value()) {
       DynamicTypeRamp dynamicTypeRamp = textAttributes.dynamicTypeRamp.value();
@@ -110,6 +113,9 @@
     CGFloat maxFontSizeMultiplier =
         !isnan(textAttributes.maxFontSizeMultiplier) ? textAttributes.maxFontSizeMultiplier : 0.0;
     return maxFontSizeMultiplier >= 1.0 ? fminf(maxFontSizeMultiplier, fontSizeMultiplier) : fontSizeMultiplier;
+#else // [macOS
+      return textAttributes.fontSizeMultiplier;
+#endif // macOS]
   } else {
     return 1.0;
   }
@@ -136,9 +142,9 @@
   return RCTFontWithFontProperties(fontProperties);
 }
 
-inline static UIColor *RCTEffectiveForegroundColorFromTextAttributes(const TextAttributes &textAttributes)
+inline static RCTUIColor *RCTEffectiveForegroundColorFromTextAttributes(const TextAttributes &textAttributes) // [macOS]
 {
-  UIColor *effectiveForegroundColor = RCTUIColorFromSharedColor(textAttributes.foregroundColor) ?: [UIColor blackColor];
+  RCTUIColor *effectiveForegroundColor = RCTUIColorFromSharedColor(textAttributes.foregroundColor) ? RCTUIColorFromSharedColor(textAttributes.foregroundColor) : [RCTUIColor blackColor]; // [macOS]
 
   if (!isnan(textAttributes.opacity)) {
     effectiveForegroundColor = [effectiveForegroundColor
@@ -148,16 +154,16 @@
   return effectiveForegroundColor;
 }
 
-inline static UIColor *RCTEffectiveBackgroundColorFromTextAttributes(const TextAttributes &textAttributes)
+inline static RCTUIColor *RCTEffectiveBackgroundColorFromTextAttributes(const TextAttributes &textAttributes) // [macOS]
 {
-  UIColor *effectiveBackgroundColor = RCTUIColorFromSharedColor(textAttributes.backgroundColor);
+  RCTUIColor *effectiveBackgroundColor = RCTUIColorFromSharedColor(textAttributes.backgroundColor); // [macOS]
 
   if (effectiveBackgroundColor && !isnan(textAttributes.opacity)) {
     effectiveBackgroundColor = [effectiveBackgroundColor
         colorWithAlphaComponent:CGColorGetAlpha(effectiveBackgroundColor.CGColor) * textAttributes.opacity];
   }
 
-  return effectiveBackgroundColor ?: [UIColor clearColor];
+  return effectiveBackgroundColor ? effectiveBackgroundColor : [RCTUIColor clearColor]; // [macOS]
 }
 
 NSMutableDictionary<NSAttributedStringKey, id> *RCTNSTextAttributesFromTextAttributes(
@@ -172,7 +178,7 @@
   }
 
   // Colors
-  UIColor *effectiveForegroundColor = RCTEffectiveForegroundColorFromTextAttributes(textAttributes);
+  RCTUIColor *effectiveForegroundColor = RCTEffectiveForegroundColorFromTextAttributes(textAttributes); // [macOS]
 
   if (textAttributes.foregroundColor || !isnan(textAttributes.opacity)) {
     attributes[NSForegroundColorAttributeName] = effectiveForegroundColor;
@@ -239,7 +245,7 @@
     NSUnderlineStyle style = RCTNSUnderlineStyleFromTextDecorationStyle(
         textAttributes.textDecorationStyle.value_or(TextDecorationStyle::Solid));
 
-    UIColor *textDecorationColor = RCTUIColorFromSharedColor(textAttributes.textDecorationColor);
+    RCTUIColor *textDecorationColor = RCTUIColorFromSharedColor(textAttributes.textDecorationColor); // [macOS]
 
     // Underline
     if (textDecorationLineType == TextDecorationLineType::Underline ||
@@ -322,7 +328,7 @@
                               return;
                             }
 
-                            maximumFontLineHeight = MAX(font.lineHeight, maximumFontLineHeight);
+                            maximumFontLineHeight = MAX(UIFontLineHeight(font), maximumFontLineHeight); // [macOS]
                           }];
 
   if (maximumLineHeight < maximumFontLineHeight) {
@@ -354,7 +360,7 @@
 
 static NSMutableAttributedString *RCTNSAttributedStringFragmentFromFragment(
     const AttributedString::Fragment &fragment,
-    UIImage *placeholderImage)
+    RCTPlatformImage *placeholderImage) // [macOS]
 {
   if (fragment.isAttachment()) {
     auto layoutMetrics = fragment.parentShadowView.layoutMetrics;
@@ -383,7 +389,7 @@
 
 static NSMutableAttributedString *RCTNSAttributedStringFragmentWithAttributesFromFragment(
     const AttributedString::Fragment &fragment,
-    UIImage *placeholderImage)
+    RCTPlatformImage *placeholderImage) // [macOS]
 {
   auto nsAttributedStringFragment = RCTNSAttributedStringFragmentFromFragment(fragment, placeholderImage);
 
@@ -402,10 +408,10 @@
 
 NSAttributedString *RCTNSAttributedStringFromAttributedString(const AttributedString &attributedString)
 {
-  static UIImage *placeholderImage;
+  static RCTPlatformImage *placeholderImage; // [macOS]
   static dispatch_once_t onceToken;
   dispatch_once(&onceToken, ^{
-    placeholderImage = [UIImage new];
+    placeholderImage = [RCTPlatformImage new]; // [macOS]
   });
 
   NSMutableAttributedString *nsAttributedString = [NSMutableAttributedString new];
