--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTFontUtils.mm	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTFontUtils.mm	2026-03-06 15:22:53
@@ -6,9 +6,7 @@
  */
 
 #import "RCTFontUtils.h"
-
 #import <CoreText/CoreText.h>
-#import <React/RCTFont+Private.h>
 #import <React/RCTFont.h>
 
 #import <algorithm>
@@ -38,7 +36,7 @@
     RCTFontProperties fontProperties,
     RCTFontProperties baseFontProperties)
 {
-  fontProperties.family = (fontProperties.family.length != 0u) ? fontProperties.family : baseFontProperties.family;
+  fontProperties.family = fontProperties.family.length ? fontProperties.family : baseFontProperties.family;
   fontProperties.size = !isnan(fontProperties.size) ? fontProperties.size : baseFontProperties.size;
   fontProperties.weight = !isnan(fontProperties.weight) ? fontProperties.weight : baseFontProperties.weight;
   fontProperties.style =
@@ -52,7 +50,7 @@
 {
   NSDictionary *traits = [font.fontDescriptor objectForKey:UIFontDescriptorTraitsAttribute];
   UIFontDescriptorSymbolicTraits symbolicTraits = [traits[UIFontSymbolicTrait] unsignedIntValue];
-  if ((symbolicTraits & UIFontDescriptorTraitItalic) != 0u) {
+  if (symbolicTraits & UIFontDescriptorTraitItalic) {
     return RCTFontStyleItalic;
   }
 
@@ -169,141 +167,112 @@
     };
   });
 
-  if ((fontVariant & RCTFontVariantSmallCaps) != 0) {
+  if (fontVariant & RCTFontVariantSmallCaps) {
     [fontFeatures addObject:mapping[RCTFontVariantSmallCaps]];
   }
-  if ((fontVariant & RCTFontVariantOldstyleNums) != 0) {
+  if (fontVariant & RCTFontVariantOldstyleNums) {
     [fontFeatures addObject:mapping[RCTFontVariantOldstyleNums]];
   }
-  if ((fontVariant & RCTFontVariantLiningNums) != 0) {
+  if (fontVariant & RCTFontVariantLiningNums) {
     [fontFeatures addObject:mapping[RCTFontVariantLiningNums]];
   }
-  if ((fontVariant & RCTFontVariantTabularNums) != 0) {
+  if (fontVariant & RCTFontVariantTabularNums) {
     [fontFeatures addObject:mapping[RCTFontVariantTabularNums]];
   }
-  if ((fontVariant & RCTFontVariantProportionalNums) != 0) {
+  if (fontVariant & RCTFontVariantProportionalNums) {
     [fontFeatures addObject:mapping[RCTFontVariantProportionalNums]];
   }
-  if ((fontVariant & RCTFontVariantStylisticOne) != 0) {
+  if (fontVariant & RCTFontVariantStylisticOne) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticOne]];
   }
-  if ((fontVariant & RCTFontVariantStylisticTwo) != 0) {
+  if (fontVariant & RCTFontVariantStylisticTwo) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticTwo]];
   }
-  if ((fontVariant & RCTFontVariantStylisticThree) != 0) {
+  if (fontVariant & RCTFontVariantStylisticThree) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticThree]];
   }
-  if ((fontVariant & RCTFontVariantStylisticFour) != 0) {
+  if (fontVariant & RCTFontVariantStylisticFour) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticFour]];
   }
-  if ((fontVariant & RCTFontVariantStylisticFive) != 0) {
+  if (fontVariant & RCTFontVariantStylisticFive) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticFive]];
   }
-  if ((fontVariant & RCTFontVariantStylisticSix) != 0) {
+  if (fontVariant & RCTFontVariantStylisticSix) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticSix]];
   }
-  if ((fontVariant & RCTFontVariantStylisticSeven) != 0) {
+  if (fontVariant & RCTFontVariantStylisticSeven) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticSeven]];
   }
-  if ((fontVariant & RCTFontVariantStylisticEight) != 0) {
+  if (fontVariant & RCTFontVariantStylisticEight) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticEight]];
   }
-  if ((fontVariant & RCTFontVariantStylisticNine) != 0) {
+  if (fontVariant & RCTFontVariantStylisticNine) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticNine]];
   }
-  if ((fontVariant & RCTFontVariantStylisticTen) != 0) {
+  if (fontVariant & RCTFontVariantStylisticTen) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticTen]];
   }
-  if ((fontVariant & RCTFontVariantStylisticEleven) != 0) {
+  if (fontVariant & RCTFontVariantStylisticEleven) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticEleven]];
   }
-  if ((fontVariant & RCTFontVariantStylisticTwelve) != 0) {
+  if (fontVariant & RCTFontVariantStylisticTwelve) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticTwelve]];
   }
-  if ((fontVariant & RCTFontVariantStylisticThirteen) != 0) {
+  if (fontVariant & RCTFontVariantStylisticThirteen) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticThirteen]];
   }
-  if ((fontVariant & RCTFontVariantStylisticFourteen) != 0) {
+  if (fontVariant & RCTFontVariantStylisticFourteen) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticFourteen]];
   }
-  if ((fontVariant & RCTFontVariantStylisticFifteen) != 0) {
+  if (fontVariant & RCTFontVariantStylisticFifteen) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticFifteen]];
   }
-  if ((fontVariant & RCTFontVariantStylisticSixteen) != 0) {
+  if (fontVariant & RCTFontVariantStylisticSixteen) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticSixteen]];
   }
-  if ((fontVariant & RCTFontVariantStylisticSeventeen) != 0) {
+  if (fontVariant & RCTFontVariantStylisticSeventeen) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticSeventeen]];
   }
-  if ((fontVariant & RCTFontVariantStylisticEighteen) != 0) {
+  if (fontVariant & RCTFontVariantStylisticEighteen) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticEighteen]];
   }
-  if ((fontVariant & RCTFontVariantStylisticNineteen) != 0) {
+  if (fontVariant & RCTFontVariantStylisticNineteen) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticNineteen]];
   }
-  if ((fontVariant & RCTFontVariantStylisticTwenty) != 0) {
+  if (fontVariant & RCTFontVariantStylisticTwenty) {
     [fontFeatures addObject:mapping[RCTFontVariantStylisticTwenty]];
   }
 
   return fontFeatures;
 }
 
-static RCTDefaultFontResolver defaultFontResolver;
-
-void RCTSetDefaultFontResolver(RCTDefaultFontResolver handler)
+static UIFont *RCTDefaultFontWithFontProperties(RCTFontProperties fontProperties)
 {
-  defaultFontResolver = handler;
-}
-
-static UIFont *RCTDefaultFontWithFontProperties(const RCTFontProperties &fontProperties)
-{
   static NSCache *fontCache;
   static std::mutex fontCacheMutex;
 
   CGFloat effectiveFontSize = fontProperties.sizeMultiplier * fontProperties.size;
-  NSString *cacheKey = [NSString stringWithFormat:@"%@/%.1f/%.2f/%ld",
-                                                  fontProperties.family,
-                                                  effectiveFontSize,
-                                                  fontProperties.weight,
-                                                  (long)fontProperties.style];
+  NSString *cacheKey = [NSString
+      stringWithFormat:@"%.1f/%.2f/%ld", effectiveFontSize, fontProperties.weight, (long)fontProperties.style];
   UIFont *font;
 
   {
     std::lock_guard<std::mutex> lock(fontCacheMutex);
-    if (fontCache == nil) {
+    if (!fontCache) {
       fontCache = [NSCache new];
     }
     font = [fontCache objectForKey:cacheKey];
   }
 
-  if (font == nil) {
-    if (defaultFontResolver != nil) {
-      font = defaultFontResolver(fontProperties);
-    }
+  if (!font) {
+    font = [UIFont systemFontOfSize:effectiveFontSize weight:fontProperties.weight];
 
-    if (font == nil) {
-      font = RCTGetLegacyDefaultFont(effectiveFontSize, fontProperties.weight);
-    }
-
-    if (font == nil) {
-      font = [UIFont systemFontOfSize:effectiveFontSize weight:fontProperties.weight];
-    }
-
-    BOOL isItalicFont = fontProperties.style == RCTFontStyleItalic;
-    BOOL isCondensedFont = [fontProperties.family isEqualToString:@"SystemCondensed"];
-
-    if (isItalicFont || isCondensedFont) {
+    if (fontProperties.style == RCTFontStyleItalic) {
       UIFontDescriptor *fontDescriptor = [font fontDescriptor];
       UIFontDescriptorSymbolicTraits symbolicTraits = fontDescriptor.symbolicTraits;
 
-      if (isItalicFont) {
-        symbolicTraits |= UIFontDescriptorTraitItalic;
-      }
+      symbolicTraits |= UIFontDescriptorTraitItalic;
 
-      if (isCondensedFont) {
-        symbolicTraits |= UIFontDescriptorTraitCondensed;
-      }
-
       fontDescriptor = [fontDescriptor fontDescriptorWithSymbolicTraits:symbolicTraits];
       font = [UIFont fontWithDescriptor:fontDescriptor size:effectiveFontSize];
     }
@@ -330,6 +299,22 @@
   return systemDesigns[family];
 }
 
+#if TARGET_OS_OSX // [macOS
+NSArray<NSString *> *RCTFontNamesForFamilyName(NSString *familyName) {
+  NSArray<NSArray *> *fontMembers = [[NSFontManager sharedFontManager] availableMembersOfFontFamily:familyName];
+  NSMutableArray<NSString *> *fontNames = [NSMutableArray array];
+
+  for (NSArray *fontMember in fontMembers) {
+      // The first object is the Post-script name of the font.
+      NSString *fontName = fontMember.firstObject;
+      [fontNames addObject:fontName];
+  }
+
+  // Return an immutable array
+  return [NSArray arrayWithArray:fontNames];
+}
+#endif // macOS]
+
 UIFont *RCTFontWithFontProperties(RCTFontProperties fontProperties)
 {
   RCTFontProperties defaultFontProperties = RCTDefaultFontProperties();
@@ -339,7 +324,7 @@
   CGFloat effectiveFontSize = fontProperties.sizeMultiplier * fontProperties.size;
   UIFont *font;
   UIFontDescriptorSystemDesign design = RCTGetFontDescriptorSystemDesign([fontProperties.family lowercaseString]);
-  if (design != nullptr) {
+  if (design) {
     // Create a system font which `-fontDescriptorWithDesign:` asks for
     // (see:
     // https://developer.apple.com/documentation/uikit/uifontdescriptor/3151797-fontdescriptorwithdesign?language=objc)
@@ -352,19 +337,27 @@
     // the specific metrics of the standard system font as closely as possible.
     font = RCTDefaultFontWithFontProperties(fontProperties);
   } else {
+#if !TARGET_OS_OSX // [macOS]
     NSArray<NSString *> *fontNames = [UIFont fontNamesForFamilyName:fontProperties.family];
+#else // [macOS
+    NSArray<NSString *> *fontNames = RCTFontNamesForFamilyName(fontProperties.family);
+#endif // macOS]
     UIFontWeight fontWeight = fontProperties.weight;
 
     if (fontNames.count == 0) {
       // Gracefully handle being given a font name rather than font family, for
       // example: "Helvetica Light Oblique" rather than just "Helvetica".
       font = [UIFont fontWithName:fontProperties.family size:effectiveFontSize];
-      if (font != nullptr) {
+      if (font) {
+#if !TARGET_OS_OSX // [macOS]
         fontNames = [UIFont fontNamesForFamilyName:font.familyName];
-        fontWeight = (fontWeight != 0.0) ?: RCTGetFontWeight(font);
+#else // [macOS
+        fontNames = RCTFontNamesForFamilyName(font.familyName);
+#endif // macOS]
+        fontWeight = fontWeight ?: RCTGetFontWeight(font);
       } else {
         // Failback to system font.
-        font = RCTDefaultFontWithFontProperties(fontProperties);
+        font = [UIFont systemFontOfSize:effectiveFontSize weight:fontProperties.weight];
       }
     }
 
@@ -385,7 +378,7 @@
         }
       }
 
-      if (font == nil) {
+      if (!font) {
         // If we still don't have a match at least return the first font in the
         // fontFamily This is to support built-in font Zapfino and other custom
         // single font families like Impact
