--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/RawText/RCTRawTextViewManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/RawText/RCTRawTextViewManager.mm	2026-03-06 15:22:52
@@ -7,17 +7,15 @@
 
 #import <React/RCTRawTextViewManager.h>
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTRawTextShadowView.h>
 
 @implementation RCTRawTextViewManager
 
 RCT_EXPORT_MODULE(RCTRawText)
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
-  return [UIView new];
+  return [RCTUIView new]; // [macOS]
 }
 
 - (RCTShadowView *)shadowView
@@ -28,5 +26,3 @@
 RCT_EXPORT_SHADOW_PROPERTY(text, NSString)
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
