--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.cpp	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.cpp	2026-03-06 15:22:53
@@ -5,6 +5,8 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#include <react/renderer/components/view/HostPlatformViewEventEmitter.h>
+
 #include "TextInputEventEmitter.h"
 
 namespace facebook::react {
@@ -131,6 +133,19 @@
   return payload;
 };
 
+#if TARGET_OS_OSX // [macOS
+static jsi::Value pasteMetricsPayload(
+    jsi::Runtime& runtime,
+    const TextInputEventEmitter::PasteMetrics& pasteMetrics) {
+  auto payload = jsi::Object(runtime);
+  auto dataTransferObject = HostPlatformViewEventEmitter::dataTransferPayload(
+    runtime,
+    pasteMetrics.dataTransfer);
+  payload.setProperty(runtime, "dataTransfer", dataTransferObject);
+  return payload;
+};
+#endif // macOS]
+
 void TextInputEventEmitter::onFocus(const Metrics& textInputMetrics) const {
   dispatchTextInputEvent("focus", textInputMetrics);
 }
@@ -177,6 +192,30 @@
   });
 }
 
+#if TARGET_OS_OSX // [macOS
+void TextInputEventEmitter::onAutoCorrectChange(
+    const Metrics& textInputMetrics) const {
+  dispatchTextInputEvent("autoCorrectChange", textInputMetrics);
+}
+
+void TextInputEventEmitter::onSpellCheckChange(
+    const Metrics& textInputMetrics) const {
+  dispatchTextInputEvent("spellCheckChange", textInputMetrics);
+}
+
+void TextInputEventEmitter::onGrammarCheckChange(
+    const Metrics& textInputMetrics) const {
+  dispatchTextInputEvent("grammarCheckChange", textInputMetrics);
+}
+
+void TextInputEventEmitter::onPaste(
+    const PasteMetrics& pasteMetrics) const {
+  dispatchEvent("paste", [pasteMetrics](jsi::Runtime& runtime) {
+    return pasteMetricsPayload(runtime, pasteMetrics);
+  });
+}
+#endif // macOS]
+
 void TextInputEventEmitter::dispatchTextInputEvent(
     const std::string& name,
     const Metrics& textInputMetrics,
