--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/components/textinput/TextInputEventEmitter.h	2026-03-06 15:22:53
@@ -9,6 +9,9 @@
 
 #include <react/renderer/attributedstring/AttributedString.h>
 #include <react/renderer/components/view/ViewEventEmitter.h>
+#if TARGET_OS_OSX // [macOS
+#include <react/renderer/components/view/MouseEvent.h>
+#endif // macOS]
 
 namespace facebook::react {
 
@@ -28,30 +31,49 @@
     Size layoutMeasurement;
     Float zoomScale;
     Tag target;
+ #if TARGET_OS_OSX // [macOS
+    bool autoCorrectEnabled;
+    bool spellCheckEnabled;
+    bool grammarCheckEnabled;
+#endif // macOS]
   };
 
   struct KeyPressMetrics {
     std::string text;
     int eventCount;
   };
+  
+#if TARGET_OS_OSX // [macOS
+  struct PasteMetrics {
+    DataTransfer dataTransfer;
+  };
+#endif // macOS]
 
-  void onFocus(const Metrics &textInputMetrics) const;
-  void onBlur(const Metrics &textInputMetrics) const;
-  void onChange(const Metrics &textInputMetrics) const;
-  void onContentSizeChange(const Metrics &textInputMetrics) const;
-  void onSelectionChange(const Metrics &textInputMetrics) const;
-  void onEndEditing(const Metrics &textInputMetrics) const;
-  void onSubmitEditing(const Metrics &textInputMetrics) const;
-  void onKeyPress(const KeyPressMetrics &keyPressMetrics) const;
-  void onScroll(const Metrics &textInputMetrics) const;
+  void onFocus(const Metrics& textInputMetrics) const;
+  void onBlur(const Metrics& textInputMetrics) const;
+  void onChange(const Metrics& textInputMetrics) const;
+  void onContentSizeChange(const Metrics& textInputMetrics) const;
+  void onSelectionChange(const Metrics& textInputMetrics) const;
+  void onEndEditing(const Metrics& textInputMetrics) const;
+  void onSubmitEditing(const Metrics& textInputMetrics) const;
+  void onKeyPress(const KeyPressMetrics& keyPressMetrics) const;
+  void onScroll(const Metrics& textInputMetrics) const;
+ #if TARGET_OS_OSX // [macOS
+  void onAutoCorrectChange(const Metrics& textInputMetrics) const;
+  void onSpellCheckChange(const Metrics& textInputMetrics) const;
+  void onGrammarCheckChange(const Metrics& textInputMetrics) const;
+  void onPaste(const PasteMetrics& pasteMetrics) const;
+ #endif // macOS]
 
  private:
   void dispatchTextInputEvent(
-      const std::string &name,
-      const Metrics &textInputMetrics,
+      const std::string& name,
+      const Metrics& textInputMetrics,
       bool includeSelectionState = false) const;
 
-  void dispatchTextInputContentSizeChangeEvent(const std::string &name, const Metrics &textInputMetrics) const;
+  void dispatchTextInputContentSizeChangeEvent(
+      const std::string& name,
+      const Metrics& textInputMetrics) const;
 };
 
 } // namespace facebook::react
