--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/RCTKeyCommands.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.knVavN0yBD	2026-03-06 18:50:47
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#if !TARGET_OS_OSX // [macOS]
 
 @interface RCTKeyCommands : NSObject
 
@@ -29,3 +29,4 @@
 - (BOOL)isKeyCommandRegisteredForInput:(NSString *)input modifierFlags:(UIKeyModifierFlags)flags;
 
 @end
+#endif // [macOS]
