--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Wrapper/RCTWrapperView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Wrapper/RCTWrapperView.h	2026-03-06 15:22:52
@@ -5,21 +5,17 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
+typedef CGSize (^RCTWrapperMeasureBlock)(CGSize minimumSize, CGSize maximumSize);
 
-typedef CGSize (^RCTWrapperMeasureBlock)(CGSize minimumSize, CGSize maximumSize)
-    __attribute__((deprecated("This API will be removed along with the legacy architecture.")));
-
 @class RCTBridge;
 
 NS_ASSUME_NONNULL_BEGIN
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTWrapperView : UIView
+@interface RCTWrapperView : RCTPlatformView // [macOS]
 
-@property (nonatomic, retain, nullable) UIView *contentView;
+@property (nonatomic, retain, nullable) RCTPlatformView *contentView; // [macOS]
 @property (nonatomic, readonly) RCTWrapperMeasureBlock measureBlock;
 
 - (instancetype)initWithBridge:(RCTBridge *)bridge NS_DESIGNATED_INITIALIZER;
@@ -30,13 +26,11 @@
 - (instancetype)initWithFrame:(CGRect)frame NS_UNAVAILABLE;
 - (instancetype)initWithCoder:(NSCoder *)decoder NS_UNAVAILABLE;
 
-- (void)addSubview:(UIView *)view NS_UNAVAILABLE;
-- (void)insertSubview:(UIView *)view atIndex:(NSInteger)index NS_UNAVAILABLE;
-- (void)insertSubview:(UIView *)view aboveSubview:(UIView *)siblingSubview NS_UNAVAILABLE;
-- (void)insertSubview:(UIView *)view belowSubview:(UIView *)siblingSubview NS_UNAVAILABLE;
+- (void)addSubview:(RCTPlatformView *)view NS_UNAVAILABLE; // [macOS]
+- (void)insertSubview:(RCTPlatformView *)view atIndex:(NSInteger)index NS_UNAVAILABLE; // [macOS]
+- (void)insertSubview:(RCTPlatformView *)view aboveSubview:(RCTPlatformView *)siblingSubview NS_UNAVAILABLE; // [macOS]
+- (void)insertSubview:(RCTPlatformView *)view belowSubview:(RCTPlatformView *)siblingSubview NS_UNAVAILABLE; // [macOS]
 
 @end
 
 NS_ASSUME_NONNULL_END
-
-#endif // RCT_REMOVE_LEGACY_ARCH
