--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Wrapper/RCTWrapper.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Wrapper/RCTWrapper.h	2026-03-06 15:22:52
@@ -5,10 +5,8 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <RCTWrapper/RCTWrapperView.h>
 #import <RCTWrapper/RCTWrapperViewControllerHostingView.h>
 #import <RCTWrapper/RCTWrapperViewManager.h>
@@ -16,29 +14,29 @@
 // Umbrella header with macros
 
 // RCT_WRAPPER_FOR_VIEW
-#define RCT_WRAPPER_FOR_VIEW(ClassName)                                                       \
-  NS_ASSUME_NONNULL_BEGIN                                                                     \
-                                                                                              \
-  __attribute__((deprecated("This API will be removed along with the legacy architecture."))) \
-  @interface ClassName                                                                        \
-  ##Manager : RCTWrapperViewManager                                                           \
-                                                                                              \
-              @end                                                                            \
-                                                                                              \
-  NS_ASSUME_NONNULL_END                                                                       \
-                                                                                              \
-  @implementation ClassName                                                                   \
-  ##Manager                                                                                   \
-                                                                                              \
-      RCT_EXPORT_MODULE()                                                                     \
-                                                                                              \
-      - (UIView *)view                                                                        \
-  {                                                                                           \
-    RCTWrapperView *wrapperView = [super view];                                               \
-    wrapperView.contentView = [ClassName new];                                                \
-    return wrapperView;                                                                       \
-  }                                                                                           \
-                                                                                              \
+#define RCT_WRAPPER_FOR_VIEW(ClassName)         \
+                                                \
+  NS_ASSUME_NONNULL_BEGIN                       \
+                                                \
+  @interface ClassName                          \
+  ##Manager : RCTWrapperViewManager             \
+                                                \
+              @end                              \
+                                                \
+  NS_ASSUME_NONNULL_END                         \
+                                                \
+  @implementation ClassName                     \
+  ##Manager                                     \
+                                                \
+      RCT_EXPORT_MODULE()                       \
+                                                \
+      - (RCTPlatformView *)view /* [macOS]*/    \
+  {                                             \
+    RCTWrapperView *wrapperView = [super view]; \
+    wrapperView.contentView = [ClassName new];  \
+    return wrapperView;                         \
+  }                                             \
+                                                \
   @end
 
 // RCT_WRAPPER_FOR_VIEW_CONTROLLER
@@ -46,7 +44,6 @@
                                                                                                                        \
   NS_ASSUME_NONNULL_BEGIN                                                                                              \
                                                                                                                        \
-  __attribute__((deprecated("This API will be removed along with the legacy architecture.")))                          \
   @interface ClassName                                                                                                 \
   ##Manager : RCTWrapperViewManager                                                                                    \
                                                                                                                        \
@@ -59,7 +56,7 @@
                                                                                                                        \
       RCT_EXPORT_MODULE()                                                                                              \
                                                                                                                        \
-      - (UIView *)view                                                                                                 \
+      - (RCTPlatformView *)view /* [macOS] */                                                                          \
   {                                                                                                                    \
     RCTWrapperViewControllerHostingView *contentViewControllerHostingView = [RCTWrapperViewControllerHostingView new]; \
     contentViewControllerHostingView.contentViewController = [[ClassName alloc] initWithNibName:nil bundle:nil];       \
@@ -69,5 +66,3 @@
   }                                                                                                                    \
                                                                                                                        \
   @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
