--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Wrapper/RCTWrapperViewControllerHostingView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Wrapper/RCTWrapperViewControllerHostingView.h	2026-03-06 15:22:52
@@ -5,26 +5,25 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 NS_ASSUME_NONNULL_BEGIN
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTWrapperViewControllerHostingView : UIView
+@interface RCTWrapperViewControllerHostingView : RCTPlatformView // [macOS]
 
+#if !TARGET_OS_OSX // [macOS]
 @property (nonatomic, retain, nullable) UIViewController *contentViewController;
+#else
+@property (nonatomic, retain, nullable) NSViewController *contentViewController;
+#endif // [macOS]
 
 #pragma mark - Restrictions
 
-- (void)addSubview:(UIView *)view NS_UNAVAILABLE;
-- (void)insertSubview:(UIView *)view atIndex:(NSInteger)index NS_UNAVAILABLE;
-- (void)insertSubview:(UIView *)view aboveSubview:(UIView *)siblingSubview NS_UNAVAILABLE;
-- (void)insertSubview:(UIView *)view belowSubview:(UIView *)siblingSubview NS_UNAVAILABLE;
+- (void)addSubview:(RCTPlatformView *)view NS_UNAVAILABLE; // [macOS]
+- (void)insertSubview:(RCTPlatformView *)view atIndex:(NSInteger)index NS_UNAVAILABLE; // [macOS]
+- (void)insertSubview:(RCTPlatformView *)view aboveSubview:(UIView *)siblingSubview NS_UNAVAILABLE; // [macOS]
+- (void)insertSubview:(RCTPlatformView *)view belowSubview:(RCTPlatformView *)siblingSubview NS_UNAVAILABLE; // [macOS]
 
 @end
 
 NS_ASSUME_NONNULL_END
-
-#endif // RCT_REMOVE_LEGACY_ARCH
