--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/TextInput/RCTSingelineTextInputNativeComponent.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Components/TextInput/RCTSingelineTextInputNativeComponent.js	2026-03-06 15:22:52
@@ -21,7 +21,7 @@
 type NativeCommands = TextInputNativeCommands<NativeType>;
 
 export const Commands: NativeCommands = codegenNativeCommands<NativeCommands>({
-  supportedCommands: ['focus', 'blur', 'setTextAndSelection'],
+  supportedCommands: ['focus', 'blur', 'setTextAndSelection', 'setGhostText'], // [macOS]
 });
 
 export const __INTERNAL_VIEW_CONFIG: PartialViewConfig = {
