--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/TextInput/TextInput.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Components/TextInput/TextInput.js	2026-03-06 15:22:52
@@ -16,6 +16,9 @@
   GestureResponderEvent,
   ScrollEvent,
 } from '../../Types/CoreEventTypes';
+// [macOS
+import type {HandledKeyEvent, KeyEvent} from '../../Types/CoreEventTypes';
+// macOS]
 import type {
   AutoCapitalize,
   EnterKeyHintType,
@@ -40,6 +43,7 @@
   TextInputContentSizeChangeEvent,
   TextInputEditingEvent,
   TextInputEndEditingEvent,
+  // [macOS]
   TextInputEvent,
   TextInputFocusEvent,
   TextInputInstance,
@@ -75,7 +79,7 @@
   AndroidTextInput = require('./AndroidTextInputNativeComponent').default;
   AndroidTextInputCommands =
     require('./AndroidTextInputNativeComponent').Commands;
-} else if (Platform.OS === 'ios') {
+} else if (Platform.OS === 'ios' || Platform.OS === 'macos' /* [macOS] */) {
   RCTSinglelineTextInputView =
     require('./RCTSingelineTextInputNativeComponent').default;
   RCTSinglelineTextInputNativeCommands =
@@ -147,9 +151,23 @@
    * noop if it wasn't focused
    */
   blurTextInput: (textField: ?HostInstance) => void,
+
+  // [macOS
+  /**
+   * @param textField ref of the text field to focus
+   * Called when the text field is focused
+   */
+  onTextInputFocus: (textField: ?HostInstance) => void, // [macOS]
+
+  /**
+   * @param textField ref of the text field to focus
+   * Called when the text field is blurred
+   */
+  onTextInputBlur: (textField: ?HostInstance) => void, // [macOS]
+  // macOS]
 }>;
 
-type ViewCommands = NonNullable<
+type ViewCommands = $NonMaybeType<
   | typeof AndroidTextInputCommands
   | typeof RCTMultilineTextInputNativeCommands
   | typeof RCTSinglelineTextInputNativeCommands,
@@ -188,8 +206,8 @@
   const [lastNativeText, setLastNativeText] = useState<?Stringish>(props.value);
   const [lastNativeSelectionState, setLastNativeSelection] =
     useState<LastNativeSelection>({
-      mostRecentEventCount,
-      selection: {end: -1, start: -1},
+      selection: {start: -1, end: -1},
+      mostRecentEventCount: mostRecentEventCount,
     });
 
   const lastNativeSelection = lastNativeSelectionState.selection;
@@ -212,7 +230,7 @@
         lastNativeSelection.end !== selection.end)
     ) {
       nativeUpdate.selection = selection;
-      setLastNativeSelection({mostRecentEventCount, selection});
+      setLastNativeSelection({selection, mostRecentEventCount});
     }
 
     if (Object.keys(nativeUpdate).length === 0) {
@@ -240,7 +258,7 @@
     viewCommands,
   ]);
 
-  return {setLastNativeSelection, setLastNativeText};
+  return {setLastNativeText, setLastNativeSelection};
 }
 
 /**
@@ -352,6 +370,19 @@
  * while keyboard is active. To avoid this behavior either specify `windowSoftInputMode`
  * in AndroidManifest.xml ( https://developer.android.com/guide/topics/manifest/activity-element.html )
  * or control this param programmatically with native code.
+ *
+ *
+ * The following values work on macOS only:
+ *
+ * - `'ortography'`
+ * - `'spelling'`
+ * - `'grammar'`
+ * - `'quote'`
+ * - `'dash'`
+ * - `'replacement'`
+ * - `'correction'`
+ * - `'regularExpression'`
+ * - `'transitInformation'`
  *
  */
 function InternalTextInput(props: TextInputProps): React.Node {
@@ -377,8 +408,8 @@
     propsSelection == null
       ? null
       : {
-          end: propsSelection.end ?? propsSelection.start,
           start: propsSelection.start,
+          end: propsSelection.end ?? propsSelection.start,
         };
 
   const text =
@@ -397,9 +428,9 @@
   const [mostRecentEventCount, setMostRecentEventCount] = useState<number>(0);
   const {setLastNativeText, setLastNativeSelection} =
     useTextInputStateSynchronization({
+      props,
       inputRef,
       mostRecentEventCount,
-      props,
       selection,
       text,
       viewCommands,
@@ -449,12 +480,6 @@
         before we can get to the long term breaking change.
       */
       if (instance != null) {
-        // Register the input immediately when the ref is set so that focus()
-        // can be called from ref callbacks
-        // Double registering during useLayoutEffect is fine, because the underlying
-        // state is a Set.
-        TextInputState.registerInput(instance);
-
         // $FlowFixMe[prop-missing] - See the explanation above.
         // $FlowFixMe[unsafe-object-assign]
         Object.assign(instance, {
@@ -469,13 +494,13 @@
               );
             }
           },
-          getNativeRef(): ?TextInputInstance {
-            return inputRef.current;
-          },
           // TODO: Fix this returning true on null === null, when no input is focused
           isFocused(): boolean {
             return TextInputState.currentlyFocusedInput() === inputRef.current;
           },
+          getNativeRef(): ?TextInputInstance {
+            return inputRef.current;
+          },
           setSelection(start: number, end: number): void {
             if (inputRef.current != null) {
               viewCommands.setTextAndSelection(
@@ -487,13 +512,20 @@
               );
             }
           },
+          // [macOS
+          setGhostText(ghostText: ?string): void {
+            if (inputRef.current != null) {
+              viewCommands.setGhostText(inputRef.current, ghostText);
+            }
+          },
+          // macOS]
         });
       }
     },
     [mostRecentEventCount, viewCommands],
   );
 
-  // $FlowExpectedError[incompatible-type]
+  // $FlowExpectedError[incompatible-call]
   const ref = useMergeRefs<HostInstance>(setLocalRef, props.forwardedRef);
 
   const _onChange = (event: TextInputChangeEvent) => {
@@ -525,8 +557,8 @@
     }
 
     setLastNativeSelection({
-      mostRecentEventCount,
       selection: event.nativeEvent.selection,
+      mostRecentEventCount,
     });
   };
 
@@ -546,7 +578,51 @@
 
   const _onScroll = (event: ScrollEvent) => {
     props.onScroll && props.onScroll(event);
+  };
+
+  // [macOS
+  const _onKeyDown = (event: KeyEvent) => {
+    const keyDownEvents = props.keyDownEvents;
+    if (keyDownEvents != null && !event.isPropagationStopped()) {
+      const isHandled = keyDownEvents.some(
+        ({key, metaKey, ctrlKey, altKey, shiftKey}: HandledKeyEvent) => {
+          return (
+            event.nativeEvent.key === key &&
+            Boolean(metaKey) === event.nativeEvent.metaKey &&
+            Boolean(ctrlKey) === event.nativeEvent.ctrlKey &&
+            Boolean(altKey) === event.nativeEvent.altKey &&
+            Boolean(shiftKey) === event.nativeEvent.shiftKey
+          );
+        },
+      );
+      if (isHandled === true) {
+        event.stopPropagation();
+      }
+    }
+    props.onKeyDown?.(event);
+  };
+
+  const _onKeyUp = (event: KeyEvent) => {
+    const keyUpEvents = props.keyUpEvents;
+    if (keyUpEvents != null && !event.isPropagationStopped()) {
+      const isHandled = keyUpEvents.some(
+        ({key, metaKey, ctrlKey, altKey, shiftKey}: HandledKeyEvent) => {
+          return (
+            event.nativeEvent.key === key &&
+            Boolean(metaKey) === event.nativeEvent.metaKey &&
+            Boolean(ctrlKey) === event.nativeEvent.ctrlKey &&
+            Boolean(altKey) === event.nativeEvent.altKey &&
+            Boolean(shiftKey) === event.nativeEvent.shiftKey
+          );
+        },
+      );
+      if (isHandled === true) {
+        event.stopPropagation();
+      }
+    }
+    props.onKeyUp?.(event);
   };
+  // macOS]
 
   let textInput = null;
 
@@ -590,7 +666,6 @@
 
   const config = useMemo(
     () => ({
-      cancelable: Platform.OS === 'ios' ? !rejectResponderTermination : null,
       hitSlop,
       onPress: (event: GestureResponderEvent) => {
         onPress?.(event);
@@ -600,8 +675,13 @@
           }
         }
       },
-      onPressIn,
-      onPressOut,
+      onPressIn: onPressIn,
+      onPressOut: onPressOut,
+      // [macOS]
+      cancelable:
+        Platform.OS === 'ios' || Platform.OS === 'macos'
+          ? !rejectResponderTermination
+          : null,
     }),
     [
       editable,
@@ -624,9 +704,6 @@
   // so omitting onBlur and onFocus pressability handlers here.
   const {onBlur, onFocus, ...eventHandlers} = usePressability(config);
 
-  const _accessibilityLabel =
-    props?.['aria-label'] ?? props?.accessibilityLabel;
-
   let _accessibilityState;
   if (
     accessibilityState != null ||
@@ -653,7 +730,7 @@
     if (typeof flattenedStyle?.fontWeight === 'number') {
       overrides = overrides || ({}: {...TextStyleInternal});
       overrides.fontWeight =
-        // $FlowFixMe[incompatible-type]
+        // $FlowFixMe[incompatible-cast]
         (flattenedStyle.fontWeight.toString(): TextStyleInternal['fontWeight']);
     }
 
@@ -670,7 +747,7 @@
     }
   }
 
-  if (Platform.OS === 'ios') {
+  if (Platform.OS === 'ios' || Platform.OS === 'macos') {
     const RCTTextInputView =
       props.multiline === true
         ? RCTMultilineTextInputView
@@ -683,9 +760,6 @@
           flattenedStyle.paddingVertical == null &&
           flattenedStyle.paddingTop == null));
 
-    const _accessibilityElementsHidden =
-      props['aria-hidden'] ?? props.accessibilityElementsHidden;
-
     textInput = (
       <RCTTextInputView
         // Figure out imperative + forward refs.
@@ -693,9 +767,7 @@
         {...otherProps}
         {...eventHandlers}
         acceptDragAndDropTypes={props.experimental_acceptDragAndDropTypes}
-        accessibilityLabel={_accessibilityLabel}
         accessibilityState={_accessibilityState}
-        accessibilityElementsHidden={_accessibilityElementsHidden}
         accessible={accessible}
         submitBehavior={submitBehavior}
         caretHidden={caretHidden}
@@ -708,6 +780,9 @@
         onChange={_onChange}
         onContentSizeChange={props.onContentSizeChange}
         onFocus={_onFocus}
+        // $FlowFixMe[exponential-spread]
+        {...(otherProps.onKeyDown && {onKeyDown: _onKeyDown})} // [macOS]
+        {...(otherProps.onKeyUp && {onKeyUp: _onKeyUp})} // [macOS]
         onScroll={_onScroll}
         onSelectionChange={_onSelectionChange}
         onSelectionChangeShouldSetResponder={emptyFunctionThatReturnsTrue}
@@ -724,10 +799,6 @@
     const autoCapitalize = props.autoCapitalize || 'sentences';
     const _accessibilityLabelledBy =
       props?.['aria-labelledby'] ?? props?.accessibilityLabelledBy;
-    const _importantForAccessibility =
-      props['aria-hidden'] === true
-        ? ('no-hide-descendants' as const)
-        : undefined;
     const placeholder = props.placeholder ?? '';
     let children = props.children;
     const childCount = React.Children.count(children);
@@ -740,12 +811,12 @@
     }
     // For consistency with iOS set cursor/selectionHandle color as selectionColor
     const colorProps = {
-      cursorColor: cursorColor === undefined ? selectionColor : cursorColor,
       selectionColor,
       selectionHandleColor:
         selectionHandleColor === undefined
           ? selectionColor
           : selectionHandleColor,
+      cursorColor: cursorColor === undefined ? selectionColor : cursorColor,
     };
     textInput = (
       /* $FlowFixMe[prop-missing] the types for AndroidTextInput don't match up
@@ -762,9 +833,8 @@
         {...otherProps}
         {...colorProps}
         {...eventHandlers}
-        accessibilityLabel={_accessibilityLabel}
-        accessibilityLabelledBy={_accessibilityLabelledBy}
         accessibilityState={_accessibilityState}
+        accessibilityLabelledBy={_accessibilityLabelledBy}
         accessible={accessible}
         acceptDragAndDropTypes={props.experimental_acceptDragAndDropTypes}
         autoCapitalize={autoCapitalize}
@@ -773,7 +843,6 @@
         children={children}
         disableFullscreenUI={props.disableFullscreenUI}
         focusable={tabIndex !== undefined ? !tabIndex : focusable}
-        importantForAccessibility={_importantForAccessibility}
         mostRecentEventCount={mostRecentEventCount}
         nativeID={id ?? props.nativeID}
         numberOfLines={props.rows ?? props.numberOfLines}
@@ -783,7 +852,7 @@
         /* $FlowFixMe[prop-missing] the types for AndroidTextInput don't match
          * up exactly with the props for TextInput. This will need to get fixed
          */
-        /* $FlowFixMe[incompatible-type] the types for AndroidTextInput
+        /* $FlowFixMe[incompatible-type-arg] the types for AndroidTextInput
          * don't match up exactly with the props for TextInput. This will need
          * to get fixed */
         onScroll={_onScroll}
@@ -799,8 +868,8 @@
 }
 
 const enterKeyHintToReturnTypeMap = {
-  done: 'done',
   enter: 'default',
+  done: 'done',
   go: 'go',
   next: 'next',
   previous: 'previous',
@@ -809,20 +878,19 @@
 } as const;
 
 const inputModeToKeyboardTypeMap = {
-  decimal: 'decimal-pad',
-  email: 'email-address',
   none: 'default',
+  text: 'default',
+  decimal: 'decimal-pad',
   numeric: 'number-pad',
+  tel: 'phone-pad',
   search:
     Platform.OS === 'ios' ? ('web-search' as const) : ('default' as const),
-  tel: 'phone-pad',
-  text: 'default',
+  email: 'email-address',
   url: 'url',
 } as const;
 
 // Map HTML autocomplete values to Android autoComplete values
 const autoCompleteWebToAutoCompleteAndroidMap = {
-  'additional-name': 'name-middle',
   'address-line1': 'postal-address-region',
   'address-line2': 'postal-address-locality',
   bday: 'birthdate-full',
@@ -837,11 +905,12 @@
   country: 'postal-address-country',
   'current-password': 'password',
   email: 'email',
-  'family-name': 'name-family',
-  'given-name': 'name-given',
   'honorific-prefix': 'name-prefix',
   'honorific-suffix': 'name-suffix',
   name: 'name',
+  'additional-name': 'name-middle',
+  'family-name': 'name-family',
+  'given-name': 'name-given',
   'new-password': 'password-new',
   off: 'off',
   'one-time-code': 'sms-otp',
@@ -856,33 +925,33 @@
 
 // Map HTML autocomplete values to iOS textContentType values
 const autoCompleteWebToTextContentTypeMap = {
-  'additional-name': 'middleName',
   'address-line1': 'streetAddressLine1',
   'address-line2': 'streetAddressLine2',
   bday: 'birthdate',
   'bday-day': 'birthdateDay',
   'bday-month': 'birthdateMonth',
   'bday-year': 'birthdateYear',
-  'cc-additional-name': 'creditCardMiddleName',
   'cc-csc': 'creditCardSecurityCode',
-  'cc-exp': 'creditCardExpiration',
   'cc-exp-month': 'creditCardExpirationMonth',
   'cc-exp-year': 'creditCardExpirationYear',
-  'cc-family-name': 'creditCardFamilyName',
+  'cc-exp': 'creditCardExpiration',
   'cc-given-name': 'creditCardGivenName',
+  'cc-additional-name': 'creditCardMiddleName',
+  'cc-family-name': 'creditCardFamilyName',
   'cc-name': 'creditCardName',
   'cc-number': 'creditCardNumber',
   'cc-type': 'creditCardType',
-  country: 'countryName',
   'current-password': 'password',
+  country: 'countryName',
   email: 'emailAddress',
+  name: 'name',
+  'additional-name': 'middleName',
   'family-name': 'familyName',
   'given-name': 'givenName',
+  nickname: 'nickname',
   'honorific-prefix': 'namePrefix',
   'honorific-suffix': 'nameSuffix',
-  name: 'name',
   'new-password': 'newPassword',
-  nickname: 'nickname',
   off: 'none',
   'one-time-code': 'oneTimeCode',
   organization: 'organizationName',
@@ -935,8 +1004,8 @@
         Platform.OS === 'android'
           ? // $FlowFixMe[invalid-computed-prop]
             // $FlowFixMe[prop-missing]
-            (autoCompleteWebToAutoCompleteAndroidMap[autoComplete] ??
-            autoComplete)
+            autoCompleteWebToAutoCompleteAndroidMap[autoComplete] ??
+            autoComplete
           : undefined
       }
       textContentType={
@@ -959,10 +1028,13 @@
 
 // $FlowFixMe[prop-missing]
 TextInput.State = {
-  blurTextInput: TextInputState.blurTextInput,
-  currentlyFocusedField: TextInputState.currentlyFocusedField,
   currentlyFocusedInput: TextInputState.currentlyFocusedInput,
+
+  currentlyFocusedField: TextInputState.currentlyFocusedField,
   focusTextInput: TextInputState.focusTextInput,
+  blurTextInput: TextInputState.blurTextInput,
+  onTextInputFocus: TextInputState.focusInput, // [macOS]
+  onTextInputBlur: TextInputState.blurInput, // [macOS]
 };
 
 export type TextInputComponentStatics = $ReadOnly<{
@@ -980,9 +1052,9 @@
 
 const verticalAlignToTextAlignVerticalMap = {
   auto: 'auto',
+  top: 'top',
   bottom: 'bottom',
   middle: 'center',
-  top: 'top',
 } as const;
 
 // $FlowFixMe[unclear-type] Unclear type. Using `any` type is not safe.
