--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/TextInput/TextInputNativeCommands.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Components/TextInput/TextInputNativeCommands.js	2026-03-06 15:22:52
@@ -22,8 +22,20 @@
     start: Int32,
     end: Int32,
   ) => void;
+  // [macOS
+  // NYI on Android
+  +setGhostText: (
+    viewRef: React.ElementRef<T>,
+    value: ?string, // in theory this is nullable
+  ) => void;
+  // macOS]
 }
 
-const supportedCommands = ['focus', 'blur', 'setTextAndSelection'] as string[];
+const supportedCommands = [
+  'focus',
+  'blur',
+  'setTextAndSelection',
+  'setGhostText',
+] as string[]; // [macOS]
 
 export default supportedCommands;
