--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/NativeAnimation/Drivers/RCTDecayAnimation.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/NativeAnimation/Drivers/RCTDecayAnimation.mm	2026-03-06 15:22:52
@@ -8,7 +8,7 @@
 #import <React/RCTDecayAnimation.h>
 
 #import <React/RCTConvert.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTAnimationUtils.h>
 #import <React/RCTValueAnimatedNode.h>
