--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/NativeAnimation/Drivers/RCTFrameAnimation.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/NativeAnimation/Drivers/RCTFrameAnimation.mm	2026-03-06 15:22:52
@@ -7,7 +7,7 @@
 
 #import <React/RCTFrameAnimation.h>
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTConvert.h>
 #import <React/RCTDefines.h>
