--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/View/ViewPropTypes.d.ts	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Components/View/ViewPropTypes.d.ts	2026-03-06 15:22:52
@@ -13,9 +13,13 @@
 import {StyleProp} from '../../StyleSheet/StyleSheet';
 import {ViewStyle} from '../../StyleSheet/StyleSheetTypes';
 import {
+  DragEvent, // [macOS]
+  HandledKeyEvent, // [macOS]
+  KeyEvent, // [macOS]
   BlurEvent,
   FocusEvent,
   LayoutChangeEvent,
+  MouseEvent, // [macOS]
   PointerEvents,
 } from '../../Types/CoreEventTypes';
 import {Touchable} from '../Touchable/Touchable';
@@ -125,12 +129,34 @@
   tabIndex?: 0 | -1 | undefined;
 }
 
+export type DraggedType = 'fileUrl' | 'image' | 'string';
+export type DraggedTypesType = DraggedType | DraggedType[];
+
+export interface ViewPropsMacOS {
+  acceptsFirstMouse?: boolean | undefined;
+  allowsVibrancy?: boolean | undefined;
+  mouseDownCanMoveWindow?: boolean | undefined;
+  enableFocusRing?: boolean | undefined;
+  onMouseEnter?: ((event: MouseEvent) => void) | undefined;
+  onMouseLeave?: ((event: MouseEvent) => void) | undefined;
+  onDoubleClick?: ((event: MouseEvent) => void) | undefined;
+  onDragEnter?: ((event: DragEvent) => void) | undefined;
+  onDragLeave?: ((event: DragEvent) => void) | undefined;
+  onDrop?: ((event: DragEvent) => void) | undefined;
+  onKeyDown?: ((event: KeyEvent) => void) | undefined;
+  onKeyUp?: ((event: KeyEvent) => void) | undefined;
+  keyDownEvents?: HandledKeyEvent[] | undefined;
+  keyUpEvents?: HandledKeyEvent[] | undefined;
+  draggedTypes?: DraggedTypesType | undefined;
+}
+
 /**
  * @see https://reactnative.dev/docs/view#props
  */
 export interface ViewProps
   extends ViewPropsAndroid,
     ViewPropsIOS,
+    ViewPropsMacOS,
     GestureResponderHandlers,
     Touchable,
     PointerEvents,
