--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/NativeAnimation/Drivers/RCTDecayAnimation.mm	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.doMW3OsE9V	2026-03-06 18:50:44
@@ -8,7 +8,7 @@
 #import <React/RCTDecayAnimation.h>
 
 #import <React/RCTConvert.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTAnimationUtils.h>
 #import <React/RCTValueAnimatedNode.h>
