--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTHermesInstance.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTHermesInstance.h	2026-03-06 15:22:53
@@ -5,35 +5,36 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#import <React/RCTUIKit.h> // [macOS]
+
+#import <cxxreact/MessageQueueThread.h>
 #import <hermes/Public/CrashManager.h>
 #import <jsi/jsi.h>
 #import <react/runtime/JSRuntimeFactory.h>
+#import <react/runtime/hermes/HermesInstance.h>
 
 namespace facebook::react {
 
-class MessageQueueThread;
+using CrashManagerProvider =
+    std::function<std::shared_ptr<::hermes::vm::CrashManager>()>;
 
-using CrashManagerProvider = std::function<std::shared_ptr<::hermes::vm::CrashManager>()>;
-
 // ObjC++ wrapper for HermesInstance.cpp
 class RCTHermesInstance : public JSRuntimeFactory {
  public:
-  RCTHermesInstance() : RCTHermesInstance(nullptr, false) {}
-  RCTHermesInstance(CrashManagerProvider crashManagerProvider)
-      : RCTHermesInstance(std::move(crashManagerProvider), false)
-  {
-  }
-  RCTHermesInstance(CrashManagerProvider crashManagerProvider, bool allocInOldGenBeforeTTI)
-      : _crashManagerProvider(std::move(crashManagerProvider)), _allocInOldGenBeforeTTI(allocInOldGenBeforeTTI)
-  {
-  }
+  RCTHermesInstance();
+  RCTHermesInstance(CrashManagerProvider crashManagerProvider);
+  RCTHermesInstance(
+      CrashManagerProvider crashManagerProvider,
+      bool allocInOldGenBeforeTTI);
 
-  std::unique_ptr<JSRuntime> createJSRuntime(std::shared_ptr<MessageQueueThread> msgQueueThread) noexcept override;
+  std::unique_ptr<JSRuntime> createJSRuntime(
+      std::shared_ptr<MessageQueueThread> msgQueueThread) noexcept override;
 
-  ~RCTHermesInstance() override {};
+  ~RCTHermesInstance() override{};
 
  private:
   CrashManagerProvider _crashManagerProvider;
+  std::unique_ptr<HermesInstance> _hermesInstance;
   bool _allocInOldGenBeforeTTI;
 };
 
