--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.h	2026-03-06 15:22:53
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTDefines.h>
 #import <React/RCTJavaScriptLoader.h>
@@ -32,7 +32,6 @@
 @class RCTPerformanceLogger;
 @class RCTSource;
 @class RCTSurfacePresenter;
-@class RCTDevMenuConfiguration;
 
 @protocol RCTTurboModuleManagerDelegate;
 
@@ -74,15 +73,6 @@
            parentInspectorTarget:(facebook::react::jsinspector_modern::HostTarget *)parentInspectorTarget
                    launchOptions:(nullable NSDictionary *)launchOptions;
 
-- (instancetype)initWithDelegate:(id<RCTInstanceDelegate>)delegate
-                jsRuntimeFactory:(std::shared_ptr<facebook::react::JSRuntimeFactory>)jsRuntimeFactory
-                   bundleManager:(RCTBundleManager *)bundleManager
-      turboModuleManagerDelegate:(id<RCTTurboModuleManagerDelegate>)turboModuleManagerDelegate
-                  moduleRegistry:(RCTModuleRegistry *)moduleRegistry
-           parentInspectorTarget:(facebook::react::jsinspector_modern::HostTarget *)parentInspectorTarget
-                   launchOptions:(nullable NSDictionary *)launchOptions
-            devMenuConfiguration:(RCTDevMenuConfiguration *__nullable)devMenuConfiguration;
-
 - (void)callFunctionOnJSModule:(NSString *)moduleName method:(NSString *)method args:(NSArray *)args;
 - (void)callFunctionOnBufferedRuntimeExecutor:(std::function<void(facebook::jsi::Runtime &runtime)> &&)executor;
 
