--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/Button.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Components/Button.js	2026-03-06 15:22:52
@@ -12,11 +12,14 @@
 
 import type {TextStyleProp, ViewStyleProp} from '../StyleSheet/StyleSheet';
 import type {GestureResponderEvent} from '../Types/CoreEventTypes';
+import type {BlurEvent, FocusEvent} from '../Types/CoreEventTypes'; // [macOS]
 import type {
   AccessibilityActionEvent,
   AccessibilityActionInfo,
+  AccessibilityRole,
+  // [macOS]
   AccessibilityState,
-} from './View/ViewAccessibility';
+} from './View/ViewAccessibility'; // [macOS]
 
 import StyleSheet, {type ColorValue} from '../StyleSheet/StyleSheet';
 import Text from '../Text/Text';
@@ -38,7 +41,7 @@
     Handler to be called when the user taps the button. The first function
     argument is an event in form of [GestureResponderEvent](pressevent).
    */
-  onPress?: (event?: GestureResponderEvent) => unknown,
+  onPress?: (event?: GestureResponderEvent) => mixed,
 
   /**
     If `true`, doesn't play system sound on touch.
@@ -126,6 +129,7 @@
     Text to display for blindness accessibility features.
    */
   accessibilityLabel?: ?string,
+
   /**
    * Alias for accessibilityLabel  https://reactnative.dev/docs/view#accessibilitylabel
    * https://github.com/facebook/react-native/issues/34424
@@ -143,12 +147,39 @@
    */
   testID?: ?string,
 
+  // [macOS
   /**
+   * Custom accessibility role -- otherwise we use button
+   */
+  accessibilityRole?: ?AccessibilityRole,
+
+  /**
+   * Accessibility action handlers
+   */
+  onAccessibilityAction?: ?(event: AccessibilityActionEvent) => mixed,
+
+  /**
+   * Handler to be called when the button receives key focus
+   */
+  onBlur?: ?(e: BlurEvent) => void,
+
+  /**
+   * Handler to be called when the button loses key focus
+   */
+  onFocus?: ?(e: FocusEvent) => void,
+
+  /*
+   * Specifies the Tooltip for the view
+   */
+  tooltip?: string,
+  // macOS]
+
+  /**
    * Accessibility props.
    */
   accessible?: ?boolean,
   accessibilityActions?: ?$ReadOnlyArray<AccessibilityActionInfo>,
-  onAccessibilityAction?: ?(event: AccessibilityActionEvent) => unknown,
+  onAccessibilityAction?: ?(event: AccessibilityActionEvent) => mixed,
   accessibilityState?: ?AccessibilityState,
 
   /**
@@ -184,6 +215,9 @@
   [button:source]:
   https://github.com/facebook/react-native/blob/HEAD/Libraries/Components/Button.js
 
+  [button:examples]:
+  https://js.coach/?menu%5Bcollections%5D=React%20Native&page=1&query=button
+
   ```jsx
   <Button
     onPress={onPressLearnMore}
@@ -291,6 +325,7 @@
 ) = ({ref, ...props}: {ref?: React.RefSetter<ButtonRef>, ...ButtonProps}) => {
   const {
     accessibilityLabel,
+    accessibilityRole, // [macOS]
     accessibilityState,
     'aria-busy': ariaBusy,
     'aria-checked': ariaChecked,
@@ -315,11 +350,17 @@
     accessibilityHint,
     accessibilityLanguage,
     onAccessibilityAction,
+    // [macOS
+    onFocus,
+    onBlur,
+    tooltip,
+    // macOS]
   } = props;
   const buttonStyles: Array<ViewStyleProp> = [styles.button];
   const textStyles: Array<TextStyleProp> = [styles.text];
   if (color) {
-    if (Platform.OS === 'ios') {
+    if (Platform.OS === 'ios' || Platform.OS === 'macos') {
+      // [macOS]
       textStyles.push({color: color});
     } else {
       buttonStyles.push({backgroundColor: color});
@@ -368,7 +409,8 @@
       accessibilityLabel={ariaLabel || accessibilityLabel}
       accessibilityHint={accessibilityHint}
       accessibilityLanguage={accessibilityLanguage}
-      accessibilityRole="button"
+      // accessibilityRole="button"
+      accessibilityRole={accessibilityRole || 'button'} // [macOS]
       accessibilityState={_accessibilityState}
       importantForAccessibility={_importantForAccessibility}
       hasTVPreferredFocus={hasTVPreferredFocus}
@@ -380,10 +422,15 @@
       testID={testID}
       disabled={disabled}
       onPress={onPress}
+      // [macOS
+      onFocus={onFocus}
+      onBlur={onBlur}
+      tooltip={tooltip}
+      // macOS]
       touchSoundDisabled={touchSoundDisabled}
       // $FlowFixMe[incompatible-exact]
       // $FlowFixMe[prop-missing]
-      // $FlowFixMe[incompatible-type]
+      // $FlowFixMe[incompatible-type-arg]
       ref={ref}>
       <View style={buttonStyles}>
         <Text style={textStyles} disabled={disabled}>
@@ -405,6 +452,7 @@
       backgroundColor: '#2196F3',
       borderRadius: 2,
     },
+    macos: {}, // [macOS]
   }),
   text: {
     textAlign: 'center',
@@ -419,6 +467,11 @@
         color: 'white',
         fontWeight: '500',
       },
+      macos: {
+        // [macOS
+        color: '#007AFF',
+        fontSize: 18,
+      }, // macOS]
     }),
   },
   buttonDisabled: Platform.select({
@@ -427,11 +480,17 @@
       elevation: 0,
       backgroundColor: '#dfdfdf',
     },
+    macos: {}, // [macOS]
   }),
   textDisabled: Platform.select({
     ios: {
       color: '#cdcdcd',
     },
+    // [macOS
+    macos: {
+      color: '#cdcdcd',
+    },
+    // macOS]
     android: {
       color: '#a1a1a1',
     },
