--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTComponentViewProtocol.h>
 #import <React/RCTConstants.h>
@@ -17,17 +17,21 @@
 #import <react/renderer/core/LayoutMetrics.h>
 #import <react/renderer/core/Props.h>
 
+#if TARGET_OS_OSX // [macOS
+#include <react/renderer/components/view/MouseEvent.h>
+#endif // macOS]
+
 NS_ASSUME_NONNULL_BEGIN
 
 /**
  * UIView class for <View> component.
  */
-@interface RCTViewComponentView : UIView <RCTComponentViewProtocol, RCTTouchableComponentViewProtocol> {
+@interface RCTViewComponentView : RCTUIView <RCTComponentViewProtocol, RCTTouchableComponentViewProtocol> {
  @protected
   facebook::react::LayoutMetrics _layoutMetrics;
   facebook::react::SharedViewProps _props;
   facebook::react::SharedViewEventEmitter _eventEmitter;
-}
+} // [macOS]
 
 /**
  * Represents the `UIView` instance that is being automatically attached to
@@ -37,7 +41,7 @@
  * to embed/bridge pure native views as component views.
  * Defaults to `nil`. Assign `nil` to remove view as subview.
  */
-@property (nonatomic, strong, nullable) UIView *contentView;
+@property (nonatomic, strong, nullable) RCTPlatformView *contentView; // [macOS]
 
 /**
  * Provides access to `nativeId` prop of the component.
@@ -56,7 +60,11 @@
  * transparent in favour of some subview.
  * Defaults to `self`.
  */
+#if !TARGET_OS_OSX // [macOS]
 @property (nonatomic, strong, nullable, readonly) NSObject *accessibilityElement;
+#else // [macOS
+@property (nonatomic, strong, nullable, readonly) NSView *accessibilityElement;
+#endif // macOS]
 
 /**
  * Insets used when hit testing inside this view.
@@ -74,7 +82,7 @@
            oldLayoutMetrics:(const facebook::react::LayoutMetrics &)oldLayoutMetrics NS_REQUIRES_SUPER;
 - (void)finalizeUpdates:(RNComponentViewUpdateMask)updateMask NS_REQUIRES_SUPER;
 - (void)prepareForRecycle NS_REQUIRES_SUPER;
-- (UIView *)betterHitTest:(CGPoint)point withEvent:(UIEvent *)event;
+- (RCTPlatformView *)betterHitTest:(CGPoint)point withEvent:(UIEvent *)event; // [macOS]
 
 /*
  * This is the label that would be coopted by another element
@@ -86,6 +94,11 @@
  */
 - (BOOL)wantsToCooptLabel;
 
+#if TARGET_OS_OSX // [macOS
+- (BOOL)handleKeyboardEvent:(NSEvent *)event;
+- (facebook::react::DataTransfer)dataTransferForPasteboard:(NSPasteboard *)pasteboard;
+#endif // macOS]
+
 /*
  * This is a fragment of temporary workaround that we need only temporary and will get rid of soon.
  */
