--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Base/Surface/SurfaceHostingView/RCTSurfaceHostingView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Base/Surface/SurfaceHostingView/RCTSurfaceHostingView.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTSurfaceDelegate.h>
 #import <React/RCTSurfaceProtocol.h>
@@ -14,8 +14,11 @@
 
 @class RCTBridge;
 @class RCTSurface;
+#if RCT_DEV_MENU // [macOS
+@class RCTDevMenu;
+#endif // macOS]
 
-typedef UIView *_Nullable (^RCTSurfaceHostingViewActivityIndicatorViewFactory)(void);
+typedef RCTPlatformView *_Nullable (^RCTSurfaceHostingViewActivityIndicatorViewFactory)(void); // [macOS]
 
 NS_ASSUME_NONNULL_BEGIN
 
@@ -25,7 +28,7 @@
  * This class can be used as easy-to-use general purpose integration point
  * of ReactNative-powered experiences in UIKit based apps.
  */
-@interface RCTSurfaceHostingView : UIView <RCTSurfaceDelegate>
+@interface RCTSurfaceHostingView : RCTUIView <RCTSurfaceDelegate> // [macOS]
 
 /**
  * Designated initializer.
@@ -63,6 +66,14 @@
  * @param disabled if `YES`, the auto-hide is disabled. Otherwise the loading view will be hidden automatically
  */
 - (void)disableActivityIndicatorAutoHide:(BOOL)disabled;
+
+#if RCT_DEV_MENU // [macOS
+/**
+ * Dev menu for macOS context menu access.
+ */
+@property (nonatomic, strong, nullable) RCTDevMenu *devMenu;
+#endif // macOS]
+
 @end
 
 NS_ASSUME_NONNULL_END
