--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/DevSupport/RCTDevLoadingViewProtocol.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/DevSupport/RCTDevLoadingViewProtocol.h	2026-03-06 15:22:52
@@ -5,16 +5,13 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 @class RCTLoadingProgress;
 
 @protocol RCTDevLoadingViewProtocol <NSObject>
 + (void)setEnabled:(BOOL)enabled;
-- (void)showMessage:(NSString *)message
-              color:(UIColor *)color
-    backgroundColor:(UIColor *)backgroundColor
-      dismissButton:(BOOL)dismissButton;
+- (void)showMessage:(NSString *)message color:(RCTUIColor *)color backgroundColor:(RCTUIColor *)backgroundColor; // [macOS]
 - (void)showWithURL:(NSURL *)URL;
 - (void)updateProgress:(RCTLoadingProgress *)progress;
 - (void)hide;
