--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/DevSupport/RCTInspectorDevServerHelper.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/DevSupport/RCTInspectorDevServerHelper.mm	2026-03-06 15:22:52
@@ -10,7 +10,7 @@
 #if RCT_DEV || RCT_REMOTE_PROFILE
 
 #import <React/RCTLog.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTCxxInspectorPackagerConnection.h>
 #import <React/RCTDefines.h>
@@ -24,14 +24,14 @@
 {
   NSNumber *port = @8081;
   NSString *portStr = [[[NSProcessInfo processInfo] environment] objectForKey:@"RCT_METRO_PORT"];
-  if ((portStr != nullptr) && [portStr length] > 0) {
+  if (portStr && [portStr length] > 0) {
     port = [NSNumber numberWithInt:[portStr intValue]];
   }
-  if ([bundleURL port] != nullptr) {
+  if ([bundleURL port]) {
     port = [bundleURL port];
   }
   NSString *host = [bundleURL host];
-  if (host == nullptr) {
+  if (!host) {
     host = @"localhost";
   }
 
@@ -103,8 +103,12 @@
   auto &inspectorFlags = facebook::react::jsinspector_modern::InspectorFlags::getInstance();
   BOOL isProfilingBuild = inspectorFlags.getIsProfilingBuild();
 
+#if !TARGET_OS_OSX // [macOS]
   NSString *escapedDeviceName = [[[UIDevice currentDevice] name]
       stringByAddingPercentEncodingWithAllowedCharacters:NSCharacterSet.URLQueryAllowedCharacterSet];
+#else // [macOS
+  NSString *escapedDeviceName = @"";
+#endif // macOS]
   NSString *escapedAppName = [[[NSBundle mainBundle] bundleIdentifier]
       stringByAddingPercentEncodingWithAllowedCharacters:NSCharacterSet.URLQueryAllowedCharacterSet];
 
@@ -125,12 +129,15 @@
 RCT_NOT_IMPLEMENTED(-(instancetype)init)
 
 static NSMutableDictionary<NSString *, id<RCTInspectorPackagerConnectionProtocol>> *socketConnections = nil;
+static NSLock *connectionsLock = [NSLock new];
 
 static void sendEventToAllConnections(NSString *event)
 {
+  [connectionsLock lock]; // [macOS]
   for (NSString *socketId in socketConnections) {
     [socketConnections[socketId] sendEventToAllConnections:event];
   }
+  [connectionsLock unlock]; // [macOS]
 }
 
 + (BOOL)isPackagerDisconnected
@@ -180,18 +187,38 @@
   // Note, using a static dictionary isn't really the greatest design, but
   // the packager connection does the same thing, so it's at least consistent.
   // This is a static map that holds different inspector clients per the inspectorURL
+  [connectionsLock lock]; // [macOS]
   if (socketConnections == nil) {
     socketConnections = [NSMutableDictionary new];
   }
+  [connectionsLock unlock]; // [macOS]
 
   NSString *key = [inspectorURL absoluteString];
-  id<RCTInspectorPackagerConnectionProtocol> connection = socketConnections[key];
-  if ((connection == nullptr) || !connection.isConnected) {
+  // [macOS safety check to avoid a crash
+  if (key == nil) {
+    RCTLogError(@"failed to get inspector URL");
+    return nil;
+  }
+  // macOS]
+
+  // [macOS Add a lock around access to connection
+  id<RCTInspectorPackagerConnectionProtocol> connection;
+  [connectionsLock lock];
+  connection = socketConnections[key];
+   // macOS]
+  if (!connection || !connection.isConnected) {
     connection = [[RCTCxxInspectorPackagerConnection alloc] initWithURL:inspectorURL];
 
-    socketConnections[key] = connection;
-    [connection connect];
+    // [macOS safety check to avoid a crash
+    if (connection != nil) {
+      socketConnections[key] = connection;
+      [connection connect];
+    } else {
+      RCTLogError(@"failed to initialize RCTInspectorPackagerConnection");
+    }
+    // macOS]
   }
+  [connectionsLock unlock]; // [macOS]
 
   return connection;
 }
