--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/Switch/RCTSwitchComponentView.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/ComponentViews/Switch/RCTSwitchComponentView.mm	2026-03-06 15:22:52
@@ -9,10 +9,10 @@
 
 #import <React/RCTConversions.h>
 
+#import <react/renderer/components/FBReactNativeSpec/ComponentDescriptors.h>
 #import <react/renderer/components/FBReactNativeSpec/EventEmitters.h>
 #import <react/renderer/components/FBReactNativeSpec/Props.h>
 #import <react/renderer/components/FBReactNativeSpec/RCTComponentViewHelpers.h>
-#import <react/renderer/components/switch/AppleSwitchComponentDescriptor.h>
 
 #import "RCTFabricComponentsPlugins.h"
 
@@ -22,7 +22,7 @@
 @end
 
 @implementation RCTSwitchComponentView {
-  UISwitch *_switchView;
+  RCTUISwitch *_switchView; // [macOS]
   BOOL _isInitialValueSet;
 }
 
@@ -31,9 +31,14 @@
   if (self = [super initWithFrame:frame]) {
     _props = SwitchShadowNode::defaultSharedProps();
 
-    _switchView = [[UISwitch alloc] initWithFrame:self.bounds];
+    _switchView = [[RCTUISwitch alloc] initWithFrame:self.bounds]; // [macOS]
 
+#if !TARGET_OS_OSX // [macOS]
     [_switchView addTarget:self action:@selector(onChange:) forControlEvents:UIControlEventValueChanged];
+#else // [macOS
+    [_switchView setTarget:self];
+    [_switchView setAction:@selector(onChange:)];
+#endif // macOS]
 
     self.contentView = _switchView;
   }
@@ -70,27 +75,29 @@
     _switchView.enabled = !newSwitchProps.disabled;
   }
 
+#if !TARGET_OS_OSX // [macOS]
   // `tintColor`
   if (oldSwitchProps.tintColor != newSwitchProps.tintColor) {
-    _switchView.tintColor = RCTUIColorFromSharedColor(newSwitchProps.tintColor);
+    _switchView.tintColor = RCTUIColorFromSharedColor(newSwitchProps.tintColor); // [macOS]
   }
 
   // `onTintColor
   if (oldSwitchProps.onTintColor != newSwitchProps.onTintColor) {
-    _switchView.onTintColor = RCTUIColorFromSharedColor(newSwitchProps.onTintColor);
+    _switchView.onTintColor = RCTUIColorFromSharedColor(newSwitchProps.onTintColor); // [macOS]
   }
 
   // `thumbTintColor`
   if (oldSwitchProps.thumbTintColor != newSwitchProps.thumbTintColor) {
-    _switchView.thumbTintColor = RCTUIColorFromSharedColor(newSwitchProps.thumbTintColor);
+    _switchView.thumbTintColor = RCTUIColorFromSharedColor(newSwitchProps.thumbTintColor); // [macOS]
   }
+#endif // [macOS]
 
   _isInitialValueSet = YES;
 
   [super updateProps:props oldProps:oldProps];
 }
 
-- (void)onChange:(UISwitch *)sender
+- (void)onChange:(RCTUISwitch *)sender // [macOS]
 {
   const auto &props = static_cast<const SwitchProps &>(*_props);
   if (props.value == sender.on) {
