--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTBorderDrawing.m	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.U30rY9ohLx	2026-03-06 18:50:52
@@ -66,8 +66,7 @@
   edgeInsets.bottom = RCTRoundPixelValue(edgeInsets.bottom);
   edgeInsets.left = RCTRoundPixelValue(edgeInsets.left);
   edgeInsets.right = RCTRoundPixelValue(edgeInsets.right);
-
-  return edgeInsets;
+    return edgeInsets;
 }
 
 static void RCTPathAddEllipticArc(
@@ -202,23 +201,23 @@
   return RCTPathCreateWithRoundedRect(rect, RCTGetCornerInsets(cornerRadii, UIEdgeInsetsZero), NULL, NO);
 }
 
-static UIGraphicsImageRenderer *
-RCTMakeUIGraphicsImageRenderer(CGSize size, UIColor *backgroundColor, BOOL hasCornerRadii, BOOL drawToEdge)
+static RCTUIGraphicsImageRenderer * // [macOS]
+RCTMakeUIGraphicsImageRenderer(CGSize size, RCTUIColor *backgroundColor, BOOL hasCornerRadii, BOOL drawToEdge) // [macOS]
 {
   const CGFloat alpha = CGColorGetAlpha(backgroundColor.CGColor);
   const BOOL opaque = (drawToEdge || !hasCornerRadii) && alpha == 1.0;
-  UIGraphicsImageRendererFormat *const rendererFormat = [UIGraphicsImageRendererFormat defaultFormat];
+  RCTUIGraphicsImageRendererFormat *const rendererFormat = [RCTUIGraphicsImageRendererFormat defaultFormat]; // [macOS]
   rendererFormat.opaque = opaque;
-  UIGraphicsImageRenderer *const renderer = [[UIGraphicsImageRenderer alloc] initWithSize:size format:rendererFormat];
+  RCTUIGraphicsImageRenderer *const renderer = [[RCTUIGraphicsImageRenderer alloc] initWithSize:size format:rendererFormat]; // [macOS]
   return renderer;
 }
 
-static UIImage *RCTGetSolidBorderImage(
+static RCTUIImage *RCTGetSolidBorderImage( // [macOS]
     RCTCornerRadii cornerRadii,
     CGSize viewSize,
     UIEdgeInsets borderInsets,
     RCTBorderColors borderColors,
-    UIColor *backgroundColor,
+    RCTUIColor *backgroundColor, // [macOS]
     BOOL drawToEdge)
 {
   const BOOL hasCornerRadii = RCTCornerRadiiAreAboveThreshold(cornerRadii);
@@ -255,10 +254,9 @@
     ceil(edgeInsets.top) + 1 + ceil(edgeInsets.bottom),
   } : viewSize;
 
-  UIGraphicsImageRenderer *const imageRenderer =
+  RCTUIGraphicsImageRenderer *const imageRenderer =
       RCTMakeUIGraphicsImageRenderer(size, backgroundColor, hasCornerRadii, drawToEdge);
-
-  UIImage *image = [imageRenderer imageWithActions:^(UIGraphicsImageRendererContext *_Nonnull rendererContext) {
+  RCTUIImage *image = [imageRenderer imageWithActions:^(RCTUIGraphicsImageRendererContext *_Nonnull rendererContext) { // [macOS]
     const CGContextRef context = rendererContext.CGContext;
     const CGRect rect = {.size = size};
     CGPathRef path = RCTPathCreateOuterOutline(drawToEdge, rect, cornerRadii);
@@ -342,7 +340,7 @@
         }
       }
 
-      UIColor *currentColor = nil;
+      RCTUIColor *currentColor = nil; // [macOS]
 
       // RIGHT
       if (borderInsets.right > 0) {
@@ -416,7 +414,14 @@
   }];
 
   if (makeStretchable) {
+#if !TARGET_OS_OSX // [macOS]
+  }];
+
+  if (makeStretchable) {
     image = [image resizableImageWithCapInsets:edgeInsets];
+#else // [macOS
+    image.capInsets = edgeInsets;
+#endif // macOS]
   }
 
   return image;
@@ -482,13 +487,13 @@
 // of gradients _along_ a path (NB: clipping a path and drawing a linear gradient
 // is _not_ equivalent).
 
-static UIImage *RCTGetDashedOrDottedBorderImage(
+static RCTUIImage *RCTGetDashedOrDottedBorderImage( // [macOS]
     RCTBorderStyle borderStyle,
     RCTCornerRadii cornerRadii,
     CGSize viewSize,
     UIEdgeInsets borderInsets,
     RCTBorderColors borderColors,
-    UIColor *backgroundColor,
+    RCTUIColor *backgroundColor, // [macOS]
     BOOL drawToEdge)
 {
   NSCParameterAssert(borderStyle == RCTBorderStyleDashed || borderStyle == RCTBorderStyleDotted);
@@ -503,10 +508,15 @@
     return nil;
   }
 
+  // [macOS viewSize must nonzero
+  if (viewSize.width <= 0 || viewSize.height <= 0) {
+    return nil;
+  } // macOS]
+
   const BOOL hasCornerRadii = RCTCornerRadiiAreAboveThreshold(cornerRadii);
-  UIGraphicsImageRenderer *const imageRenderer =
+  RCTUIGraphicsImageRenderer *const imageRenderer = // [macOS]
       RCTMakeUIGraphicsImageRenderer(viewSize, backgroundColor, hasCornerRadii, drawToEdge);
-  return [imageRenderer imageWithActions:^(UIGraphicsImageRendererContext *_Nonnull rendererContext) {
+  return [imageRenderer imageWithActions:^(RCTUIGraphicsImageRendererContext *_Nonnull rendererContext) { // [macOS]
     const CGContextRef context = rendererContext.CGContext;
     const CGRect rect = {.size = viewSize};
 
@@ -560,7 +570,7 @@
     CGContextSetLineWidth(context, lineWidth);
     CGContextSetLineDash(context, 0, dashLengths, sizeof(dashLengths) / sizeof(*dashLengths));
 
-    CGContextSetStrokeColorWithColor(context, [UIColor yellowColor].CGColor);
+    CGContextSetStrokeColorWithColor(context, [RCTUIColor yellowColor].CGColor); // [macOS]
 
     CGContextAddPath(context, path);
     CGContextSetStrokeColorWithColor(context, borderColors.top.CGColor);
@@ -570,14 +580,13 @@
   }];
 }
 
-UIImage *RCTGetBorderImage(
+RCTUIImage *RCTGetBorderImage( // [macOS]
     RCTBorderStyle borderStyle,
     CGSize viewSize,
     RCTCornerRadii cornerRadii,
     UIEdgeInsets borderInsets,
     RCTBorderColors borderColors,
-    UIColor *backgroundColor,
-    BOOL drawToEdge)
+    RCTUIColor *backgroundColor, // [macOS]
 {
   switch (borderStyle) {
     case RCTBorderStyleSolid:
