--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/RCTComponentData.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.o2LEc4NNBw	2026-03-06 18:50:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <Foundation/Foundation.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTComponent.h>
 #import <React/RCTDefines.h>
@@ -13,7 +13,7 @@
 
 @class RCTBridge;
 @class RCTShadowView;
-@class UIView;
+@class RCTUIView; // [macOS]
 @class RCTEventDispatcherProtocol;
 
 NS_ASSUME_NONNULL_BEGIN
@@ -34,7 +34,7 @@
                               bridge:(RCTBridge *)bridge
                      eventDispatcher:(id<RCTEventDispatcherProtocol>)eventDispatcher NS_DESIGNATED_INITIALIZER;
 
-- (UIView *)createViewWithTag:(nullable NSNumber *)tag rootTag:(nullable NSNumber *)rootTag;
+- (RCTPlatformView *)createViewWithTag:(nullable NSNumber *)tag rootTag:(nullable NSNumber *)rootTag; // [macOS]
 - (RCTShadowView *)createShadowViewWithTag:(NSNumber *)tag;
 - (void)setProps:(NSDictionary<NSString *, id> *)props forView:(id<RCTComponent>)view;
 - (void)setProps:(NSDictionary<NSString *, id> *)props forShadowView:(RCTShadowView *)shadowView;
