--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/graphics/React-graphics.podspec	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/graphics/React-graphics.podspec	2026-03-06 15:22:53
@@ -32,14 +32,18 @@
   s.source                 = source
   s.source_files           = podspec_sources(source_files, ["*.h", "platform/ios/**/*.h"])
   s.header_dir             = "react/renderer/graphics"
-  s.framework = "UIKit"
+  # [macOS Restrict UIKit to iOS and visionOS
+  s.ios.framework = "UIKit" 
+  s.visionos.framework = "UIKit" 
+  s.osx.framework = "Appkit" 
+  # macOS]
 
   if ENV['USE_FRAMEWORKS']
+    s.module_name            = "React_graphics"
+    s.header_mappings_dir  = "../../.."
     header_search_paths = header_search_paths + ["\"$(PODS_TARGET_SRCROOT)/platform/ios\""]
   end
 
-  resolve_use_frameworks(s, header_mappings_dir: "../../..", module_name: "React_graphics")
-
   s.pod_target_xcconfig  = { "USE_HEADERMAP" => "NO",
                              "HEADER_SEARCH_PATHS" => header_search_paths.join(" "),
                              "DEFINES_MODULE" => "YES",
@@ -48,8 +52,7 @@
   s.dependency "React-jsi"
   s.dependency "React-jsiexecutor"
   s.dependency "React-utils"
-
+  s.dependency "React-Core" # [macOS] Needed for RCTUIKit
   depend_on_js_engine(s)
   add_rn_third_party_dependencies(s)
-  add_rncore_dependency(s)
 end
