--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/Touchable/TouchableWithoutFeedback.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Components/Touchable/TouchableWithoutFeedback.js	2026-03-06 15:22:52
@@ -16,6 +16,10 @@
   GestureResponderEvent,
   LayoutChangeEvent,
 } from '../../Types/CoreEventTypes';
+// [macOS
+import type {DragEvent, MouseEvent} from '../../Types/CoreEventTypes';
+// macOS]
+import type {DraggedTypesType} from '../View/DraggedType'; // [macOS]
 
 import View from '../../Components/View/View';
 import {type AccessibilityProps} from '../../Components/View/ViewAccessibility';
@@ -25,7 +29,19 @@
 import * as React from 'react';
 import {cloneElement, useMemo} from 'react';
 
-export type TouchableWithoutFeedbackPropsIOS = {};
+export type TouchableWithoutFeedbackPropsIOS = {
+  // [macOS
+  acceptsFirstMouse?: ?boolean,
+  enableFocusRing?: ?boolean,
+  tooltip?: ?string,
+  onMouseEnter?: (event: MouseEvent) => void,
+  onMouseLeave?: (event: MouseEvent) => void,
+  onDragEnter?: (event: DragEvent) => void,
+  onDragLeave?: (event: DragEvent) => void,
+  onDrop?: (event: DragEvent) => void,
+  draggedTypes?: ?DraggedTypesType,
+  // macOS]
+};
 
 export type TouchableWithoutFeedbackPropsAndroid = {
   /**
@@ -79,32 +95,32 @@
       | 'no-hide-descendants'
     ),
     nativeID?: ?string,
-    onAccessibilityAction?: ?(event: AccessibilityActionEvent) => unknown,
+    onAccessibilityAction?: ?(event: AccessibilityActionEvent) => mixed,
     /**
      * When `accessible` is true (which is the default) this may be called when
      * the OS-specific concept of "blur" occurs, meaning the element lost focus.
      * Some platforms may not have the concept of blur.
      */
-    onBlur?: ?(event: BlurEvent) => unknown,
+    onBlur?: ?(event: BlurEvent) => void, // [macOS]
     /**
      * When `accessible` is true (which is the default) this may be called when
      * the OS-specific concept of "focus" occurs. Some platforms may not have
      * the concept of focus.
      */
-    onFocus?: ?(event: FocusEvent) => unknown,
+    onFocus?: ?(event: FocusEvent) => void, // [macOS]
     /**
      * Invoked on mount and layout changes with
      * {nativeEvent: {layout: {x, y, width, height}}}
      */
-    onLayout?: ?(event: LayoutChangeEvent) => unknown,
-    onLongPress?: ?(event: GestureResponderEvent) => unknown,
+    onLayout?: ?(event: LayoutChangeEvent) => mixed,
+    onLongPress?: ?(event: GestureResponderEvent) => mixed,
     /**
      * Called when the touch is released,
      * but not if cancelled (e.g. by a scroll that steals the responder lock).
      */
-    onPress?: ?(event: GestureResponderEvent) => unknown,
-    onPressIn?: ?(event: GestureResponderEvent) => unknown,
-    onPressOut?: ?(event: GestureResponderEvent) => unknown,
+    onPress?: ?(event: GestureResponderEvent) => mixed,
+    onPressIn?: ?(event: GestureResponderEvent) => mixed,
+    onPressOut?: ?(event: GestureResponderEvent) => mixed,
     /**
      * When the scroll view is disabled, this defines how far your
      * touch may move off of the button, before deactivating the button.
@@ -151,6 +167,15 @@
   'onBlur',
   'onFocus',
   'onLayout',
+  // [macOS
+  'onMouseEnter',
+  'onMouseLeave',
+  'onDragEnter',
+  'onDragLeave',
+  'onDrop',
+  'draggedTypes',
+  'tooltip',
+  // macOS]
   'testID',
 ] as const;
 
@@ -189,7 +214,7 @@
       disabled:
         disabled !== null
           ? disabled
-          : (ariaDisabled ?? accessibilityState?.disabled),
+          : ariaDisabled ?? accessibilityState?.disabled,
       hitSlop: hitSlop,
       delayLongPress: delayLongPress,
       delayPressIn: delayPressIn,
@@ -250,7 +275,7 @@
   // adopting `Pressability`, so preserve that behavior.
   const {onBlur, onFocus, ...eventHandlersWithoutBlurAndFocus} = eventHandlers;
 
-  const elementProps: {[string]: unknown, ...} = {
+  const elementProps: {[string]: mixed, ...} = {
     ...eventHandlersWithoutBlurAndFocus,
     accessible: props.accessible !== false,
     accessibilityState:
@@ -264,6 +289,16 @@
       props.focusable !== false &&
       props.onPress !== undefined &&
       !props.disabled,
+    // [macOS
+    acceptsFirstMouse:
+      props.acceptsFirstMouse !== false &&
+      props.onPress !== undefined &&
+      !props.disabled,
+    enableFocusRing:
+      props.enableFocusRing !== false &&
+      props.onPress !== undefined &&
+      !props.disabled,
+    // macOS]
 
     accessibilityElementsHidden:
       props['aria-hidden'] ?? props.accessibilityElementsHidden,
@@ -272,7 +307,7 @@
         ? 'no-hide-descendants'
         : props.importantForAccessibility,
     accessibilityLiveRegion:
-      ariaLive === 'off' ? 'none' : (ariaLive ?? props.accessibilityLiveRegion),
+      ariaLive === 'off' ? 'none' : ariaLive ?? props.accessibilityLiveRegion,
     nativeID: props.id ?? props.nativeID,
   };
 
@@ -282,6 +317,6 @@
     }
   }
 
-  // $FlowFixMe[incompatible-type]
+  // $FlowFixMe[incompatible-call]
   return cloneElement(element, elementProps, ...children);
 }
