--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/Touchable/TouchableOpacity.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Components/Touchable/TouchableOpacity.js	2026-03-06 15:22:52
@@ -261,8 +261,13 @@
   render(): React.Node {
     // BACKWARD-COMPATIBILITY: Focus and blur events were never supported before
     // adopting `Pressability`, so preserve that behavior.
-    const {onBlur, onFocus, ...eventHandlersWithoutBlurAndFocus} =
-      this.state.pressability.getEventHandlers();
+    const {
+      onBlur,
+      onFocus,
+      onMouseEnter, // [macOS]
+      onMouseLeave, // [macOS]
+      ...eventHandlersWithoutBlurAndFocus
+    } = this.state.pressability.getEventHandlers();
 
     let _accessibilityState = {
       busy: this.props['aria-busy'] ?? this.props.accessibilityState?.busy,
@@ -294,7 +299,7 @@
     const accessibilityLiveRegion =
       this.props['aria-live'] === 'off'
         ? 'none'
-        : (this.props['aria-live'] ?? this.props.accessibilityLiveRegion);
+        : this.props['aria-live'] ?? this.props.accessibilityLiveRegion;
 
     const accessibilityLabel =
       this.props['aria-label'] ?? this.props.accessibilityLabel;
@@ -336,8 +341,29 @@
         focusable={
           this.props.focusable !== false &&
           this.props.onPress !== undefined &&
+          !this.props.disabled
+        }
+        // [macOS
+        enableFocusRing={
+          (this.props.enableFocusRing === undefined ||
+            this.props.enableFocusRing === true) &&
           !this.props.disabled
         }
+        acceptsFirstMouse={
+          this.props.acceptsFirstMouse !== false &&
+          this.props.acceptsFirstMouse !== undefined &&
+          !this.props.disabled
+        }
+        tooltip={this.props.tooltip}
+        onMouseEnter={this.props.onMouseEnter}
+        onMouseLeave={this.props.onMouseLeave}
+        onDragEnter={this.props.onDragEnter}
+        onDragLeave={this.props.onDragLeave}
+        onDrop={this.props.onDrop}
+        onFocus={this.props.onFocus}
+        onBlur={this.props.onBlur}
+        draggedTypes={this.props.draggedTypes}
+        // macOS]
         // $FlowFixMe[prop-missing]
         ref={this.props.hostRef}
         {...eventHandlersWithoutBlurAndFocus}>
