--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/components/textinput/platform/ios/react/renderer/components/iostextinput/primitives.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.t2m8JMLqe4	2026-03-06 18:50:54
@@ -83,23 +83,23 @@
   int end{0};
 };
 
-/*
- * Controls features of text inputs.
- */
-class TextInputTraits final {
- public:
-  /*
-   * iOS & Android
-   * Default value: `Sentences`.
-   */
-  AutocapitalizationType autocapitalizationType{AutocapitalizationType::Sentences};
+#if TARGET_OS_OSX // [macOS
+enum class PastedTypesType {
+  FileUrl,
+  Image,
+  String,
+};
 
-  /*
-   * Can be empty (`null` in JavaScript) which means `default`.
-   * iOS & Android
-   * Default value: `empty` (`null`).
-   */
-  std::optional<bool> autoCorrect{};
+class SubmitKeyEvent final {
+  public:
+    std::string key{};
+    bool altKey{false};
+    bool shiftKey{false};
+    bool ctrlKey{false};
+    bool metaKey{false};
+    bool functionKey{false};
+};
+#endif // macOS]
 
   /*
    * iOS & Android
@@ -217,6 +217,33 @@
    * Default value: `empty` (`null`).
    */
   std::optional<bool> smartInsertDelete{};
-};
 
-} // namespace facebook::react
+#if TARGET_OS_OSX // [macOS
+  /*
+   * Can be empty (`null` in JavaScript) which means `default`.
+   * macOS
+   * Default value: `empty` (`null`).
+   */
+  std::optional<bool> grammarCheck{};
+
+  /*
+   * List of pastable types
+   * macOS-only
+   * Default value: `empty list`
+   */
+  std::vector<PastedTypesType> pastedTypes{};
+
+  /*
+   * List of key combinations that should submit.
+   * macOS-only
+   * Default value: `empty list` applies as 'Enter' key.
+  */
+  std::vector<SubmitKeyEvent> submitKeyEvents{};
+
+  /*
+   * When set to `true`, the text will be cleared after the submit.
+   * macOS-only
+   * Default value: `false`
+   */
+   bool clearTextOnSubmit{false};
+#endif // macOS]
