--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Views/ScrollView/RCTScrollableProtocol.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.xG8WE0GaWA	2026-03-06 18:50:53
@@ -6,7 +6,7 @@
  */
 
 #import <React/RCTComponent.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 /**
  * Contains any methods related to scrolling. Any `RCTView` that has scrolling
@@ -25,8 +25,8 @@
 - (void)scrollToEnd:(BOOL)animated;
 - (void)zoomToRect:(CGRect)rect animated:(BOOL)animated;
 
-- (void)addScrollListener:(NSObject<UIScrollViewDelegate> *)scrollListener;
-- (void)removeScrollListener:(NSObject<UIScrollViewDelegate> *)scrollListener;
+- (void)addScrollListener:(NSObject<RCTUIScrollViewDelegate> *)scrollListener; // [macOS]
+- (void)removeScrollListener:(NSObject<RCTUIScrollViewDelegate> *)scrollListener; // [macOS]
 
 @end
 
@@ -38,9 +38,9 @@
 @property (nonatomic, copy) RCTDirectEventBlock onRefresh;
 @property (nonatomic, readonly, getter=isRefreshing) BOOL refreshing;
 
-@optional
-@property (nonatomic, weak) UIScrollView *scrollView;
-
+#if !TARGET_OS_OSX // [macOS]
+@property (nonatomic, weak) RCTUIScrollView *scrollView;
+#endif // [macOS]
 @end
 
 __attribute__((deprecated("Use RCTCustomRefreshControlProtocol instead")))
