--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Surface/RCTFabricSurface.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Surface/RCTFabricSurface.mm	2026-03-06 15:22:52
@@ -57,7 +57,7 @@
   if (self = [super init]) {
     _surfacePresenter = surfacePresenter;
 
-    _surfaceHandler.emplace(RCTStringFromNSString(moduleName), getNextRootViewTag());
+    _surfaceHandler = SurfaceHandler{RCTStringFromNSString(moduleName), getNextRootViewTag()};
     _surfaceHandler->setProps(convertIdToFollyDynamic(initialProperties));
 
     [_surfacePresenter registerSurface:self];
@@ -66,10 +66,12 @@
 
     [self _updateLayoutContext];
 
+#if !TARGET_OS_OSX // [macOS]
     [[NSNotificationCenter defaultCenter] addObserver:self
                                              selector:@selector(handleContentSizeCategoryDidChangeNotification:)
                                                  name:UIContentSizeCategoryDidChangeNotification
                                                object:nil];
+#endif // [macOS]
   }
 
   return self;
