--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/StatusBar/StatusBar.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Components/StatusBar/StatusBar.js	2026-03-06 15:22:52
@@ -28,11 +28,11 @@
   /**
    * Dark background, white texts and icons
    */
-  'light-content': string,
+  'light-content': ColorValue, // [macOS]
   /**
    * Light background, dark texts and icons
    */
-  'dark-content': string,
+  'dark-content': ColorValue, // [macOS]
   ...
 }>;
 
@@ -235,8 +235,8 @@
   static _defaultProps: any = createStackEntry({
     backgroundColor:
       Platform.OS === 'android'
-        ? (NativeStatusBarManagerAndroid.getConstants()
-            .DEFAULT_BACKGROUND_COLOR ?? 'black')
+        ? NativeStatusBarManagerAndroid.getConstants()
+            .DEFAULT_BACKGROUND_COLOR ?? 'black'
         : 'black',
     barStyle: 'default',
     translucent: false,
