--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/RCTConvert+Text.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/RCTConvert+Text.mm	2026-03-06 15:22:52
@@ -34,11 +34,13 @@
     RCTTextTransformUndefined,
     integerValue)
 
+#if !TARGET_OS_OSX // [macOS]
 + (UITextSmartInsertDeleteType)UITextSmartInsertDeleteType:(id)json
 {
   return json == nil           ? UITextSmartInsertDeleteTypeDefault
       : [RCTConvert BOOL:json] ? UITextSmartInsertDeleteTypeYes
                                : UITextSmartInsertDeleteTypeNo;
 }
+#endif// macOS]
 
 @end
