--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/React-RCTText.podspec	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/React-RCTText.podspec	2026-03-06 15:22:52
@@ -27,9 +27,14 @@
   s.platforms              = min_supported_versions
   s.source                 = source
   s.source_files           = podspec_sources("**/*.{h,m,mm}", "**/*.h")
+  s.ios.exclude_files      = "**/macOS/*" # [macOS]
+  s.visionos.exclude_files = "**/macOS/*" # [visionOS]
   s.preserve_paths         = "package.json", "LICENSE", "LICENSE-docs"
   s.header_dir             = "RCTText"
-  s.framework              = ["MobileCoreServices"]
+  # [macOS MobileCoreServices Not available on macOS
+  s.ios.frameworks         = "MobileCoreServices" 
+  s.visionos.frameworks    = "MobileCoreServices"
+  # macOS]
   s.pod_target_xcconfig    = { "CLANG_CXX_LANGUAGE_STANDARD" => rct_cxx_language_standard() }
 
   s.dependency "Yoga"
