--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/TextNativeComponent.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/TextNativeComponent.js	2026-03-06 15:22:52
@@ -21,7 +21,7 @@
   ...TextProps,
   isHighlighted?: ?boolean,
   selectionColor?: ?ProcessedColorValue,
-  onClick?: ?(event: GestureResponderEvent) => unknown,
+  onClick?: ?(event: GestureResponderEvent) => mixed,
   // This is only needed for platforms that optimize text hit testing, e.g.,
   // react-native-windows. It can be used to only hit test virtual text spans
   // that have pressable events attached to them.
@@ -47,6 +47,7 @@
     dataDetectorType: true,
     android_hyphenationFrequency: true,
     lineBreakStrategyIOS: true,
+    tooltip: true, // [macOS]
   },
   directEventTypes: {
     topTextLayout: {
@@ -65,19 +66,9 @@
   uiViewClassName: 'RCTVirtualText',
 };
 
-/**
- * `NativeText` is an internal React Native host component, and is exported to
- * provide lower-level access for libraries.
- *
- * @warning `<unstable_NativeText>` provides no semver guarantees and is not
- *   intended to be used in app code. Please use
- *   [`<Text>`](https://reactnative.dev/docs/text) instead.
- */
-// Additional note: Our long term plan is to reduce the overhead of the <Text>
-// and <View> wrappers so that we no longer have any reason to export these APIs.
 export const NativeText: HostComponent<NativeTextProps> =
   (createReactNativeComponentClass('RCTText', () =>
-    /* $FlowFixMe[incompatible-type] Natural Inference rollout. See
+    /* $FlowFixMe[incompatible-call] Natural Inference rollout. See
      * https://fburl.com/workplace/6291gfvu */
     createViewConfig(textViewConfig),
   ): any);
@@ -86,7 +77,7 @@
   !global.RN$Bridgeless && !UIManager.hasViewManagerConfig('RCTVirtualText')
     ? NativeText
     : (createReactNativeComponentClass('RCTVirtualText', () =>
-        /* $FlowFixMe[incompatible-type] Natural Inference rollout. See
+        /* $FlowFixMe[incompatible-call] Natural Inference rollout. See
          * https://fburl.com/workplace/6291gfvu */
         createViewConfig(virtualTextViewConfig),
       ): any);
