--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/Text.d.ts	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/Text.d.ts	2026-03-06 15:22:52
@@ -10,12 +10,13 @@
 import type * as React from 'react';
 import {Constructor} from '../../types/private/Utilities';
 import {AccessibilityProps} from '../Components/View/ViewAccessibility';
-import {HostInstance} from '../../types/public/ReactNativeTypes';
+import {NativeMethods} from '../../types/public/ReactNativeTypes';
 import {ColorValue, StyleProp} from '../StyleSheet/StyleSheet';
 import {TextStyle, ViewStyle} from '../StyleSheet/StyleSheetTypes';
 import {
   GestureResponderEvent,
   LayoutChangeEvent,
+  MouseEvent, // [macOS]
   TextLayoutEvent,
 } from '../Types/CoreEventTypes';
 
@@ -224,7 +225,7 @@
  * A React component for displaying text which supports nesting, styling, and touch handling.
  */
 declare class TextComponent extends React.Component<TextProps> {}
-declare const TextBase: Constructor<HostInstance> & typeof TextComponent;
+declare const TextBase: Constructor<NativeMethods> & typeof TextComponent;
 export class Text extends TextBase {}
 
 export const unstable_TextAncestorContext: React.Context<boolean>;
