--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/NativeAnimation/RCTAnimationUtils.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/NativeAnimation/RCTAnimationUtils.mm	2026-03-06 15:22:52
@@ -84,21 +84,15 @@
   return RCTInterpolateValue(value, inputMin, inputMax, outputMin, outputMax, extrapolateLeft, extrapolateRight);
 }
 
-uint32_t RCTInterpolateColorInRange(CGFloat value, NSArray<NSNumber *> *inputRange, NSArray<UIColor *> *outputRange)
+uint32_t RCTInterpolateColorInRange(CGFloat value, NSArray<NSNumber *> *inputRange, NSArray<RCTUIColor *> *outputRange) // [macOS]
 {
   NSUInteger rangeIndex = RCTFindIndexOfNearestValue(value, inputRange);
   CGFloat inputMin = inputRange[rangeIndex].doubleValue;
   CGFloat inputMax = inputRange[rangeIndex + 1].doubleValue;
 
-  CGFloat redMin;
-  CGFloat greenMin;
-  CGFloat blueMin;
-  CGFloat alphaMin;
+  CGFloat redMin, greenMin, blueMin, alphaMin;
   [outputRange[rangeIndex] getRed:&redMin green:&greenMin blue:&blueMin alpha:&alphaMin];
-  CGFloat redMax;
-  CGFloat greenMax;
-  CGFloat blueMax;
-  CGFloat alphaMax;
+  CGFloat redMax, greenMax, blueMax, alphaMax;
   [outputRange[rangeIndex + 1] getRed:&redMax green:&greenMax blue:&blueMax alpha:&alphaMax];
 
   return RCTColorFromComponents(
