--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/NativeAnimation/RCTNativeAnimatedNodesManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/NativeAnimation/RCTNativeAnimatedNodesManager.mm	2026-03-06 15:22:52
@@ -53,13 +53,13 @@
   // there will be only one driver per mapping so all code code should be optimized around that.
   NSMutableDictionary<NSString *, NSMutableArray<RCTEventAnimation *> *> *_eventDrivers;
   NSMutableSet<id<RCTAnimationDriver>> *_activeAnimations;
-  CADisplayLink *_displayLink;
+  RCTPlatformDisplayLink *_displayLink; // [macOS]
 }
 
 - (instancetype)initWithBridge:(nullable RCTBridge *)bridge
               surfacePresenter:(id<RCTSurfacePresenterStub>)surfacePresenter
 {
-  if ((self = [super init]) != nullptr) {
+  if ((self = [super init])) {
     _bridge = bridge;
     _surfacePresenter = surfacePresenter;
     _animationNodes = [NSMutableDictionary new];
@@ -72,7 +72,7 @@
 - (BOOL)isNodeManagedByFabric:(NSNumber *)tag
 {
   RCTAnimatedNode *node = _animationNodes[tag];
-  if (node != nullptr) {
+  if (node) {
     return [node isManagedByFabric];
   }
   return false;
@@ -106,7 +106,7 @@
   NSString *nodeType = [RCTConvert NSString:config[@"type"]];
 
   Class nodeClass = map[nodeType];
-  if (nodeClass == nullptr) {
+  if (!nodeClass) {
     RCTLogError(@"Animated node type %@ not supported natively", nodeType);
     return;
   }
@@ -187,7 +187,7 @@
 - (void)dropAnimatedNode:(NSNumber *)tag
 {
   RCTAnimatedNode *node = _animationNodes[tag];
-  if (node != nullptr) {
+  if (node) {
     [node detachNode];
     [_animationNodes removeObjectForKey:tag];
   }
@@ -345,7 +345,7 @@
   NSNumber *nodeTag = [RCTConvert NSNumber:eventMapping[@"animatedValueTag"]];
   RCTAnimatedNode *node = _animationNodes[nodeTag];
 
-  if (node == nullptr) {
+  if (!node) {
     RCTLogError(@"Animated node with tag %@ does not exist", nodeTag);
     return;
   }
@@ -407,7 +407,7 @@
 
   NSString *key = [NSString stringWithFormat:@"%@%@", event.viewTag, RCTNormalizeAnimatedEventName(event.eventName)];
   NSMutableArray<RCTEventAnimation *> *driversForKey = _eventDrivers[key];
-  if (driversForKey != nullptr) {
+  if (driversForKey) {
     for (RCTEventAnimation *driver in driversForKey) {
       [self stopAnimationsForNode:driver.valueNode];
       [driver updateWithEvent:event];
@@ -439,8 +439,8 @@
 
 - (void)startAnimationLoopIfNeeded
 {
-  if ((_displayLink == nullptr) && _activeAnimations.count > 0) {
-    _displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(stepAnimations:)];
+  if (!_displayLink && _activeAnimations.count > 0) {
+    _displayLink = [RCTPlatformDisplayLink displayLinkWithTarget:self selector:@selector(stepAnimations:)]; // [macOS]
     [_displayLink addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
   }
 }
@@ -454,13 +454,13 @@
 
 - (void)stopAnimationLoop
 {
-  if (_displayLink != nullptr) {
+  if (_displayLink) {
     [_displayLink invalidate];
     _displayLink = nil;
   }
 }
 
-- (void)stepAnimations:(CADisplayLink *)displaylink
+- (void)stepAnimations:(RCTPlatformDisplayLink *)displaylink // [macOS]
 {
   NSTimeInterval time = displaylink.timestamp;
   for (id<RCTAnimationDriver> animationDriver in _activeAnimations) {
@@ -486,7 +486,7 @@
   NSArray<RCTEventAnimation *> *eventAnimations = _eventDrivers[key];
   for (RCTEventAnimation *animation in eventAnimations) {
     NSNumber *nodeTag = [animation.valueNode nodeTag];
-    if (nodeTag != nullptr) {
+    if (nodeTag) {
       [tags addObject:nodeTag];
     }
     for (NSNumber *childNodeKey in [animation.valueNode childNodes]) {
