--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/NativeAnimation/RCTNativeAnimatedNodesManager.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/NativeAnimation/RCTNativeAnimatedNodesManager.h	2026-03-06 15:22:52
@@ -8,6 +8,7 @@
 #import <Foundation/Foundation.h>
 #import <React/RCTBridgeModule.h>
 #import <React/RCTEventDispatcherProtocol.h>
+#import <React/RCTPlatformDisplayLink.h> // [macOS]
 #import <React/RCTSurfacePresenterStub.h>
 #import <React/RCTUIManager.h>
 
@@ -22,7 +23,7 @@
 
 - (void)updateAnimations;
 
-- (void)stepAnimations:(CADisplayLink *)displaylink;
+- (void)stepAnimations:(RCTPlatformDisplayLink *)displaylink; // [macOS]
 
 - (BOOL)isNodeManagedByFabric:(NSNumber *)tag;
 
