--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/EventEmitter/NativeEventEmitter.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/EventEmitter/NativeEventEmitter.js	2026-03-06 15:22:52
@@ -53,7 +53,8 @@
   _nativeModule: ?NativeModule;
 
   constructor(nativeModule?: ?NativeModule) {
-    if (Platform.OS === 'ios') {
+    if (Platform.OS === 'ios' || Platform.OS === 'macos') {
+      // [macOS]
       invariant(
         nativeModule != null,
         '`new NativeEventEmitter()` requires a non-null argument.',
@@ -85,8 +86,8 @@
 
   addListener<TEvent: $Keys<TEventToArgsMap>>(
     eventType: TEvent,
-    listener: (...args: TEventToArgsMap[TEvent]) => unknown,
-    context?: unknown,
+    listener: (...args: TEventToArgsMap[TEvent]) => mixed,
+    context?: mixed,
   ): EventSubscription {
     this._nativeModule?.addListener(eventType);
     let subscription: ?EventSubscription = RCTDeviceEventEmitter.addListener(
