--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/Text/NSTextStorage+FontScaling.m	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/Text/NSTextStorage+FontScaling.m	2026-03-06 15:22:52
@@ -122,7 +122,7 @@
 
                   CGFloat fontSize = MAX(MIN(font.pointSize * ratio, maximumFontSize), minimumFontSize);
 
-                  [self addAttribute:NSFontAttributeName value:[font fontWithSize:fontSize] range:range];
+                  [self addAttribute:NSFontAttributeName value:UIFontWithSize(font, fontSize) /* [macOS] */ range:range];
                 }];
 
   [self endEditing];
