--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/Text/RCTDynamicTypeRamp.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/Text/RCTDynamicTypeRamp.mm	2026-03-06 15:22:52
@@ -29,6 +29,7 @@
 
 @end
 
+#if !TARGET_OS_OSX // [macOS]
 UIFontMetrics *RCTUIFontMetricsForDynamicTypeRamp(RCTDynamicTypeRamp dynamicTypeRamp)
 {
   static NSDictionary<NSNumber *, UIFontTextStyle> *mapping;
@@ -80,3 +81,4 @@
       mapping[@(dynamicTypeRamp)] ?: @17; // Default to body size if we don't recognize the specified ramp
   return [baseSize doubleValue];
 }
+#endif // [macOS]
