--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/Text/RCTDynamicTypeRamp.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/Text/RCTDynamicTypeRamp.h	2026-03-06 15:22:52
@@ -30,8 +30,10 @@
 
 @end
 
+#if !TARGET_OS_OSX // [macOS]
 /// Generates a `UIFontMetrics` instance representing a particular Dynamic Type ramp.
 UIFontMetrics *_Nonnull RCTUIFontMetricsForDynamicTypeRamp(RCTDynamicTypeRamp dynamicTypeRamp);
 /// The "reference" size for a particular font scale ramp, equal to a text element's size under default text size
 /// settings.
 CGFloat RCTBaseSizeForDynamicTypeRamp(RCTDynamicTypeRamp dynamicTypeRamp);
+#endif // [macOS]
