--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/Text/RCTTextViewManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/Text/RCTTextViewManager.mm	2026-03-06 15:22:52
@@ -7,8 +7,6 @@
 
 #import <React/RCTTextViewManager.h>
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTShadowView+Layout.h>
 #import <React/RCTShadowView.h>
 #import <React/RCTUIManager.h>
@@ -44,23 +42,27 @@
 
   [bridge.uiManager.observerCoordinator addObserver:self];
 
+#if !TARGET_OS_OSX // [macOS]
   [[NSNotificationCenter defaultCenter] addObserver:self
                                            selector:@selector(handleDidUpdateMultiplierNotification)
                                                name:@"RCTAccessibilityManagerDidUpdateMultiplierNotification"
                                              object:[bridge moduleForName:@"AccessibilityManager"
                                                         lazilyLoadIfNecessary:YES]];
+#endif // [macOS]
 }
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
-  return [RCTTextView new];
+  return [[RCTTextView alloc] initWithEventDispatcher:self.bridge.eventDispatcher]; // [macOS]
 }
 
 - (RCTShadowView *)shadowView
 {
   RCTTextShadowView *shadowView = [[RCTTextShadowView alloc] initWithBridge:self.bridge];
+#if !TARGET_OS_OSX // [macOS]
   shadowView.textAttributes.fontSizeMultiplier =
       [[[self.bridge moduleForName:@"AccessibilityManager"] valueForKey:@"multiplier"] floatValue];
+#endif // [macOS]
   [_shadowViews addObject:shadowView];
   return shadowView;
 }
@@ -93,5 +95,3 @@
 }
 
 @end
-
-#endif // RCT_REMOVE_LEGACY_ARCH
