--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/Text/RCTTextView.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/Text/RCTTextView.h	2026-03-06 15:22:52
@@ -6,21 +6,21 @@
  */
 
 #import <React/RCTComponent.h>
+#import <React/RCTEventDispatcher.h> // [macOS]
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
+#import <React/RCTUIKit.h> // [macOS]
 
-#import <UIKit/UIKit.h>
-
 NS_ASSUME_NONNULL_BEGIN
 
-__attribute__((deprecated("This API will be removed along with the legacy architecture.")))
-@interface RCTTextView : UIView
+@interface RCTTextView : RCTUIView // [macOS]
 
+- (instancetype)initWithEventDispatcher:(id<RCTEventDispatcherProtocol>)eventDispatcher; // [macOS]
+
 @property (nonatomic, assign) BOOL selectable;
 
 - (void)setTextStorage:(NSTextStorage *)textStorage
           contentFrame:(CGRect)contentFrame
-       descendantViews:(NSArray<UIView *> *)descendantViews;
+       descendantViews:(NSArray<RCTPlatformView *> *)descendantViews; // [macOS]
 
 /**
  * (Experimental and unused for Paper) Pointer event handlers.
@@ -30,5 +30,3 @@
 @end
 
 NS_ASSUME_NONNULL_END
-
-#endif // RCT_REMOVE_LEGACY_ARCH
