--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Linking/Linking.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Linking/Linking.js	2026-03-06 15:22:52
@@ -23,7 +23,11 @@
 
 class LinkingImpl extends NativeEventEmitter<LinkingEventDefinitions> {
   constructor() {
-    super(Platform.OS === 'ios' ? nullthrows(NativeLinkingManager) : undefined);
+    super(
+      Platform.OS === 'ios' || Platform.OS === 'macos' // [macOS]
+        ? nullthrows(NativeLinkingManager)
+        : undefined,
+    );
   }
 
   /**
@@ -34,7 +38,7 @@
    */
   addEventListener<K: $Keys<LinkingEventDefinitions>>(
     eventType: K,
-    listener: (...LinkingEventDefinitions[K]) => unknown,
+    listener: (...LinkingEventDefinitions[K]) => mixed,
   ): EventSubscription {
     return this.addListener(eventType, listener);
   }
