--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/jsc/JSCRuntime.cpp	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/jsc/JSCRuntime.cpp	2026-03-06 15:22:53
@@ -27,14 +27,18 @@
 class JSCRuntime;
 
 struct Lock {
-  void lock(const jsc::JSCRuntime& /*unused*/) const {}
-  void unlock(const jsc::JSCRuntime& /*unused*/) const {}
+  void lock(const jsc::JSCRuntime&) const {}
+  void unlock(const jsc::JSCRuntime&) const {}
 };
 
 class JSCRuntime : public jsi::Runtime {
  public:
   // Creates new context in new context group
   JSCRuntime();
+  // [macOS
+  // Creates new context in new context group with config
+  JSCRuntime(const facebook::jsc::RuntimeConfig& rc);
+  // macOS]
   // Retains ctx
   JSCRuntime(JSGlobalContextRef ctx);
   ~JSCRuntime();
@@ -158,88 +162,78 @@
       override;
   jsi::PropNameID createPropNameIDFromString(const jsi::String& str) override;
   jsi::PropNameID createPropNameIDFromSymbol(const jsi::Symbol& sym) override;
-  std::string utf8(const jsi::PropNameID& /*sym*/) override;
-  bool compare(const jsi::PropNameID& /*a*/, const jsi::PropNameID& /*b*/)
-      override;
+  std::string utf8(const jsi::PropNameID&) override;
+  bool compare(const jsi::PropNameID&, const jsi::PropNameID&) override;
 
-  std::string symbolToString(const jsi::Symbol& /*sym*/) override;
+  std::string symbolToString(const jsi::Symbol&) override;
 
-  jsi::BigInt createBigIntFromInt64(int64_t /*unused*/) override;
-  jsi::BigInt createBigIntFromUint64(uint64_t /*unused*/) override;
-  bool bigintIsInt64(const jsi::BigInt& /*unused*/) override;
-  bool bigintIsUint64(const jsi::BigInt& /*unused*/) override;
-  uint64_t truncate(const jsi::BigInt& /*unused*/) override;
-  jsi::String bigintToString(const jsi::BigInt& /*unused*/, int /*unused*/)
-      override;
+  jsi::BigInt createBigIntFromInt64(int64_t) override;
+  jsi::BigInt createBigIntFromUint64(uint64_t) override;
+  bool bigintIsInt64(const jsi::BigInt&) override;
+  bool bigintIsUint64(const jsi::BigInt&) override;
+  uint64_t truncate(const jsi::BigInt&) override;
+  jsi::String bigintToString(const jsi::BigInt&, int) override;
 
   jsi::String createStringFromAscii(const char* str, size_t length) override;
   jsi::String createStringFromUtf8(const uint8_t* utf8, size_t length) override;
-  std::string utf8(const jsi::String& /*str*/) override;
+  std::string utf8(const jsi::String&) override;
 
   jsi::Object createObject() override;
   jsi::Object createObject(std::shared_ptr<jsi::HostObject> ho) override;
   virtual std::shared_ptr<jsi::HostObject> getHostObject(
-      const jsi::Object& /*obj*/) override;
-  jsi::HostFunctionType& getHostFunction(const jsi::Function& /*obj*/) override;
+      const jsi::Object&) override;
+  jsi::HostFunctionType& getHostFunction(const jsi::Function&) override;
 
-  bool hasNativeState(const jsi::Object& /*obj*/) override;
-  std::shared_ptr<jsi::NativeState> getNativeState(
-      const jsi::Object& /*obj*/) override;
-  void setNativeState(
-      const jsi::Object& /*obj*/,
-      std::shared_ptr<jsi::NativeState> /*nativeState*/) override;
+  bool hasNativeState(const jsi::Object&) override;
+  std::shared_ptr<jsi::NativeState> getNativeState(const jsi::Object&) override;
+  void setNativeState(const jsi::Object&, std::shared_ptr<jsi::NativeState>)
+      override;
 
-  jsi::Value getProperty(const jsi::Object& /*obj*/, const jsi::String& name)
+  jsi::Value getProperty(const jsi::Object&, const jsi::String& name) override;
+  jsi::Value getProperty(const jsi::Object&, const jsi::PropNameID& name)
       override;
-  jsi::Value getProperty(
-      const jsi::Object& /*obj*/,
-      const jsi::PropNameID& name) override;
-  bool hasProperty(const jsi::Object& /*obj*/, const jsi::String& name)
-      override;
-  bool hasProperty(const jsi::Object& /*obj*/, const jsi::PropNameID& name)
-      override;
+  bool hasProperty(const jsi::Object&, const jsi::String& name) override;
+  bool hasProperty(const jsi::Object&, const jsi::PropNameID& name) override;
   void setPropertyValue(
-      const jsi::Object& /*object*/,
+      const jsi::Object&,
       const jsi::String& name,
       const jsi::Value& value) override;
   void setPropertyValue(
-      const jsi::Object& /*object*/,
+      const jsi::Object&,
       const jsi::PropNameID& name,
       const jsi::Value& value) override;
-  bool isArray(const jsi::Object& /*obj*/) const override;
-  bool isArrayBuffer(const jsi::Object& /*obj*/) const override;
-  bool isFunction(const jsi::Object& /*obj*/) const override;
-  bool isHostObject(const jsi::Object& /*obj*/) const override;
-  bool isHostFunction(const jsi::Function& /*obj*/) const override;
-  jsi::Array getPropertyNames(const jsi::Object& /*obj*/) override;
+  bool isArray(const jsi::Object&) const override;
+  bool isArrayBuffer(const jsi::Object&) const override;
+  bool isFunction(const jsi::Object&) const override;
+  bool isHostObject(const jsi::Object&) const override;
+  bool isHostFunction(const jsi::Function&) const override;
+  jsi::Array getPropertyNames(const jsi::Object&) override;
 
   // TODO: revisit this implementation
-  jsi::WeakObject createWeakObject(const jsi::Object& /*obj*/) override;
-  jsi::Value lockWeakObject(const jsi::WeakObject& /*obj*/) override;
+  jsi::WeakObject createWeakObject(const jsi::Object&) override;
+  jsi::Value lockWeakObject(const jsi::WeakObject&) override;
 
   jsi::Array createArray(size_t length) override;
   jsi::ArrayBuffer createArrayBuffer(
       std::shared_ptr<jsi::MutableBuffer> buffer) override;
-  size_t size(const jsi::Array& /*arr*/) override;
-  size_t size(const jsi::ArrayBuffer& /*obj*/) override;
-  uint8_t* data(const jsi::ArrayBuffer& /*obj*/) override;
-  jsi::Value getValueAtIndex(const jsi::Array& /*arr*/, size_t i) override;
-  void setValueAtIndexImpl(
-      const jsi::Array& /*arr*/,
-      size_t i,
-      const jsi::Value& value) override;
+  size_t size(const jsi::Array&) override;
+  size_t size(const jsi::ArrayBuffer&) override;
+  uint8_t* data(const jsi::ArrayBuffer&) override;
+  jsi::Value getValueAtIndex(const jsi::Array&, size_t i) override;
+  void setValueAtIndexImpl(const jsi::Array&, size_t i, const jsi::Value& value)
+      override;
 
   jsi::Function createFunctionFromHostFunction(
       const jsi::PropNameID& name,
       unsigned int paramCount,
       jsi::HostFunctionType func) override;
   jsi::Value call(
-      const jsi::Function& /*f*/,
+      const jsi::Function&,
       const jsi::Value& jsThis,
       const jsi::Value* args,
       size_t count) override;
   jsi::Value callAsConstructor(
-      const jsi::Function& /*f*/,
+      const jsi::Function&,
       const jsi::Value* args,
       size_t count) override;
 
@@ -248,8 +242,7 @@
   bool strictEquals(const jsi::String& a, const jsi::String& b) const override;
   bool strictEquals(const jsi::Object& a, const jsi::Object& b) const override;
   bool instanceOf(const jsi::Object& o, const jsi::Function& f) override;
-  void setExternalMemoryPressure(const jsi::Object& /*obj*/, size_t /*amount*/)
-      override;
+  void setExternalMemoryPressure(const jsi::Object&, size_t) override;
 
  private:
   // Basically convenience casts
@@ -308,11 +301,18 @@
     }                          \
   } while (0)
 
-#if defined(__IPHONE_OS_VERSION_MIN_REQUIRED)
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) // [macOS]
 #if __IPHONE_OS_VERSION_MAX_ALLOWED >= 160400
 #define _JSC_HAS_INSPECTABLE
 #endif
 #endif
+// [macOS
+#if defined(__MAC_OS_X_VERSION_MAX_ALLOWED)
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 130300
+#define _JSC_HAS_INSPECTABLE
+#endif
+#endif
+// macOS]
 
 // JSStringRef utilities
 namespace {
@@ -331,7 +331,7 @@
     buffer = heapBuffer.get();
   }
   size_t actualBytes = JSStringGetUTF8CString(str, buffer, maxBytes);
-  if (actualBytes == 0u) {
+  if (!actualBytes) {
     // Happens if maxBytes == 0 (never the case here) or if str contains
     // invalid UTF-16 data, since JSStringGetUTF8CString attempts a strict
     // conversion.
@@ -374,7 +374,20 @@
 JSCRuntime::JSCRuntime()
     : JSCRuntime(JSGlobalContextCreateInGroup(nullptr, nullptr)) {
   JSGlobalContextRelease(ctx_);
+}
+
+// [macOS
+JSCRuntime::JSCRuntime(const facebook::jsc::RuntimeConfig& rc)
+	: JSCRuntime() {
+#ifdef _JSC_HAS_INSPECTABLE
+  if (__builtin_available(macOS 13.3, iOS 16.4, tvOS 16.4, *)) {
+    JSGlobalContextSetInspectable(ctx_, rc.enableDebugger);
+  }
+#endif
+  JSGlobalContextSetName(ctx_, JSStringCreateWithUTF8CString(rc.debuggerName.c_str()));
+
 }
+// macOS]
 
 JSCRuntime::JSCRuntime(JSGlobalContextRef ctx)
     : ctx_(JSGlobalContextRetain(ctx)),
@@ -448,7 +461,7 @@
   JSValueRef res =
       JSEvaluateScript(ctx_, sourceRef, nullptr, sourceURLRef, 0, &exc);
   JSStringRelease(sourceRef);
-  if (sourceURLRef != nullptr) {
+  if (sourceURLRef) {
     JSStringRelease(sourceURLRef);
   }
   checkException(res, exc);
@@ -608,7 +621,7 @@
 
 jsi::Runtime::PointerValue* JSCRuntime::cloneSymbol(
     const jsi::Runtime::PointerValue* pv) {
-  if (pv == nullptr) {
+  if (!pv) {
     return nullptr;
   }
   const JSCSymbolValue* symbol = static_cast<const JSCSymbolValue*>(pv);
@@ -616,13 +629,13 @@
 }
 
 jsi::Runtime::PointerValue* JSCRuntime::cloneBigInt(
-    const Runtime::PointerValue* /*pv*/) {
+    const Runtime::PointerValue*) {
   throw std::logic_error("Not implemented");
 }
 
 jsi::Runtime::PointerValue* JSCRuntime::cloneString(
     const jsi::Runtime::PointerValue* pv) {
-  if (pv == nullptr) {
+  if (!pv) {
     return nullptr;
   }
   const JSCStringValue* string = static_cast<const JSCStringValue*>(pv);
@@ -631,7 +644,7 @@
 
 jsi::Runtime::PointerValue* JSCRuntime::cloneObject(
     const jsi::Runtime::PointerValue* pv) {
-  if (pv == nullptr) {
+  if (!pv) {
     return nullptr;
   }
   const JSCObjectValue* object = static_cast<const JSCObjectValue*>(pv);
@@ -643,7 +656,7 @@
 
 jsi::Runtime::PointerValue* JSCRuntime::clonePropNameID(
     const jsi::Runtime::PointerValue* pv) {
-  if (pv == nullptr) {
+  if (!pv) {
     return nullptr;
   }
   const JSCStringValue* string = static_cast<const JSCStringValue*>(pv);
@@ -694,29 +707,27 @@
   return jsi::Value(*this, sym).toString(*this).utf8(*this);
 }
 
-jsi::BigInt JSCRuntime::createBigIntFromInt64(int64_t /*unused*/) {
+jsi::BigInt JSCRuntime::createBigIntFromInt64(int64_t) {
   throw std::logic_error("Not implemented");
 }
 
-jsi::BigInt JSCRuntime::createBigIntFromUint64(uint64_t /*unused*/) {
+jsi::BigInt JSCRuntime::createBigIntFromUint64(uint64_t) {
   throw std::logic_error("Not implemented");
 }
 
-bool JSCRuntime::bigintIsInt64(const jsi::BigInt& /*unused*/) {
+bool JSCRuntime::bigintIsInt64(const jsi::BigInt&) {
   throw std::logic_error("Not implemented");
 }
 
-bool JSCRuntime::bigintIsUint64(const jsi::BigInt& /*unused*/) {
+bool JSCRuntime::bigintIsUint64(const jsi::BigInt&) {
   throw std::logic_error("Not implemented");
 }
 
-uint64_t JSCRuntime::truncate(const jsi::BigInt& /*unused*/) {
+uint64_t JSCRuntime::truncate(const jsi::BigInt&) {
   throw std::logic_error("Not implemented");
 }
 
-jsi::String JSCRuntime::bigintToString(
-    const jsi::BigInt& /*unused*/,
-    int /*unused*/) {
+jsi::String JSCRuntime::bigintToString(const jsi::BigInt&, int) {
   throw std::logic_error("Not implemented");
 }
 
@@ -927,7 +938,7 @@
 } // namespace
 
 JSValueRef JSCRuntime::getNativeStateSymbol() {
-  if (nativeStateSymbol_ == nullptr) {
+  if (!nativeStateSymbol_) {
     JSStringRef symbolName =
         JSStringCreateWithUTF8CString("__internal_nativeState");
     JSValueRef symbol = JSValueMakeSymbol(ctx_, symbolName);
@@ -1060,14 +1071,7 @@
 }
 
 bool JSCRuntime::isArray(const jsi::Object& obj) const {
-  // JSValueIsArray only returns true if the object is a JS array. There's no
-  // public JSC API equivalent to Array.isArray, so we call the JS function
-  // through JSI directly.
-  auto constRt = const_cast<JSCRuntime*>(this);
-  auto isArrayFn = constRt->global()
-                       .getPropertyAsObject(*constRt, "Array")
-                       .getPropertyAsFunction(*constRt, "isArray");
-  return isArrayFn.call(*constRt, obj).asBool();
+  return JSValueIsArray(ctx_, objectRef(obj));
 }
 
 bool JSCRuntime::isArrayBuffer(const jsi::Object& obj) const {
@@ -1202,7 +1206,7 @@
           kJSPropertyAttributeReadOnly | kJSPropertyAttributeDontEnum |
               kJSPropertyAttributeDontDelete,
           &exc);
-      if (exc != nullptr) {
+      if (exc) {
         // Silently fail to set length
         exc = nullptr;
       }
@@ -1218,7 +1222,7 @@
               kJSPropertyAttributeDontDelete,
           &exc);
       JSStringRelease(name);
-      if (exc != nullptr) {
+      if (exc) {
         // Silently fail to set name
         exc = nullptr;
       }
@@ -1231,7 +1235,7 @@
         abort();
       }
       JSObjectRef funcCtor = JSValueToObject(ctx, value, &exc);
-      if (funcCtor == nullptr) {
+      if (!funcCtor) {
         // We can't do anything if Function is not an object
         return;
       }
@@ -1439,9 +1443,7 @@
   return res;
 }
 
-void JSCRuntime::setExternalMemoryPressure(
-    const jsi::Object& /*obj*/,
-    size_t /*amount*/) {}
+void JSCRuntime::setExternalMemoryPressure(const jsi::Object&, size_t) {}
 
 jsi::Runtime::PointerValue* JSCRuntime::makeSymbolValue(
     JSValueRef symbolRef) const {
@@ -1461,7 +1463,7 @@
 
 jsi::Runtime::PointerValue* JSCRuntime::makeStringValue(
     JSStringRef stringRef) const {
-  if (stringRef == nullptr) {
+  if (!stringRef) {
     stringRef = getEmptyString();
   }
 #ifndef NDEBUG
@@ -1485,7 +1487,7 @@
 
 jsi::Runtime::PointerValue* JSCRuntime::makeObjectValue(
     JSObjectRef objectRef) const {
-  if (objectRef == nullptr) {
+  if (!objectRef) {
     objectRef = JSObjectMake(ctx_, nullptr, nullptr);
   }
 #ifndef NDEBUG
@@ -1523,7 +1525,6 @@
     }
     case kJSTypeSymbol:
       return jsi::Value(createSymbol(value));
-    case kJSTypeBigInt:
     default:
       // WHAT ARE YOU
       abort();
@@ -1604,4 +1605,10 @@
   return std::make_unique<JSCRuntime>();
 }
 
+// [macOS
+std::unique_ptr<jsi::Runtime> makeJSCRuntime(const facebook::jsc::RuntimeConfig& rc) {
+  return std::make_unique<JSCRuntime>(rc);
+}
+// macOS]
+
 } // namespace facebook::jsc
