--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/jsc/JSCRuntime.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/jsc/JSCRuntime.h	2026-03-06 15:22:53
@@ -12,6 +12,15 @@
 
 namespace facebook::jsc {
 
+// [macOS
+struct RuntimeConfig {
+  bool enableDebugger;
+  std::string debuggerName;
+};
+// macOS]
+
 std::unique_ptr<jsi::Runtime> makeJSCRuntime();
 
+std::unique_ptr<jsi::Runtime> makeJSCRuntime(const facebook::jsc::RuntimeConfig& rc); // [macOS]
+
 } // namespace facebook::jsc
