--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Utils/RCTLinearGradient.mm	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.9kiVvyuxXG	2026-03-06 18:50:51
@@ -7,8 +7,7 @@
 
 #import "RCTLinearGradient.h"
 
-#import <React/RCTAnimationUtils.h>
-#import <React/RCTConversions.h>
+#import <QuartzCore/QuartzCore.h> // [macOS]
 #include <react/renderer/graphics/ValueUnit.h>
 #import <react/utils/FloatComparison.h>
 #import "RCTGradientUtils.h"
