--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Utils/RCTReactTaggedView.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.rbIhZltvyt	2026-03-06 18:50:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 NS_ASSUME_NONNULL_BEGIN
 
@@ -16,14 +16,14 @@
  * changed from the one provided at initialization time (i.e. recycled).
  */
 @interface RCTReactTaggedView : NSObject {
-  UIView *_view;
+  RCTPlatformView *_view; // [macOS]
   NSInteger _tag;
 }
 
-+ (RCTReactTaggedView *)wrap:(UIView *)view;
++ (RCTReactTaggedView *)wrap:(RCTPlatformView *)view; // [macOS]
 
-- (instancetype)initWithView:(UIView *)view;
-- (nullable UIView *)view;
+- (instancetype)initWithView:(RCTPlatformView *)view; // [macOS]
+- (nullable RCTPlatformView *)view; // [macOS]
 - (NSInteger)tag;
 
 - (BOOL)isEqual:(id)other;
