--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/graphics/platform/ios/react/renderer/graphics/RCTPlatformColorUtils.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/graphics/platform/ios/react/renderer/graphics/RCTPlatformColorUtils.h	2026-03-06 15:22:53
@@ -7,7 +7,7 @@
 
 #pragma once
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h>
 #import <vector>
 
 namespace facebook::react {
@@ -15,6 +15,8 @@
 struct Color;
 } // namespace facebook::react
 
-facebook::react::ColorComponents RCTPlatformColorComponentsFromSemanticItems(std::vector<std::string> &semanticItems);
-UIColor *RCTPlatformColorFromSemanticItems(std::vector<std::string> &semanticItems);
-UIColor *RCTPlatformColorFromColor(const facebook::react::Color &color);
+facebook::react::ColorComponents RCTPlatformColorComponentsFromSemanticItems(
+    std::vector<std::string>& semanticItems);
+RCTUIColor* RCTPlatformColorFromSemanticItems( // [macOS]
+    std::vector<std::string>& semanticItems);
+RCTUIColor* RCTPlatformColorFromColor(const facebook::react::Color& color); // [macOS]
