--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Utils/RCTReactTaggedView.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Utils/RCTReactTaggedView.mm	2026-03-06 15:22:52
@@ -7,14 +7,16 @@
 
 #import "RCTReactTaggedView.h"
 
+#import <React/RCTUIKit.h>
+
 @implementation RCTReactTaggedView
 
-+ (RCTReactTaggedView *)wrap:(UIView *)view
++ (RCTReactTaggedView *)wrap:(RCTPlatformView *)view // [macOS]
 {
   return [[RCTReactTaggedView alloc] initWithView:view];
 }
 
-- (instancetype)initWithView:(UIView *)view
+- (instancetype)initWithView:(RCTPlatformView *)view // [macOS]
 {
   if (self = [super init]) {
     _view = view;
@@ -23,7 +25,7 @@
   return self;
 }
 
-- (nullable UIView *)view
+- (nullable RCTPlatformView *)view // [macOS]
 {
   if (_view.tag == _tag) {
     return _view;
