--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Components/Keyboard/Keyboard.js	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Components/Keyboard/Keyboard.js	2026-03-06 15:22:52
@@ -110,7 +110,9 @@
     new NativeEventEmitter(
       // T88715063: NativeEventEmitter only used this parameter on iOS. Now it uses it on all platforms, so this code was modified automatically to preserve its behavior
       // If you want to use the native module on other platforms, please remove this condition and test its behavior
-      Platform.OS !== 'ios' ? null : NativeKeyboardObserver,
+      Platform.OS !== 'ios' && Platform.OS !== 'macos' // [macOS] Also use this parameter on macOS
+        ? null
+        : NativeKeyboardObserver,
     );
 
   constructor() {
@@ -148,8 +150,8 @@
    */
   addListener<K: $Keys<KeyboardEventDefinitions>>(
     eventType: K,
-    listener: (...KeyboardEventDefinitions[K]) => unknown,
-    context?: unknown,
+    listener: (...KeyboardEventDefinitions[K]) => mixed,
+    context?: mixed,
   ): EventSubscription {
     return this._emitter.addListener(eventType, listener);
   }
