--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/LegacyViewManagerInterop/RCTLegacyViewManagerInteropComponentView.mm	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.KfSWKhHsCD	2026-03-06 18:50:49
@@ -23,7 +23,7 @@
 
 @implementation RCTLegacyViewManagerInteropComponentView {
   NSMutableArray<NSDictionary *> *_viewsToBeMounted;
-  NSMutableArray<UIView *> *_viewsToBeUnmounted;
+  NSMutableArray<RCTUIView *> *_viewsToBeUnmounted; // [macOS]
   RCTLegacyViewManagerInteropCoordinatorAdapter *_adapter;
   LegacyViewManagerInteropShadowNode::ConcreteState::Shared _state;
   BOOL _hasInvokedForwardingWarning;
@@ -168,7 +168,7 @@
   [super prepareForRecycle];
 }
 
-- (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
+- (void)mountChildComponentView:(RCTUIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index // [macOS]
 {
   if (_adapter && index == _adapter.paperView.reactSubviews.count) {
     // This is a new child view that is being added to the end of the children array.
@@ -176,7 +176,7 @@
     // Without this change, the new child will not be rendered right away because the didUpdateReactSubviews is not
     // called and the `finalizeUpdate` is not invoked.
     if ([childComponentView isKindOfClass:[RCTLegacyViewManagerInteropComponentView class]]) {
-      UIView *target = ((RCTLegacyViewManagerInteropComponentView *)childComponentView).contentView;
+      RCTUIView *target = ((RCTLegacyViewManagerInteropComponentView *)childComponentView).contentView; // [macOS]
       [_adapter.paperView insertReactSubview:target atIndex:index];
     } else {
       [_adapter.paperView insertReactSubview:childComponentView atIndex:index];
@@ -190,7 +190,7 @@
   }
 }
 
-- (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
+- (void)unmountChildComponentView:(RCTUIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index // [macOS]
 {
   if (_adapter && index < _adapter.paperView.reactSubviews.count) {
     [_adapter.paperView removeReactSubview:_adapter.paperView.reactSubviews[index]];
@@ -224,13 +224,12 @@
   };
 
   if (!_adapter) {
-    _adapter = [[RCTLegacyViewManagerInteropCoordinatorAdapter alloc] initWithCoordinator:[self _coordinator]
+#if !TARGET_OS_OSX // [macOS]
                                                                                  reactTag:self.tag];
-    _adapter.eventInterceptor = ^(std::string eventName, folly::dynamic &&event) {
-      if (weakSelf) {
-        __typeof(self) strongSelf = weakSelf;
-        const auto &eventEmitter = static_cast<const ViewEventEmitter &>(*strongSelf->_eventEmitter);
-        eventEmitter.dispatchEvent(eventName, std::move(event));
+#else // [macOS
+    _adapter = [[RCTLegacyViewManagerInteropCoordinatorAdapter alloc] initWithCoordinator:[self _coordinator]
+                                                                                 reactTag:self.reactTag.integerValue];
+#endif // macOS]
       }
     };
     // Set props immediately. This is required to set the initial state of the view.
@@ -245,9 +244,9 @@
 
   for (NSDictionary *mountInstruction in _viewsToBeMounted) {
     NSNumber *index = mountInstruction[kRCTLegacyInteropChildIndexKey];
-    UIView *childView = mountInstruction[kRCTLegacyInteropChildComponentKey];
+    RCTUIView *childView = mountInstruction[kRCTLegacyInteropChildComponentKey]; // [macOS]
     if ([childView isKindOfClass:[RCTLegacyViewManagerInteropComponentView class]]) {
-      UIView *target = ((RCTLegacyViewManagerInteropComponentView *)childView).contentView;
+      RCTPlatformView *target = ((RCTLegacyViewManagerInteropComponentView *)childView).contentView; // [macOS]
       [_adapter.paperView insertReactSubview:target atIndex:index.integerValue];
     } else {
       [_adapter.paperView insertReactSubview:childView atIndex:index.integerValue];
@@ -256,7 +255,7 @@
 
   [_viewsToBeMounted removeAllObjects];
 
-  for (UIView *view in _viewsToBeUnmounted) {
+  for (RCTUIView *view in _viewsToBeUnmounted) { // [macOS]
     [_adapter.paperView removeReactSubview:view];
   }
 
@@ -275,8 +274,7 @@
   }
 }
 
-- (UIView *)paperView
-{
+- (RCTPlatformView *)paperView // [macOS]
   return _adapter.paperView;
 }
 
