--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Core/setUpAlert.js	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.yef2JujU8J	2026-03-06 18:50:42
@@ -18,6 +18,7 @@
   global.alert = function (text: string) {
     // Require Alert on demand. Requiring it too early can lead to issues
     // with things like Platform not being fully initialized.
+    // @flow // [macOS]
     require('../Alert/Alert').default.alert('Alert', '' + text);
   };
 }
