--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/LegacyViewManagerInterop/RCTLegacyViewManagerInteropComponentView.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/ComponentViews/LegacyViewManagerInterop/RCTLegacyViewManagerInteropComponentView.mm	2026-03-06 15:22:52
@@ -23,7 +23,7 @@
 
 @implementation RCTLegacyViewManagerInteropComponentView {
   NSMutableArray<NSDictionary *> *_viewsToBeMounted;
-  NSMutableArray<UIView *> *_viewsToBeUnmounted;
+  NSMutableArray<RCTUIView *> *_viewsToBeUnmounted; // [macOS]
   RCTLegacyViewManagerInteropCoordinatorAdapter *_adapter;
   LegacyViewManagerInteropShadowNode::ConcreteState::Shared _state;
   BOOL _hasInvokedForwardingWarning;
@@ -168,7 +168,7 @@
   [super prepareForRecycle];
 }
 
-- (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
+- (void)mountChildComponentView:(RCTUIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index // [macOS]
 {
   if (_adapter && index == _adapter.paperView.reactSubviews.count) {
     // This is a new child view that is being added to the end of the children array.
@@ -176,7 +176,7 @@
     // Without this change, the new child will not be rendered right away because the didUpdateReactSubviews is not
     // called and the `finalizeUpdate` is not invoked.
     if ([childComponentView isKindOfClass:[RCTLegacyViewManagerInteropComponentView class]]) {
-      UIView *target = ((RCTLegacyViewManagerInteropComponentView *)childComponentView).contentView;
+      RCTUIView *target = ((RCTLegacyViewManagerInteropComponentView *)childComponentView).contentView; // [macOS]
       [_adapter.paperView insertReactSubview:target atIndex:index];
     } else {
       [_adapter.paperView insertReactSubview:childComponentView atIndex:index];
@@ -190,7 +190,7 @@
   }
 }
 
-- (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
+- (void)unmountChildComponentView:(RCTUIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index // [macOS]
 {
   if (_adapter && index < _adapter.paperView.reactSubviews.count) {
     [_adapter.paperView removeReactSubview:_adapter.paperView.reactSubviews[index]];
@@ -224,8 +224,13 @@
   };
 
   if (!_adapter) {
+#if !TARGET_OS_OSX // [macOS]
     _adapter = [[RCTLegacyViewManagerInteropCoordinatorAdapter alloc] initWithCoordinator:[self _coordinator]
                                                                                  reactTag:self.tag];
+#else // [macOS
+    _adapter = [[RCTLegacyViewManagerInteropCoordinatorAdapter alloc] initWithCoordinator:[self _coordinator]
+                                                                                 reactTag:self.reactTag.integerValue];
+#endif // macOS]
     _adapter.eventInterceptor = ^(std::string eventName, folly::dynamic &&event) {
       if (weakSelf) {
         __typeof(self) strongSelf = weakSelf;
@@ -245,9 +250,9 @@
 
   for (NSDictionary *mountInstruction in _viewsToBeMounted) {
     NSNumber *index = mountInstruction[kRCTLegacyInteropChildIndexKey];
-    UIView *childView = mountInstruction[kRCTLegacyInteropChildComponentKey];
+    RCTUIView *childView = mountInstruction[kRCTLegacyInteropChildComponentKey]; // [macOS]
     if ([childView isKindOfClass:[RCTLegacyViewManagerInteropComponentView class]]) {
-      UIView *target = ((RCTLegacyViewManagerInteropComponentView *)childView).contentView;
+      RCTPlatformView *target = ((RCTLegacyViewManagerInteropComponentView *)childView).contentView; // [macOS]
       [_adapter.paperView insertReactSubview:target atIndex:index.integerValue];
     } else {
       [_adapter.paperView insertReactSubview:childView atIndex:index.integerValue];
@@ -256,7 +261,7 @@
 
   [_viewsToBeMounted removeAllObjects];
 
-  for (UIView *view in _viewsToBeUnmounted) {
+  for (RCTUIView *view in _viewsToBeUnmounted) { // [macOS]
     [_adapter.paperView removeReactSubview:view];
   }
 
@@ -275,7 +280,7 @@
   }
 }
 
-- (UIView *)paperView
+- (RCTPlatformView *)paperView // [macOS]
 {
   return _adapter.paperView;
 }
@@ -287,15 +292,4 @@
   [_adapter handleCommand:(NSString *)commandName args:(NSArray *)args];
 }
 
-- (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
-{
-  UIView *result = [super hitTest:point withEvent:event];
-
-  if (result == self) {
-    return nil;
-  }
-
-  return result;
-}
-
 @end
