--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/components/textinput/platform/ios/react/renderer/components/iostextinput/conversions.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/components/textinput/platform/ios/react/renderer/components/iostextinput/conversions.h	2026-03-06 15:22:53
@@ -13,8 +13,10 @@
 
 namespace facebook::react {
 
-inline void fromRawValue(const PropsParserContext &context, const RawValue &value, AutocapitalizationType &result)
-{
+inline void fromRawValue(
+    const PropsParserContext& context,
+    const RawValue& value,
+    AutocapitalizationType& result) {
   auto string = (std::string)value;
   if (string == "none") {
     result = AutocapitalizationType::None;
@@ -35,8 +37,10 @@
   result = AutocapitalizationType::None;
 }
 
-inline void fromRawValue(const PropsParserContext &context, const RawValue &value, KeyboardAppearance &result)
-{
+inline void fromRawValue(
+    const PropsParserContext& context,
+    const RawValue& value,
+    KeyboardAppearance& result) {
   auto string = (std::string)value;
   if (string == "default") {
     result = KeyboardAppearance::Default;
@@ -53,8 +57,10 @@
   result = KeyboardAppearance::Default;
 }
 
-inline void fromRawValue(const PropsParserContext &context, const RawValue &value, ReturnKeyType &result)
-{
+inline void fromRawValue(
+    const PropsParserContext& context,
+    const RawValue& value,
+    ReturnKeyType& result) {
   auto string = (std::string)value;
   if (string == "default") {
     result = ReturnKeyType::Default;
@@ -119,9 +125,10 @@
   result = ReturnKeyType::Default;
 }
 
-inline void
-fromRawValue(const PropsParserContext &context, const RawValue &value, TextInputAccessoryVisibilityMode &result)
-{
+inline void fromRawValue(
+    const PropsParserContext& context,
+    const RawValue& value,
+    TextInputAccessoryVisibilityMode& result) {
   auto string = (std::string)value;
   if (string == "never") {
     result = TextInputAccessoryVisibilityMode::Never;
@@ -142,8 +149,10 @@
   result = TextInputAccessoryVisibilityMode::Never;
 }
 
-inline void fromRawValue(const PropsParserContext &context, const RawValue &value, KeyboardType &result)
-{
+inline void fromRawValue(
+    const PropsParserContext& context,
+    const RawValue& value,
+    KeyboardType& result) {
   auto string = (std::string)value;
   if (string == "default") {
     result = KeyboardType::Default;
@@ -208,4 +217,26 @@
   result = KeyboardType::Default;
 }
 
+#if TARGET_OS_OSX // [macOS
+inline void fromRawValue(
+    const PropsParserContext &context,
+    const RawValue &value,
+    PastedTypesType &result) {
+  auto string = (std::string)value;
+  if (string == "fileUrl") {
+    result = PastedTypesType::FileUrl;
+    return;
+  }
+  if (string == "image") {
+    result = PastedTypesType::Image;
+    return;
+  }
+  if (string == "string") {
+    result = PastedTypesType::String;
+    return;
+  }
+  abort();
+}
+#endif // macOS]
+
 } // namespace facebook::react
