--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/ReactCommon/react/renderer/components/textinput/platform/ios/react/renderer/components/iostextinput/propsConversions.h	2026-03-06 15:51:55
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/ReactCommon/react/renderer/components/textinput/platform/ios/react/renderer/components/iostextinput/propsConversions.h	2026-03-06 15:22:53
@@ -15,20 +15,36 @@
 namespace facebook::react {
 
 static TextInputTraits convertRawProp(
-    const PropsParserContext &context,
-    const RawProps &rawProps,
-    const TextInputTraits &sourceTraits,
-    const TextInputTraits &defaultTraits)
-{
+    const PropsParserContext& context,
+    const RawProps& rawProps,
+    const TextInputTraits& sourceTraits,
+    const TextInputTraits& defaultTraits) {
   auto traits = TextInputTraits{};
 
   traits.autocapitalizationType = convertRawProp(
-      context, rawProps, "autoCapitalize", sourceTraits.autocapitalizationType, defaultTraits.autocapitalizationType);
-  traits.autoCorrect =
-      convertRawProp(context, rawProps, "autoCorrect", sourceTraits.autoCorrect, defaultTraits.autoCorrect);
+      context,
+      rawProps,
+      "autoCapitalize",
+      sourceTraits.autocapitalizationType,
+      defaultTraits.autocapitalizationType);
+  traits.autoCorrect = convertRawProp(
+      context,
+      rawProps,
+      "autoCorrect",
+      sourceTraits.autoCorrect,
+      defaultTraits.autoCorrect);
   traits.contextMenuHidden = convertRawProp(
-      context, rawProps, "contextMenuHidden", sourceTraits.contextMenuHidden, defaultTraits.contextMenuHidden);
-  traits.editable = convertRawProp(context, rawProps, "editable", sourceTraits.editable, defaultTraits.editable);
+      context,
+      rawProps,
+      "contextMenuHidden",
+      sourceTraits.contextMenuHidden,
+      defaultTraits.contextMenuHidden);
+  traits.editable = convertRawProp(
+      context,
+      rawProps,
+      "editable",
+      sourceTraits.editable,
+      defaultTraits.editable);
   traits.enablesReturnKeyAutomatically = convertRawProp(
       context,
       rawProps,
@@ -36,45 +52,122 @@
       sourceTraits.enablesReturnKeyAutomatically,
       defaultTraits.enablesReturnKeyAutomatically);
   traits.keyboardAppearance = convertRawProp(
-      context, rawProps, "keyboardAppearance", sourceTraits.keyboardAppearance, defaultTraits.keyboardAppearance);
-  traits.spellCheck =
-      convertRawProp(context, rawProps, "spellCheck", sourceTraits.spellCheck, defaultTraits.spellCheck);
-  traits.caretHidden =
-      convertRawProp(context, rawProps, "caretHidden", sourceTraits.caretHidden, defaultTraits.caretHidden);
-  traits.clearButtonMode =
-      convertRawProp(context, rawProps, "clearButtonMode", sourceTraits.clearButtonMode, defaultTraits.clearButtonMode);
-  traits.scrollEnabled =
-      convertRawProp(context, rawProps, "scrollEnabled", sourceTraits.scrollEnabled, defaultTraits.scrollEnabled);
-  traits.secureTextEntry =
-      convertRawProp(context, rawProps, "secureTextEntry", sourceTraits.secureTextEntry, defaultTraits.secureTextEntry);
+      context,
+      rawProps,
+      "keyboardAppearance",
+      sourceTraits.keyboardAppearance,
+      defaultTraits.keyboardAppearance);
+  traits.spellCheck = convertRawProp(
+      context,
+      rawProps,
+      "spellCheck",
+      sourceTraits.spellCheck,
+      defaultTraits.spellCheck);
+  traits.caretHidden = convertRawProp(
+      context,
+      rawProps,
+      "caretHidden",
+      sourceTraits.caretHidden,
+      defaultTraits.caretHidden);
+  traits.clearButtonMode = convertRawProp(
+      context,
+      rawProps,
+      "clearButtonMode",
+      sourceTraits.clearButtonMode,
+      defaultTraits.clearButtonMode);
+  traits.scrollEnabled = convertRawProp(
+      context,
+      rawProps,
+      "scrollEnabled",
+      sourceTraits.scrollEnabled,
+      defaultTraits.scrollEnabled);
+  traits.secureTextEntry = convertRawProp(
+      context,
+      rawProps,
+      "secureTextEntry",
+      sourceTraits.secureTextEntry,
+      defaultTraits.secureTextEntry);
   traits.clearTextOnFocus = convertRawProp(
-      context, rawProps, "clearTextOnFocus", sourceTraits.clearTextOnFocus, defaultTraits.clearTextOnFocus);
-  traits.keyboardType =
-      convertRawProp(context, rawProps, "keyboardType", sourceTraits.keyboardType, defaultTraits.keyboardType);
+      context,
+      rawProps,
+      "clearTextOnFocus",
+      sourceTraits.clearTextOnFocus,
+      defaultTraits.clearTextOnFocus);
+  traits.keyboardType = convertRawProp(
+      context,
+      rawProps,
+      "keyboardType",
+      sourceTraits.keyboardType,
+      defaultTraits.keyboardType);
   traits.showSoftInputOnFocus = convertRawProp(
-      context, rawProps, "showSoftInputOnFocus", sourceTraits.showSoftInputOnFocus, defaultTraits.showSoftInputOnFocus);
-  traits.returnKeyType =
-      convertRawProp(context, rawProps, "returnKeyType", sourceTraits.returnKeyType, defaultTraits.returnKeyType);
+      context,
+      rawProps,
+      "showSoftInputOnFocus",
+      sourceTraits.showSoftInputOnFocus,
+      defaultTraits.showSoftInputOnFocus);
+  traits.returnKeyType = convertRawProp(
+      context,
+      rawProps,
+      "returnKeyType",
+      sourceTraits.returnKeyType,
+      defaultTraits.returnKeyType);
   traits.selectTextOnFocus = convertRawProp(
-      context, rawProps, "selectTextOnFocus", sourceTraits.selectTextOnFocus, defaultTraits.selectTextOnFocus);
-  traits.textContentType =
-      convertRawProp(context, rawProps, "textContentType", sourceTraits.textContentType, defaultTraits.textContentType);
-  traits.passwordRules =
-      convertRawProp(context, rawProps, "passwordRules", sourceTraits.passwordRules, defaultTraits.passwordRules);
+      context,
+      rawProps,
+      "selectTextOnFocus",
+      sourceTraits.selectTextOnFocus,
+      defaultTraits.selectTextOnFocus);
+  traits.textContentType = convertRawProp(
+      context,
+      rawProps,
+      "textContentType",
+      sourceTraits.textContentType,
+      defaultTraits.textContentType);
+  traits.passwordRules = convertRawProp(
+      context,
+      rawProps,
+      "passwordRules",
+      sourceTraits.passwordRules,
+      defaultTraits.passwordRules);
   traits.smartInsertDelete = convertRawProp(
-      context, rawProps, "smartInsertDelete", sourceTraits.smartInsertDelete, defaultTraits.smartInsertDelete);
+      context,
+      rawProps,
+      "smartInsertDelete",
+      sourceTraits.smartInsertDelete,
+      defaultTraits.smartInsertDelete);
 
+#if TARGET_OS_OSX // [macOS
+  traits.grammarCheck = convertRawProp(
+      context,
+      rawProps,
+      "grammarCheck",
+      sourceTraits.grammarCheck,
+      defaultTraits.grammarCheck);
+  traits.pastedTypes = convertRawProp(
+      context,
+      rawProps,
+      "pastedTypes",
+      sourceTraits.pastedTypes,
+      defaultTraits.pastedTypes);
+#endif // macOS]
+
   traits.dataDetectorTypes = convertRawProp(
-      context, rawProps, "dataDetectorTypes", sourceTraits.dataDetectorTypes, defaultTraits.dataDetectorTypes);
+      context,
+      rawProps,
+      "dataDetectorTypes",
+      sourceTraits.dataDetectorTypes,
+      defaultTraits.dataDetectorTypes);
 
   return traits;
 }
 
-inline void fromRawValue(const PropsParserContext &context, const RawValue &value, Selection &result)
-{
+inline void fromRawValue(
+    const PropsParserContext& context,
+    const RawValue& value,
+    Selection& result) {
   if (value.hasType<std::unordered_map<std::string, int>>()) {
     auto map = (std::unordered_map<std::string, int>)value;
-    for (const auto &pair : map) {
+    for (const auto& pair : map) {
       if (pair.first == "start") {
         result.start = pair.second;
       } else if (pair.first == "end") {
@@ -101,4 +194,50 @@
     LOG(ERROR) << "Unsupported Selection type";
   }
 }
+
+#if TARGET_OS_OSX // [macOS
+inline void fromRawValue(
+    const PropsParserContext &context,
+    const RawValue& value,
+    SubmitKeyEvent& result) {
+  auto map = (std::unordered_map<std::string, RawValue>)value;
+
+  auto tmp_key = map.find("key");
+  if (tmp_key != map.end()) {
+    fromRawValue(context, tmp_key->second, result.key);
+  }
+  auto tmp_altKey = map.find("altKey");
+  if (tmp_altKey != map.end()) {
+    fromRawValue(context, tmp_altKey->second, result.altKey);
+  }
+  auto tmp_shiftKey = map.find("shiftKey");
+  if (tmp_shiftKey != map.end()) {
+    fromRawValue(context, tmp_shiftKey->second, result.shiftKey);
+  }
+  auto tmp_ctrlKey = map.find("ctrlKey");
+  if (tmp_ctrlKey != map.end()) {
+    fromRawValue(context, tmp_ctrlKey->second, result.ctrlKey);
+  }
+  auto tmp_metaKey = map.find("metaKey");
+  if (tmp_metaKey != map.end()) {
+    fromRawValue(context, tmp_metaKey->second, result.metaKey);
+  }
+  auto tmp_functionKey = map.find("functionKey");
+  if (tmp_functionKey != map.end()) {
+    fromRawValue(context, tmp_functionKey->second, result.functionKey);
+  }
+}
+
+inline void fromRawValue(
+    const PropsParserContext& context,
+    const RawValue& value,
+    std::vector<SubmitKeyEvent>& result) {
+  auto items = (std::vector<RawValue>)value;
+  for (const auto &item : items) {
+    SubmitKeyEvent newItem;
+    fromRawValue(context, item, newItem);
+    result.emplace_back(newItem);
+  }
+}
+#endif // macOS]
 } // namespace facebook::react
