--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/Libraries/Text/TextInput/Singleline/RCTSinglelineTextInputViewManager.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/Libraries/Text/TextInput/Singleline/RCTSinglelineTextInputViewManager.mm	2026-03-06 15:22:52
@@ -7,8 +7,6 @@
 
 #import <React/RCTSinglelineTextInputViewManager.h>
 
-#ifndef RCT_REMOVE_LEGACY_ARCH
-
 #import <React/RCTBaseTextInputShadowView.h>
 #import <React/RCTSinglelineTextInputView.h>
 
@@ -25,11 +23,11 @@
   return shadowView;
 }
 
-- (UIView *)view
+- (RCTPlatformView *)view // [macOS]
 {
   return [[RCTSinglelineTextInputView alloc] initWithBridge:self.bridge];
 }
 
-@end
+RCT_REMAP_OSX_VIEW_PROPERTY(secureTextEntry, useSecureTextField, BOOL) // [macOS]
 
-#endif // RCT_REMOVE_LEGACY_ARCH
+@end
