--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/Mounting/ComponentViews/ActivityIndicator/RCTActivityIndicatorViewComponentView.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/Mounting/ComponentViews/ActivityIndicator/RCTActivityIndicatorViewComponentView.mm	2026-03-06 15:22:52
@@ -8,6 +8,7 @@
 #import "RCTActivityIndicatorViewComponentView.h"
 
 #import <React/RCTConversions.h>
+#import <React/RCTActivityIndicatorView.h> // [macOS]
 
 #import <react/renderer/components/FBReactNativeSpec/ComponentDescriptors.h>
 #import <react/renderer/components/FBReactNativeSpec/EventEmitters.h>
@@ -28,7 +29,7 @@
 }
 
 @implementation RCTActivityIndicatorViewComponentView {
-  UIActivityIndicatorView *_activityIndicatorView;
+  RCTUIActivityIndicatorView *_activityIndicatorView; // [macOS]
 }
 
 #pragma mark - RCTComponentViewProtocol
@@ -44,7 +45,7 @@
     const auto &defaultProps = ActivityIndicatorViewShadowNode::defaultSharedProps();
     _props = defaultProps;
 
-    _activityIndicatorView = [[UIActivityIndicatorView alloc] initWithFrame:self.bounds];
+    _activityIndicatorView = [[RCTUIActivityIndicatorView alloc] initWithFrame:self.bounds]; // [macOS]
     _activityIndicatorView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
 
     if (defaultProps->animating) {
