--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/RCTSurfaceTouchHandler.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.fOXiIZ1bMA	2026-03-06 18:50:51
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 NS_ASSUME_NONNULL_BEGIN
 
@@ -15,8 +15,8 @@
  * Attaches (and detaches) a view to the touch handler.
  * The receiver does not retain the provided view.
  */
-- (void)attachToView:(UIView *)view;
-- (void)detachFromView:(UIView *)view;
+- (void)attachToView:(RCTPlatformView *)view; // [macOS]
+- (void)detachFromView:(RCTPlatformView *)view; // [macOS]
 
 /*
  * Offset of the attached view relative to the root component in points.
