--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/RCTSurfacePresenterBridgeAdapter.h	2026-03-06 18:50:06
+++ /var/folders/jp/b_h7tjf10rg_htfhxp6tds100000gn/T/tmp.dFed5a36Cp	2026-03-06 18:50:51
@@ -7,7 +7,7 @@
 
 #import <Foundation/Foundation.h>
 #import <ReactCommon/RuntimeExecutor.h>
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 #import <react/utils/ContextContainer.h>
 
 NS_ASSUME_NONNULL_BEGIN
