--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/RCTConversions.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/RCTConversions.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <react/renderer/components/view/AccessibilityPrimitives.h>
 #import <react/renderer/components/view/primitives.h>
@@ -14,6 +14,10 @@
 #import <react/renderer/graphics/RCTPlatformColorUtils.h>
 #import <react/renderer/graphics/Transform.h>
 
+#if TARGET_OS_OSX // [macOS
+#import <react/renderer/components/iostextinput/primitives.h>
+#endif // macOS]
+
 NS_ASSUME_NONNULL_BEGIN
 
 inline NSString *RCTNSStringFromString(
@@ -35,7 +39,7 @@
   return std::string{string.UTF8String ?: ""};
 }
 
-inline UIColor *_Nullable RCTUIColorFromSharedColor(const facebook::react::SharedColor &sharedColor)
+inline RCTUIColor *_Nullable RCTUIColorFromSharedColor(const facebook::react::SharedColor &sharedColor) // [macOS]
 {
   return RCTPlatformColorFromColor(*sharedColor);
 }
@@ -66,69 +70,73 @@
   return {edgeInsets.top, edgeInsets.left, edgeInsets.bottom, edgeInsets.right};
 }
 
+#if !TARGET_OS_OSX // [macOS]
 const UIAccessibilityTraits AccessibilityTraitSwitch = 0x20000000000001;
+#endif // [macOS]
 
-inline UIAccessibilityTraits RCTUIAccessibilityTraitsFromAccessibilityTraits(
+// [macOS
+inline RCTUIAccessibilityTraits RCTUIAccessibilityTraitsFromAccessibilityTraits(
     facebook::react::AccessibilityTraits accessibilityTraits)
 {
   using AccessibilityTraits = facebook::react::AccessibilityTraits;
-  UIAccessibilityTraits result = UIAccessibilityTraitNone;
+  RCTUIAccessibilityTraits result = RCTUIAccessibilityTraitNone;
   if ((accessibilityTraits & AccessibilityTraits::Button) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitButton;
+    result |= RCTUIAccessibilityTraitButton;
   }
   if ((accessibilityTraits & AccessibilityTraits::Link) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitLink;
+    result |= RCTUIAccessibilityTraitLink;
   }
   if ((accessibilityTraits & AccessibilityTraits::Image) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitImage;
+    result |= RCTUIAccessibilityTraitImage;
   }
   if ((accessibilityTraits & AccessibilityTraits::Selected) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitSelected;
+    result |= RCTUIAccessibilityTraitSelected;
   }
   if ((accessibilityTraits & AccessibilityTraits::PlaysSound) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitPlaysSound;
+    result |= RCTUIAccessibilityTraitPlaysSound;
   }
   if ((accessibilityTraits & AccessibilityTraits::KeyboardKey) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitKeyboardKey;
+    result |= RCTUIAccessibilityTraitKeyboardKey;
   }
   if ((accessibilityTraits & AccessibilityTraits::StaticText) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitStaticText;
+    result |= RCTUIAccessibilityTraitStaticText;
   }
   if ((accessibilityTraits & AccessibilityTraits::SummaryElement) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitSummaryElement;
+    result |= RCTUIAccessibilityTraitSummaryElement;
   }
   if ((accessibilityTraits & AccessibilityTraits::NotEnabled) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitNotEnabled;
+    result |= RCTUIAccessibilityTraitNotEnabled;
   }
   if ((accessibilityTraits & AccessibilityTraits::UpdatesFrequently) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitUpdatesFrequently;
+    result |= RCTUIAccessibilityTraitUpdatesFrequently;
   }
   if ((accessibilityTraits & AccessibilityTraits::SearchField) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitSearchField;
+    result |= RCTUIAccessibilityTraitSearchField;
   }
   if ((accessibilityTraits & AccessibilityTraits::StartsMediaSession) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitStartsMediaSession;
+    result |= RCTUIAccessibilityTraitStartsMediaSession;
   }
   if ((accessibilityTraits & AccessibilityTraits::Adjustable) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitAdjustable;
+    result |= RCTUIAccessibilityTraitAdjustable;
   }
   if ((accessibilityTraits & AccessibilityTraits::AllowsDirectInteraction) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitAllowsDirectInteraction;
+    result |= RCTUIAccessibilityTraitAllowsDirectInteraction;
   }
   if ((accessibilityTraits & AccessibilityTraits::CausesPageTurn) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitCausesPageTurn;
+    result |= RCTUIAccessibilityTraitCausesPageTurn;
   }
   if ((accessibilityTraits & AccessibilityTraits::Header) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitHeader;
+    result |= RCTUIAccessibilityTraitHeader;
   }
   if ((accessibilityTraits & AccessibilityTraits::Switch) != AccessibilityTraits::None) {
-    result |= AccessibilityTraitSwitch;
+    result |= RCTUIAccessibilityTraitSwitch;
   }
   if ((accessibilityTraits & AccessibilityTraits::TabBar) != AccessibilityTraits::None) {
-    result |= UIAccessibilityTraitTabBar;
+    result |= RCTUIAccessibilityTraitTabBar;
   }
   return result;
-};
+}
+// macOS]
 
 inline CATransform3D RCTCATransform3DFromTransformMatrix(const facebook::react::Transform &transformMatrix)
 {
@@ -153,7 +161,7 @@
 
 inline facebook::react::Point RCTPointFromCGPoint(const CGPoint &point)
 {
-  return {.x = point.x, .y = point.y};
+  return {point.x, point.y};
 }
 
 inline facebook::react::Float RCTFloatFromCGFloat(CGFloat value)
@@ -166,12 +174,12 @@
 
 inline facebook::react::Size RCTSizeFromCGSize(const CGSize &size)
 {
-  return {.width = RCTFloatFromCGFloat(size.width), .height = RCTFloatFromCGFloat(size.height)};
+  return {RCTFloatFromCGFloat(size.width), RCTFloatFromCGFloat(size.height)};
 }
 
 inline facebook::react::Rect RCTRectFromCGRect(const CGRect &rect)
 {
-  return {.origin = RCTPointFromCGPoint(rect.origin), .size = RCTSizeFromCGSize(rect.size)};
+  return {RCTPointFromCGPoint(rect.origin), RCTSizeFromCGSize(rect.size)};
 }
 
 inline facebook::react::EdgeInsets RCTEdgeInsetsFromUIEdgeInsets(const UIEdgeInsets &edgeInsets)
@@ -184,4 +192,29 @@
   return isRTL ? facebook::react::LayoutDirection::RightToLeft : facebook::react::LayoutDirection::LeftToRight;
 }
 
+#if TARGET_OS_OSX // [macOS
+inline NSArray<NSPasteboardType> *RCTPasteboardTypeArrayFromProps(const std::vector<facebook::react::PastedTypesType> &pastedTypes)
+{
+  NSMutableArray<NSPasteboardType> *types = [NSMutableArray new];
+  
+  for (const auto &type : pastedTypes) {
+    switch (type) {
+      case facebook::react::PastedTypesType::FileUrl:
+        [types addObjectsFromArray:@[NSFilenamesPboardType]];
+        break;
+      case facebook::react::PastedTypesType::Image:
+        [types addObjectsFromArray:@[NSPasteboardTypePNG, NSPasteboardTypeTIFF]];
+        break;
+      case facebook::react::PastedTypesType::String:
+        [types addObjectsFromArray:@[NSPasteboardTypeString]];
+        break;
+      default:
+        break;
+    }
+  }
+    
+  return [types copy];
+}
+#endif // macOS]
+
 NS_ASSUME_NONNULL_END
