--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/RCTSurfaceTouchHandler.mm	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/RCTSurfaceTouchHandler.mm	2026-03-06 15:22:52
@@ -10,6 +10,7 @@
 #import <React/RCTIdentifierPool.h>
 #import <React/RCTUtils.h>
 #import <React/RCTViewComponentView.h>
+#import <React/RCTUIKit.h>
 
 #import "RCTConversions.h"
 #import "RCTSurfacePointerHandler.h"
@@ -31,7 +32,7 @@
   /*
    * A component view on which the touch was begun.
    */
-  __strong UIView<RCTComponentViewProtocol> *componentView = nil;
+  __strong RCTPlatformView<RCTComponentViewProtocol> *componentView = nil; // [macOS]
 
   struct Hasher {
     size_t operator()(const ActiveTouch &activeTouch) const
@@ -50,15 +51,25 @@
 
 static void UpdateActiveTouchWithUITouch(
     ActiveTouch &activeTouch,
-    UITouch *uiTouch,
-    UIView *rootComponentView,
-    CGPoint rootViewOriginOffset)
+    RCTUITouch *uiTouch, // [macOS]
+    RCTPlatformView *rootComponentView,  // [macOS]
+    CGPoint rootViewOriginOffset) // [macOS]
 {
+#if !TARGET_OS_OSX // [macOS]
   CGPoint offsetPoint = [uiTouch locationInView:activeTouch.componentView];
   CGPoint pagePoint = [uiTouch locationInView:rootComponentView];
+#if !TARGET_OS_VISION // [visionOS]
   CGPoint screenPoint = [rootComponentView convertPoint:pagePoint
                                       toCoordinateSpace:rootComponentView.window.screen.coordinateSpace];
+#else // [visionOS
+  CGPoint screenPoint = CGPointZero;
+#endif // visionOS]
   pagePoint = CGPointMake(pagePoint.x + rootViewOriginOffset.x, pagePoint.y + rootViewOriginOffset.y);
+#else // [macOS
+  CGPoint offsetPoint = [activeTouch.componentView convertPoint:uiTouch.locationInWindow fromView:nil];
+  CGPoint screenPoint = uiTouch.locationInWindow;
+  CGPoint pagePoint = CGPointMake(screenPoint.x, CGRectGetHeight(rootComponentView.window.frame) - screenPoint.y);
+#endif // macOS]
 
   activeTouch.touch.offsetPoint = RCTPointFromCGPoint(offsetPoint);
   activeTouch.touch.screenPoint = RCTPointFromCGPoint(screenPoint);
@@ -66,22 +77,59 @@
 
   activeTouch.touch.timestamp = uiTouch.timestamp;
 
+#if !TARGET_OS_OSX // [macOS]
   if (RCTForceTouchAvailable()) {
     activeTouch.touch.force = RCTZeroIfNaN(uiTouch.force / uiTouch.maximumPossibleForce);
   }
+#else // [macOS
+  NSEventType type = uiTouch.type;
+  if (type == NSEventTypeLeftMouseDown || type == NSEventTypeLeftMouseUp || type == NSEventTypeLeftMouseDragged) {
+    activeTouch.touch.button = 0;
+  } else if (type == NSEventTypeRightMouseDown || type == NSEventTypeRightMouseUp || type == NSEventTypeRightMouseDragged) {
+    activeTouch.touch.button = 2;
+  }
+
+  NSEventModifierFlags modifierFlags = uiTouch.modifierFlags;
+  if (modifierFlags & NSEventModifierFlagOption) {
+    activeTouch.touch.altKey = true;
+  }
+  if (modifierFlags & NSEventModifierFlagControl) {
+    activeTouch.touch.ctrlKey = true;
+  }
+  if (modifierFlags & NSEventModifierFlagShift) {
+    activeTouch.touch.shiftKey = true;
+  }
+  if (modifierFlags & NSEventModifierFlagCommand) {
+    activeTouch.touch.metaKey = true;
+  }
+#endif // macOS]
 }
 
-static ActiveTouch CreateTouchWithUITouch(UITouch *uiTouch, UIView *rootComponentView, CGPoint rootViewOriginOffset)
+static ActiveTouch CreateTouchWithUITouch(RCTUITouch *uiTouch, RCTPlatformView *rootComponentView, CGPoint rootViewOriginOffset) // [macOS]
 {
   ActiveTouch activeTouch = {};
 
   // Find closest Fabric-managed touchable view
-  UIView *componentView = uiTouch.view;
+#if !TARGET_OS_OSX // [macOS]
+  RCTUIView *componentView = uiTouch.view; // [macOS]
+#else // [macOS
+  CGPoint touchLocation = [rootComponentView.superview convertPoint:uiTouch.locationInWindow fromView:nil];
+  RCTUIView *componentView = (RCTUIView *) [rootComponentView hitTest:touchLocation]; // [macOS]
+#endif // macOS]
   while (componentView) {
+#if !TARGET_OS_OSX // [macOS]
+    CGPoint offsetPoint = [uiTouch locationInView:componentView];
+#else // [macOS
+    CGPoint offsetPoint = [componentView convertPoint:uiTouch.locationInWindow fromView:nil];
+#endif // macOS]
     if ([componentView respondsToSelector:@selector(touchEventEmitterAtPoint:)]) {
       activeTouch.eventEmitter = [(id<RCTTouchableComponentViewProtocol>)componentView
-          touchEventEmitterAtPoint:[uiTouch locationInView:componentView]];
+          touchEventEmitterAtPoint:offsetPoint];
+#if !TARGET_OS_OSX // [macOS]
       activeTouch.touch.target = (Tag)componentView.tag;
+#else // [macOS
+      activeTouch.touch.target = (Tag)(componentView.reactTag.intValue);
+#endif // macOS]
       activeTouch.componentView = componentView;
       break;
     }
@@ -92,9 +140,10 @@
   return activeTouch;
 }
 
+#if !TARGET_OS_OSX // [macOS]
 static BOOL AllTouchesAreCancelledOrEnded(NSSet<UITouch *> *touches)
 {
-  for (UITouch *touch in touches) {
+  for (RCTUITouch *touch in touches) { // [macOS]
     if (touch.phase == UITouchPhaseBegan || touch.phase == UITouchPhaseMoved || touch.phase == UITouchPhaseStationary) {
       return NO;
     }
@@ -102,15 +151,16 @@
   return YES;
 }
 
-static BOOL AnyTouchesChanged(NSSet<UITouch *> *touches)
+static BOOL AnyTouchesChanged(NSSet<RCTUITouch *> *touches) // [macOS]
 {
-  for (UITouch *touch in touches) {
+  for (RCTUITouch *touch in touches) { // [macOS]
     if (touch.phase == UITouchPhaseBegan || touch.phase == UITouchPhaseMoved) {
       return YES;
     }
   }
   return NO;
 }
+#endif // [macOS]
 
 /**
  * Surprisingly, `__unsafe_unretained id` pointers are not regular pointers
@@ -130,14 +180,18 @@
 @end
 
 @implementation RCTSurfaceTouchHandler {
-  std::unordered_map<__unsafe_unretained UITouch *, ActiveTouch, PointerHasher<__unsafe_unretained UITouch *>>
+#if !TARGET_OS_OSX // [macOS]
+  std::unordered_map<__unsafe_unretained RCTUITouch *, ActiveTouch, PointerHasher<__unsafe_unretained RCTUITouch *>>
       _activeTouches;
+#else // [macOS
+  std::unordered_map<NSInteger, ActiveTouch> _activeTouches;
+#endif // macOS]
 
   /*
    * We hold the view weakly to prevent a retain cycle.
    */
-  __weak UIView *_rootComponentView;
-  RCTIdentifierPool<17> _identifierPool;
+  __weak RCTPlatformView *_rootComponentView; // [macOS]
+  RCTIdentifierPool<11> _identifierPool;
 
   RCTSurfacePointerHandler *_pointerHandler;
 }
@@ -149,9 +203,11 @@
     // to be used as a top level event delegated recognizer.
     // Otherwise, lower-level components not built using React Native,
     // will fail to recognize gestures.
+#if !TARGET_OS_OSX // [macOS]
     self.cancelsTouchesInView = NO;
     self.delaysTouchesBegan = NO; // This is default value.
     self.delaysTouchesEnded = NO;
+#endif // [macOS]
 
     self.delegate = self;
 
@@ -165,7 +221,7 @@
 
 RCT_NOT_IMPLEMENTED(-(instancetype)initWithTarget : (id)target action : (SEL)action)
 
-- (void)attachToView:(UIView *)view
+- (void)attachToView:(RCTPlatformView *)view // [macOS]
 {
   RCTAssert(self.view == nil, @"RCTTouchHandler already has attached view.");
 
@@ -177,7 +233,7 @@
   }
 }
 
-- (void)detachFromView:(UIView *)view
+- (void)detachFromView:(RCTPlatformView *)view // [macOS]
 {
   RCTAssertParam(view);
   RCTAssert(self.view == view, @"RCTTouchHandler attached to another view.");
@@ -190,19 +246,27 @@
   }
 }
 
-- (void)_registerTouches:(NSSet<UITouch *> *)touches
+- (void)_registerTouches:(NSSet<RCTUITouch *> *)touches // [macOS]
 {
-  for (UITouch *touch in touches) {
-    auto activeTouch = CreateTouchWithUITouch(touch, _rootComponentView, _viewOriginOffset);
+  for (RCTUITouch *touch in touches) { // [macOS]
+		auto activeTouch = CreateTouchWithUITouch(touch, _rootComponentView, _viewOriginOffset);
     activeTouch.touch.identifier = _identifierPool.dequeue();
+#if !TARGET_OS_OSX // [macOS]
     _activeTouches.emplace(touch, activeTouch);
+#else // [macOS
+    _activeTouches.emplace(touch.eventNumber, activeTouch);
+#endif // macOS]
   }
 }
 
-- (void)_updateTouches:(NSSet<UITouch *> *)touches
+- (void)_updateTouches:(NSSet<RCTUITouch *> *)touches // [macOS]
 {
-  for (UITouch *touch in touches) {
+  for (RCTUITouch *touch in touches) { // [macOS]
+#if !TARGET_OS_OSX // [macOS]
     auto iterator = _activeTouches.find(touch);
+#else // [macOS
+    auto iterator = _activeTouches.find(touch.eventNumber);
+#endif // macOS]
     RCTAssert(iterator != _activeTouches.end(), @"Inconsistency between local and UIKit touch registries");
     if (iterator == _activeTouches.end()) {
       continue;
@@ -212,27 +276,39 @@
   }
 }
 
-- (void)_unregisterTouches:(NSSet<UITouch *> *)touches
+- (void)_unregisterTouches:(NSSet<RCTUITouch *> *)touches // [macOS]
 {
-  for (UITouch *touch in touches) {
+  for (RCTUITouch *touch in touches) { // [macOS]
+#if !TARGET_OS_OSX // [macOS]
     auto iterator = _activeTouches.find(touch);
+#else // [macOS
+    auto iterator = _activeTouches.find(touch.eventNumber);
+#endif // macOS]
     RCTAssert(iterator != _activeTouches.end(), @"Inconsistency between local and UIKit touch registries");
     if (iterator == _activeTouches.end()) {
       continue;
     }
     auto &activeTouch = iterator->second;
     _identifierPool.enqueue(activeTouch.touch.identifier);
+#if !TARGET_OS_OSX // [macOS]
     _activeTouches.erase(touch);
+#else // [macOS
+    _activeTouches.erase(touch.eventNumber);
+#endif // macOS]
   }
 }
 
-- (std::vector<ActiveTouch>)_activeTouchesFromTouches:(NSSet<UITouch *> *)touches
+- (std::vector<ActiveTouch>)_activeTouchesFromTouches:(NSSet<RCTUITouch *> *)touches // [macOS]
 {
   std::vector<ActiveTouch> activeTouches;
   activeTouches.reserve(touches.count);
 
-  for (UITouch *touch in touches) {
+  for (RCTUITouch *touch in touches) {
+#if !TARGET_OS_OSX // [macOS]
     auto iterator = _activeTouches.find(touch);
+#else // [macOS
+    auto iterator = _activeTouches.find(touch.eventNumber);
+#endif // macOS]
     RCTAssert(iterator != _activeTouches.end(), @"Inconsistency between local and UIKit touch registries");
     if (iterator == _activeTouches.end()) {
       continue;
@@ -300,7 +376,9 @@
 
 #pragma mark - `UIResponder`-ish touch-delivery methods
 
-- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
+#if !TARGET_OS_OSX // [macOS]
+
+- (void)touchesBegan:(NSSet<RCTUITouch *> *)touches withEvent:(UIEvent *)event // [macOS]
 {
   [super touchesBegan:touches withEvent:event];
 
@@ -314,7 +392,7 @@
   }
 }
 
-- (void)touchesMoved:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
+- (void)touchesMoved:(NSSet<RCTUITouch *> *)touches withEvent:(UIEvent *)event
 {
   [super touchesMoved:touches withEvent:event];
 
@@ -324,7 +402,7 @@
   self.state = UIGestureRecognizerStateChanged;
 }
 
-- (void)touchesEnded:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
+- (void)touchesEnded:(NSSet<RCTUITouch *> *)touches withEvent:(UIEvent *)event
 {
   [super touchesEnded:touches withEvent:event];
 
@@ -339,7 +417,7 @@
   }
 }
 
-- (void)touchesCancelled:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
+- (void)touchesCancelled:(NSSet<RCTUITouch *> *)touches withEvent:(UIEvent *)event
 {
   [super touchesCancelled:touches withEvent:event];
 
@@ -354,6 +432,97 @@
   }
 }
 
+#else // [macOS
+
+- (BOOL)acceptsFirstMouse:(NSEvent *)event
+{
+  // This will only be called if the hit-tested view returns YES for acceptsFirstMouse,
+  // therefore asking it again would be redundant.
+  return YES;
+}
+
+- (void)mouseDown:(NSEvent *)event
+{
+  [super mouseDown:event];
+
+  {
+    NSSet* touches = [NSSet setWithObject:event];
+    [self _registerTouches:touches]; // [macOS]
+    [self _dispatchActiveTouches:[self _activeTouchesFromTouches:touches] eventType:RCTTouchEventTypeTouchStart];
+
+    if (self.state == NSGestureRecognizerStatePossible) {
+      self.state = NSGestureRecognizerStateBegan;
+    } else if (self.state == NSGestureRecognizerStateBegan) {
+      self.state = NSGestureRecognizerStateChanged;
+    }
+  }
+}
+
+- (void)rightMouseDown:(NSEvent *)event
+{
+  [super rightMouseDown:event];
+
+  {
+    NSSet* touches = [NSSet setWithObject:event];
+		[self _registerTouches:touches]; // [macOS]
+    [self _dispatchActiveTouches:[self _activeTouchesFromTouches:touches] eventType:RCTTouchEventTypeTouchStart];
+
+    if (self.state == NSGestureRecognizerStatePossible) {
+      self.state = NSGestureRecognizerStateBegan;
+    } else if (self.state == NSGestureRecognizerStateBegan) {
+      self.state = NSGestureRecognizerStateChanged;
+    }
+  }
+}
+
+- (void)mouseDragged:(NSEvent *)event
+{
+  [super mouseDragged:event];
+
+  NSSet* touches = [NSSet setWithObject:event];
+  [self _updateTouches:touches]; // [macOS]
+  [self _dispatchActiveTouches:[self _activeTouchesFromTouches:touches] eventType:RCTTouchEventTypeTouchMove];
+
+  self.state = NSGestureRecognizerStateChanged;
+}
+
+- (void)rightMouseDragged:(NSEvent *)event
+{
+  [super rightMouseDragged:event];
+
+  NSSet* touches = [NSSet setWithObject:event];
+  [self _updateTouches:touches]; // [macOS]
+  [self _dispatchActiveTouches:[self _activeTouchesFromTouches:touches] eventType:RCTTouchEventTypeTouchMove];
+
+  self.state = NSGestureRecognizerStateChanged;
+}
+
+- (void)mouseUp:(NSEvent *)event
+{
+  [super mouseUp:event];
+
+  NSSet* touches = [NSSet setWithObject:event];
+  [self _updateTouches:touches]; // [macOS]
+  [self _dispatchActiveTouches:[self _activeTouchesFromTouches:touches] eventType:RCTTouchEventTypeTouchEnd];
+  [self _unregisterTouches:touches];
+
+  self.state = NSGestureRecognizerStateEnded;
+}
+
+- (void)rightMouseUp:(NSEvent *)event
+{
+  [super rightMouseUp:event];
+
+  NSSet* touches = [NSSet setWithObject:event];
+  [self _updateTouches:touches]; // [macOS]
+  [self _dispatchActiveTouches:[self _activeTouchesFromTouches:touches] eventType:RCTTouchEventTypeTouchEnd];
+  [self _unregisterTouches:touches];
+
+  self.state = NSGestureRecognizerStateEnded;
+}
+
+#endif // macOS]
+
 - (void)reset
 {
   [super reset];
@@ -381,9 +550,13 @@
 
 - (BOOL)canBePreventedByGestureRecognizer:(UIGestureRecognizer *)preventingGestureRecognizer
 {
+#if !TARGET_OS_OSX // [macOS]
   // We fail in favour of other external gesture recognizers.
   // iOS will ask `delegate`'s opinion about this gesture recognizer little bit later.
   return ![preventingGestureRecognizer.view isDescendantOfView:self.view];
+#else  // [macOS
+  return NO;
+#endif // macOS]
 }
 
 #pragma mark - UIGestureRecognizerDelegate
@@ -399,7 +572,7 @@
     shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer
 {
   BOOL canBePrevented = [self canBePreventedByGestureRecognizer:otherGestureRecognizer];
-  if (canBePrevented && otherGestureRecognizer.cancelsTouchesInView) {
+  if (canBePrevented) {
     [self _cancelTouches];
   }
   return NO;
