--- /Users/sethagent/Developer/react-native-upstream/packages/react-native/React/Fabric/RCTSurfacePresenter.h	2026-03-06 15:51:54
+++ /Users/sethagent/Developer/react-native-macos-fork/packages/react-native/React/Fabric/RCTSurfacePresenter.h	2026-03-06 15:22:52
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#import <UIKit/UIKit.h>
+#import <React/RCTUIKit.h> // [macOS]
 
 #import <React/RCTPrimitives.h>
 #import <React/RCTSurfacePresenterStub.h>
@@ -27,11 +27,11 @@
  */
 @interface RCTSurfacePresenter : NSObject
 
-- (instancetype)initWithContextContainer:(std::shared_ptr<const facebook::react::ContextContainer>)contextContainer
+- (instancetype)initWithContextContainer:(facebook::react::ContextContainer::Shared)contextContainer
                          runtimeExecutor:(facebook::react::RuntimeExecutor)runtimeExecutor
               bridgelessBindingsExecutor:(std::optional<facebook::react::RuntimeExecutor>)bridgelessBindingsExecutor;
 
-@property (nonatomic) std::shared_ptr<const facebook::react::ContextContainer> contextContainer;
+@property (nonatomic) facebook::react::ContextContainer::Shared contextContainer;
 @property (nonatomic) facebook::react::RuntimeExecutor runtimeExecutor;
 
 /*
@@ -80,7 +80,7 @@
 /*
  * Please do not use this, this will be deleted soon.
  */
-- (nullable UIView *)findComponentViewWithTag_DO_NOT_USE_DEPRECATED:(NSInteger)tag;
+- (nullable RCTUIView *)findComponentViewWithTag_DO_NOT_USE_DEPRECATED:(NSInteger)tag; // [macOS]
 
 @end
 
